<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kathleen Durkin">
<meta name="dcterms.date" content="2025-04-23">

<title>Kathleen’s Lab Notebook - Timeseries molecular: A.pul, epigenetic features to predict expression. Machine learning – Part 2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kathleen’s Lab Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../daily_log.html" rel="" target="">
 <span class="menu-text">Daily Notebook Posts</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../projects.html" rel="" target="">
 <span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/shedurkin" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:kdurkin1@uw.edu" rel="" target=""><i class="bi bi-envelope" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <h1 class="title">Timeseries molecular: A.pul, epigenetic features to predict expression. Machine learning – Part 2</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">E5-coral</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Kathleen Durkin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 23, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#set-up" id="toc-set-up" class="nav-link active" data-scroll-target="#set-up">1 Set up</a></li>
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link" data-scroll-target="#load-libraries">2 Load libraries</a></li>
  <li><a href="#load-and-format-data" id="toc-load-and-format-data" class="nav-link" data-scroll-target="#load-and-format-data">3 Load and format data</a>
  <ul class="collapse">
  <li><a href="#rna-seq-data-mrna-mirna-lncrna" id="toc-rna-seq-data-mrna-mirna-lncrna" class="nav-link" data-scroll-target="#rna-seq-data-mrna-mirna-lncrna">3.1 RNA-seq data (mRNA, miRNA, lncRNA)</a></li>
  <li><a href="#wgbs-data" id="toc-wgbs-data" class="nav-link" data-scroll-target="#wgbs-data">3.2 WGBS data</a></li>
  <li><a href="#filter-data-sets" id="toc-filter-data-sets" class="nav-link" data-scroll-target="#filter-data-sets">3.3 Filter data sets</a></li>
  <li><a href="#transform-data" id="toc-transform-data" class="nav-link" data-scroll-target="#transform-data">3.4 Transform data</a></li>
  <li><a href="#islolate-gene-sets" id="toc-islolate-gene-sets" class="nav-link" data-scroll-target="#islolate-gene-sets">3.5 Islolate gene sets</a></li>
  </ul></li>
  <li><a href="#merge-predictor-features" id="toc-merge-predictor-features" class="nav-link" data-scroll-target="#merge-predictor-features">5 Merge predictor features</a>
  <ul class="collapse">
  <li><a href="#merge-raw-datsets" id="toc-merge-raw-datsets" class="nav-link" data-scroll-target="#merge-raw-datsets">5.2 Merge raw datsets</a></li>
  </ul></li>
  <li><a href="#host-biomass-host_afdw" id="toc-host-biomass-host_afdw" class="nav-link" data-scroll-target="#host-biomass-host_afdw">14 Host biomass (Host_AFDW)</a>
  <ul class="collapse">
  <li><a href="#the-model" id="toc-the-model" class="nav-link" data-scroll-target="#the-model">14.1 The model</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">14.2 Results</a></li>
  </ul></li>
  <li><a href="#symbiont-photsynthesis-am" id="toc-symbiont-photsynthesis-am" class="nav-link" data-scroll-target="#symbiont-photsynthesis-am">15 Symbiont photsynthesis (Am)</a>
  <ul class="collapse">
  <li><a href="#the-model-1" id="toc-the-model-1" class="nav-link" data-scroll-target="#the-model-1">15.1 The model</a></li>
  <li><a href="#results-1" id="toc-results-1" class="nav-link" data-scroll-target="#results-1">15.2 Results</a></li>
  </ul></li>
  <li><a href="#atp-production-go-terms" id="toc-atp-production-go-terms" class="nav-link" data-scroll-target="#atp-production-go-terms">16 ATP production (GO terms)</a>
  <ul class="collapse">
  <li><a href="#the-model-2" id="toc-the-model-2" class="nav-link" data-scroll-target="#the-model-2">16.1 The model</a></li>
  <li><a href="#results-2" id="toc-results-2" class="nav-link" data-scroll-target="#results-2">16.2 Results</a></li>
  </ul></li>
  <li><a href="#energy-usagestorage-go-terms" id="toc-energy-usagestorage-go-terms" class="nav-link" data-scroll-target="#energy-usagestorage-go-terms">17 Energy Usage/Storage (GO terms)</a>
  <ul class="collapse">
  <li><a href="#the-model-3" id="toc-the-model-3" class="nav-link" data-scroll-target="#the-model-3">17.1 The model</a></li>
  <li><a href="#results-3" id="toc-results-3" class="nav-link" data-scroll-target="#results-3">17.2 Results</a></li>
  </ul></li>
  <li><a href="#compare" id="toc-compare" class="nav-link" data-scroll-target="#compare">18 Compare</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">




<p><a href="https://github.com/urol-e5/timeseries_molecular/blob/main/D-Apul/code/22.2-Apul-multiomic-machine-learning.Rmd">Code</a></p>
<p><a href="https://github.com/urol-e5/timeseries_molecular/blob/main/D-Apul/code/22.2-Apul-multiomic-machine-learning.md">Rendered Code</a></p>
<p>Applying ML model using multiomic predictors – miRNA + lncRNA + methylation as predictors of gene expression. This time, instead of reducing the dimensionality of all sets using PCs, I’m using vsd-transformed miRNA/lncRNA/CpG counts as predictors, and vsd-transformed gene counts as responses.</p>
<p>Excluding the dimensionality-reduction makes the model training much more computationally intensive, so I only ran it for the smaller gene sets (Host biomass, ATP production GO terms, Energy usage/storage GO terms).</p>
<p>Inputs:</p>
<ul>
<li><p>RNA counts matrix (raw): <code>../output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-gene_count_matrix.csv</code></p></li>
<li><p>Gene sets of interest: <code>../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/</code></p></li>
<li><p>sRNA/miRNA counts matrix (raw): <code>../output/03.10-D-Apul-sRNAseq-expression-DESeq2/Apul_miRNA_ShortStack_counts_formatted.txt</code></p></li>
<li><p>lncRNA counts matrix (raw): <code>../output/08-Apul-lncNRA/counts.txt</code></p></li>
<li><p>WGBS data (processed): Performed in <code>/timeseries_molecular/D-Apul/output/15.5-Apul-bismark/</code>, data in <a href="https://gannet.fish.washington.edu/seashell/bu-github/timeseries_molecular/D-Apul/output/15.5-Apul-bismark/">large-file storage</a>.</p></li>
<li><p>sample metadata: <code>../../M-multi-species/data/rna_metadata.csv</code></p></li>
</ul>
<section id="set-up" class="level1">
<h1>1 Set up</h1>
</section>
<section id="load-libraries" class="level1">
<h1>2 Load libraries</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(psych)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidygraph)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rvest)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genefilter)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The model includes random processes, so set a seed for reproducability. Note: for final results, should we consider bootstrapping to ensure reported results are representative?</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">703</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="load-and-format-data" class="level1">
<h1>3 Load and format data</h1>
<section id="rna-seq-data-mrna-mirna-lncrna" class="level2">
<h2 class="anchored" data-anchor-id="rna-seq-data-mrna-mirna-lncrna">3.1 RNA-seq data (mRNA, miRNA, lncRNA)</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">### mRNA </span><span class="al">###</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># raw gene counts data (will filter and variance stabilize)</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>Apul_genes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-gene_count_matrix.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Rows: 44371 Columns: 41
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: ","
## chr  (1): gene_id
## dbl (40): 1A1, 1A10, 1A12, 1A2, 1A8, 1A9, 1B1, 1B10, 1B2, 1B5, 1B9, 1C10, 1C...
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>Apul_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_genes)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># format gene IDs as rownames (instead of a column)</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Apul_genes) <span class="ot">&lt;-</span> Apul_genes<span class="sc">$</span>gene_id</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>Apul_genes <span class="ot">&lt;-</span> Apul_genes<span class="sc">%&gt;%</span><span class="fu">select</span>(<span class="sc">!</span>gene_id)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="do">### miRNA </span><span class="al">###</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># raw miRNA counts (will filter and variance stabilize)</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>Apul_miRNA <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">"../output/03.10-D-Apul-sRNAseq-expression-DESeq2/Apul_miRNA_ShortStack_counts_formatted.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="do">### lncRNA </span><span class="al">###</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co"># raw lncRNA counts (will filter and variance stabilize)</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA_full <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/08-Apul-lncRNA/counts.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove info on genomic location, set lncRNA IDs as rownames</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Apul_lncRNA_full) <span class="ot">&lt;-</span> Apul_lncRNA_full<span class="sc">$</span>Geneid</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA <span class="ot">&lt;-</span> Apul_lncRNA_full <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>Geneid, <span class="sc">-</span>Chr, <span class="sc">-</span>Start, <span class="sc">-</span>End, <span class="sc">-</span>Strand, <span class="sc">-</span>Length)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="do">### load and format metadata </span><span class="al">###</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../../M-multi-species/data/rna_metadata.csv"</span>)<span class="sc">%&gt;%</span><span class="fu">select</span>(AzentaSampleName, ColonyID, Timepoint) <span class="sc">%&gt;%</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"ACR"</span>, ColonyID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## New names:
## Rows: 117 Columns: 19
## ── Column specification
## ──────────────────────────────────────────────────────── Delimiter: "," chr
## (13): SampleName, WellNumber, AzentaSampleName, ColonyID, Timepoint, Sam... dbl
## (5): SampleNumber, Plate, TotalAmount-ng, Volume-uL, Conc-ng.uL lgl (1):
## MethodUsedForSpectrophotometry
## ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
## Specify the column types or set `show_col_types = FALSE` to quiet this message.
## • `` -&gt; `...19`</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">paste0</span>(metadata<span class="sc">$</span>ColonyID, <span class="st">"-"</span>, metadata<span class="sc">$</span>Timepoint)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Sample</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Setting row names on a tibble is deprecated.</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>colonies <span class="ot">&lt;-</span> <span class="fu">unique</span>(metadata<span class="sc">$</span>ColonyID)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename gene column names to include full sample info</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_genes) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Sample[<span class="fu">match</span>(<span class="fu">colnames</span>(Apul_genes), metadata<span class="sc">$</span>AzentaSampleName)]</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename miRNA column names to match formatting</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_miRNA) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(Apul_miRNA))</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_miRNA) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Sample[<span class="fu">match</span>(<span class="fu">colnames</span>(Apul_miRNA), metadata<span class="sc">$</span>AzentaSampleName)]</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co"># rename lncRNA colin names to include full sample info</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_lncRNA) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"...data."</span>, <span class="st">""</span>, <span class="fu">colnames</span>(Apul_lncRNA))</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_lncRNA) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".sorted.bam"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(Apul_lncRNA))</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_lncRNA) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Sample[<span class="fu">match</span>(<span class="fu">colnames</span>(Apul_lncRNA), metadata<span class="sc">$</span>AzentaSampleName)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="wgbs-data" class="level2">
<h2 class="anchored" data-anchor-id="wgbs-data">3.2 WGBS data</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#pull processed files from Gannet </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: Unfortunately we can't use the `cache` feature to make this process more time efficient, as it doesn't support long vectors</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the base URL</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>base_url <span class="ot">&lt;-</span> <span class="st">"https://gannet.fish.washington.edu/seashell/bu-github/timeseries_molecular/D-Apul/output/15.5-Apul-bismark/"</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the HTML page</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>page <span class="ot">&lt;-</span> <span class="fu">read_html</span>(base_url)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract links to files</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>file_links <span class="ot">&lt;-</span> page <span class="sc">%&gt;%</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">html_nodes</span>(<span class="st">"a"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">html_attr</span>(<span class="st">"href"</span>)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for files ending in "processed.txt"</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>processed_files <span class="ot">&lt;-</span> file_links[<span class="fu">grepl</span>(<span class="st">"processed</span><span class="sc">\\</span><span class="st">.txt$"</span>, file_links)]</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Create full URLs</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>file_urls <span class="ot">&lt;-</span> <span class="fu">paste0</span>(base_url, processed_files)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to read a file from URL</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>read_processed_file <span class="ot">&lt;-</span> <span class="cf">function</span>(url) {</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_table</span>(url, <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default =</span> <span class="st">"c"</span>))  <span class="co"># Read as character to avoid parsing issues</span></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Import all processed files into a list</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>processed_data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(file_urls, read_processed_file)</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Name the list elements by file name</span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_data) <span class="ot">&lt;-</span> processed_files</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Print structure of imported data</span></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(processed_data)</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a><span class="co"># add a header row that has "CpG" for the first column and "sample" for the second column, which will be populated by the file name </span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a>processed_data <span class="ot">&lt;-</span> <span class="fu">Map</span>(<span class="cf">function</span>(df, filename) {</span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CpG"</span>, filename)  <span class="co"># Rename columns</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>}, processed_data, <span class="fu">names</span>(processed_data))  <span class="co"># Use stored file names</span></span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a><span class="co">#merge files together by "CpG"</span></span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">reduce</span>(processed_data, full_join, <span class="at">by =</span> <span class="st">"CpG"</span>)</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Print structure of final merged data</span></span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(merged_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Replace any NA with 0.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert all columns (except "CpG") to numeric and replace NAs with 0</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> merged_data <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="sc">-</span>CpG, as.numeric)) <span class="sc">%&gt;%</span>  <span class="co"># Convert all except CpG to numeric</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="sc">-</span>CpG, <span class="sc">~</span> <span class="fu">replace_na</span>(.x, <span class="dv">0</span>)))  <span class="co"># Replace NA with 0 in numeric columns</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="filter-data-sets" class="level2">
<h2 class="anchored" data-anchor-id="filter-data-sets">3.3 Filter data sets</h2>
<p>Only keep CpGs that have a non-zero value in all samples.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> merged_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">if_all</span>(<span class="sc">-</span>CpG, <span class="sc">~</span> .x <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure it's formatted as a data frame</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(filtered_wgbs)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Only keep the sample information in the column name. </span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(filtered_wgbs) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^(.*?)_.*$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">colnames</span>(filtered_wgbs))</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Set CpG IDs to rownames</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(filtered_wgbs) <span class="ot">&lt;-</span> filtered_wgbs<span class="sc">$</span>CpG</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> filtered_wgbs <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>CpG)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(merged_data)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(filtered_wgbs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>We had 12,093,025 CpGs before filtering and have only 507 after filtering. This makes sense because most CpGs were not methylated in all samples.</p>
<p>Save filtered set to make code reruns/knitting quicker</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(filtered_wgbs, <span class="st">"../output/22.2-Apul-multiomic-machine-learning/filtered-WGBS-CpG-counts.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>If knitting/rerunning code, we can load in this filtered data here, instead of loading raw counts and reprocessing.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/22.2-Apul-multiomic-machine-learning/filtered-WGBS-CpG-counts.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only keep genes, miRNA, and lncRNA that are present in at least one sample</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># genes</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>Apul_genes_red <span class="ot">&lt;-</span> Apul_genes[<span class="fu">rowSums</span>(Apul_genes) <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co"># miRNA</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>Apul_miRNA_red <span class="ot">&lt;-</span> Apul_miRNA[<span class="fu">rowSums</span>(Apul_miRNA) <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># lncRNA</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA_red <span class="ot">&lt;-</span> Apul_lncRNA[<span class="fu">rowSums</span>(Apul_lncRNA) <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Retained "</span>, <span class="fu">nrow</span>(Apul_genes_red), <span class="st">" of "</span>, <span class="fu">nrow</span>(Apul_genes), <span class="st">"genes; "</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>       <span class="fu">nrow</span>(Apul_miRNA_red), <span class="st">" of "</span>, <span class="fu">nrow</span>(Apul_miRNA), <span class="st">" miRNA; and "</span>, </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>       <span class="fu">nrow</span>(Apul_lncRNA_red), <span class="st">" of "</span>, <span class="fu">nrow</span>(Apul_lncRNA), <span class="st">" lncRNA"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Retained  35869  of  44371 genes;  51  of  51  miRNA; and  20559  of  24181  lncRNA</code></pre>
<p><em>pOverA</em>: Specifying the minimum count for a proportion of samples for each gene. Setting 3/40 = 0.08. This would retain genes that are only expressed in a single season in a couple of the colonies. Additionally, setting the minimum count so that the minimum number of samples must have a gene count above a certain threshold.</p>
<p>genes:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>filt <span class="ot">&lt;-</span> <span class="fu">filterfun</span>(<span class="fu">pOverA</span>(<span class="fl">0.08</span>, <span class="dv">5</span>))</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#create filter for the counts data</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>gfilt <span class="ot">&lt;-</span> <span class="fu">genefilter</span>(Apul_genes_red, filt)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">#identify genes to keep by count filter</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>gkeep <span class="ot">&lt;-</span> Apul_genes_red[gfilt,]</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co">#identify gene lists</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>gn.keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gkeep)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co">#gene count data filtered in PoverA, P percent of the samples have counts over A</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>Apul_genes_filt <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_genes_red[<span class="fu">which</span>(<span class="fu">rownames</span>(Apul_genes_red) <span class="sc">%in%</span> gn.keep),])</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="co">#How many rows do we have before and after filtering?</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Pre-filtering:"</span>, <span class="fu">nrow</span>(Apul_genes_red), <span class="st">"; Post-filtering:"</span>, <span class="fu">nrow</span>(Apul_genes_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Pre-filtering: 35869 ; Post-filtering: 25730</code></pre>
<p>miRNA:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>mifilt <span class="ot">&lt;-</span> <span class="fu">filterfun</span>(<span class="fu">pOverA</span>(<span class="fl">0.08</span>, <span class="dv">5</span>))</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co">#create filter for the counts data</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>mifilt <span class="ot">&lt;-</span> <span class="fu">genefilter</span>(Apul_miRNA_red, mifilt)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co">#identify miRNA to keep by count filter</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>mikeep <span class="ot">&lt;-</span> Apul_miRNA_red[mifilt,]</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co">#identify miRNA to keep by count filter</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>mikeep <span class="ot">&lt;-</span> Apul_miRNA_red[mifilt,]</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co">#identify miRNA lists</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>mi.keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mikeep)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="co">#miRNA count data filtered in PoverA, P percent of the samples have counts over A</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>Apul_miRNA_filt <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_miRNA_red[<span class="fu">which</span>(<span class="fu">rownames</span>(Apul_miRNA_red) <span class="sc">%in%</span> mi.keep),])</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co">#How many rows do we have before and after filtering?</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Pre-filtering:"</span>, <span class="fu">nrow</span>(Apul_miRNA_red), <span class="st">"; Post-filtering:"</span>, <span class="fu">nrow</span>(Apul_miRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Pre-filtering: 51 ; Post-filtering: 47</code></pre>
<p>Of the 51 miRNA, 47 were retained. Which were removed?</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(<span class="fu">rownames</span>(Apul_miRNA_red), <span class="fu">rownames</span>(Apul_miRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] "Cluster_5685"  "Cluster_11565" "Cluster_13647" "Cluster_14633"</code></pre>
<p>lncRNA:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>lncfilt <span class="ot">&lt;-</span> <span class="fu">filterfun</span>(<span class="fu">pOverA</span>(<span class="fl">0.08</span>, <span class="dv">5</span>))</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#create filter for the counts data</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>lncfilt <span class="ot">&lt;-</span> <span class="fu">genefilter</span>(Apul_lncRNA_red, lncfilt)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="co">#identify lncRNA to keep by count filter</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>lnckeep <span class="ot">&lt;-</span> Apul_lncRNA_red[lncfilt,]</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="co">#identify lncRNA to keep by count filter</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>lnckeep <span class="ot">&lt;-</span> Apul_lncRNA_red[lncfilt,]</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a><span class="co">#identify lncRNA lists</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>lnc.keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(lnckeep)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a><span class="co">#lncRNA count data filtered in PoverA, P percent of the samples have counts over A</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA_filt <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_lncRNA_red[<span class="fu">which</span>(<span class="fu">rownames</span>(Apul_lncRNA_red) <span class="sc">%in%</span> lnc.keep),])</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a><span class="co">#How many rows do we have before and after filtering?</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Pre-filtering:"</span>, <span class="fu">nrow</span>(Apul_lncRNA_red), <span class="st">"; Post-filtering:"</span>, <span class="fu">nrow</span>(Apul_lncRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Pre-filtering: 20559 ; Post-filtering: 16006</code></pre>
</section>
<section id="transform-data" class="level2">
<h2 class="anchored" data-anchor-id="transform-data">3.4 Transform data</h2>
<p>Set the order of genes, miRNA, lncRNA, wgbs, and metadata to all be the same.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure rownames of metadata are used as the desired column order</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>desired_order <span class="ot">&lt;-</span> <span class="fu">rownames</span>(metadata)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder data frame columns</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>Apul_genes_filt <span class="ot">&lt;-</span> Apul_genes_filt[, desired_order]</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>Apul_miRNA_filt <span class="ot">&lt;-</span> Apul_miRNA_filt[, desired_order]</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA_filt <span class="ot">&lt;-</span> Apul_lncRNA_filt[, desired_order]</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> filtered_wgbs[, desired_order]</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Check they all match</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(Apul_genes_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(Apul_miRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(Apul_lncRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(filtered_wgbs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>Use a variance stabilized transformation for all four data sets. Variance stabilization essentially tries to make variance independent of the mean</p>
<p>(Is this the most appropriate design to use?)</p>
<p>genes:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>dds_genes <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> Apul_genes_filt, </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span>Timepoint<span class="sc">+</span>ColonyID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## converting counts to integer mode

## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors

##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance Stabilizing Transformation</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>vsd_genes <span class="ot">&lt;-</span> <span class="fu">assay</span>(<span class="fu">vst</span>(dds_genes, <span class="at">blind =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>miRNA:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>dds_miRNA <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> Apul_miRNA_filt, </span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span>Timepoint<span class="sc">+</span>ColonyID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors

##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance Stabilizing Transformation</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>vsd_miRNA <span class="ot">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span>(dds_miRNA, <span class="at">blind=</span><span class="cn">TRUE</span>) <span class="co"># Must use varianceStabilizingTransformation() instead of vst() due to few input genes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>vsd_miRNA <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_miRNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>lncRNA:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>dds_lncRNA <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> Apul_lncRNA_filt, </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span>Timepoint<span class="sc">+</span>ColonyID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors

##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance Stabilizing Transformation</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>vsd_lncRNA <span class="ot">&lt;-</span> <span class="fu">assay</span>(<span class="fu">vst</span>(dds_lncRNA, <span class="at">blind =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Must round wgbs data to whole integers for normalization - need to return to this to decide if this is appropriate.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co">#round to integers </span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>filtered_wgbs<span class="ot">&lt;-</span>filtered_wgbs <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), round))</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>dds_wgbs <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> filtered_wgbs, </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> Timepoint<span class="sc">+</span>ColonyID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## converting counts to integer mode

## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors

##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance Stabilizing Transformation</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>vsd_wgbs <span class="ot">&lt;-</span> <span class="fu">assay</span>(<span class="fu">varianceStabilizingTransformation</span>(dds_wgbs, <span class="at">blind =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]

## -- note: fitType='parametric', but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType='local' or 'mean' to avoid this message next time.</code></pre>
</section>
<section id="islolate-gene-sets" class="level2">
<h2 class="anchored" data-anchor-id="islolate-gene-sets">3.5 Islolate gene sets</h2>
<p>Read in gene set tables</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># genes from WGCNA modules significantly correlated with host biomass</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>Host_AFDW <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/Host_AFDW.mg.cm2_gene_counts.tab"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co"># genes from WGCNA modules significantly correlated with symbiont photosynthesis</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>Am <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/Am_gene_counts.tab"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co"># GO terms related to energy production/usage</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>ATP_production_GO <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/ATP_production_GO_terms_gene_counts.tab"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>energy_GO <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/energy_usage_storage_GO_terms_gene_counts.tab"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Isolate filtered counts by gene set</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>vsd_Host_AFDW <span class="ot">&lt;-</span> vsd_genes[<span class="fu">rownames</span>(vsd_genes) <span class="sc">%in%</span> Host_AFDW<span class="sc">$</span>gene_id,]</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>vsd_Am <span class="ot">&lt;-</span> vsd_genes[<span class="fu">rownames</span>(vsd_genes) <span class="sc">%in%</span> Am<span class="sc">$</span>gene_id,]</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>vsd_ATP_production_GO <span class="ot">&lt;-</span> vsd_genes[<span class="fu">rownames</span>(vsd_genes) <span class="sc">%in%</span> ATP_production_GO<span class="sc">$</span>gene_id,]</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>vsd_energy_GO <span class="ot">&lt;-</span> vsd_genes[<span class="fu">rownames</span>(vsd_genes) <span class="sc">%in%</span> energy_GO<span class="sc">$</span>gene_id,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="merge-predictor-features" class="level1">
<h1>5 Merge predictor features</h1>
<section id="merge-raw-datsets" class="level2">
<h2 class="anchored" data-anchor-id="merge-raw-datsets">5.2 Merge raw datsets</h2>
<p>I also want to try using individiual miRNA/lncRNA/CpGs as predictors, instead of the reduced PCs. Let’s also create a merged dataset that contains the variance stabilized counts for all miRNA, lncRNA, and CpGs.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Triple check that all three data frames have sample names in the same order</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">colnames</span>(vsd_lncRNA), <span class="fu">colnames</span>(vsd_miRNA))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">colnames</span>(vsd_lncRNA), <span class="fu">colnames</span>(vsd_wgbs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Bind (stack dataframes vertically, so that they match by column/sample)</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>full_pred_counts <span class="ot">&lt;-</span> <span class="fu">rbind</span>(vsd_lncRNA, vsd_miRNA, vsd_wgbs)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform so that samples are on rows and features are in columns</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>full_pred_counts <span class="ot">&lt;-</span> <span class="fu">t</span>(full_pred_counts)</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(full_pred_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1]    40 16560</code></pre>
<p>Ok, now we can run the model!</p>
<p>The below code uses both predictors (miRNA+lncRNA+CpGs) and responses (genes) that have been <em>not</em> been reduced using PCs.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform the counts matrices so that samples are on the rows and gene IDs on the columns</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>vsd_genes_t <span class="ot">&lt;-</span> <span class="fu">t</span>(vsd_genes)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>vsd_Host_AFDW_t <span class="ot">&lt;-</span> <span class="fu">t</span>(vsd_Host_AFDW)</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>vsd_Am_t <span class="ot">&lt;-</span> <span class="fu">t</span>(vsd_Am)</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>vsd_ATP_production_GO_t <span class="ot">&lt;-</span> <span class="fu">t</span>(vsd_ATP_production_GO)</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>vsd_energy_GO_t <span class="ot">&lt;-</span> <span class="fu">t</span>(vsd_energy_GO)</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure both are formatted as data frames</span></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>full_pred_counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(full_pred_counts)</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>vsd_genes_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(vsd_genes_t)</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>vsd_Host_AFDW_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(vsd_Host_AFDW_t)</span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>vsd_Am_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(vsd_Am_t)</span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>vsd_ATP_production_GO_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(vsd_ATP_production_GO_t)</span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>vsd_energy_GO_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(vsd_energy_GO_t)</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure sample matching between gene and epigenetic dfs</span></span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(vsd_genes_t), <span class="fu">rownames</span>(full_pred_counts))</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>full_pred_counts <span class="ot">&lt;-</span> full_pred_counts[common_samples, ]</span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a>vsd_genes_t <span class="ot">&lt;-</span> vsd_genes_t[common_samples, ]</span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>vsd_Host_AFDW_t <span class="ot">&lt;-</span> vsd_Host_AFDW_t[common_samples,]</span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a>vsd_Am_t <span class="ot">&lt;-</span> vsd_Am_t[common_samples,]</span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a>vsd_ATP_production_GO_t <span class="ot">&lt;-</span> vsd_ATP_production_GO_t[common_samples,]</span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a>vsd_energy_GO_t <span class="ot">&lt;-</span> vsd_energy_GO_t[common_samples,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="host-biomass-host_afdw" class="level1">
<h1>14 Host biomass (Host_AFDW)</h1>
<section id="the-model" class="level2">
<h2 class="anchored" data-anchor-id="the-model">14.1 The model</h2>
<p>Train elastic models to predict gene expression from miRNA/lncRNA/methylation.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>train_models <span class="ot">&lt;-</span> <span class="cf">function</span>(response_features, predictor_features) {</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  models <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (feature <span class="cf">in</span> <span class="fu">colnames</span>(response_features)) {</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> response_features[[feature]]  <span class="co"># Gene expression</span></span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a>    X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(predictor_features)  <span class="co"># miRNA/lncRNA/methylation as predictors</span></span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Train elastic net model (alpha = 0.5 for mix of LASSO &amp; Ridge)</span></span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>    model <span class="ot">&lt;-</span> <span class="fu">cv.glmnet</span>(X, y, <span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a>    models[[feature]] <span class="ot">&lt;-</span> model</span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb57-13"><a href="#cb57-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb57-14"><a href="#cb57-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(models)</span>
<span id="cb57-15"><a href="#cb57-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb57-16"><a href="#cb57-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-17"><a href="#cb57-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression from miRNA/lncRNA/methylation</span></span>
<span id="cb57-18"><a href="#cb57-18" aria-hidden="true" tabindex="-1"></a>models_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">train_models</span>(vsd_Host_AFDW_t, full_pred_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance <span class="ot">&lt;-</span> <span class="cf">function</span>(models) {</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  importance_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(models, <span class="cf">function</span>(model) {</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>    coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Convert to regular matrix &amp; remove intercept</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Convert to data frame</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>    coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">as.numeric</span>(coefs))</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(coefs_df)</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine feature importance across all predicted genes</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  importance_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(importance_list) <span class="sc">%&gt;%</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">MeanImportance =</span> <span class="fu">mean</span>(<span class="fu">abs</span>(Importance)), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(MeanImportance))</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(importance_df)</span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a>feature_importance_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_Host_AFDW)</span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_Host_AFDW, <span class="dv">20</span>)  <span class="co"># Top predictive features</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 2
##    Feature    MeanImportance
##    &lt;chr&gt;               &lt;dbl&gt;
##  1 PC20_miRNA         0.0906
##  2 PC18_miRNA         0.0771
##  3 PC17_miRNA         0.0762
##  4 PC19_miRNA         0.0760
##  5 PC16_miRNA         0.0753
##  6 PC8_miRNA          0.0689
##  7 PC3_miRNA          0.0674
##  8 PC13_miRNA         0.0672
##  9 PC11_miRNA         0.0669
## 10 PC7_miRNA          0.0663
## 11 PC9_miRNA          0.0581
## 12 PC27_WGBS          0.0527
## 13 PC26_WGBS          0.0516
## 14 PC2_miRNA          0.0505
## 15 PC14_miRNA         0.0471
## 16 PC5_miRNA          0.0463
## 17 PC12_miRNA         0.0440
## 18 PC21_WGBS          0.0430
## 19 PC13_WGBS          0.0430
## 20 PC11_WGBS          0.0354</code></pre>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>evaluate_model_performance <span class="ot">&lt;-</span> <span class="cf">function</span>(models, response_features, predictor_features) {</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>  results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">colnames</span>(response_features), <span class="at">R2 =</span> <span class="cn">NA</span>)</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (feature <span class="cf">in</span> <span class="fu">colnames</span>(response_features)) {</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> response_features[[feature]]</span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>    X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(predictor_features)</span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>    model <span class="ot">&lt;-</span> models[[feature]]</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a>    preds <span class="ot">&lt;-</span> <span class="fu">predict</span>(model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>    R2 <span class="ot">&lt;-</span> <span class="fu">cor</span>(y, preds)<span class="sc">^</span><span class="dv">2</span>  <span class="co"># R-squared metric</span></span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a>    results[results<span class="sc">$</span>Feature <span class="sc">==</span> feature, <span class="st">"R2"</span>] <span class="ot">&lt;-</span> R2</span>
<span id="cb60-13"><a href="#cb60-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb60-14"><a href="#cb60-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-15"><a href="#cb60-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(results)</span>
<span id="cb60-16"><a href="#cb60-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb60-17"><a href="#cb60-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-18"><a href="#cb60-18" aria-hidden="true" tabindex="-1"></a>performance_results_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_Host_AFDW, vsd_Host_AFDW_t, full_pred_counts)</span>
<span id="cb60-19"><a href="#cb60-19" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_Host_AFDW<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
##  0.0000  0.9554  0.9919  0.9498  0.9996  0.9998      14</code></pre>
</section>
<section id="results" class="level2">
<h2 class="anchored" data-anchor-id="results">14.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top predictive features</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>top_features_Host_AFDW <span class="ot">&lt;-</span> feature_importance_Host_AFDW <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">50</span>, MeanImportance)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_Host_AFDW, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"miRNA/lncRNA/methylation as predictive features"</span>,</span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Features"</span>,</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-90-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_Host_AFDW, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(Feature), <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_Host_AFDW<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Genes"</span>,</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene"</span>,</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Removed 14 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-91-1.png" class="img-fluid"><!-- --></p>
<p>We can also look at which feature(s) contributed most to predicting genes of interest</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Will keep this unevaluated to reduce size of knitted document</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>all_features_Host_AFDW <span class="ot">&lt;-</span> performance_results_Host_AFDW <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;</span> <span class="fl">0.95</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Feature)</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>get_feature_importance_for_feature <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>  coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Remove intercept</span></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>  coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">abs</span>(<span class="fu">as.numeric</span>(coefs)))</span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(coefs_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Importance)))  <span class="co"># Sort by importance</span></span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (feature <span class="cf">in</span> all_features_Host_AFDW) {</span>
<span id="cb65-12"><a href="#cb65-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract feature importance for the most predictable gene</span></span>
<span id="cb65-13"><a href="#cb65-13" aria-hidden="true" tabindex="-1"></a>  best_feature_model <span class="ot">&lt;-</span> models_Host_AFDW[[feature]]</span>
<span id="cb65-14"><a href="#cb65-14" aria-hidden="true" tabindex="-1"></a>  best_feature_importance <span class="ot">&lt;-</span> <span class="fu">get_feature_importance_for_feature</span>(best_feature_model)</span>
<span id="cb65-15"><a href="#cb65-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb65-16"><a href="#cb65-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot top most important miRNA/lncRNA/methylation for predicting this gene</span></span>
<span id="cb65-17"><a href="#cb65-17" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(best_feature_importance <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, Importance), <span class="at">y =</span> Importance)) <span class="sc">+</span></span>
<span id="cb65-18"><a href="#cb65-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb65-19"><a href="#cb65-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb65-20"><a href="#cb65-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb65-21"><a href="#cb65-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top miRNA/lncRNA/methylation Predictors for"</span>, feature),</span>
<span id="cb65-22"><a href="#cb65-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb65-23"><a href="#cb65-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Importance Score"</span>)</span>
<span id="cb65-24"><a href="#cb65-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb65-25"><a href="#cb65-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb65-26"><a href="#cb65-26" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute correlation between actual and predicted genes</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> <span class="fu">sapply</span>(models_Host_AFDW, <span class="cf">function</span>(m) <span class="fu">predict</span>(m, <span class="fu">as.matrix</span>(full_pred_counts), <span class="at">s =</span> <span class="st">"lambda.min"</span>))</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure matrices are the same size</span></span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> predicted_matrix[, <span class="fu">colnames</span>(vsd_Host_AFDW_t), drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Align columns</span></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="co"># remove 0 variance columns</span></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> predicted_matrix[, <span class="fu">apply</span>(predicted_matrix, <span class="dv">2</span>, <span class="cf">function</span>(col) <span class="fu">sd</span>(col, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>)]</span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute correlation matrix, handling missing values</span></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>cor_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(predicted_matrix, <span class="fu">as.matrix</span>(vsd_Host_AFDW_t), <span class="at">use =</span> <span class="st">"complete.obs"</span>)</span>
<span id="cb66-12"><a href="#cb66-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-13"><a href="#cb66-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace NA or Inf values with zero</span></span>
<span id="cb66-14"><a href="#cb66-14" aria-hidden="true" tabindex="-1"></a>cor_matrix[<span class="fu">is.na</span>(cor_matrix) <span class="sc">|</span> <span class="fu">is.infinite</span>(cor_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span>  </span>
<span id="cb66-15"><a href="#cb66-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-16"><a href="#cb66-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb66-17"><a href="#cb66-17" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(cor_matrix, <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">100</span>),</span>
<span id="cb66-18"><a href="#cb66-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"Correlation Between Actual and Predicted Genes"</span>,</span>
<span id="cb66-19"><a href="#cb66-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb66-20"><a href="#cb66-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb66-21"><a href="#cb66-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-93-1.png" class="img-fluid"><!-- --></p>
</section>
</section>
<section id="symbiont-photsynthesis-am" class="level1">
<h1>15 Symbiont photsynthesis (Am)</h1>
<section id="the-model-1" class="level2">
<h2 class="anchored" data-anchor-id="the-model-1">15.1 The model</h2>
<p>Train elastic models to predict gene expression from miRNA/lncRNA/methylation</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression from miRNA/lncRNA/methylation</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>models_Am <span class="ot">&lt;-</span> <span class="fu">train_models</span>(vsd_Am_t, full_pred_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>feature_importance_Am <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_Am)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_Am, <span class="dv">20</span>)  <span class="co"># Top predictive features</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>performance_results_Am <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_Am, vsd_Am_t, full_pred_counts)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_Am<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="results-1" class="level2">
<h2 class="anchored" data-anchor-id="results-1">15.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top predictive features</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>top_features_Am <span class="ot">&lt;-</span> feature_importance_Am <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">50</span>, MeanImportance)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_Am, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"miRNA/lncRNA/methylation as Predictive Features"</span>,</span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_Am, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(Feature), <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_Am<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Gene Expression PCs"</span>,</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene Expression PC"</span>,</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>View predicted vs actual gene expression values to evaluate model.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a gene expression PC to visualize (e.g., the most predictable one)</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>best_feature_Am <span class="ot">&lt;-</span> performance_results_Am<span class="sc">$</span>Feature[<span class="fu">which.max</span>(performance_results_Am<span class="sc">$</span>R2)]</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>actual_values_Am <span class="ot">&lt;-</span> vsd_Am_t[[best_feature_Am]]</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>predicted_values_Am <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_Am[[best_feature_Am]], <span class="fu">as.matrix</span>(full_pred_counts), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame</span></span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>prediction_df_Am <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Actual =</span> actual_values_Am,</span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Predicted =</span> predicted_values_Am</span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot with regression line</span></span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(prediction_df_Am, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, best_feature_Am),</span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values_Am), <span class="at">y =</span> <span class="fu">max</span>(predicted_values_Am), </span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_Am<span class="sc">$</span>R2, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb72-25"><a href="#cb72-25" aria-hidden="true" tabindex="-1"></a><span class="do">## `geom_smooth()` using formula = 'y ~ x'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>View top 20 genes associated with the PC with the highest R^2</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_Am<span class="sc">%&gt;%</span><span class="fu">filter</span>(Am_Feature<span class="sc">==</span>best_feature_Am))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Plot performance for all PCs</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all PCs with R^2 values above 0.75</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>all_features_Am <span class="ot">&lt;-</span> performance_results_Am <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;</span> <span class="fl">0.75</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Feature)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (feature <span class="cf">in</span> all_features_Am) {</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>  actual_values <span class="ot">&lt;-</span> vsd_Am_t[[feature]]</span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a>  predicted_values <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_Am[[feature]], <span class="fu">as.matrix</span>(full_pred_counts), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create data frame</span></span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>  prediction_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">Actual =</span> actual_values,</span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">Predicted =</span> predicted_values</span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scatter plot with regression line</span></span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(prediction_df, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb74-20"><a href="#cb74-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb74-21"><a href="#cb74-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, feature),</span>
<span id="cb74-22"><a href="#cb74-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb74-23"><a href="#cb74-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb74-24"><a href="#cb74-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values), <span class="at">y =</span> <span class="fu">max</span>(predicted_values), </span>
<span id="cb74-25"><a href="#cb74-25" aria-hidden="true" tabindex="-1"></a>             <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_Am[performance_results_Am<span class="sc">$</span>Feature<span class="sc">==</span>feature,<span class="dv">2</span>], <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb74-26"><a href="#cb74-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb74-27"><a href="#cb74-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb74-28"><a href="#cb74-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb74-29"><a href="#cb74-29" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>We can also look at which miRNA(s) contributed most to predicting gene PCs of interest</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance_for_feature <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>  coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Remove intercept</span></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>  coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">abs</span>(<span class="fu">as.numeric</span>(coefs)))</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(coefs_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Importance)))  <span class="co"># Sort by importance</span></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (feature <span class="cf">in</span> all_features_Am) {</span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract feature importance for the most predictable PC</span></span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a>  best_feature_model <span class="ot">&lt;-</span> models_Am[[feature]]</span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a>  best_feature_importance <span class="ot">&lt;-</span> <span class="fu">get_feature_importance_for_feature</span>(best_feature_model)</span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot top most important miRNA/lncRNA/methylation for predicting this PC</span></span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(best_feature_importance <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, Importance), <span class="at">y =</span> Importance)) <span class="sc">+</span></span>
<span id="cb75-15"><a href="#cb75-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb75-16"><a href="#cb75-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb75-17"><a href="#cb75-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb75-18"><a href="#cb75-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top miRNA/lncRNA/methylation Predictors for"</span>, feature),</span>
<span id="cb75-19"><a href="#cb75-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb75-20"><a href="#cb75-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Importance Score"</span>)</span>
<span id="cb75-21"><a href="#cb75-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb75-22"><a href="#cb75-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb75-23"><a href="#cb75-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="atp-production-go-terms" class="level1">
<h1>16 ATP production (GO terms)</h1>
<section id="the-model-2" class="level2">
<h2 class="anchored" data-anchor-id="the-model-2">16.1 The model</h2>
<p>Train elastic models to predict gene expression PCs from miRNA/lncRNA/methylation</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression PCs from miRNA/lncRNA/methylation</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>models_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">train_models</span>(vsd_ATP_production_GO_t, full_pred_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>feature_importance_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_ATP_prod_GO)</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_ATP_prod_GO, <span class="dv">20</span>)  <span class="co"># Top predictive miRNA</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 2
##    Feature                MeanImportance
##    &lt;chr&gt;                           &lt;dbl&gt;
##  1 lncRNA_16172                  0.0151 
##  2 lncRNA_19034                  0.0137 
##  3 lncRNA_3645                   0.0125 
##  4 lncRNA_20796                  0.0110 
##  5 lncRNA_4454                   0.00999
##  6 lncRNA_8058                   0.00920
##  7 lncRNA_18277                  0.00918
##  8 CpG_ptg000024l_4334252        0.00893
##  9 lncRNA_9242                   0.00818
## 10 lncRNA_263                    0.00782
## 11 lncRNA_3622                   0.00781
## 12 lncRNA_9252                   0.00773
## 13 CpG_ptg000031l_5273398        0.00716
## 14 lncRNA_19773                  0.00687
## 15 CpG_ptg000022l_2983280        0.00683
## 16 lncRNA_11719                  0.00667
## 17 CpG_ptg000024l_213653         0.00665
## 18 lncRNA_23085                  0.00664
## 19 lncRNA_18158                  0.00655
## 20 lncRNA_7887                   0.00652</code></pre>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>performance_results_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_ATP_prod_GO, vsd_ATP_production_GO_t, full_pred_counts)</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_ATP_prod_GO<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.0000  0.9644  0.9957  0.9366  0.9996  0.9997</code></pre>
</section>
<section id="results-2" class="level2">
<h2 class="anchored" data-anchor-id="results-2">16.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top predictive features</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>top_features_ATP_prod_GO <span class="ot">&lt;-</span> feature_importance_ATP_prod_GO <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">50</span>, MeanImportance)</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_ATP_prod_GO, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"miRNA/lncRNA/methylation as Predictive Features"</span>,</span>
<span id="cb81-10"><a href="#cb81-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb81-11"><a href="#cb81-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-106-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_ATP_prod_GO, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(Feature), <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_ATP_prod_GO<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Gene Expression PCs"</span>,</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene Expression PC"</span>,</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-107-1.png" class="img-fluid"><!-- --></p>
<p>View predicted vs actual gene expression values to evaluate model.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a gene expression PC to visualize (e.g., the most predictable one)</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>best_feature_ATP_prod_GO <span class="ot">&lt;-</span> performance_results_ATP_prod_GO<span class="sc">$</span>Feature[<span class="fu">which.max</span>(performance_results_ATP_prod_GO<span class="sc">$</span>R2)]</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>actual_values_ATP_prod_GO <span class="ot">&lt;-</span> vsd_ATP_production_GO_t[[best_feature_ATP_prod_GO]]</span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>predicted_values_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_ATP_prod_GO[[best_feature_ATP_prod_GO]], <span class="fu">as.matrix</span>(full_pred_counts), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame</span></span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>prediction_df_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Actual =</span> actual_values_ATP_prod_GO,</span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Predicted =</span> predicted_values_ATP_prod_GO</span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot with regression line</span></span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(prediction_df_ATP_prod_GO, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb83-16"><a href="#cb83-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb83-17"><a href="#cb83-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb83-18"><a href="#cb83-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb83-19"><a href="#cb83-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, best_feature_ATP_prod_GO),</span>
<span id="cb83-20"><a href="#cb83-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb83-21"><a href="#cb83-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb83-22"><a href="#cb83-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values_ATP_prod_GO), <span class="at">y =</span> <span class="fu">max</span>(predicted_values_ATP_prod_GO), </span>
<span id="cb83-23"><a href="#cb83-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_ATP_prod_GO<span class="sc">$</span>R2, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb83-24"><a href="#cb83-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-108-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="do">## `geom_smooth()` using formula = 'y ~ x'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Plot performance for all PCs</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all PCs with R^2 values above line in plot</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>all_features_ATP_prod_GO <span class="ot">&lt;-</span> performance_results_ATP_prod_GO <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;</span> <span class="fl">0.75</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Feature)</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (feature <span class="cf">in</span> all_features_ATP_prod_GO) {</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>  actual_values <span class="ot">&lt;-</span> vsd_ATP_production_GO_t[[feature]]</span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>  predicted_values <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_ATP_prod_GO[[feature]], <span class="fu">as.matrix</span>(full_pred_counts), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create data frame</span></span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a>  prediction_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">Actual =</span> actual_values,</span>
<span id="cb86-13"><a href="#cb86-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">Predicted =</span> predicted_values</span>
<span id="cb86-14"><a href="#cb86-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb86-15"><a href="#cb86-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb86-16"><a href="#cb86-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scatter plot with regression line</span></span>
<span id="cb86-17"><a href="#cb86-17" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(prediction_df, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb86-18"><a href="#cb86-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb86-19"><a href="#cb86-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb86-20"><a href="#cb86-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb86-21"><a href="#cb86-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, feature),</span>
<span id="cb86-22"><a href="#cb86-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb86-23"><a href="#cb86-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb86-24"><a href="#cb86-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values), <span class="at">y =</span> <span class="fu">max</span>(predicted_values), </span>
<span id="cb86-25"><a href="#cb86-25" aria-hidden="true" tabindex="-1"></a>             <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_ATP_prod_GO[performance_results_ATP_prod_GO<span class="sc">$</span>Feature<span class="sc">==</span>feature,<span class="dv">2</span>], <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb86-26"><a href="#cb86-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb86-27"><a href="#cb86-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb86-28"><a href="#cb86-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb86-29"><a href="#cb86-29" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-1.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-2.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-3.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-4.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-5.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-6.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-7.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-8.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-9.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-10.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-11.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-12.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-13.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-14.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-15.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-16.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-17.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-18.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-19.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-20.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-109-21.png" class="img-fluid"><!-- --></p>
<p>We can also look at which miRNA(s) contributed most to predicting gene PCs of interest</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance_for_feature <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>  coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Remove intercept</span></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>  coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">abs</span>(<span class="fu">as.numeric</span>(coefs)))</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(coefs_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Importance)))  <span class="co"># Sort by importance</span></span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (feature <span class="cf">in</span> all_features_ATP_prod_GO) {</span>
<span id="cb108-9"><a href="#cb108-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract feature importance for the most predictable PC</span></span>
<span id="cb108-10"><a href="#cb108-10" aria-hidden="true" tabindex="-1"></a>  best_feature_model <span class="ot">&lt;-</span> models_ATP_prod_GO[[feature]]</span>
<span id="cb108-11"><a href="#cb108-11" aria-hidden="true" tabindex="-1"></a>  best_feature_importance <span class="ot">&lt;-</span> <span class="fu">get_feature_importance_for_feature</span>(best_feature_model)</span>
<span id="cb108-12"><a href="#cb108-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-13"><a href="#cb108-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot top most important miRNA/lncRNA/methylation for predicting this PC</span></span>
<span id="cb108-14"><a href="#cb108-14" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(best_feature_importance <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, Importance), <span class="at">y =</span> Importance)) <span class="sc">+</span></span>
<span id="cb108-15"><a href="#cb108-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb108-16"><a href="#cb108-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb108-17"><a href="#cb108-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb108-18"><a href="#cb108-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top miRNA/lncRNA/methylation Predictors for"</span>, feature),</span>
<span id="cb108-19"><a href="#cb108-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb108-20"><a href="#cb108-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Importance Score"</span>)</span>
<span id="cb108-21"><a href="#cb108-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-22"><a href="#cb108-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb108-23"><a href="#cb108-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-1.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-2.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-3.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-4.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-5.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-6.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-7.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-8.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-9.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-10.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-11.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-12.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-13.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-14.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-15.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-16.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-17.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-18.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-19.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-20.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-110-21.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute correlation between actual and predicted gene expression PCs</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> <span class="fu">sapply</span>(models_ATP_prod_GO, <span class="cf">function</span>(m) <span class="fu">predict</span>(m, <span class="fu">as.matrix</span>(full_pred_counts), <span class="at">s =</span> <span class="st">"lambda.min"</span>))</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure matrices are the same size</span></span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> predicted_matrix[, <span class="fu">colnames</span>(vsd_ATP_production_GO_t), drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Align columns</span></span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a><span class="co"># remove 0 variance columns</span></span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> predicted_matrix[, <span class="fu">apply</span>(predicted_matrix, <span class="dv">2</span>, <span class="cf">function</span>(col) <span class="fu">sd</span>(col, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>)]</span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute correlation matrix, handling missing values</span></span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a>cor_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(predicted_matrix, <span class="fu">as.matrix</span>(vsd_ATP_production_GO_t), <span class="at">use =</span> <span class="st">"complete.obs"</span>)</span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-13"><a href="#cb109-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace NA or Inf values with zero</span></span>
<span id="cb109-14"><a href="#cb109-14" aria-hidden="true" tabindex="-1"></a>cor_matrix[<span class="fu">is.na</span>(cor_matrix) <span class="sc">|</span> <span class="fu">is.infinite</span>(cor_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span>  </span>
<span id="cb109-15"><a href="#cb109-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-16"><a href="#cb109-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb109-17"><a href="#cb109-17" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(cor_matrix, <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">100</span>),</span>
<span id="cb109-18"><a href="#cb109-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"Correlation Between Actual and Predicted Gene Expression PCs"</span>,</span>
<span id="cb109-19"><a href="#cb109-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb109-20"><a href="#cb109-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb109-21"><a href="#cb109-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-111-1.png" class="img-fluid"><!-- --></p>
</section>
</section>
<section id="energy-usagestorage-go-terms" class="level1">
<h1>17 Energy Usage/Storage (GO terms)</h1>
<section id="the-model-3" class="level2">
<h2 class="anchored" data-anchor-id="the-model-3">17.1 The model</h2>
<p>Train elastic models to predict gene expression PCs from miRNA/lncRNA/methylation</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression PCs from miRNA/lncRNA/methylation</span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>models_energy_GO <span class="ot">&lt;-</span> <span class="fu">train_models</span>(vsd_energy_GO_t, full_pred_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>feature_importance_energy_GO <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_energy_GO)</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_energy_GO, <span class="dv">20</span>)  <span class="co"># Top predictive miRNA</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 2
##    Feature                 MeanImportance
##    &lt;chr&gt;                            &lt;dbl&gt;
##  1 lncRNA_19034                   0.00555
##  2 lncRNA_4454                    0.00373
##  3 lncRNA_2998                    0.00302
##  4 lncRNA_12064                   0.00248
##  5 CpG_ptg000025l_18378678        0.00244
##  6 lncRNA_12212                   0.00237
##  7 lncRNA_8086                    0.00221
##  8 lncRNA_20796                   0.00216
##  9 lncRNA_13960                   0.00215
## 10 lncRNA_263                     0.00214
## 11 lncRNA_21989                   0.00210
## 12 lncRNA_15222                   0.00210
## 13 lncRNA_7887                    0.00208
## 14 CpG_ptg000024l_4334565         0.00206
## 15 lncRNA_16584                   0.00205
## 16 lncRNA_6545                    0.00203
## 17 lncRNA_1395                    0.00199
## 18 lncRNA_22939                   0.00199
## 19 lncRNA_16417                   0.00198
## 20 CpG_ptg000021l_8434453         0.00194</code></pre>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>performance_results_energy_GO <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_energy_GO, vsd_energy_GO_t, full_pred_counts)</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_energy_GO<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
##  0.3597  0.9565  0.9950  0.9574  0.9996  0.9998       3</code></pre>
</section>
<section id="results-3" class="level2">
<h2 class="anchored" data-anchor-id="results-3">17.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top predictive features</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>top_features_energy_GO <span class="ot">&lt;-</span> feature_importance_energy_GO <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">50</span>, MeanImportance)</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_energy_GO, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"miRNA/lncRNA/methylation as Predictive Features"</span>,</span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-115-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_energy_GO, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(Feature), <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_energy_GO<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Gene Expression PCs"</span>,</span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene Expression PC"</span>,</span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Removed 3 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-116-1.png" class="img-fluid"><!-- --></p>
<p>We can also look at which miRNA(s) contributed most to predicting gene PCs of interest</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Will keep this unevaluated to reduce size of knitted document</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>get_feature_importance_for_feature <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>  coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Remove intercept</span></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>  coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">abs</span>(<span class="fu">as.numeric</span>(coefs)))</span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(coefs_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Importance)))  <span class="co"># Sort by importance</span></span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>all_features_energy_GO <span class="ot">&lt;-</span> performance_results_energy_GO <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;</span> <span class="fl">0.75</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Feature)</span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-11"><a href="#cb118-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (feature <span class="cf">in</span> all_features_energy_GO) {</span>
<span id="cb118-12"><a href="#cb118-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract feature importance for the most predictable PC</span></span>
<span id="cb118-13"><a href="#cb118-13" aria-hidden="true" tabindex="-1"></a>  best_feature_model <span class="ot">&lt;-</span> models_energy_GO[[feature]]</span>
<span id="cb118-14"><a href="#cb118-14" aria-hidden="true" tabindex="-1"></a>  best_feature_importance <span class="ot">&lt;-</span> <span class="fu">get_feature_importance_for_feature</span>(best_feature_model)</span>
<span id="cb118-15"><a href="#cb118-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb118-16"><a href="#cb118-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot top most important miRNA/lncRNA/methylation for predicting this PC</span></span>
<span id="cb118-17"><a href="#cb118-17" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(best_feature_importance <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, Importance), <span class="at">y =</span> Importance)) <span class="sc">+</span></span>
<span id="cb118-18"><a href="#cb118-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb118-19"><a href="#cb118-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb118-20"><a href="#cb118-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb118-21"><a href="#cb118-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top miRNA/lncRNA/methylation Predictors for"</span>, feature),</span>
<span id="cb118-22"><a href="#cb118-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb118-23"><a href="#cb118-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Importance Score"</span>)</span>
<span id="cb118-24"><a href="#cb118-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb118-25"><a href="#cb118-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb118-26"><a href="#cb118-26" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="compare" class="level1">
<h1>18 Compare</h1>
<p>Visualize the relative importance of miRNA/lncRNA/methylation in predicting expression for these different gene sets:</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform min-max normalization on the mean importance of miRNA/lncRNA/methylation for each group</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="co"># This will place all along a 0-1 range for comparison purposes</span></span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>normalize <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>  (x <span class="sc">-</span> <span class="fu">min</span>(x)) <span class="sc">/</span> (<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x))</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize</span></span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a>top_features_Host_AFDW<span class="sc">$</span>MeanImportance_norm <span class="ot">&lt;-</span> <span class="fu">normalize</span>(top_features_Host_AFDW<span class="sc">$</span>MeanImportance)</span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a><span class="co">#top_features_Am$MeanImportance_norm &lt;- normalize(top_features_Am$MeanImportance)</span></span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a>top_features_ATP_prod_GO<span class="sc">$</span>MeanImportance_norm <span class="ot">&lt;-</span> <span class="fu">normalize</span>(top_features_ATP_prod_GO<span class="sc">$</span>MeanImportance)</span>
<span id="cb119-11"><a href="#cb119-11" aria-hidden="true" tabindex="-1"></a>top_features_energy_GO<span class="sc">$</span>MeanImportance_norm <span class="ot">&lt;-</span> <span class="fu">normalize</span>(top_features_energy_GO<span class="sc">$</span>MeanImportance)</span>
<span id="cb119-12"><a href="#cb119-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-13"><a href="#cb119-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Add group labels</span></span>
<span id="cb119-14"><a href="#cb119-14" aria-hidden="true" tabindex="-1"></a>top_features_Host_AFDW <span class="ot">&lt;-</span> top_features_Host_AFDW <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"Host_AFDW"</span>)</span>
<span id="cb119-15"><a href="#cb119-15" aria-hidden="true" tabindex="-1"></a><span class="co">#top_features_Am &lt;- top_features_Am %&gt;% mutate(group = "Am")</span></span>
<span id="cb119-16"><a href="#cb119-16" aria-hidden="true" tabindex="-1"></a>top_features_ATP_prod_GO <span class="ot">&lt;-</span> top_features_ATP_prod_GO <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"ATP_prod_GO"</span>)</span>
<span id="cb119-17"><a href="#cb119-17" aria-hidden="true" tabindex="-1"></a>top_features_energy_GO <span class="ot">&lt;-</span> top_features_energy_GO <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"energy_GO"</span>)</span>
<span id="cb119-18"><a href="#cb119-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-19"><a href="#cb119-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Set rows in same order</span></span>
<span id="cb119-20"><a href="#cb119-20" aria-hidden="true" tabindex="-1"></a><span class="co">#top_features_Am &lt;- top_features_Am[rownames(top_features_Host_AFDW),]</span></span>
<span id="cb119-21"><a href="#cb119-21" aria-hidden="true" tabindex="-1"></a>top_features_ATP_prod_GO <span class="ot">&lt;-</span> top_features_ATP_prod_GO[<span class="fu">rownames</span>(top_features_Host_AFDW),]</span>
<span id="cb119-22"><a href="#cb119-22" aria-hidden="true" tabindex="-1"></a>top_features_energy_GO <span class="ot">&lt;-</span> top_features_energy_GO[<span class="fu">rownames</span>(top_features_Host_AFDW),]</span>
<span id="cb119-23"><a href="#cb119-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-24"><a href="#cb119-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine</span></span>
<span id="cb119-25"><a href="#cb119-25" aria-hidden="true" tabindex="-1"></a>all_gene_sets <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(top_features_Host_AFDW, </span>
<span id="cb119-26"><a href="#cb119-26" aria-hidden="true" tabindex="-1"></a>                           <span class="co">#top_features_Am, </span></span>
<span id="cb119-27"><a href="#cb119-27" aria-hidden="true" tabindex="-1"></a>                           top_features_ATP_prod_GO, top_features_energy_GO)</span>
<span id="cb119-28"><a href="#cb119-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove raw mean importance</span></span>
<span id="cb119-29"><a href="#cb119-29" aria-hidden="true" tabindex="-1"></a>all_gene_sets <span class="ot">&lt;-</span> all_gene_sets <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">!</span>MeanImportance)</span>
<span id="cb119-30"><a href="#cb119-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-31"><a href="#cb119-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Wide format: rows = miRNAs, columns = groups</span></span>
<span id="cb119-32"><a href="#cb119-32" aria-hidden="true" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> all_gene_sets <span class="sc">%&gt;%</span></span>
<span id="cb119-33"><a href="#cb119-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> group, <span class="at">values_from =</span> MeanImportance_norm)</span>
<span id="cb119-34"><a href="#cb119-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-35"><a href="#cb119-35" aria-hidden="true" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(heatmap_df)</span>
<span id="cb119-36"><a href="#cb119-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-37"><a href="#cb119-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Melt into long format for ggplot</span></span>
<span id="cb119-38"><a href="#cb119-38" aria-hidden="true" tabindex="-1"></a>heatmap_long <span class="ot">&lt;-</span> <span class="fu">melt</span>(heatmap_df, <span class="at">id.vars =</span> <span class="st">"Feature"</span>)</span>
<span id="cb119-39"><a href="#cb119-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-40"><a href="#cb119-40" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(heatmap_long, <span class="fu">aes</span>(<span class="at">x =</span> variable, <span class="at">y =</span> Feature, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb119-41"><a href="#cb119-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb119-42"><a href="#cb119-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb119-43"><a href="#cb119-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb119-44"><a href="#cb119-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Group"</span>, <span class="at">y =</span> <span class="st">"Feature"</span>, <span class="at">fill =</span> <span class="st">"Importance"</span>) <span class="sc">+</span></span>
<span id="cb119-45"><a href="#cb119-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-118-1.png" class="img-fluid"><!-- --></p>
<p>Cluster by miRNA/lncRNA/methylation importance</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make Feature column the rownames and convert to matrix</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatmap_df) <span class="ot">&lt;-</span> heatmap_df<span class="sc">$</span>Feature</span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>heatmap_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(heatmap_df[, <span class="sc">-</span><span class="dv">1</span>])  <span class="co"># Removes the 'Feature' column</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace NAs with 0</span></span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>heatmap_matrix[<span class="fu">is.na</span>(heatmap_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-8"><a href="#cb120-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb120-9"><a href="#cb120-9" aria-hidden="true" tabindex="-1"></a>  heatmap_matrix, </span>
<span id="cb120-10"><a href="#cb120-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,  <span class="co"># Clustering miRNAs (rows) by similarity in importance</span></span>
<span id="cb120-11"><a href="#cb120-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,  <span class="co"># Clustering groups (columns)</span></span>
<span id="cb120-12"><a href="#cb120-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"none"</span>,  <span class="co"># No scaling (since data is already normalized)</span></span>
<span id="cb120-13"><a href="#cb120-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">TRUE</span>,  <span class="co"># Show miRNA/lncRNA/methylation names</span></span>
<span id="cb120-14"><a href="#cb120-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_colnames =</span> <span class="cn">TRUE</span>,  <span class="co"># Show group names</span></span>
<span id="cb120-15"><a href="#cb120-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">100</span>),  <span class="co"># Red gradient for importance</span></span>
<span id="cb120-16"><a href="#cb120-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"miRNAs Importance Across Groups"</span>  <span class="co"># Title of the heatmap</span></span>
<span id="cb120-17"><a href="#cb120-17" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-119-1.png" class="img-fluid"><!-- --></p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">This notebook is built using <a href="https://quarto.org/">Quarto</a></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>