<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kathleen Durkin">
<meta name="dcterms.date" content="2025-04-10">

<title>Deep-Dive Expression: Functional comparisons for known miRNA – Kathleen’s Lab Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-2fef5ea3f8957b3e4ecc936fc74692ca.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-b515dea5baab39ffcfcd4bae179e3bdb.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark-195886546b2eb1f67ede645fd062ce19.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../../site_libs/bootstrap/bootstrap-b515dea5baab39ffcfcd4bae179e3bdb.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kathleen’s Lab Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../daily_log.html"> 
<span class="menu-text">Daily Notebook Posts</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../projects.html"> 
<span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/shedurkin"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:kdurkin1@uw.edu"> <i class="bi bi-envelope" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <h1 class="title">Deep-Dive Expression: Functional comparisons for known miRNA</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">E5-coral</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-left">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Kathleen Durkin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 10, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#mir-2022" id="toc-mir-2022" class="nav-link active" data-scroll-target="#mir-2022">mir-2022</a></li>
  <li><a href="#mir-2030" id="toc-mir-2030" class="nav-link" data-scroll-target="#mir-2030">mir-2030</a></li>
  <li><a href="#mir-100" id="toc-mir-100" class="nav-link" data-scroll-target="#mir-100">mir-100</a></li>
  <li><a href="#mir-2050" id="toc-mir-2050" class="nav-link" data-scroll-target="#mir-2050">mir-2050</a></li>
  <li><a href="#mir-2023" id="toc-mir-2023" class="nav-link" data-scroll-target="#mir-2023">mir-2023</a></li>
  <li><a href="#mir-2028" id="toc-mir-2028" class="nav-link" data-scroll-target="#mir-2028">mir-2028</a></li>
  <li><a href="#mir-2036" id="toc-mir-2036" class="nav-link" data-scroll-target="#mir-2036">mir-2036</a></li>
  <li><a href="#mir-2025" id="toc-mir-2025" class="nav-link" data-scroll-target="#mir-2025">mir-2025</a></li>
  <li><a href="#mir-9425" id="toc-mir-9425" class="nav-link" data-scroll-target="#mir-9425">mir-9425</a></li>
  <li><a href="#apul-mir-novel-34-mir-2037" id="toc-apul-mir-novel-34-mir-2037" class="nav-link" data-scroll-target="#apul-mir-novel-34-mir-2037">apul-mir-novel-34 / miR-2037</a></li>
  <li><a href="#apul-mir-novel-9" id="toc-apul-mir-novel-9" class="nav-link" data-scroll-target="#apul-mir-novel-9">apul-mir-novel-9</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">






<p>We’ve determined putative function of our miRNAs by <a href="./2025_04_09_dde_miranda_miRNA_FE">running functional enrichment on each miRNA’s putative targets</a>. Now I’d like to compare our putative functions to the function(s) of any miRNA that are previously described.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyr' was built under R version 4.2.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dplyr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in miRNA with assigned names</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>miRNA_named <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/77b55212aaca4763917f67188d9e5e0fd1000c11/D-Apul/output/11-Apul-sRNA-ShortStack_4.1.0-pulchra_genome/ShortStack_out/Apul_Results_mature_named_miRNAs.csv"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(Name, given_miRNA_name)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(miRNA_named <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"novel"</span>, given_miRNA_name)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           Name given_miRNA_name
1  Cluster_2859    apul-mir-9425
2  Cluster_3437    apul-mir-2030
3  Cluster_4254    apul-mir-2050
4  Cluster_5012    apul-mir-2023
5 Cluster_10057    apul-mir-2028
6 Cluster_10228    apul-mir-2022
7 Cluster_15316    apul-mir-2025
8 Cluster_18728     apul-mir-100
9 Cluster_18772    apul-mir-2036</code></pre>
</div>
</div>
<p>9 of our 39 miRNA identified by Shortstack are previously described.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in functional enrichment analysis results</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>miRNA_FE <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/77b55212aaca4763917f67188d9e5e0fd1000c11/D-Apul/output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/miRNA_targets_topGO_FE.csv"</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Function for pulling encriched terms for input miRNA</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>FE <span class="ot">&lt;-</span> <span class="cf">function</span>(miRNA_name){</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  cluster <span class="ot">&lt;-</span> miRNA_named <span class="sc">%&gt;%</span> <span class="fu">filter</span>(given_miRNA_name <span class="sc">==</span> miRNA_name) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Name)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  terms <span class="ot">&lt;-</span> miRNA_FE <span class="sc">%&gt;%</span> <span class="fu">filter</span>(miRNA <span class="sc">==</span> cluster) <span class="sc">%&gt;%</span> <span class="fu">select</span>(GO.ID, Term, type)</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(terms)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in functional annotation table</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>miRNA_FA <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"https://raw.githubusercontent.com/urol-e5/deep-dive-expression/refs/heads/main/D-Apul/output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/miRNA_targets_FA.csv"</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to significant correlation of expression</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>miRNA_sig_FA <span class="ot">&lt;-</span> miRNA_FA <span class="sc">%&gt;%</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_value <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Gene.Ontology.IDs))</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to pull annotated terms for input miRNA</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>sig_FA <span class="ot">&lt;-</span> <span class="cf">function</span>(miRNA_name){</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  cluster <span class="ot">&lt;-</span> miRNA_named <span class="sc">%&gt;%</span> <span class="fu">filter</span>(given_miRNA_name <span class="sc">==</span> miRNA_name) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(Name)</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  terms <span class="ot">&lt;-</span> miRNA_sig_FA <span class="sc">%&gt;%</span> <span class="fu">filter</span>(miRNA <span class="sc">==</span> cluster) <span class="sc">%&gt;%</span> <span class="fu">select</span>(Gene.Ontology..biological.process., Gene.Ontology.IDs, mRNA)</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(terms)</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="mir-2022" class="level2">
<h2 class="anchored" data-anchor-id="mir-2022">mir-2022</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-2022"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                                 Term               type
1 GO:0001964                     startle response Biological.Process
2 GO:0002218 activation of innate immune response Biological.Process
3 GO:0005496                      steroid binding Molecular.Function
4 GO:0004995         tachykinin receptor activity Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-2022"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                                                                                                                                                                                                                                                            Gene.Ontology..biological.process.
1 fibroblast growth factor receptor signaling pathway [GO:0008543]; hatching gland development [GO:0048785]; nervous system development [GO:0007399]; neural crest cell fate commitment [GO:0014034]; neural crest formation [GO:0014029]; positive regulation of DNA-templated transcription [GO:0045893]; Wnt signaling pathway [GO:0016055]
                                                                                                       Gene.Ontology.IDs
1 GO:0000981; GO:0003677; GO:0005634; GO:0007399; GO:0008543; GO:0014029; GO:0014034; GO:0016055; GO:0045893; GO:0048785
        mRNA
1 FUN_040720</code></pre>
</div>
</div>
<p><span class="citation" data-cites="teefy2020">Teefy (<a href="#ref-teefy2020" role="doc-biblioref">2020</a>)</span> found that predicted targets of miR-2022 are predominantly expressed in nematocytes and that miR-2022 is controlled by PAX-A transcription factor, a regulator of nematogenesis in cnidarians. They hypothesize that “miR-2022 participates in an incoherent feedforward loop to control gene dosage during nematocyte development.”</p>
<p><span class="citation" data-cites="fridrich_ancient_2023">Fridrich et al. (<a href="#ref-fridrich_ancient_2023" role="doc-biblioref">2023</a>)</span> showed that miR-2022 is predominantly expressed in stinging cells, and that miR-2022 knockdown in <em>Nematostella</em> inhibits cnidogenisis (development of stinging cnidocyte cells) and defensive capabilities in juvenile <em>Nematostella</em></p>
<p><span class="citation" data-cites="liew_identification_2014">Liew et al. (<a href="#ref-liew_identification_2014" role="doc-biblioref">2014</a>)</span> found that predicted targets of miR-2022 in <em>Stylophera pistillata</em> are enriched (topGO, FDR pval &lt; 0.01) for the Biological Processes:</p>
<p><strong>Immunity:</strong> “GO:0044003: modification by symbiont of host morphology or physiology”, “GO:0019048: virus-host interaction”<br>
<strong>Cell cycle:</strong> “GO:0010564 : regulation of cell cycle process”, “GO:0000281: cytokinesis after mitosis”, “GO:2001237: negative regulation of extrinsic apoptotic signaling pathway”<strong><br>
Transport/Signaling:</strong> “GO:0042147: retrograde transport, endosome to Golgi”, “GO:0006886: intracellular protein transport”, “GO:0051904: pigment granule transport”<br>
<strong>Differentiation/development:</strong> “**GO:0001822: kidney development”, “GO:0030858: positive regulation of epithelial cell differentiation”, “GO:0001942: hair follicle development”, “GO:0048864: stem cell development”, “GO:0021670: lateral ventricle development”, “GO:0043588: skin development”</p>
<p>Our enriched terms seem to support the previously described function of miR-2022, since both “startle response” and “tachykinin receptor activity” are directly related to sensation/response, a primary function of nematocytes/cnidocytes!</p>
</section>
<section id="mir-2030" class="level2">
<h2 class="anchored" data-anchor-id="mir-2030">mir-2030</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-2030"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                                  Term               type
1 GO:0000226 microtubule cytoskeleton organization Biological.Process</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-2030"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] Gene.Ontology..biological.process. Gene.Ontology.IDs                 
[3] mRNA                              
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<p>(None of apul-mir-2030’s significantly correlated targets are annotated)</p>
<p>Highly conserved across bilateria</p>
<p><span class="citation" data-cites="krishna_deep_2013">Krishna et al. (<a href="#ref-krishna_deep_2013" role="doc-biblioref">2013</a>)</span> found that miR-2030 is expressed throughout the body column of <em>Hydra magnipapillata</em> during head regeneration, but *not* in the tentacles</p>
<p>however, <span class="citation" data-cites="moran_cnidarian_2014">Moran et al. (<a href="#ref-moran_cnidarian_2014" role="doc-biblioref">2014</a>)</span> found that miR-2030 is expressed <em>only</em> in the tentacles in <em>Nematostella vectensis</em></p>
<p><span class="citation" data-cites="liew_identification_2014">Liew et al. (<a href="#ref-liew_identification_2014" role="doc-biblioref">2014</a>)</span> found that predicted targets of miR-2030 in <em>Stylophera pistillata</em> are enriched (topGO, FDR pval &lt; 0.01) for the Biological Processes:</p>
<p><strong>Differentiation/Development:</strong> “GO:0009952: anterior/posterior pattern specification”, “GO:0048538: thymus development”, “GO:0007283: spermatogenesis”,<br>
<strong>Miscellaneous:</strong> “GO:0007030: Golgi organization”</p>
</section>
<section id="mir-100" class="level2">
<h2 class="anchored" data-anchor-id="mir-100">mir-100</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-100"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                                                   Term
1 GO:0002224                   toll-like receptor signaling pathway
2 GO:0006598                            polyamine catabolic process
3 GO:0001578                           microtubule bundle formation
4 GO:0006699                         bile acid biosynthetic process
5 GO:0005471                            ATP:ADP antiporter activity
6 GO:0003960                       NADPH:quinone reductase activity
7 GO:0004675 transmembrane receptor protein serine/threonine kin...
8 GO:0004860                      protein kinase inhibitor activity
9 GO:0004857                              enzyme inhibitor activity
                type
1 Biological.Process
2 Biological.Process
3 Biological.Process
4 Biological.Process
5 Molecular.Function
6 Molecular.Function
7 Molecular.Function
8 Molecular.Function
9 Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-100"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       Gene.Ontology..biological.process.
1 embryonic body morphogenesis [GO:0010172]; fibroblast growth factor receptor signaling pathway [GO:0008543]; G1/S transition of mitotic cell cycle [GO:0000082]; miRNA metabolic process [GO:0010586]; miRNA-mediated gene silencing by inhibition of translation [GO:0035278]; negative regulation of translation [GO:0017148]; neural tube development [GO:0021915]; proteasome-mediated ubiquitin-dependent protein catabolic process [GO:0043161]; protein autoubiquitination [GO:0051865]; protein polyubiquitination [GO:0000209]; regulation of miRNA-mediated gene silencing [GO:0060964]; regulation of neural precursor cell proliferation [GO:2000177]; stem cell proliferation [GO:0072089]
2 embryonic body morphogenesis [GO:0010172]; fibroblast growth factor receptor signaling pathway [GO:0008543]; G1/S transition of mitotic cell cycle [GO:0000082]; miRNA metabolic process [GO:0010586]; miRNA-mediated gene silencing by inhibition of translation [GO:0035278]; negative regulation of translation [GO:0017148]; neural tube development [GO:0021915]; proteasome-mediated ubiquitin-dependent protein catabolic process [GO:0043161]; protein autoubiquitination [GO:0051865]; protein polyubiquitination [GO:0000209]; regulation of miRNA-mediated gene silencing [GO:0060964]; regulation of neural precursor cell proliferation [GO:2000177]; stem cell proliferation [GO:0072089]
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             bile acid biosynthetic process [GO:0006699]
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                transposition, DNA-mediated [GO:0006313]
5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      cell cycle [GO:0007049]; cell division [GO:0051301]; regulation of mitotic cell cycle [GO:0007346]
                                                                                                                                                                                                                   Gene.Ontology.IDs
1 GO:0000082; GO:0000209; GO:0000932; GO:0004842; GO:0008270; GO:0008543; GO:0010172; GO:0010586; GO:0017148; GO:0021915; GO:0030371; GO:0035198; GO:0035278; GO:0043161; GO:0051865; GO:0060964; GO:0061630; GO:0072089; GO:2000177
2 GO:0000082; GO:0000209; GO:0000932; GO:0004842; GO:0008270; GO:0008543; GO:0010172; GO:0010586; GO:0017148; GO:0021915; GO:0030371; GO:0035198; GO:0035278; GO:0043161; GO:0051865; GO:0060964; GO:0061630; GO:0072089; GO:2000177
3                                                                                                                                                                                                 GO:0006699; GO:0045302; GO:0047742
4                                                                                                                                                                                                             GO:0003964; GO:0006313
5                                                                                                                                     GO:0000922; GO:0005737; GO:0005815; GO:0007049; GO:0007346; GO:0019900; GO:0030496; GO:0051301
        mRNA
1 FUN_029594
2 FUN_029602
3 FUN_011056
4 FUN_034127
5 FUN_018376</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>in humans, miR-100 has been shown to target genes involved in growth and development. Examples include Plk1, a key mitotic checkpoint regulatory protein [26]; RBSP3, involved in cell proliferation and myeloid cell differentiation [27]; BMPR2, involved in osteogenesis [94]; and FRAP1/mTOR, which regulates cell growth [95]. (Liew et al, 2014)</p>
</blockquote>
<p><span class="citation" data-cites="liew_identification_2014">Liew et al. (<a href="#ref-liew_identification_2014" role="doc-biblioref">2014</a>)</span> found that predicted targets of miR-100 in <em>Stylophera pistillata</em> are enriched (topGO, FDR pval &lt; 0.01) for the Biological Processes:</p>
<p><strong>Biomineralisation:</strong> “GO:0060348: bone development”<br>
<strong>Transcription:</strong> “GO:0000122: negative regulation of transcription from RNA polymerase II promoter”<br>
<strong>Transport/Signaling:</strong> “GO:0006895: Golgi to endosome transport”<br>
<strong>Differentiation/Development:</strong> “GO:0035115: embryonic forelimb morphogenesis”<br>
<strong>Miscellaneous:</strong> “GO:0009987: cellular process”, “GO:0045176: apical protein localization”</p>
</section>
<section id="mir-2050" class="level2">
<h2 class="anchored" data-anchor-id="mir-2050">mir-2050</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-2050"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        GO.ID                                                   Term
1  GO:0006836                             neurotransmitter transport
2  GO:0001843                                    neural tube closure
3  GO:0015969             guanosine tetraphosphate metabolic process
4  GO:0000413                  protein peptidyl-prolyl isomerization
5  GO:0004064                                  arylesterase activity
6  GO:0005262                               calcium channel activity
7  GO:0004312                           fatty acid synthase activity
8  GO:0005539                              glycosaminoglycan binding
9  GO:0004560                            alpha-L-fucosidase activity
10 GO:0000977 RNA polymerase II transcription regulatory region s...
                 type
1  Biological.Process
2  Biological.Process
3  Biological.Process
4  Biological.Process
5  Molecular.Function
6  Molecular.Function
7  Molecular.Function
8  Molecular.Function
9  Molecular.Function
10 Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-2050"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] Gene.Ontology..biological.process. Gene.Ontology.IDs                 
[3] mRNA                              
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<p>(None of apul-mir-2050’s significantly correlated targets are annotated)</p>
<p><span class="citation" data-cites="grinblat_biogeography_2021">Grinblat (<a href="#ref-grinblat_biogeography_2021" role="doc-biblioref">2021</a>)</span> found miR-2050 was highly expressed during spawning in <em>Fungia fungites</em></p>
</section>
<section id="mir-2023" class="level2">
<h2 class="anchored" data-anchor-id="mir-2023">mir-2023</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-2023"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                                Term               type
1 GO:0004736       pyruvate carboxylase activity Molecular.Function
2 GO:0005388 P-type calcium transporter activity Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-2023"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] Gene.Ontology..biological.process. Gene.Ontology.IDs                 
[3] mRNA                              
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<p>(None of apul-mir-2023’s significantly correlated targets are annotated)</p>
<p><span class="citation" data-cites="liew_identification_2014">Liew et al. (<a href="#ref-liew_identification_2014" role="doc-biblioref">2014</a>)</span> found that predicted targets of miR-2023 in <em>Stylophera pistillata</em> are enriched (topGO, FDR pval &lt; 0.01) for the Biological Processes:</p>
<p><strong>Transport/signaling:</strong> “GO:0032869: cellular response to insulin stimulus”<br>
<strong>Cell cycle:</strong> “GO:0007049: cell cycle”<br>
<strong>Transcription:</strong> “GO:0006397: mRNA processing”<br>
<strong>Miscellaneous:</strong> “GO:0042177: negative regulation of protein catabolic process”, “GO:0007030: Golgi organization”</p>
</section>
<section id="mir-2028" class="level2">
<h2 class="anchored" data-anchor-id="mir-2028">mir-2028</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-2028"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                                       Term               type
1 GO:0000278                         mitotic cell cycle Biological.Process
2 GO:0004345 glucose-6-phosphate dehydrogenase activity Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-2028"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                        Gene.Ontology..biological.process.
1 mitotic cell cycle [GO:0000278]; positive regulation of NLRP3 inflammasome complex assembly [GO:1900227]
                                                                                                       Gene.Ontology.IDs
1 GO:0000278; GO:0000775; GO:0000922; GO:0004722; GO:0005634; GO:0005829; GO:0017018; GO:0046872; GO:0046982; GO:1900227
        mRNA
1 FUN_001214</code></pre>
</div>
</div>
</section>
<section id="mir-2036" class="level2">
<h2 class="anchored" data-anchor-id="mir-2036">mir-2036</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-2036"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                                                   Term
1 GO:0000209                             protein polyubiquitination
2 GO:0004591 oxoglutarate dehydrogenase (succinyl-transferring) ...
                type
1 Biological.Process
2 Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-2036"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] Gene.Ontology..biological.process. Gene.Ontology.IDs                 
[3] mRNA                              
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<p>(None of apul-mir-2036’s significantly correlated targets are annotated)</p>
<p><span class="citation" data-cites="liew_identification_2014">Liew et al. (<a href="#ref-liew_identification_2014" role="doc-biblioref">2014</a>)</span> found that predicted targets of miR-2036 in <em>Stylophera pistillata</em> are enriched (topGO, FDR pval &lt; 0.01) for the Biological Processes:</p>
<p><strong>Cytoskeleton:</strong> “GO:0030199: collagen fibril organization”<br>
<strong>Differentiation/development:</strong> “GO:0021670: lateral ventricle development”, “GO:0003323: type B pancreatic cell development”, “GO:0045638: negative regulation of myeloid cell differentiation”<br>
<strong>Miscellaneous:</strong> “GO:0045766: positive regulation of angiogenesis”</p>
</section>
<section id="mir-2025" class="level2">
<h2 class="anchored" data-anchor-id="mir-2025">mir-2025</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-2025"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                       Term               type
1 GO:0001649 osteoblast differentiation Biological.Process</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-2025"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] Gene.Ontology..biological.process. Gene.Ontology.IDs                 
[3] mRNA                              
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
</div>
</div>
<p>(None of apul-mir-2025’s significantly correlated targets are annotated)</p>
</section>
<section id="mir-9425" class="level2">
<h2 class="anchored" data-anchor-id="mir-9425">mir-9425</h2>
<p>Functions that are enriched in this miRNA’s putative targets:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-9425"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                                              Term
1 GO:0000082             G1/S transition of mitotic cell cycle
2 GO:0001523                        retinoid metabolic process
3 GO:0015101 organic cation transmembrane transporter activity
                type
1 Biological.Process
2 Biological.Process
3 Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-9425"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                                                                                                                              Gene.Ontology..biological.process.
1 beta-carotene metabolic process [GO:1901810]; carotene catabolic process [GO:0016121]; retinal metabolic process [GO:0042574]; retinoid metabolic process [GO:0001523]; retinol metabolic process [GO:0042572]
                                                                                           Gene.Ontology.IDs
1 GO:0001523; GO:0003834; GO:0005829; GO:0010436; GO:0016121; GO:0042572; GO:0042574; GO:0046872; GO:1901810
        mRNA
1 FUN_008226</code></pre>
</div>
</div>
</section>
<section id="apul-mir-novel-34-mir-2037" class="level2">
<h2 class="anchored" data-anchor-id="apul-mir-novel-34-mir-2037">apul-mir-novel-34 / miR-2037</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-novel-34"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        GO.ID                                                   Term
1  GO:0000266                                  mitochondrial fission
2  GO:0003360                                  brainstem development
3  GO:0000413                  protein peptidyl-prolyl isomerization
4  GO:0002291 T cell activation via T cell receptor contact with ...
5  GO:0002221         pattern recognition receptor signaling pathway
6  GO:0004013                        adenosylhomocysteinase activity
7  GO:0004402                     histone acetyltransferase activity
8  GO:0005200                 structural constituent of cytoskeleton
9  GO:0005109                                       frizzled binding
10 GO:0004758                 serine C-palmitoyltransferase activity
11 GO:0005245                 voltage-gated calcium channel activity
                 type
1  Biological.Process
2  Biological.Process
3  Biological.Process
4  Biological.Process
5  Biological.Process
6  Molecular.Function
7  Molecular.Function
8  Molecular.Function
9  Molecular.Function
10 Molecular.Function
11 Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-novel-34"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Gene.Ontology..biological.process.
1 activation of cysteine-type endopeptidase activity involved in apoptotic process [GO:0006919]; apoptotic process [GO:0006915]; cellular response to muramyl dipeptide [GO:0071225]; defense response to bacterium [GO:0042742]; defense response to Gram-positive bacterium [GO:0050830]; detection of bacterium [GO:0016045]; ERK1 and ERK2 cascade [GO:0070371]; I-kappaB kinase/NF-kappaB signaling [GO:0007249]; innate immune response [GO:0045087]; intracellular signal transduction [GO:0035556]; JNK cascade [GO:0007254]; pattern recognition receptor signaling pathway [GO:0002221]; positive regulation of cell death [GO:0010942]; positive regulation of cysteine-type endopeptidase activity involved in apoptotic process [GO:0043280]; positive regulation of dendritic cell antigen processing and presentation [GO:0002606]; positive regulation of ERK1 and ERK2 cascade [GO:0070374]; positive regulation of I-kappaB kinase/NF-kappaB signaling [GO:0043123]; positive regulation of interleukin-1 beta production [GO:0032731]; positive regulation of interleukin-6 production [GO:0032755]; positive regulation of interleukin-8 production [GO:0032757]; positive regulation of JNK cascade [GO:0046330]; positive regulation of macrophage cytokine production [GO:0060907]; positive regulation of NF-kappaB transcription factor activity [GO:0051092]; positive regulation of NIK/NF-kappaB signaling [GO:1901224]; positive regulation of nitric-oxide synthase activity [GO:0051000]; positive regulation of stress-activated MAPK cascade [GO:0032874]; positive regulation of tumor necrosis factor production [GO:0032760]; positive regulation of xenophagy [GO:1904417]; regulation of apoptotic process [GO:0042981]; stress-activated MAPK cascade [GO:0051403]; xenophagy [GO:0098792]
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   regulation of biological quality [GO:0065008]
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        proteolysis [GO:0006508]
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Gene.Ontology.IDs
1 GO:0002221; GO:0002606; GO:0005524; GO:0005737; GO:0005829; GO:0005886; GO:0006915; GO:0006919; GO:0007249; GO:0007254; GO:0010942; GO:0016045; GO:0016323; GO:0016324; GO:0032731; GO:0032755; GO:0032757; GO:0032760; GO:0032874; GO:0035556; GO:0038187; GO:0042742; GO:0042802; GO:0042803; GO:0042981; GO:0043123; GO:0043280; GO:0044877; GO:0045087; GO:0045335; GO:0046330; GO:0050700; GO:0050830; GO:0051000; GO:0051092; GO:0051403; GO:0060907; GO:0070371; GO:0070374; GO:0071225; GO:0098792; GO:1901224; GO:1904417
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 GO:0004930; GO:0005886; GO:0065008
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 GO:0004252; GO:0005576; GO:0006508
        mRNA
1 FUN_035124
2 FUN_001557
3 FUN_037296</code></pre>
</div>
</div>
<p><span class="citation" data-cites="liew_identification_2014">Liew et al. (<a href="#ref-liew_identification_2014" role="doc-biblioref">2014</a>)</span> identify a matching miRNA in <em>Stylophora pistillata</em> (spi-miR-temp-20) with targets enriched (topGO, FDR pval &lt; 0.01) for the Biological Processes:</p>
<p><strong>Immunity:</strong> “GO:0032088: negative regulation of NF-kappaB transcription factor activity”<br>
<strong>Cell cycle:</strong> “GO:0042981: regulation of apoptotic process”<br>
<strong>Transport/signaling:</strong> “GO:0046330: positive regulation of JNK cascade”<br>
<strong>Differentiation/development:</strong> “GO:0061146: Peyer’s patch morphogenesis”<strong><br>
Miscellaneous:</strong> “GO:0048169: regulation of long-term neuronal synaptic plasticity”, “GO:0008542: visual learning”, “GO:0006290: pyrimidine dimer repair”</p>
</section>
<section id="apul-mir-novel-9" class="level2">
<h2 class="anchored" data-anchor-id="apul-mir-novel-9">apul-mir-novel-9</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FE</span>(<span class="st">"apul-mir-novel-9"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       GO.ID                                                   Term
1 GO:0001745                             compound eye morphogenesis
2 GO:0001933         negative regulation of protein phosphorylation
3 GO:0009734                      auxin-activated signaling pathway
4 GO:0004459                       L-lactate dehydrogenase activity
5 GO:0004760                  serine-pyruvate transaminase activity
6 GO:0004742 dihydrolipoyllysine-residue acetyltransferase activ...
7 GO:0003779                                          actin binding
                type
1 Biological.Process
2 Biological.Process
3 Biological.Process
4 Molecular.Function
5 Molecular.Function
6 Molecular.Function
7 Molecular.Function</code></pre>
</div>
</div>
<p>Functions of targets with significantly correlated expression:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sig_FA</span>(<span class="st">"apul-mir-novel-9"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Gene.Ontology..biological.process.
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  cell cycle [GO:0007049]; cell division [GO:0051301]; regulation of mitotic cell cycle [GO:0007346]
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    signal transduction [GO:0007165]
3 I-kappaB kinase/NF-kappaB signaling [GO:0007249]; intracellular signal transduction [GO:0035556]; negative regulation of cytokine production involved in inflammatory response [GO:1900016]; negative regulation of epithelial cell proliferation [GO:0050680]; negative regulation of fibroblast proliferation [GO:0048147]; negative regulation of I-kappaB kinase/NF-kappaB signaling [GO:0043124]; negative regulation of inflammatory response [GO:0050728]; negative regulation of innate immune response [GO:0045824]; negative regulation of interferon-alpha production [GO:0032687]; negative regulation of interferon-beta production [GO:0032688]; negative regulation of interleukin-1 beta production [GO:0032691]; negative regulation of interleukin-12 production [GO:0032695]; negative regulation of interleukin-6 production [GO:0032715]; negative regulation of NF-kappaB transcription factor activity [GO:0032088]; negative regulation of NIK/NF-kappaB signaling [GO:1901223]; negative regulation of NLRP3 inflammasome complex assembly [GO:1900226]; negative regulation of phosphatidylinositol 3-kinase signaling [GO:0014067]; negative regulation of tumor necrosis factor production [GO:0032720]; regulation of protein ubiquitination [GO:0031396]; response to lipopolysaccharide [GO:0032496]; T cell activation [GO:0042110]
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      carboxylic acid metabolic process [GO:0019752]
5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  cell cycle [GO:0007049]; cell division [GO:0051301]; regulation of mitotic cell cycle [GO:0007346]
                                                                                                                                                                                                                                                                                                                               Gene.Ontology.IDs
1                                                                                                                                                                                                                         GO:0005737; GO:0005813; GO:0007049; GO:0007346; GO:0019900; GO:0030496; GO:0051301; GO:0072686; GO:0097431; GO:1990023
2                                                                                                                                                                                                                                                                                     GO:0003924; GO:0005525; GO:0005886; GO:0007165; GO:0019003
3 GO:0005524; GO:0005737; GO:0005829; GO:0007249; GO:0014067; GO:0031396; GO:0032088; GO:0032496; GO:0032687; GO:0032688; GO:0032691; GO:0032695; GO:0032715; GO:0032720; GO:0034451; GO:0035556; GO:0036312; GO:0042110; GO:0043124; GO:0043231; GO:0045824; GO:0048147; GO:0048471; GO:0050680; GO:0050728; GO:1900016; GO:1900226; GO:1901223
4                                                                                                                                                                                                                                                                                                             GO:0004459; GO:0005737; GO:0019752
5                                                                                                                                                                                                                                                 GO:0000922; GO:0005737; GO:0005815; GO:0007049; GO:0007346; GO:0019900; GO:0030496; GO:0051301
        mRNA
1 FUN_024529
2 FUN_038089
3 FUN_040522
4 FUN_006649
5 FUN_018376</code></pre>
</div>
</div>
<p><span class="citation" data-cites="gajigan_microrna_2017">Gajigan and Conaco (<a href="#ref-gajigan_microrna_2017" role="doc-biblioref">2017</a>)</span> identify a miRNA in <em>Acropora digitifera</em> (Adi-Mir-Novel-5_3p) that matches our “apul-mir-novel-9”. They found this miRNA was differentially expressed in response to thermal stress, and it’s predicted targets were enriched (topGO, FDR pval &lt; 0.05) for:</p>
<p><strong>Regulation of expression:</strong> “RNA polymerase II transcription corepressor activity”, “Histone dephosphorylation”, “DNA demethylation”, “Positive regulation of transcription RNA poIII promoter”, “Transcription, DNA-templeted”, “Circadian regulation of gene expression”<strong><br>
Stress response:</strong> “Cytokine production”, “Oxidative stress-induced premature senescence”, “MyD88-dependent toll-like receptor”, “Protein disulfide isomerase activity”, “Secretion of lysosomal enzymes”, “Adaptive immune response”, “Negative regulation of cell cycle”<br>
<strong>Repair:</strong> “Positive regulation of DNA repair”, “Regulation of nucleotide-excision”, “RNA repair” “Oxidative demethylation”<br>
<strong>Signaling:</strong> “Receptor catabolic process”, “G-protein coupled receptor kinase”, “Phosphatidylinositol binding”, “GABA receptor binding”, “Protein serine/threonine”, “cAMP-dependent protein kinase”<br>
<strong>Tissue morphogenisis:</strong> “Type I TGFbeta receptor binding”, “Embryonic skeletal system morphogene”, “Zonula adherens maintenance”<br>
</p>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-fridrich_ancient_2023" class="csl-entry" role="listitem">
Fridrich, Arie, Miguel Salinas-Saaverda, Itamar Kozlolvski, Joachim M. Surm, Eleni Chrysostomou, Abhinandan M. Tripathi, Uri Frank, and Yehu Moran. 2023. <span>“An Ancient Pan-Cnidarian <span class="nocase">microRNA</span> Regulates Stinging Capsule Biogenesis in <em><span>Nematostella</span> Vectensis</em>.”</span> <em>Cell Reports</em> 42 (9): 113072. <a href="https://doi.org/10.1016/j.celrep.2023.113072">https://doi.org/10.1016/j.celrep.2023.113072</a>.
</div>
<div id="ref-gajigan_microrna_2017" class="csl-entry" role="listitem">
Gajigan, Andrian P., and Cecilia Conaco. 2017. <span>“A <span class="nocase">microRNA</span> Regulates the Response of Corals to Thermal Stress.”</span> <em>Molecular Ecology</em> 26 (13): 3472–83. <a href="https://doi.org/10.1111/mec.14130">https://doi.org/10.1111/mec.14130</a>.
</div>
<div id="ref-grinblat_biogeography_2021" class="csl-entry" role="listitem">
Grinblat, Mila. 2021. <span>“Biogeography, Reproductive Biology and Early Development in Scleractinian Corals.”</span> Phd, James Cook University. <a href="https://doi.org/10.25903/5v26-zb37">https://doi.org/10.25903/5v26-zb37</a>.
</div>
<div id="ref-krishna_deep_2013" class="csl-entry" role="listitem">
Krishna, Srikar, Aparna Nair, Sirisha Cheedipudi, Deepak Poduval, Jyotsna Dhawan, Dasaradhi Palakodeti, and Yashoda Ghanekar. 2013. <span>“Deep Sequencing Reveals Unique Small <span>RNA</span> Repertoire That Is Regulated During Head Regeneration in <span>Hydra</span> Magnipapillata.”</span> <em>Nucleic Acids Research</em> 41 (1): 599–616. <a href="https://doi.org/10.1093/nar/gks1020">https://doi.org/10.1093/nar/gks1020</a>.
</div>
<div id="ref-liew_identification_2014" class="csl-entry" role="listitem">
Liew, Yi Jin, Manuel Aranda, Adrian Carr, Sebastian Baumgarten, Didier Zoccola, Sylvie Tambutté, Denis Allemand, Gos Micklem, and Christian R. Voolstra. 2014. <span>“Identification of <span>MicroRNAs</span> in the <span>Coral</span> <span>Stylophora</span> Pistillata.”</span> <em>PLOS ONE</em> 9 (3): e91101. <a href="https://doi.org/10.1371/journal.pone.0091101">https://doi.org/10.1371/journal.pone.0091101</a>.
</div>
<div id="ref-moran_cnidarian_2014" class="csl-entry" role="listitem">
Moran, Yehu, David Fredman, Daniela Praher, Xin Li, Liang Wee, Fabian Rentzsch, Phillip Zamore, Ulrich Technau, and Hervé Seitz. 2014. <span>“Cnidarian <span class="nocase">microRNAs</span> Frequently Regulate Targets by Cleavage.”</span> <em>Genome Research</em> 24 (March). <a href="https://doi.org/10.1101/gr.162503.113">https://doi.org/10.1101/gr.162503.113</a>.
</div>
<div id="ref-teefy2020" class="csl-entry" role="listitem">
Teefy, Bryan Barrett. 2020. <span>“The Function of Small RNA Pathways in Cnidarians.”</span> PhD thesis, United States – California. <a href="https://www.proquest.com/docview/2503663922/abstract/885C0525BC204266PQ/1">https://www.proquest.com/docview/2503663922/abstract/885C0525BC204266PQ/1</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/shedurkin\.github\.io\/Roberts-LabNotebook");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>This notebook is built using <a href="https://quarto.org/">Quarto</a></p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>