<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kathleen Durkin">
<meta name="dcterms.date" content="2025-12-08">

<title>Kathleen’s Lab Notebook - Deep Dive Expression: Figure edits</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kathleen’s Lab Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../daily_log.html" rel="" target="">
 <span class="menu-text">Daily Notebook Posts</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../projects.html" rel="" target="">
 <span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/shedurkin" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:kdurkin1@uw.edu" rel="" target=""><i class="bi bi-envelope" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <h1 class="title">Deep Dive Expression: Figure edits</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">E5-coral</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Kathleen Durkin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">December 8, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#circos-plot" id="toc-circos-plot" class="nav-link active" data-scroll-target="#circos-plot">CIRCOS plot</a></li>
  <li><a href="#version-3-compact-layout-with-overlapping-unconserved-targets" id="toc-version-3-compact-layout-with-overlapping-unconserved-targets" class="nav-link" data-scroll-target="#version-3-compact-layout-with-overlapping-unconserved-targets">Version 3: Compact layout with overlapping unconserved targets</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">




<p>Trying out a circos plot via the <code>circlize</code> R package. Hoping to get a figure something like this:</p>
<p><img src="./images/image0.jpeg" class="img-fluid"></p>
<p>Final plot stored <a href="https://github.com/shedurkin/Roberts-LabNotebook/tree/master/posts/projects/E5_coral/output/2025_12_08_dde_plot_edits/circos_plot.png">here</a></p>
<p>Load libraries</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'dplyr' was built under R version 4.2.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'tidyr' was built under R version 4.2.3</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
circlize version 0.4.17
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
</div>
</div>
<p>Load edges files and select only miR-100 targets</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>Apul_edges <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"https://github.com/urol-e5/deep-dive-expression/raw/refs/heads/main/D-Apul/output/33-Apul-miRNA-mRNA-lncRNA-network/edges_miRNA_mRNA_lncRNA_network_p0.05_unique.csv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>X) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(source <span class="sc">==</span> <span class="st">"apul-mir-100"</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>Peve_edges <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"https://github.com/urol-e5/deep-dive-expression/raw/refs/heads/main/E-Peve/output/31-Peve-miRNA-mRNA-lncRNA-network/edges_miRNA_mRNA_lncRNA_network_p0.05_unique.csv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>X) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(source <span class="sc">==</span> <span class="st">"peve-mir-100"</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>Ptuh_edges <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"https://github.com/urol-e5/deep-dive-expression/raw/refs/heads/main/F-Ptuh/output/31-Ptuh-miRNA-mRNA-lncRNA-network/edges_miRNA_mRNA_lncRNA_network_p0.05_unique.csv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>X) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(source <span class="sc">==</span> <span class="st">"ptuh-mir-100"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Read in <a href="https://github.com/urol-e5/timeseries_molecular/blob/main/M-multi-species/scripts/33-orthofinder_E5.md">Hollie’s orthogroup analysis</a> output:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>h_ortho <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(<span class="st">"https://github.com/urol-e5/timeseries_molecular/raw/1b3da407f438e9ecf6fbc43b083b4b94897f2f99/M-multi-species/output/33-orthologs/shared_orthologs_all_species.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 52280 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (3): Orthogroup, Species, Proteins

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>Add column showing whether a given orthogroup contains proteins from all species or only a subset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Standardize species names for simplicity</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>h_ortho <span class="ot">&lt;-</span> h_ortho <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">simple_species =</span> <span class="fu">case_when</span>(</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"Apulchra"</span>, Species) <span class="sc">~</span> <span class="st">"apul"</span>,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"Porites_evermanni"</span>, Species) <span class="sc">~</span> <span class="st">"peve"</span>,</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">grepl</span>(<span class="st">"Pocillopora_meandrina"</span>, Species) <span class="sc">~</span> <span class="st">"ptuh"</span>,</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"other"</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarize species presence per orthogroup</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>h_ortho_summary <span class="ot">&lt;-</span> h_ortho <span class="sc">%&gt;%</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Orthogroup) <span class="sc">%&gt;%</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">type =</span> <span class="fu">case_when</span>(</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>      <span class="fu">all</span>(<span class="fu">c</span>(<span class="st">"apul"</span>, <span class="st">"peve"</span>, <span class="st">"ptuh"</span>) <span class="sc">%in%</span> simple_species) <span class="sc">~</span> <span class="st">"three_way"</span>,</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>      <span class="fu">all</span>(<span class="fu">c</span>(<span class="st">"apul"</span>, <span class="st">"peve"</span>) <span class="sc">%in%</span> simple_species) <span class="sc">&amp;</span> <span class="sc">!</span>(<span class="st">"ptuh"</span> <span class="sc">%in%</span> simple_species) <span class="sc">~</span> <span class="st">"apul_peve"</span>,</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>      <span class="fu">all</span>(<span class="fu">c</span>(<span class="st">"apul"</span>, <span class="st">"ptuh"</span>) <span class="sc">%in%</span> simple_species) <span class="sc">&amp;</span> <span class="sc">!</span>(<span class="st">"peve"</span> <span class="sc">%in%</span> simple_species) <span class="sc">~</span> <span class="st">"apul_ptuh"</span>,</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">all</span>(<span class="fu">c</span>(<span class="st">"peve"</span>, <span class="st">"ptuh"</span>) <span class="sc">%in%</span> simple_species) <span class="sc">&amp;</span> <span class="sc">!</span>(<span class="st">"apul"</span> <span class="sc">%in%</span> simple_species) <span class="sc">~</span> <span class="st">"peve_ptuh"</span>,</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"other"</span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Join back to original</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>h_ortho <span class="ot">&lt;-</span> h_ortho <span class="sc">%&gt;%</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(h_ortho_summary, <span class="at">by =</span> <span class="st">"Orthogroup"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Huh, looks like Hollie’s analysis only outputs orthogroups which contain at least one protein from each species</p>
<p>Read in <a href="https://github.com/urol-e5/timeseries_molecular/blob/1b3da407f438e9ecf6fbc43b083b4b94897f2f99/M-multi-species/scripts/11.1-orthology.md#L4">Steven’s orthogroup analysis</a> output:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>s_ortho <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"https://github.com/urol-e5/timeseries_molecular/raw/refs/heads/main/M-multi-species/output/11-orthology-analysis/ortholog_groups.csv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Annotate each edges file with orthogroup info from both analyses</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotate each target with orthogroup type</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>Apul_edges_ortho <span class="ot">&lt;-</span> Apul_edges <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">h_ortho_type =</span> {</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>      match_row <span class="ot">&lt;-</span> h_ortho <span class="sc">%&gt;%</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(<span class="fu">str_detect</span>(Proteins, <span class="fu">fixed</span>(target)))</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">nrow</span>(match_row) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If multiple matches, pick the first</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>        match_row<span class="sc">$</span>type[<span class="dv">1</span>]</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"none"</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">s_ortho_type =</span> {</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>      match_row <span class="ot">&lt;-</span> s_ortho <span class="sc">%&gt;%</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(<span class="fu">str_detect</span>(apul, <span class="fu">fixed</span>(target)))</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">nrow</span>(match_row) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If multiple matches, pick the first</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>        match_row<span class="sc">$</span>type[<span class="dv">1</span>]</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>        <span class="st">"none"</span></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">species =</span> <span class="st">"Apul"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>Peve_edges_ortho <span class="ot">&lt;-</span> Peve_edges <span class="sc">%&gt;%</span></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">h_ortho_type =</span> {</span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>      match_row <span class="ot">&lt;-</span> h_ortho <span class="sc">%&gt;%</span></span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(<span class="fu">str_detect</span>(Proteins, <span class="fu">fixed</span>(target)))</span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">nrow</span>(match_row) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If multiple matches, pick the first</span></span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>        match_row<span class="sc">$</span>type[<span class="dv">1</span>]</span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>        <span class="st">"none"</span></span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">s_ortho_type =</span> {</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>      match_row <span class="ot">&lt;-</span> s_ortho <span class="sc">%&gt;%</span></span>
<span id="cb15-44"><a href="#cb15-44" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(<span class="fu">str_detect</span>(peve, <span class="fu">fixed</span>(target)))</span>
<span id="cb15-45"><a href="#cb15-45" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">nrow</span>(match_row) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb15-46"><a href="#cb15-46" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If multiple matches, pick the first</span></span>
<span id="cb15-47"><a href="#cb15-47" aria-hidden="true" tabindex="-1"></a>        match_row<span class="sc">$</span>type[<span class="dv">1</span>]</span>
<span id="cb15-48"><a href="#cb15-48" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb15-49"><a href="#cb15-49" aria-hidden="true" tabindex="-1"></a>        <span class="st">"none"</span></span>
<span id="cb15-50"><a href="#cb15-50" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb15-51"><a href="#cb15-51" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb15-52"><a href="#cb15-52" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb15-53"><a href="#cb15-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">species =</span> <span class="st">"Peve"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-54"><a href="#cb15-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb15-55"><a href="#cb15-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-56"><a href="#cb15-56" aria-hidden="true" tabindex="-1"></a>Ptuh_edges_ortho <span class="ot">&lt;-</span> Ptuh_edges <span class="sc">%&gt;%</span></span>
<span id="cb15-57"><a href="#cb15-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-58"><a href="#cb15-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb15-59"><a href="#cb15-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">h_ortho_type =</span> {</span>
<span id="cb15-60"><a href="#cb15-60" aria-hidden="true" tabindex="-1"></a>      match_row <span class="ot">&lt;-</span> h_ortho <span class="sc">%&gt;%</span></span>
<span id="cb15-61"><a href="#cb15-61" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(<span class="fu">str_detect</span>(Proteins, <span class="fu">fixed</span>(target)))</span>
<span id="cb15-62"><a href="#cb15-62" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">nrow</span>(match_row) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb15-63"><a href="#cb15-63" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If multiple matches, pick the first</span></span>
<span id="cb15-64"><a href="#cb15-64" aria-hidden="true" tabindex="-1"></a>        match_row<span class="sc">$</span>type[<span class="dv">1</span>]</span>
<span id="cb15-65"><a href="#cb15-65" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb15-66"><a href="#cb15-66" aria-hidden="true" tabindex="-1"></a>        <span class="st">"none"</span></span>
<span id="cb15-67"><a href="#cb15-67" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb15-68"><a href="#cb15-68" aria-hidden="true" tabindex="-1"></a>    },</span>
<span id="cb15-69"><a href="#cb15-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">s_ortho_type =</span> {</span>
<span id="cb15-70"><a href="#cb15-70" aria-hidden="true" tabindex="-1"></a>      match_row <span class="ot">&lt;-</span> s_ortho <span class="sc">%&gt;%</span></span>
<span id="cb15-71"><a href="#cb15-71" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(<span class="fu">str_detect</span>(ptua, <span class="fu">fixed</span>(target)))</span>
<span id="cb15-72"><a href="#cb15-72" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">nrow</span>(match_row) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb15-73"><a href="#cb15-73" aria-hidden="true" tabindex="-1"></a>        <span class="co"># If multiple matches, pick the first</span></span>
<span id="cb15-74"><a href="#cb15-74" aria-hidden="true" tabindex="-1"></a>        match_row<span class="sc">$</span>type[<span class="dv">1</span>]</span>
<span id="cb15-75"><a href="#cb15-75" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb15-76"><a href="#cb15-76" aria-hidden="true" tabindex="-1"></a>        <span class="st">"none"</span></span>
<span id="cb15-77"><a href="#cb15-77" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb15-78"><a href="#cb15-78" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb15-79"><a href="#cb15-79" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb15-80"><a href="#cb15-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">species =</span> <span class="st">"Ptuh"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-81"><a href="#cb15-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Since Steven’s includes the possibility of orthologs in species subsets (e.g.&nbsp;genes shared between only A. pulchra and P. evermanni), I’ll use his ortholog set now for plotting.</p>
<section id="circos-plot" class="level2">
<h2 class="anchored" data-anchor-id="circos-plot">CIRCOS plot</h2>
<p>Note: Holy crappoly this plot was exceedingly difficult to put together, even when using ChatGPT and Claude Sonnet 4.5 for help – I was troubleshooting problems for <em>hours</em>. <strong>But</strong>, once I tried using Lucas’s paid version (Claude Opus 4.5, integrated into VS code terminal, which he has for his work as a software developer), the agent was able to immediately generate exactly what I was envisioning and make requested changes with no back-and-forth or error handling on my end – holy jinkies Claude Opus is powerful!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(Apul_edges_ortho, Peve_edges_ortho, Ptuh_edges_ortho)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Add target_type column to distinguish lncRNA from genes</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Also fix any typos in s_ortho_type (ptua -&gt; ptuh)</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^lncRNA_"</span>, target), <span class="st">"lncRNA"</span>, <span class="st">"gene"</span>),</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">s_ortho_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>      s_ortho_type <span class="sc">==</span> <span class="st">"apul_ptua"</span> <span class="sc">~</span> <span class="st">"apul_ptuh"</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>      s_ortho_type <span class="sc">==</span> <span class="st">"peve_ptua"</span> <span class="sc">~</span> <span class="st">"peve_ptuh"</span>,</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> s_ortho_type</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the segment order and which species appear in each</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="co"># We'll have segments for both lncRNA (left half) and genes (right half)</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>segment_info <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># lncRNA segments (left half of circle)</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_all_3"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"All 3"</span>,</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"three_way"</span>,</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_apul_peve"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul+Peve"</span>,</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>),</span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"apul_peve"</span>,</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_peve_ptuh"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Peve+Ptuh"</span>,</span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"peve_ptuh"</span>,</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span></span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_apul_ptuh"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul+Ptuh"</span>,</span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"apul_ptuh"</span>,</span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span></span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_apul_only"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul"</span>,</span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>),</span>
<span id="cb16-46"><a href="#cb16-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"none"</span>,</span>
<span id="cb16-47"><a href="#cb16-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span></span>
<span id="cb16-48"><a href="#cb16-48" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-49"><a href="#cb16-49" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_peve_only"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-50"><a href="#cb16-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Peve"</span>,</span>
<span id="cb16-51"><a href="#cb16-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Peve"</span>),</span>
<span id="cb16-52"><a href="#cb16-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"none"</span>,</span>
<span id="cb16-53"><a href="#cb16-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span></span>
<span id="cb16-54"><a href="#cb16-54" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-55"><a href="#cb16-55" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_ptuh_only"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-56"><a href="#cb16-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Ptuh"</span>,</span>
<span id="cb16-57"><a href="#cb16-57" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Ptuh"</span>),</span>
<span id="cb16-58"><a href="#cb16-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"none"</span>,</span>
<span id="cb16-59"><a href="#cb16-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span></span>
<span id="cb16-60"><a href="#cb16-60" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-61"><a href="#cb16-61" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gene segments (right half of circle)</span></span>
<span id="cb16-62"><a href="#cb16-62" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_all_3"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-63"><a href="#cb16-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"All 3"</span>,</span>
<span id="cb16-64"><a href="#cb16-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb16-65"><a href="#cb16-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"three_way"</span>,</span>
<span id="cb16-66"><a href="#cb16-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span></span>
<span id="cb16-67"><a href="#cb16-67" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-68"><a href="#cb16-68" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_apul_peve"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-69"><a href="#cb16-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul+Peve"</span>,</span>
<span id="cb16-70"><a href="#cb16-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>),</span>
<span id="cb16-71"><a href="#cb16-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"apul_peve"</span>,</span>
<span id="cb16-72"><a href="#cb16-72" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span></span>
<span id="cb16-73"><a href="#cb16-73" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-74"><a href="#cb16-74" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_peve_ptuh"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-75"><a href="#cb16-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Peve+Ptuh"</span>,</span>
<span id="cb16-76"><a href="#cb16-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb16-77"><a href="#cb16-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"peve_ptuh"</span>,</span>
<span id="cb16-78"><a href="#cb16-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span></span>
<span id="cb16-79"><a href="#cb16-79" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-80"><a href="#cb16-80" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_apul_ptuh"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-81"><a href="#cb16-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul+Ptuh"</span>,</span>
<span id="cb16-82"><a href="#cb16-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb16-83"><a href="#cb16-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"apul_ptuh"</span>,</span>
<span id="cb16-84"><a href="#cb16-84" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span></span>
<span id="cb16-85"><a href="#cb16-85" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-86"><a href="#cb16-86" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_apul_only"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-87"><a href="#cb16-87" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul"</span>,</span>
<span id="cb16-88"><a href="#cb16-88" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>),</span>
<span id="cb16-89"><a href="#cb16-89" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"none"</span>,</span>
<span id="cb16-90"><a href="#cb16-90" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span></span>
<span id="cb16-91"><a href="#cb16-91" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-92"><a href="#cb16-92" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_peve_only"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-93"><a href="#cb16-93" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Peve"</span>,</span>
<span id="cb16-94"><a href="#cb16-94" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Peve"</span>),</span>
<span id="cb16-95"><a href="#cb16-95" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"none"</span>,</span>
<span id="cb16-96"><a href="#cb16-96" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span></span>
<span id="cb16-97"><a href="#cb16-97" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb16-98"><a href="#cb16-98" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_ptuh_only"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb16-99"><a href="#cb16-99" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Ptuh"</span>,</span>
<span id="cb16-100"><a href="#cb16-100" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Ptuh"</span>),</span>
<span id="cb16-101"><a href="#cb16-101" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"none"</span>,</span>
<span id="cb16-102"><a href="#cb16-102" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span></span>
<span id="cb16-103"><a href="#cb16-103" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-104"><a href="#cb16-104" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb16-105"><a href="#cb16-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-106"><a href="#cb16-106" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate segment sizes based on actual data</span></span>
<span id="cb16-107"><a href="#cb16-107" aria-hidden="true" tabindex="-1"></a>calc_segment_sizes <span class="ot">&lt;-</span> <span class="cf">function</span>(data, segment_info) {</span>
<span id="cb16-108"><a href="#cb16-108" aria-hidden="true" tabindex="-1"></a>  sizes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb16-109"><a href="#cb16-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-110"><a href="#cb16-110" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (seg_name <span class="cf">in</span> <span class="fu">names</span>(segment_info)) {</span>
<span id="cb16-111"><a href="#cb16-111" aria-hidden="true" tabindex="-1"></a>    seg <span class="ot">&lt;-</span> segment_info[[seg_name]]</span>
<span id="cb16-112"><a href="#cb16-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-113"><a href="#cb16-113" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (seg<span class="sc">$</span>ortho_type <span class="sc">==</span> <span class="st">"none"</span>) {</span>
<span id="cb16-114"><a href="#cb16-114" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Species-specific: filter by species and target type</span></span>
<span id="cb16-115"><a href="#cb16-115" aria-hidden="true" tabindex="-1"></a>      n <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb16-116"><a href="#cb16-116" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> <span class="st">"none"</span>,</span>
<span id="cb16-117"><a href="#cb16-117" aria-hidden="true" tabindex="-1"></a>               species <span class="sc">%in%</span> seg<span class="sc">$</span>species,</span>
<span id="cb16-118"><a href="#cb16-118" aria-hidden="true" tabindex="-1"></a>               target_type <span class="sc">==</span> seg<span class="sc">$</span>target_type) <span class="sc">%&gt;%</span></span>
<span id="cb16-119"><a href="#cb16-119" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(target) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span>
<span id="cb16-120"><a href="#cb16-120" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb16-121"><a href="#cb16-121" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Conserved: filter by ortho type and target type</span></span>
<span id="cb16-122"><a href="#cb16-122" aria-hidden="true" tabindex="-1"></a>      n <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb16-123"><a href="#cb16-123" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> seg<span class="sc">$</span>ortho_type,</span>
<span id="cb16-124"><a href="#cb16-124" aria-hidden="true" tabindex="-1"></a>               target_type <span class="sc">==</span> seg<span class="sc">$</span>target_type) <span class="sc">%&gt;%</span></span>
<span id="cb16-125"><a href="#cb16-125" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(target) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span>
<span id="cb16-126"><a href="#cb16-126" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb16-127"><a href="#cb16-127" aria-hidden="true" tabindex="-1"></a>    sizes[[seg_name]] <span class="ot">&lt;-</span> <span class="fu">max</span>(n, <span class="dv">1</span>)  <span class="co"># Minimum size of 1 for visual consistency</span></span>
<span id="cb16-128"><a href="#cb16-128" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-129"><a href="#cb16-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-130"><a href="#cb16-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sizes)</span>
<span id="cb16-131"><a href="#cb16-131" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-132"><a href="#cb16-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-133"><a href="#cb16-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Create improved circos plot with lncRNA/gene split</span></span>
<span id="cb16-134"><a href="#cb16-134" aria-hidden="true" tabindex="-1"></a>create_circos_plot_v2 <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb16-135"><a href="#cb16-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-136"><a href="#cb16-136" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Species order (outer to inner): Apul, Peve, Ptuh</span></span>
<span id="cb16-137"><a href="#cb16-137" aria-hidden="true" tabindex="-1"></a>  species_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>)</span>
<span id="cb16-138"><a href="#cb16-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-139"><a href="#cb16-139" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate segment sizes</span></span>
<span id="cb16-140"><a href="#cb16-140" aria-hidden="true" tabindex="-1"></a>  seg_sizes <span class="ot">&lt;-</span> <span class="fu">calc_segment_sizes</span>(data, segment_info)</span>
<span id="cb16-141"><a href="#cb16-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-142"><a href="#cb16-142" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define sector order: lncRNA segments first (will be left half), then gene segments (right half)</span></span>
<span id="cb16-143"><a href="#cb16-143" aria-hidden="true" tabindex="-1"></a>  lnc_segments <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"lnc_all_3"</span>, <span class="st">"lnc_apul_peve"</span>, <span class="st">"lnc_peve_ptuh"</span>, <span class="st">"lnc_apul_ptuh"</span>,</span>
<span id="cb16-144"><a href="#cb16-144" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"lnc_apul_only"</span>, <span class="st">"lnc_peve_only"</span>, <span class="st">"lnc_ptuh_only"</span>)</span>
<span id="cb16-145"><a href="#cb16-145" aria-hidden="true" tabindex="-1"></a>  gene_segments <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene_all_3"</span>, <span class="st">"gene_apul_peve"</span>, <span class="st">"gene_peve_ptuh"</span>, <span class="st">"gene_apul_ptuh"</span>,</span>
<span id="cb16-146"><a href="#cb16-146" aria-hidden="true" tabindex="-1"></a>                     <span class="st">"gene_apul_only"</span>, <span class="st">"gene_peve_only"</span>, <span class="st">"gene_ptuh_only"</span>)</span>
<span id="cb16-147"><a href="#cb16-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-148"><a href="#cb16-148" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter to only segments with data</span></span>
<span id="cb16-149"><a href="#cb16-149" aria-hidden="true" tabindex="-1"></a>  lnc_segments <span class="ot">&lt;-</span> lnc_segments[<span class="fu">sapply</span>(lnc_segments, <span class="cf">function</span>(s) {</span>
<span id="cb16-150"><a href="#cb16-150" aria-hidden="true" tabindex="-1"></a>    seg <span class="ot">&lt;-</span> segment_info[[s]]</span>
<span id="cb16-151"><a href="#cb16-151" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (seg<span class="sc">$</span>ortho_type <span class="sc">==</span> <span class="st">"none"</span>) {</span>
<span id="cb16-152"><a href="#cb16-152" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> <span class="st">"none"</span>, species <span class="sc">%in%</span> seg<span class="sc">$</span>species, target_type <span class="sc">==</span> <span class="st">"lncRNA"</span>)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb16-153"><a href="#cb16-153" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb16-154"><a href="#cb16-154" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> seg<span class="sc">$</span>ortho_type, target_type <span class="sc">==</span> <span class="st">"lncRNA"</span>)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb16-155"><a href="#cb16-155" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb16-156"><a href="#cb16-156" aria-hidden="true" tabindex="-1"></a>  })]</span>
<span id="cb16-157"><a href="#cb16-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-158"><a href="#cb16-158" aria-hidden="true" tabindex="-1"></a>  gene_segments <span class="ot">&lt;-</span> gene_segments[<span class="fu">sapply</span>(gene_segments, <span class="cf">function</span>(s) {</span>
<span id="cb16-159"><a href="#cb16-159" aria-hidden="true" tabindex="-1"></a>    seg <span class="ot">&lt;-</span> segment_info[[s]]</span>
<span id="cb16-160"><a href="#cb16-160" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (seg<span class="sc">$</span>ortho_type <span class="sc">==</span> <span class="st">"none"</span>) {</span>
<span id="cb16-161"><a href="#cb16-161" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> <span class="st">"none"</span>, species <span class="sc">%in%</span> seg<span class="sc">$</span>species, target_type <span class="sc">==</span> <span class="st">"gene"</span>)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb16-162"><a href="#cb16-162" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb16-163"><a href="#cb16-163" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> seg<span class="sc">$</span>ortho_type, target_type <span class="sc">==</span> <span class="st">"gene"</span>)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb16-164"><a href="#cb16-164" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb16-165"><a href="#cb16-165" aria-hidden="true" tabindex="-1"></a>  })]</span>
<span id="cb16-166"><a href="#cb16-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-167"><a href="#cb16-167" aria-hidden="true" tabindex="-1"></a>  <span class="co"># All sectors in order</span></span>
<span id="cb16-168"><a href="#cb16-168" aria-hidden="true" tabindex="-1"></a>  all_sectors <span class="ot">&lt;-</span> <span class="fu">c</span>(lnc_segments, gene_segments)</span>
<span id="cb16-169"><a href="#cb16-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-170"><a href="#cb16-170" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(all_sectors) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb16-171"><a href="#cb16-171" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"No data to plot"</span>)</span>
<span id="cb16-172"><a href="#cb16-172" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">invisible</span>(<span class="cn">NULL</span>))</span>
<span id="cb16-173"><a href="#cb16-173" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-174"><a href="#cb16-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-175"><a href="#cb16-175" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Build xlims</span></span>
<span id="cb16-176"><a href="#cb16-176" aria-hidden="true" tabindex="-1"></a>  xlims <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb16-177"><a href="#cb16-177" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (seg_name <span class="cf">in</span> all_sectors) {</span>
<span id="cb16-178"><a href="#cb16-178" aria-hidden="true" tabindex="-1"></a>    xlims[[seg_name]] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, seg_sizes[[seg_name]])</span>
<span id="cb16-179"><a href="#cb16-179" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-180"><a href="#cb16-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-181"><a href="#cb16-181" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Initialize circos</span></span>
<span id="cb16-182"><a href="#cb16-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.clear</span>()</span>
<span id="cb16-183"><a href="#cb16-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-184"><a href="#cb16-184" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate gap degrees - larger gaps between lncRNA and gene halves</span></span>
<span id="cb16-185"><a href="#cb16-185" aria-hidden="true" tabindex="-1"></a>  n_lnc <span class="ot">&lt;-</span> <span class="fu">length</span>(lnc_segments)</span>
<span id="cb16-186"><a href="#cb16-186" aria-hidden="true" tabindex="-1"></a>  n_gene <span class="ot">&lt;-</span> <span class="fu">length</span>(gene_segments)</span>
<span id="cb16-187"><a href="#cb16-187" aria-hidden="true" tabindex="-1"></a>  n_total <span class="ot">&lt;-</span> n_lnc <span class="sc">+</span> n_gene</span>
<span id="cb16-188"><a href="#cb16-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-189"><a href="#cb16-189" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create gap vector: small gaps within each half, large gap between halves</span></span>
<span id="cb16-190"><a href="#cb16-190" aria-hidden="true" tabindex="-1"></a>  gap_degrees <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">4</span>, n_total)</span>
<span id="cb16-191"><a href="#cb16-191" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (n_lnc <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;&amp;</span> n_gene <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb16-192"><a href="#cb16-192" aria-hidden="true" tabindex="-1"></a>    gap_degrees[n_lnc] <span class="ot">&lt;-</span> <span class="dv">25</span>  <span class="co"># Gap after last lncRNA segment (before genes)</span></span>
<span id="cb16-193"><a href="#cb16-193" aria-hidden="true" tabindex="-1"></a>    gap_degrees[n_total] <span class="ot">&lt;-</span> <span class="dv">25</span>  <span class="co"># Gap after last gene segment (before lncRNA)</span></span>
<span id="cb16-194"><a href="#cb16-194" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-195"><a href="#cb16-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-196"><a href="#cb16-196" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.par</span>(</span>
<span id="cb16-197"><a href="#cb16-197" aria-hidden="true" tabindex="-1"></a>    <span class="at">start.degree =</span> <span class="dv">90</span>,</span>
<span id="cb16-198"><a href="#cb16-198" aria-hidden="true" tabindex="-1"></a>    <span class="at">gap.degree =</span> gap_degrees,</span>
<span id="cb16-199"><a href="#cb16-199" aria-hidden="true" tabindex="-1"></a>    <span class="at">track.margin =</span> <span class="fu">c</span>(<span class="fl">0.02</span>, <span class="fl">0.02</span>),</span>
<span id="cb16-200"><a href="#cb16-200" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell.padding =</span> <span class="fu">c</span>(<span class="fl">0.02</span>, <span class="fl">0.1</span>, <span class="fl">0.02</span>, <span class="fl">0.1</span>)</span>
<span id="cb16-201"><a href="#cb16-201" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-202"><a href="#cb16-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-203"><a href="#cb16-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.initialize</span>(</span>
<span id="cb16-204"><a href="#cb16-204" aria-hidden="true" tabindex="-1"></a>    <span class="at">factors =</span> <span class="fu">factor</span>(all_sectors, <span class="at">levels =</span> all_sectors),</span>
<span id="cb16-205"><a href="#cb16-205" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">do.call</span>(rbind, xlims[all_sectors])</span>
<span id="cb16-206"><a href="#cb16-206" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-207"><a href="#cb16-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-208"><a href="#cb16-208" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Colors</span></span>
<span id="cb16-209"><a href="#cb16-209" aria-hidden="true" tabindex="-1"></a>  highlight_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.6</span>, <span class="fl">0.85</span>, <span class="dv">1</span>, <span class="fl">0.6</span>)  <span class="co"># Light blue for conserved</span></span>
<span id="cb16-210"><a href="#cb16-210" aria-hidden="true" tabindex="-1"></a>  no_highlight_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.95</span>, <span class="fl">0.95</span>, <span class="fl">0.95</span>, <span class="dv">1</span>)  <span class="co"># Light grey for species-specific</span></span>
<span id="cb16-211"><a href="#cb16-211" aria-hidden="true" tabindex="-1"></a>  pos_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.2</span>, <span class="fl">0.7</span>, <span class="fl">0.2</span>, <span class="fl">0.9</span>)  <span class="co"># Green for positive PCC</span></span>
<span id="cb16-212"><a href="#cb16-212" aria-hidden="true" tabindex="-1"></a>  neg_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.9</span>, <span class="fl">0.4</span>, <span class="fl">0.6</span>, <span class="fl">0.9</span>)  <span class="co"># Pink for negative PCC</span></span>
<span id="cb16-213"><a href="#cb16-213" aria-hidden="true" tabindex="-1"></a>  border_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.3</span>, <span class="fl">0.5</span>, <span class="fl">0.7</span>, <span class="dv">1</span>)  <span class="co"># Blue border</span></span>
<span id="cb16-214"><a href="#cb16-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-215"><a href="#cb16-215" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create 3 tracks (one per species, outer to inner)</span></span>
<span id="cb16-216"><a href="#cb16-216" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (sp_idx <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>) {</span>
<span id="cb16-217"><a href="#cb16-217" aria-hidden="true" tabindex="-1"></a>    sp <span class="ot">&lt;-</span> species_order[sp_idx]</span>
<span id="cb16-218"><a href="#cb16-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-219"><a href="#cb16-219" aria-hidden="true" tabindex="-1"></a>    <span class="fu">circos.track</span>(</span>
<span id="cb16-220"><a href="#cb16-220" aria-hidden="true" tabindex="-1"></a>      <span class="at">factors =</span> <span class="fu">factor</span>(all_sectors, <span class="at">levels =</span> all_sectors),</span>
<span id="cb16-221"><a href="#cb16-221" aria-hidden="true" tabindex="-1"></a>      <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb16-222"><a href="#cb16-222" aria-hidden="true" tabindex="-1"></a>      <span class="at">track.height =</span> <span class="fl">0.15</span>,</span>
<span id="cb16-223"><a href="#cb16-223" aria-hidden="true" tabindex="-1"></a>      <span class="at">bg.border =</span> <span class="st">"grey60"</span>,</span>
<span id="cb16-224"><a href="#cb16-224" aria-hidden="true" tabindex="-1"></a>      <span class="at">bg.lwd =</span> <span class="fl">1.5</span>,</span>
<span id="cb16-225"><a href="#cb16-225" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.fun =</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb16-226"><a href="#cb16-226" aria-hidden="true" tabindex="-1"></a>        sector_name <span class="ot">&lt;-</span> CELL_META<span class="sc">$</span>sector.index</span>
<span id="cb16-227"><a href="#cb16-227" aria-hidden="true" tabindex="-1"></a>        seg <span class="ot">&lt;-</span> segment_info[[sector_name]]</span>
<span id="cb16-228"><a href="#cb16-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-229"><a href="#cb16-229" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Check if this species appears in this segment</span></span>
<span id="cb16-230"><a href="#cb16-230" aria-hidden="true" tabindex="-1"></a>        sp_in_segment <span class="ot">&lt;-</span> sp <span class="sc">%in%</span> seg<span class="sc">$</span>species</span>
<span id="cb16-231"><a href="#cb16-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-232"><a href="#cb16-232" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Background color</span></span>
<span id="cb16-233"><a href="#cb16-233" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (sp_in_segment <span class="sc">&amp;&amp;</span> seg<span class="sc">$</span>ortho_type <span class="sc">!=</span> <span class="st">"none"</span>) {</span>
<span id="cb16-234"><a href="#cb16-234" aria-hidden="true" tabindex="-1"></a>          bg_col <span class="ot">&lt;-</span> highlight_col</span>
<span id="cb16-235"><a href="#cb16-235" aria-hidden="true" tabindex="-1"></a>          border <span class="ot">&lt;-</span> border_col</span>
<span id="cb16-236"><a href="#cb16-236" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (sp_in_segment) {</span>
<span id="cb16-237"><a href="#cb16-237" aria-hidden="true" tabindex="-1"></a>          bg_col <span class="ot">&lt;-</span> no_highlight_col</span>
<span id="cb16-238"><a href="#cb16-238" aria-hidden="true" tabindex="-1"></a>          border <span class="ot">&lt;-</span> <span class="st">"grey60"</span></span>
<span id="cb16-239"><a href="#cb16-239" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb16-240"><a href="#cb16-240" aria-hidden="true" tabindex="-1"></a>          bg_col <span class="ot">&lt;-</span> <span class="st">"white"</span></span>
<span id="cb16-241"><a href="#cb16-241" aria-hidden="true" tabindex="-1"></a>          border <span class="ot">&lt;-</span> <span class="st">"grey80"</span></span>
<span id="cb16-242"><a href="#cb16-242" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb16-243"><a href="#cb16-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-244"><a href="#cb16-244" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Draw background rectangle</span></span>
<span id="cb16-245"><a href="#cb16-245" aria-hidden="true" tabindex="-1"></a>        <span class="fu">circos.rect</span>(</span>
<span id="cb16-246"><a href="#cb16-246" aria-hidden="true" tabindex="-1"></a>          CELL_META<span class="sc">$</span>xlim[<span class="dv">1</span>], <span class="dv">0</span>,</span>
<span id="cb16-247"><a href="#cb16-247" aria-hidden="true" tabindex="-1"></a>          CELL_META<span class="sc">$</span>xlim[<span class="dv">2</span>], <span class="dv">1</span>,</span>
<span id="cb16-248"><a href="#cb16-248" aria-hidden="true" tabindex="-1"></a>          <span class="at">col =</span> bg_col,</span>
<span id="cb16-249"><a href="#cb16-249" aria-hidden="true" tabindex="-1"></a>          <span class="at">border =</span> border,</span>
<span id="cb16-250"><a href="#cb16-250" aria-hidden="true" tabindex="-1"></a>          <span class="at">lwd =</span> <span class="fl">1.5</span></span>
<span id="cb16-251"><a href="#cb16-251" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb16-252"><a href="#cb16-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-253"><a href="#cb16-253" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot points if species is in segment</span></span>
<span id="cb16-254"><a href="#cb16-254" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (sp_in_segment) {</span>
<span id="cb16-255"><a href="#cb16-255" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Get targets for this segment</span></span>
<span id="cb16-256"><a href="#cb16-256" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (seg<span class="sc">$</span>ortho_type <span class="sc">==</span> <span class="st">"none"</span>) {</span>
<span id="cb16-257"><a href="#cb16-257" aria-hidden="true" tabindex="-1"></a>            seg_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb16-258"><a href="#cb16-258" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter</span>(species <span class="sc">==</span> sp,</span>
<span id="cb16-259"><a href="#cb16-259" aria-hidden="true" tabindex="-1"></a>                     s_ortho_type <span class="sc">==</span> <span class="st">"none"</span>,</span>
<span id="cb16-260"><a href="#cb16-260" aria-hidden="true" tabindex="-1"></a>                     target_type <span class="sc">==</span> seg<span class="sc">$</span>target_type) <span class="sc">%&gt;%</span></span>
<span id="cb16-261"><a href="#cb16-261" aria-hidden="true" tabindex="-1"></a>              <span class="fu">distinct</span>(target, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-262"><a href="#cb16-262" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb16-263"><a href="#cb16-263" aria-hidden="true" tabindex="-1"></a>            seg_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb16-264"><a href="#cb16-264" aria-hidden="true" tabindex="-1"></a>              <span class="fu">filter</span>(species <span class="sc">==</span> sp,</span>
<span id="cb16-265"><a href="#cb16-265" aria-hidden="true" tabindex="-1"></a>                     s_ortho_type <span class="sc">==</span> seg<span class="sc">$</span>ortho_type,</span>
<span id="cb16-266"><a href="#cb16-266" aria-hidden="true" tabindex="-1"></a>                     target_type <span class="sc">==</span> seg<span class="sc">$</span>target_type) <span class="sc">%&gt;%</span></span>
<span id="cb16-267"><a href="#cb16-267" aria-hidden="true" tabindex="-1"></a>              <span class="fu">distinct</span>(target, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-268"><a href="#cb16-268" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb16-269"><a href="#cb16-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-270"><a href="#cb16-270" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (<span class="fu">nrow</span>(seg_data) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb16-271"><a href="#cb16-271" aria-hidden="true" tabindex="-1"></a>            n_points <span class="ot">&lt;-</span> <span class="fu">nrow</span>(seg_data)</span>
<span id="cb16-272"><a href="#cb16-272" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> (n_points <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb16-273"><a href="#cb16-273" aria-hidden="true" tabindex="-1"></a>              x_positions <span class="ot">&lt;-</span> CELL_META<span class="sc">$</span>xlim[<span class="dv">2</span>] <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb16-274"><a href="#cb16-274" aria-hidden="true" tabindex="-1"></a>            } <span class="cf">else</span> {</span>
<span id="cb16-275"><a href="#cb16-275" aria-hidden="true" tabindex="-1"></a>              x_positions <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.5</span>, CELL_META<span class="sc">$</span>xlim[<span class="dv">2</span>] <span class="sc">-</span> <span class="fl">0.5</span>, <span class="at">length.out =</span> n_points)</span>
<span id="cb16-276"><a href="#cb16-276" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb16-277"><a href="#cb16-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-278"><a href="#cb16-278" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_points) {</span>
<span id="cb16-279"><a href="#cb16-279" aria-hidden="true" tabindex="-1"></a>              pt_col <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(seg_data<span class="sc">$</span>PCC_direction[i] <span class="sc">&gt;</span> <span class="dv">0</span>, pos_col, neg_col)</span>
<span id="cb16-280"><a href="#cb16-280" aria-hidden="true" tabindex="-1"></a>              <span class="fu">circos.points</span>(</span>
<span id="cb16-281"><a href="#cb16-281" aria-hidden="true" tabindex="-1"></a>                x_positions[i], <span class="fl">0.5</span>,</span>
<span id="cb16-282"><a href="#cb16-282" aria-hidden="true" tabindex="-1"></a>                <span class="at">col =</span> pt_col,</span>
<span id="cb16-283"><a href="#cb16-283" aria-hidden="true" tabindex="-1"></a>                <span class="at">pch =</span> <span class="dv">21</span>,</span>
<span id="cb16-284"><a href="#cb16-284" aria-hidden="true" tabindex="-1"></a>                <span class="at">bg =</span> pt_col,</span>
<span id="cb16-285"><a href="#cb16-285" aria-hidden="true" tabindex="-1"></a>                <span class="at">cex =</span> <span class="fl">1.5</span></span>
<span id="cb16-286"><a href="#cb16-286" aria-hidden="true" tabindex="-1"></a>              )</span>
<span id="cb16-287"><a href="#cb16-287" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb16-288"><a href="#cb16-288" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb16-289"><a href="#cb16-289" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb16-290"><a href="#cb16-290" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb16-291"><a href="#cb16-291" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-292"><a href="#cb16-292" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-293"><a href="#cb16-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-294"><a href="#cb16-294" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add segment labels on the outside</span></span>
<span id="cb16-295"><a href="#cb16-295" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (seg_name <span class="cf">in</span> all_sectors) {</span>
<span id="cb16-296"><a href="#cb16-296" aria-hidden="true" tabindex="-1"></a>    seg <span class="ot">&lt;-</span> segment_info[[seg_name]]</span>
<span id="cb16-297"><a href="#cb16-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-298"><a href="#cb16-298" aria-hidden="true" tabindex="-1"></a>    <span class="fu">circos.text</span>(</span>
<span id="cb16-299"><a href="#cb16-299" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">mean</span>(xlims[[seg_name]]),</span>
<span id="cb16-300"><a href="#cb16-300" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fl">1.8</span>,</span>
<span id="cb16-301"><a href="#cb16-301" aria-hidden="true" tabindex="-1"></a>      <span class="at">sector.index =</span> seg_name,</span>
<span id="cb16-302"><a href="#cb16-302" aria-hidden="true" tabindex="-1"></a>      <span class="at">track.index =</span> <span class="dv">1</span>,</span>
<span id="cb16-303"><a href="#cb16-303" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> seg<span class="sc">$</span>label,</span>
<span id="cb16-304"><a href="#cb16-304" aria-hidden="true" tabindex="-1"></a>      <span class="at">facing =</span> <span class="st">"bending.outside"</span>,</span>
<span id="cb16-305"><a href="#cb16-305" aria-hidden="true" tabindex="-1"></a>      <span class="at">niceFacing =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-306"><a href="#cb16-306" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">0.7</span>,</span>
<span id="cb16-307"><a href="#cb16-307" aria-hidden="true" tabindex="-1"></a>      <span class="at">font =</span> <span class="dv">2</span>,</span>
<span id="cb16-308"><a href="#cb16-308" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> <span class="st">"grey30"</span></span>
<span id="cb16-309"><a href="#cb16-309" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-310"><a href="#cb16-310" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-311"><a href="#cb16-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-312"><a href="#cb16-312" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add "lncRNA" and "Genes" labels for each half</span></span>
<span id="cb16-313"><a href="#cb16-313" aria-hidden="true" tabindex="-1"></a>  <span class="co"># These need to be positioned at the center of each half</span></span>
<span id="cb16-314"><a href="#cb16-314" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (n_lnc <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb16-315"><a href="#cb16-315" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get middle lncRNA segment for positioning</span></span>
<span id="cb16-316"><a href="#cb16-316" aria-hidden="true" tabindex="-1"></a>    mid_lnc <span class="ot">&lt;-</span> lnc_segments[<span class="fu">ceiling</span>(n_lnc <span class="sc">/</span> <span class="dv">2</span>)]</span>
<span id="cb16-317"><a href="#cb16-317" aria-hidden="true" tabindex="-1"></a>    <span class="fu">circos.text</span>(</span>
<span id="cb16-318"><a href="#cb16-318" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">mean</span>(xlims[[mid_lnc]]),</span>
<span id="cb16-319"><a href="#cb16-319" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fl">2.8</span>,</span>
<span id="cb16-320"><a href="#cb16-320" aria-hidden="true" tabindex="-1"></a>      <span class="at">sector.index =</span> mid_lnc,</span>
<span id="cb16-321"><a href="#cb16-321" aria-hidden="true" tabindex="-1"></a>      <span class="at">track.index =</span> <span class="dv">1</span>,</span>
<span id="cb16-322"><a href="#cb16-322" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="st">"lncRNA Targets"</span>,</span>
<span id="cb16-323"><a href="#cb16-323" aria-hidden="true" tabindex="-1"></a>      <span class="at">facing =</span> <span class="st">"bending.outside"</span>,</span>
<span id="cb16-324"><a href="#cb16-324" aria-hidden="true" tabindex="-1"></a>      <span class="at">niceFacing =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-325"><a href="#cb16-325" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">1.1</span>,</span>
<span id="cb16-326"><a href="#cb16-326" aria-hidden="true" tabindex="-1"></a>      <span class="at">font =</span> <span class="dv">2</span>,</span>
<span id="cb16-327"><a href="#cb16-327" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> <span class="st">"darkblue"</span></span>
<span id="cb16-328"><a href="#cb16-328" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-329"><a href="#cb16-329" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-330"><a href="#cb16-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-331"><a href="#cb16-331" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (n_gene <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb16-332"><a href="#cb16-332" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Get middle gene segment for positioning</span></span>
<span id="cb16-333"><a href="#cb16-333" aria-hidden="true" tabindex="-1"></a>    mid_gene <span class="ot">&lt;-</span> gene_segments[<span class="fu">ceiling</span>(n_gene <span class="sc">/</span> <span class="dv">2</span>)]</span>
<span id="cb16-334"><a href="#cb16-334" aria-hidden="true" tabindex="-1"></a>    <span class="fu">circos.text</span>(</span>
<span id="cb16-335"><a href="#cb16-335" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">mean</span>(xlims[[mid_gene]]),</span>
<span id="cb16-336"><a href="#cb16-336" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fl">2.8</span>,</span>
<span id="cb16-337"><a href="#cb16-337" aria-hidden="true" tabindex="-1"></a>      <span class="at">sector.index =</span> mid_gene,</span>
<span id="cb16-338"><a href="#cb16-338" aria-hidden="true" tabindex="-1"></a>      <span class="at">track.index =</span> <span class="dv">1</span>,</span>
<span id="cb16-339"><a href="#cb16-339" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="st">"Gene Targets"</span>,</span>
<span id="cb16-340"><a href="#cb16-340" aria-hidden="true" tabindex="-1"></a>      <span class="at">facing =</span> <span class="st">"bending.outside"</span>,</span>
<span id="cb16-341"><a href="#cb16-341" aria-hidden="true" tabindex="-1"></a>      <span class="at">niceFacing =</span> <span class="cn">TRUE</span>,</span>
<span id="cb16-342"><a href="#cb16-342" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">1.1</span>,</span>
<span id="cb16-343"><a href="#cb16-343" aria-hidden="true" tabindex="-1"></a>      <span class="at">font =</span> <span class="dv">2</span>,</span>
<span id="cb16-344"><a href="#cb16-344" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> <span class="st">"darkred"</span></span>
<span id="cb16-345"><a href="#cb16-345" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb16-346"><a href="#cb16-346" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb16-347"><a href="#cb16-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-348"><a href="#cb16-348" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add species labels in the center</span></span>
<span id="cb16-349"><a href="#cb16-349" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="dv">0</span>, <span class="fl">0.08</span>, <span class="st">"Apul"</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">1.0</span>, <span class="at">col =</span> <span class="st">"grey30"</span>)</span>
<span id="cb16-350"><a href="#cb16-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="st">"Peve"</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">1.0</span>, <span class="at">col =</span> <span class="st">"grey30"</span>)</span>
<span id="cb16-351"><a href="#cb16-351" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="dv">0</span>, <span class="sc">-</span><span class="fl">0.08</span>, <span class="st">"Ptuh"</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">1.0</span>, <span class="at">col =</span> <span class="st">"grey30"</span>)</span>
<span id="cb16-352"><a href="#cb16-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-353"><a href="#cb16-353" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add legend</span></span>
<span id="cb16-354"><a href="#cb16-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">legend</span>(</span>
<span id="cb16-355"><a href="#cb16-355" aria-hidden="true" tabindex="-1"></a>    <span class="st">"bottomright"</span>,</span>
<span id="cb16-356"><a href="#cb16-356" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"Positive PCC"</span>, <span class="st">"Negative PCC"</span>, <span class="st">"Conserved region"</span>),</span>
<span id="cb16-357"><a href="#cb16-357" aria-hidden="true" tabindex="-1"></a>    <span class="at">pch =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">21</span>, <span class="dv">22</span>),</span>
<span id="cb16-358"><a href="#cb16-358" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.bg =</span> <span class="fu">c</span>(pos_col, neg_col, highlight_col),</span>
<span id="cb16-359"><a href="#cb16-359" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(pos_col, neg_col, border_col),</span>
<span id="cb16-360"><a href="#cb16-360" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.cex =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="fl">1.5</span>, <span class="fl">2.2</span>),</span>
<span id="cb16-361"><a href="#cb16-361" aria-hidden="true" tabindex="-1"></a>    <span class="at">bty =</span> <span class="st">"n"</span>,</span>
<span id="cb16-362"><a href="#cb16-362" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.85</span></span>
<span id="cb16-363"><a href="#cb16-363" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb16-364"><a href="#cb16-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-365"><a href="#cb16-365" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add title</span></span>
<span id="cb16-366"><a href="#cb16-366" aria-hidden="true" tabindex="-1"></a>  <span class="fu">title</span>(<span class="st">"miR-100 Targets by Conservation"</span>, <span class="at">line =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">cex.main =</span> <span class="fl">1.3</span>)</span>
<span id="cb16-367"><a href="#cb16-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-368"><a href="#cb16-368" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.clear</span>()</span>
<span id="cb16-369"><a href="#cb16-369" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-370"><a href="#cb16-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-371"><a href="#cb16-371" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the plot</span></span>
<span id="cb16-372"><a href="#cb16-372" aria-hidden="true" tabindex="-1"></a><span class="fu">create_circos_plot_v2</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_apul_only',
track '1'.
Note: 1 point is out of plotting region in sector 'lnc_apul_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 4 points are out of plotting region in sector 'lnc_apul_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_peve_only',
track '1'.
Note: 1 point is out of plotting region in sector 'lnc_peve_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 4 points are out of plotting region in sector 'lnc_peve_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_ptuh_only',
track '1'.
Note: 1 point is out of plotting region in sector 'lnc_ptuh_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 4 points are out of plotting region in sector 'lnc_ptuh_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_all_3', track
'1'.
Note: 1 point is out of plotting region in sector 'gene_all_3', track
'1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 5 points are out of plotting region in sector 'gene_all_3', track
'1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_peve',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_peve',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_apul_peve',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_peve_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_peve_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_peve_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_only',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 4 points are out of plotting region in sector 'gene_apul_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_peve_only',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_peve_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 4 points are out of plotting region in sector 'gene_peve_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_ptuh_only',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_ptuh_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 4 points are out of plotting region in sector 'gene_ptuh_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_peve_only',
track '1'.
Note: 1 point is out of plotting region in sector 'lnc_peve_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 14 points are out of plotting region in sector 'lnc_peve_only',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 12 points are out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2025_12_08_dde_plot_edits_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="version-3-compact-layout-with-overlapping-unconserved-targets" class="level2">
<h2 class="anchored" data-anchor-id="version-3-compact-layout-with-overlapping-unconserved-targets">Version 3: Compact layout with overlapping unconserved targets</h2>
<p>This version combines all species-specific (unconserved) targets into a single segment per target type, where each species’ targets overlap in the same angular space but are not shaded to indicate conservation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set margins to allow room for outer labels</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define segments for compact version - unconserved targets share same segment</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>segment_info_v3 <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># lncRNA segments (left half of circle)</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_all_3"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"All 3"</span>,</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"three_way"</span>,</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">TRUE</span></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_apul_peve"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul+Peve"</span>,</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>),</span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"apul_peve"</span>,</span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">TRUE</span></span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_peve_ptuh"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Peve+Ptuh"</span>,</span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"peve_ptuh"</span>,</span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">TRUE</span></span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_apul_ptuh"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul+Ptuh"</span>,</span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"apul_ptuh"</span>,</span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">TRUE</span></span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a>  <span class="st">"lnc_unconserved"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Unconserved"</span>,</span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"none"</span>,</span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"lncRNA"</span>,</span>
<span id="cb41-40"><a href="#cb41-40" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">FALSE</span></span>
<span id="cb41-41"><a href="#cb41-41" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-42"><a href="#cb41-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Gene segments (right half of circle)</span></span>
<span id="cb41-43"><a href="#cb41-43" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_all_3"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-44"><a href="#cb41-44" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"All 3"</span>,</span>
<span id="cb41-45"><a href="#cb41-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb41-46"><a href="#cb41-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"three_way"</span>,</span>
<span id="cb41-47"><a href="#cb41-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span>,</span>
<span id="cb41-48"><a href="#cb41-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">TRUE</span></span>
<span id="cb41-49"><a href="#cb41-49" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-50"><a href="#cb41-50" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_apul_peve"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-51"><a href="#cb41-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul+Peve"</span>,</span>
<span id="cb41-52"><a href="#cb41-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>),</span>
<span id="cb41-53"><a href="#cb41-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"apul_peve"</span>,</span>
<span id="cb41-54"><a href="#cb41-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span>,</span>
<span id="cb41-55"><a href="#cb41-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">TRUE</span></span>
<span id="cb41-56"><a href="#cb41-56" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-57"><a href="#cb41-57" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_peve_ptuh"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-58"><a href="#cb41-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Peve+Ptuh"</span>,</span>
<span id="cb41-59"><a href="#cb41-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb41-60"><a href="#cb41-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"peve_ptuh"</span>,</span>
<span id="cb41-61"><a href="#cb41-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span>,</span>
<span id="cb41-62"><a href="#cb41-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">TRUE</span></span>
<span id="cb41-63"><a href="#cb41-63" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-64"><a href="#cb41-64" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_apul_ptuh"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-65"><a href="#cb41-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Apul+Ptuh"</span>,</span>
<span id="cb41-66"><a href="#cb41-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb41-67"><a href="#cb41-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"apul_ptuh"</span>,</span>
<span id="cb41-68"><a href="#cb41-68" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span>,</span>
<span id="cb41-69"><a href="#cb41-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">TRUE</span></span>
<span id="cb41-70"><a href="#cb41-70" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb41-71"><a href="#cb41-71" aria-hidden="true" tabindex="-1"></a>  <span class="st">"gene_unconserved"</span> <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb41-72"><a href="#cb41-72" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="st">"Unconserved"</span>,</span>
<span id="cb41-73"><a href="#cb41-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"Apul"</span>, <span class="st">"Peve"</span>, <span class="st">"Ptuh"</span>),</span>
<span id="cb41-74"><a href="#cb41-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">ortho_type =</span> <span class="st">"none"</span>,</span>
<span id="cb41-75"><a href="#cb41-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">target_type =</span> <span class="st">"gene"</span>,</span>
<span id="cb41-76"><a href="#cb41-76" aria-hidden="true" tabindex="-1"></a>    <span class="at">conserved =</span> <span class="cn">FALSE</span></span>
<span id="cb41-77"><a href="#cb41-77" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-78"><a href="#cb41-78" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-79"><a href="#cb41-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-80"><a href="#cb41-80" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate segment sizes for v3 - unconserved segments need max across all species</span></span>
<span id="cb41-81"><a href="#cb41-81" aria-hidden="true" tabindex="-1"></a>calc_segment_sizes_v3 <span class="ot">&lt;-</span> <span class="cf">function</span>(data, segment_info) {</span>
<span id="cb41-82"><a href="#cb41-82" aria-hidden="true" tabindex="-1"></a>  sizes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb41-83"><a href="#cb41-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-84"><a href="#cb41-84" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (seg_name <span class="cf">in</span> <span class="fu">names</span>(segment_info)) {</span>
<span id="cb41-85"><a href="#cb41-85" aria-hidden="true" tabindex="-1"></a>    seg <span class="ot">&lt;-</span> segment_info[[seg_name]]</span>
<span id="cb41-86"><a href="#cb41-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-87"><a href="#cb41-87" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (seg<span class="sc">$</span>ortho_type <span class="sc">==</span> <span class="st">"none"</span>) {</span>
<span id="cb41-88"><a href="#cb41-88" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Unconserved: find the max count across all species</span></span>
<span id="cb41-89"><a href="#cb41-89" aria-hidden="true" tabindex="-1"></a>      max_n <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb41-90"><a href="#cb41-90" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span> (sp <span class="cf">in</span> seg<span class="sc">$</span>species) {</span>
<span id="cb41-91"><a href="#cb41-91" aria-hidden="true" tabindex="-1"></a>        n <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb41-92"><a href="#cb41-92" aria-hidden="true" tabindex="-1"></a>          <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> <span class="st">"none"</span>,</span>
<span id="cb41-93"><a href="#cb41-93" aria-hidden="true" tabindex="-1"></a>                 species <span class="sc">==</span> sp,</span>
<span id="cb41-94"><a href="#cb41-94" aria-hidden="true" tabindex="-1"></a>                 target_type <span class="sc">==</span> seg<span class="sc">$</span>target_type) <span class="sc">%&gt;%</span></span>
<span id="cb41-95"><a href="#cb41-95" aria-hidden="true" tabindex="-1"></a>          <span class="fu">select</span>(target) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span>
<span id="cb41-96"><a href="#cb41-96" aria-hidden="true" tabindex="-1"></a>        max_n <span class="ot">&lt;-</span> <span class="fu">max</span>(max_n, n)</span>
<span id="cb41-97"><a href="#cb41-97" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb41-98"><a href="#cb41-98" aria-hidden="true" tabindex="-1"></a>      sizes[[seg_name]] <span class="ot">&lt;-</span> <span class="fu">max</span>(max_n, <span class="dv">1</span>)</span>
<span id="cb41-99"><a href="#cb41-99" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb41-100"><a href="#cb41-100" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Conserved: filter by ortho type and target type</span></span>
<span id="cb41-101"><a href="#cb41-101" aria-hidden="true" tabindex="-1"></a>      n <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb41-102"><a href="#cb41-102" aria-hidden="true" tabindex="-1"></a>        <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> seg<span class="sc">$</span>ortho_type,</span>
<span id="cb41-103"><a href="#cb41-103" aria-hidden="true" tabindex="-1"></a>               target_type <span class="sc">==</span> seg<span class="sc">$</span>target_type) <span class="sc">%&gt;%</span></span>
<span id="cb41-104"><a href="#cb41-104" aria-hidden="true" tabindex="-1"></a>        <span class="fu">select</span>(target) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">nrow</span>()</span>
<span id="cb41-105"><a href="#cb41-105" aria-hidden="true" tabindex="-1"></a>      sizes[[seg_name]] <span class="ot">&lt;-</span> <span class="fu">max</span>(n, <span class="dv">1</span>)</span>
<span id="cb41-106"><a href="#cb41-106" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb41-107"><a href="#cb41-107" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-108"><a href="#cb41-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-109"><a href="#cb41-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sizes)</span>
<span id="cb41-110"><a href="#cb41-110" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-111"><a href="#cb41-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-112"><a href="#cb41-112" aria-hidden="true" tabindex="-1"></a><span class="co"># Create compact circos plot with overlapping unconserved targets</span></span>
<span id="cb41-113"><a href="#cb41-113" aria-hidden="true" tabindex="-1"></a>create_circos_plot_v3 <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb41-114"><a href="#cb41-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-115"><a href="#cb41-115" aria-hidden="true" tabindex="-1"></a> <span class="co"># Species order (outer to inner): Peve (most), Apul, Ptuh (fewest)</span></span>
<span id="cb41-116"><a href="#cb41-116" aria-hidden="true" tabindex="-1"></a>  species_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Peve"</span>, <span class="st">"Apul"</span>, <span class="st">"Ptuh"</span>)</span>
<span id="cb41-117"><a href="#cb41-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-118"><a href="#cb41-118" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate segment sizes</span></span>
<span id="cb41-119"><a href="#cb41-119" aria-hidden="true" tabindex="-1"></a>  seg_sizes <span class="ot">&lt;-</span> <span class="fu">calc_segment_sizes_v3</span>(data, segment_info_v3)</span>
<span id="cb41-120"><a href="#cb41-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-121"><a href="#cb41-121" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define sector order: lncRNA segments first, then gene segments</span></span>
<span id="cb41-122"><a href="#cb41-122" aria-hidden="true" tabindex="-1"></a>  lnc_segments <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"lnc_all_3"</span>, <span class="st">"lnc_apul_peve"</span>, <span class="st">"lnc_peve_ptuh"</span>, <span class="st">"lnc_apul_ptuh"</span>, <span class="st">"lnc_unconserved"</span>)</span>
<span id="cb41-123"><a href="#cb41-123" aria-hidden="true" tabindex="-1"></a>  gene_segments <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene_all_3"</span>, <span class="st">"gene_apul_peve"</span>, <span class="st">"gene_apul_ptuh"</span>,  <span class="st">"gene_peve_ptuh"</span>, <span class="st">"gene_unconserved"</span>)</span>
<span id="cb41-124"><a href="#cb41-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-125"><a href="#cb41-125" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter to only segments with data</span></span>
<span id="cb41-126"><a href="#cb41-126" aria-hidden="true" tabindex="-1"></a>  lnc_segments <span class="ot">&lt;-</span> lnc_segments[<span class="fu">sapply</span>(lnc_segments, <span class="cf">function</span>(s) {</span>
<span id="cb41-127"><a href="#cb41-127" aria-hidden="true" tabindex="-1"></a>    seg <span class="ot">&lt;-</span> segment_info_v3[[s]]</span>
<span id="cb41-128"><a href="#cb41-128" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (seg<span class="sc">$</span>ortho_type <span class="sc">==</span> <span class="st">"none"</span>) {</span>
<span id="cb41-129"><a href="#cb41-129" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> <span class="st">"none"</span>, target_type <span class="sc">==</span> <span class="st">"lncRNA"</span>)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb41-130"><a href="#cb41-130" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb41-131"><a href="#cb41-131" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> seg<span class="sc">$</span>ortho_type, target_type <span class="sc">==</span> <span class="st">"lncRNA"</span>)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb41-132"><a href="#cb41-132" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb41-133"><a href="#cb41-133" aria-hidden="true" tabindex="-1"></a>  })]</span>
<span id="cb41-134"><a href="#cb41-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-135"><a href="#cb41-135" aria-hidden="true" tabindex="-1"></a>  gene_segments <span class="ot">&lt;-</span> gene_segments[<span class="fu">sapply</span>(gene_segments, <span class="cf">function</span>(s) {</span>
<span id="cb41-136"><a href="#cb41-136" aria-hidden="true" tabindex="-1"></a>    seg <span class="ot">&lt;-</span> segment_info_v3[[s]]</span>
<span id="cb41-137"><a href="#cb41-137" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (seg<span class="sc">$</span>ortho_type <span class="sc">==</span> <span class="st">"none"</span>) {</span>
<span id="cb41-138"><a href="#cb41-138" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> <span class="st">"none"</span>, target_type <span class="sc">==</span> <span class="st">"gene"</span>)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb41-139"><a href="#cb41-139" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb41-140"><a href="#cb41-140" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>(data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s_ortho_type <span class="sc">==</span> seg<span class="sc">$</span>ortho_type, target_type <span class="sc">==</span> <span class="st">"gene"</span>)) <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb41-141"><a href="#cb41-141" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb41-142"><a href="#cb41-142" aria-hidden="true" tabindex="-1"></a>  })]</span>
<span id="cb41-143"><a href="#cb41-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-144"><a href="#cb41-144" aria-hidden="true" tabindex="-1"></a>  <span class="co"># All sectors in order</span></span>
<span id="cb41-145"><a href="#cb41-145" aria-hidden="true" tabindex="-1"></a>  all_sectors <span class="ot">&lt;-</span> <span class="fu">c</span>(lnc_segments, gene_segments)</span>
<span id="cb41-146"><a href="#cb41-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-147"><a href="#cb41-147" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(all_sectors) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb41-148"><a href="#cb41-148" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">"No data to plot"</span>)</span>
<span id="cb41-149"><a href="#cb41-149" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">invisible</span>(<span class="cn">NULL</span>))</span>
<span id="cb41-150"><a href="#cb41-150" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-151"><a href="#cb41-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-152"><a href="#cb41-152" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Build xlims</span></span>
<span id="cb41-153"><a href="#cb41-153" aria-hidden="true" tabindex="-1"></a>  xlims <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb41-154"><a href="#cb41-154" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (seg_name <span class="cf">in</span> all_sectors) {</span>
<span id="cb41-155"><a href="#cb41-155" aria-hidden="true" tabindex="-1"></a>    xlims[[seg_name]] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, seg_sizes[[seg_name]])</span>
<span id="cb41-156"><a href="#cb41-156" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-157"><a href="#cb41-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-158"><a href="#cb41-158" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Initialize circos</span></span>
<span id="cb41-159"><a href="#cb41-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.clear</span>()</span>
<span id="cb41-160"><a href="#cb41-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-161"><a href="#cb41-161" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate gap degrees - larger gaps between lncRNA and gene halves</span></span>
<span id="cb41-162"><a href="#cb41-162" aria-hidden="true" tabindex="-1"></a>  n_lnc <span class="ot">&lt;-</span> <span class="fu">length</span>(lnc_segments)</span>
<span id="cb41-163"><a href="#cb41-163" aria-hidden="true" tabindex="-1"></a>  n_gene <span class="ot">&lt;-</span> <span class="fu">length</span>(gene_segments)</span>
<span id="cb41-164"><a href="#cb41-164" aria-hidden="true" tabindex="-1"></a>  n_total <span class="ot">&lt;-</span> n_lnc <span class="sc">+</span> n_gene</span>
<span id="cb41-165"><a href="#cb41-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-166"><a href="#cb41-166" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create gap vector: small gaps within each half, large gap between halves</span></span>
<span id="cb41-167"><a href="#cb41-167" aria-hidden="true" tabindex="-1"></a>  gap_degrees <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">3</span>, n_total)</span>
<span id="cb41-168"><a href="#cb41-168" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (n_lnc <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;&amp;</span> n_gene <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb41-169"><a href="#cb41-169" aria-hidden="true" tabindex="-1"></a>    gap_degrees[n_lnc] <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb41-170"><a href="#cb41-170" aria-hidden="true" tabindex="-1"></a>    gap_degrees[n_total] <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb41-171"><a href="#cb41-171" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-172"><a href="#cb41-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-173"><a href="#cb41-173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.par</span>(</span>
<span id="cb41-174"><a href="#cb41-174" aria-hidden="true" tabindex="-1"></a>    <span class="at">start.degree =</span> <span class="dv">90</span>,</span>
<span id="cb41-175"><a href="#cb41-175" aria-hidden="true" tabindex="-1"></a>    <span class="at">gap.degree =</span> gap_degrees,</span>
<span id="cb41-176"><a href="#cb41-176" aria-hidden="true" tabindex="-1"></a>    <span class="at">track.margin =</span> <span class="fu">c</span>(<span class="fl">0.02</span>, <span class="fl">0.02</span>),</span>
<span id="cb41-177"><a href="#cb41-177" aria-hidden="true" tabindex="-1"></a>    <span class="at">cell.padding =</span> <span class="fu">c</span>(<span class="fl">0.02</span>, <span class="fl">0.1</span>, <span class="fl">0.02</span>, <span class="fl">0.1</span>),</span>
<span id="cb41-178"><a href="#cb41-178" aria-hidden="true" tabindex="-1"></a>    <span class="at">canvas.xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.4</span>, <span class="fl">1.4</span>),</span>
<span id="cb41-179"><a href="#cb41-179" aria-hidden="true" tabindex="-1"></a>    <span class="at">canvas.ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.4</span>, <span class="fl">1.4</span>)</span>
<span id="cb41-180"><a href="#cb41-180" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-181"><a href="#cb41-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-182"><a href="#cb41-182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.initialize</span>(</span>
<span id="cb41-183"><a href="#cb41-183" aria-hidden="true" tabindex="-1"></a>    <span class="at">factors =</span> <span class="fu">factor</span>(all_sectors, <span class="at">levels =</span> all_sectors),</span>
<span id="cb41-184"><a href="#cb41-184" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">do.call</span>(rbind, xlims[all_sectors])</span>
<span id="cb41-185"><a href="#cb41-185" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-186"><a href="#cb41-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-187"><a href="#cb41-187" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Colors</span></span>
<span id="cb41-188"><a href="#cb41-188" aria-hidden="true" tabindex="-1"></a>  highlight_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.6</span>, <span class="fl">0.85</span>, <span class="dv">1</span>, <span class="fl">0.6</span>)  <span class="co"># Light blue for conserved</span></span>
<span id="cb41-189"><a href="#cb41-189" aria-hidden="true" tabindex="-1"></a>  no_highlight_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.95</span>, <span class="fl">0.95</span>, <span class="fl">0.95</span>, <span class="dv">1</span>)  <span class="co"># Light grey for unconserved</span></span>
<span id="cb41-190"><a href="#cb41-190" aria-hidden="true" tabindex="-1"></a>  pos_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.2</span>, <span class="fl">0.7</span>, <span class="fl">0.2</span>, <span class="fl">0.9</span>)  <span class="co"># Green for positive PCC</span></span>
<span id="cb41-191"><a href="#cb41-191" aria-hidden="true" tabindex="-1"></a>  neg_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.9</span>, <span class="fl">0.4</span>, <span class="fl">0.6</span>, <span class="fl">0.9</span>)  <span class="co"># Pink for negative PCC</span></span>
<span id="cb41-192"><a href="#cb41-192" aria-hidden="true" tabindex="-1"></a>  border_col <span class="ot">&lt;-</span> <span class="fu">rgb</span>(<span class="fl">0.3</span>, <span class="fl">0.5</span>, <span class="fl">0.7</span>, <span class="dv">1</span>)  <span class="co"># Blue border</span></span>
<span id="cb41-193"><a href="#cb41-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-194"><a href="#cb41-194" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create 3 tracks (one per species, outer to inner)</span></span>
<span id="cb41-195"><a href="#cb41-195" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (sp_idx <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>) {</span>
<span id="cb41-196"><a href="#cb41-196" aria-hidden="true" tabindex="-1"></a>    sp <span class="ot">&lt;-</span> species_order[sp_idx]</span>
<span id="cb41-197"><a href="#cb41-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-198"><a href="#cb41-198" aria-hidden="true" tabindex="-1"></a>    <span class="fu">circos.track</span>(</span>
<span id="cb41-199"><a href="#cb41-199" aria-hidden="true" tabindex="-1"></a>      <span class="at">factors =</span> <span class="fu">factor</span>(all_sectors, <span class="at">levels =</span> all_sectors),</span>
<span id="cb41-200"><a href="#cb41-200" aria-hidden="true" tabindex="-1"></a>      <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb41-201"><a href="#cb41-201" aria-hidden="true" tabindex="-1"></a>      <span class="at">track.height =</span> <span class="fl">0.15</span>,</span>
<span id="cb41-202"><a href="#cb41-202" aria-hidden="true" tabindex="-1"></a>      <span class="at">bg.border =</span> <span class="st">"grey60"</span>,</span>
<span id="cb41-203"><a href="#cb41-203" aria-hidden="true" tabindex="-1"></a>      <span class="at">bg.lwd =</span> <span class="fl">1.5</span>,</span>
<span id="cb41-204"><a href="#cb41-204" aria-hidden="true" tabindex="-1"></a>      <span class="at">panel.fun =</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb41-205"><a href="#cb41-205" aria-hidden="true" tabindex="-1"></a>        sector_name <span class="ot">&lt;-</span> CELL_META<span class="sc">$</span>sector.index</span>
<span id="cb41-206"><a href="#cb41-206" aria-hidden="true" tabindex="-1"></a>        seg <span class="ot">&lt;-</span> segment_info_v3[[sector_name]]</span>
<span id="cb41-207"><a href="#cb41-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-208"><a href="#cb41-208" aria-hidden="true" tabindex="-1"></a>        <span class="co"># For conserved segments, check if species is in the conservation group</span></span>
<span id="cb41-209"><a href="#cb41-209" aria-hidden="true" tabindex="-1"></a>        <span class="co"># For unconserved, all species appear but no highlighting</span></span>
<span id="cb41-210"><a href="#cb41-210" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (seg<span class="sc">$</span>conserved) {</span>
<span id="cb41-211"><a href="#cb41-211" aria-hidden="true" tabindex="-1"></a>          sp_in_segment <span class="ot">&lt;-</span> sp <span class="sc">%in%</span> seg<span class="sc">$</span>species</span>
<span id="cb41-212"><a href="#cb41-212" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb41-213"><a href="#cb41-213" aria-hidden="true" tabindex="-1"></a>          sp_in_segment <span class="ot">&lt;-</span> <span class="cn">TRUE</span>  <span class="co"># All species appear in unconserved segment</span></span>
<span id="cb41-214"><a href="#cb41-214" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb41-215"><a href="#cb41-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-216"><a href="#cb41-216" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Background color</span></span>
<span id="cb41-217"><a href="#cb41-217" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (seg<span class="sc">$</span>conserved <span class="sc">&amp;&amp;</span> sp_in_segment) {</span>
<span id="cb41-218"><a href="#cb41-218" aria-hidden="true" tabindex="-1"></a>          bg_col <span class="ot">&lt;-</span> highlight_col</span>
<span id="cb41-219"><a href="#cb41-219" aria-hidden="true" tabindex="-1"></a>          border <span class="ot">&lt;-</span> border_col</span>
<span id="cb41-220"><a href="#cb41-220" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (sp_in_segment) {</span>
<span id="cb41-221"><a href="#cb41-221" aria-hidden="true" tabindex="-1"></a>          bg_col <span class="ot">&lt;-</span> no_highlight_col</span>
<span id="cb41-222"><a href="#cb41-222" aria-hidden="true" tabindex="-1"></a>          border <span class="ot">&lt;-</span> <span class="st">"grey60"</span></span>
<span id="cb41-223"><a href="#cb41-223" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb41-224"><a href="#cb41-224" aria-hidden="true" tabindex="-1"></a>          bg_col <span class="ot">&lt;-</span> <span class="st">"white"</span></span>
<span id="cb41-225"><a href="#cb41-225" aria-hidden="true" tabindex="-1"></a>          border <span class="ot">&lt;-</span> <span class="st">"grey80"</span></span>
<span id="cb41-226"><a href="#cb41-226" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb41-227"><a href="#cb41-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-228"><a href="#cb41-228" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Draw background rectangle</span></span>
<span id="cb41-229"><a href="#cb41-229" aria-hidden="true" tabindex="-1"></a>        <span class="fu">circos.rect</span>(</span>
<span id="cb41-230"><a href="#cb41-230" aria-hidden="true" tabindex="-1"></a>          CELL_META<span class="sc">$</span>xlim[<span class="dv">1</span>], <span class="dv">0</span>,</span>
<span id="cb41-231"><a href="#cb41-231" aria-hidden="true" tabindex="-1"></a>          CELL_META<span class="sc">$</span>xlim[<span class="dv">2</span>], <span class="dv">1</span>,</span>
<span id="cb41-232"><a href="#cb41-232" aria-hidden="true" tabindex="-1"></a>          <span class="at">col =</span> bg_col,</span>
<span id="cb41-233"><a href="#cb41-233" aria-hidden="true" tabindex="-1"></a>          <span class="at">border =</span> border,</span>
<span id="cb41-234"><a href="#cb41-234" aria-hidden="true" tabindex="-1"></a>          <span class="at">lwd =</span> <span class="fl">1.5</span></span>
<span id="cb41-235"><a href="#cb41-235" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb41-236"><a href="#cb41-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-237"><a href="#cb41-237" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Plot points</span></span>
<span id="cb41-238"><a href="#cb41-238" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (seg<span class="sc">$</span>ortho_type <span class="sc">==</span> <span class="st">"none"</span>) {</span>
<span id="cb41-239"><a href="#cb41-239" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Unconserved: each species plots its own targets</span></span>
<span id="cb41-240"><a href="#cb41-240" aria-hidden="true" tabindex="-1"></a>          seg_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb41-241"><a href="#cb41-241" aria-hidden="true" tabindex="-1"></a>            <span class="fu">filter</span>(species <span class="sc">==</span> sp,</span>
<span id="cb41-242"><a href="#cb41-242" aria-hidden="true" tabindex="-1"></a>                   s_ortho_type <span class="sc">==</span> <span class="st">"none"</span>,</span>
<span id="cb41-243"><a href="#cb41-243" aria-hidden="true" tabindex="-1"></a>                   target_type <span class="sc">==</span> seg<span class="sc">$</span>target_type) <span class="sc">%&gt;%</span></span>
<span id="cb41-244"><a href="#cb41-244" aria-hidden="true" tabindex="-1"></a>            <span class="fu">distinct</span>(target, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-245"><a href="#cb41-245" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> <span class="cf">if</span> (sp_in_segment) {</span>
<span id="cb41-246"><a href="#cb41-246" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Conserved: filter by ortho type</span></span>
<span id="cb41-247"><a href="#cb41-247" aria-hidden="true" tabindex="-1"></a>          seg_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb41-248"><a href="#cb41-248" aria-hidden="true" tabindex="-1"></a>            <span class="fu">filter</span>(species <span class="sc">==</span> sp,</span>
<span id="cb41-249"><a href="#cb41-249" aria-hidden="true" tabindex="-1"></a>                   s_ortho_type <span class="sc">==</span> seg<span class="sc">$</span>ortho_type,</span>
<span id="cb41-250"><a href="#cb41-250" aria-hidden="true" tabindex="-1"></a>                   target_type <span class="sc">==</span> seg<span class="sc">$</span>target_type) <span class="sc">%&gt;%</span></span>
<span id="cb41-251"><a href="#cb41-251" aria-hidden="true" tabindex="-1"></a>            <span class="fu">distinct</span>(target, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-252"><a href="#cb41-252" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb41-253"><a href="#cb41-253" aria-hidden="true" tabindex="-1"></a>          seg_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb41-254"><a href="#cb41-254" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb41-255"><a href="#cb41-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-256"><a href="#cb41-256" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> (<span class="fu">nrow</span>(seg_data) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb41-257"><a href="#cb41-257" aria-hidden="true" tabindex="-1"></a>          n_points <span class="ot">&lt;-</span> <span class="fu">nrow</span>(seg_data)</span>
<span id="cb41-258"><a href="#cb41-258" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (n_points <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb41-259"><a href="#cb41-259" aria-hidden="true" tabindex="-1"></a>            x_positions <span class="ot">&lt;-</span> CELL_META<span class="sc">$</span>xlim[<span class="dv">2</span>] <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb41-260"><a href="#cb41-260" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> {</span>
<span id="cb41-261"><a href="#cb41-261" aria-hidden="true" tabindex="-1"></a>            x_positions <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.5</span>, CELL_META<span class="sc">$</span>xlim[<span class="dv">2</span>] <span class="sc">-</span> <span class="fl">0.5</span>, <span class="at">length.out =</span> n_points)</span>
<span id="cb41-262"><a href="#cb41-262" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb41-263"><a href="#cb41-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-264"><a href="#cb41-264" aria-hidden="true" tabindex="-1"></a>          <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n_points) {</span>
<span id="cb41-265"><a href="#cb41-265" aria-hidden="true" tabindex="-1"></a>            pt_col <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(seg_data<span class="sc">$</span>PCC_direction[i] <span class="sc">&gt;</span> <span class="dv">0</span>, pos_col, neg_col)</span>
<span id="cb41-266"><a href="#cb41-266" aria-hidden="true" tabindex="-1"></a>            <span class="fu">circos.points</span>(</span>
<span id="cb41-267"><a href="#cb41-267" aria-hidden="true" tabindex="-1"></a>              x_positions[i], <span class="fl">0.5</span>,</span>
<span id="cb41-268"><a href="#cb41-268" aria-hidden="true" tabindex="-1"></a>              <span class="at">col =</span> pt_col,</span>
<span id="cb41-269"><a href="#cb41-269" aria-hidden="true" tabindex="-1"></a>              <span class="at">pch =</span> <span class="dv">21</span>,</span>
<span id="cb41-270"><a href="#cb41-270" aria-hidden="true" tabindex="-1"></a>              <span class="at">bg =</span> pt_col,</span>
<span id="cb41-271"><a href="#cb41-271" aria-hidden="true" tabindex="-1"></a>              <span class="at">cex =</span> <span class="fl">0.75</span></span>
<span id="cb41-272"><a href="#cb41-272" aria-hidden="true" tabindex="-1"></a>            )</span>
<span id="cb41-273"><a href="#cb41-273" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb41-274"><a href="#cb41-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-275"><a href="#cb41-275" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Add count label above the feature points</span></span>
<span id="cb41-276"><a href="#cb41-276" aria-hidden="true" tabindex="-1"></a>          <span class="fu">circos.text</span>(</span>
<span id="cb41-277"><a href="#cb41-277" aria-hidden="true" tabindex="-1"></a>            <span class="at">x =</span> <span class="fu">mean</span>(x_positions),</span>
<span id="cb41-278"><a href="#cb41-278" aria-hidden="true" tabindex="-1"></a>            <span class="at">y =</span> <span class="fl">0.85</span>,</span>
<span id="cb41-279"><a href="#cb41-279" aria-hidden="true" tabindex="-1"></a>            <span class="at">labels =</span> <span class="fu">as.character</span>(n_points),</span>
<span id="cb41-280"><a href="#cb41-280" aria-hidden="true" tabindex="-1"></a>            <span class="at">facing =</span> <span class="st">"bending.inside"</span>,</span>
<span id="cb41-281"><a href="#cb41-281" aria-hidden="true" tabindex="-1"></a>            <span class="at">niceFacing =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-282"><a href="#cb41-282" aria-hidden="true" tabindex="-1"></a>            <span class="at">cex =</span> <span class="fl">0.75</span>,</span>
<span id="cb41-283"><a href="#cb41-283" aria-hidden="true" tabindex="-1"></a>            <span class="at">col =</span> <span class="st">"black"</span>,</span>
<span id="cb41-284"><a href="#cb41-284" aria-hidden="true" tabindex="-1"></a>            <span class="at">font =</span> <span class="dv">2</span></span>
<span id="cb41-285"><a href="#cb41-285" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb41-286"><a href="#cb41-286" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb41-287"><a href="#cb41-287" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb41-288"><a href="#cb41-288" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb41-289"><a href="#cb41-289" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-290"><a href="#cb41-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-291"><a href="#cb41-291" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add segment labels on the outside</span></span>
<span id="cb41-292"><a href="#cb41-292" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (seg_name <span class="cf">in</span> all_sectors) {</span>
<span id="cb41-293"><a href="#cb41-293" aria-hidden="true" tabindex="-1"></a>    seg <span class="ot">&lt;-</span> segment_info_v3[[seg_name]]</span>
<span id="cb41-294"><a href="#cb41-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-295"><a href="#cb41-295" aria-hidden="true" tabindex="-1"></a>    <span class="fu">circos.text</span>(</span>
<span id="cb41-296"><a href="#cb41-296" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">mean</span>(xlims[[seg_name]]),</span>
<span id="cb41-297"><a href="#cb41-297" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fl">1.8</span>,</span>
<span id="cb41-298"><a href="#cb41-298" aria-hidden="true" tabindex="-1"></a>      <span class="at">sector.index =</span> seg_name,</span>
<span id="cb41-299"><a href="#cb41-299" aria-hidden="true" tabindex="-1"></a>      <span class="at">track.index =</span> <span class="dv">1</span>,</span>
<span id="cb41-300"><a href="#cb41-300" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> seg<span class="sc">$</span>label,</span>
<span id="cb41-301"><a href="#cb41-301" aria-hidden="true" tabindex="-1"></a>      <span class="at">facing =</span> <span class="st">"bending.outside"</span>,</span>
<span id="cb41-302"><a href="#cb41-302" aria-hidden="true" tabindex="-1"></a>      <span class="at">niceFacing =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-303"><a href="#cb41-303" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">0.7</span>,</span>
<span id="cb41-304"><a href="#cb41-304" aria-hidden="true" tabindex="-1"></a>      <span class="at">font =</span> <span class="dv">2</span>,</span>
<span id="cb41-305"><a href="#cb41-305" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> <span class="st">"grey30"</span></span>
<span id="cb41-306"><a href="#cb41-306" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb41-307"><a href="#cb41-307" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-308"><a href="#cb41-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-309"><a href="#cb41-309" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add "lncRNA" and "Genes" labels for each half</span></span>
<span id="cb41-310"><a href="#cb41-310" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (n_lnc <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb41-311"><a href="#cb41-311" aria-hidden="true" tabindex="-1"></a>    mid_lnc <span class="ot">&lt;-</span> lnc_segments[<span class="fu">ceiling</span>(n_lnc <span class="sc">/</span> <span class="dv">2</span>)]</span>
<span id="cb41-312"><a href="#cb41-312" aria-hidden="true" tabindex="-1"></a>    <span class="fu">circos.text</span>(</span>
<span id="cb41-313"><a href="#cb41-313" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">mean</span>(xlims[[mid_lnc]]),</span>
<span id="cb41-314"><a href="#cb41-314" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fl">2.5</span>,</span>
<span id="cb41-315"><a href="#cb41-315" aria-hidden="true" tabindex="-1"></a>      <span class="at">sector.index =</span> mid_lnc,</span>
<span id="cb41-316"><a href="#cb41-316" aria-hidden="true" tabindex="-1"></a>      <span class="at">track.index =</span> <span class="dv">1</span>,</span>
<span id="cb41-317"><a href="#cb41-317" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="st">"lncRNA Targets"</span>,</span>
<span id="cb41-318"><a href="#cb41-318" aria-hidden="true" tabindex="-1"></a>      <span class="at">facing =</span> <span class="st">"bending.outside"</span>,</span>
<span id="cb41-319"><a href="#cb41-319" aria-hidden="true" tabindex="-1"></a>      <span class="at">niceFacing =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-320"><a href="#cb41-320" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">1.0</span>,</span>
<span id="cb41-321"><a href="#cb41-321" aria-hidden="true" tabindex="-1"></a>      <span class="at">font =</span> <span class="dv">2</span>,</span>
<span id="cb41-322"><a href="#cb41-322" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> <span class="st">"darkblue"</span></span>
<span id="cb41-323"><a href="#cb41-323" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb41-324"><a href="#cb41-324" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-325"><a href="#cb41-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-326"><a href="#cb41-326" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (n_gene <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb41-327"><a href="#cb41-327" aria-hidden="true" tabindex="-1"></a>    mid_gene <span class="ot">&lt;-</span> gene_segments[<span class="fu">ceiling</span>(n_gene <span class="sc">/</span> <span class="dv">2</span>)]</span>
<span id="cb41-328"><a href="#cb41-328" aria-hidden="true" tabindex="-1"></a>    <span class="fu">circos.text</span>(</span>
<span id="cb41-329"><a href="#cb41-329" aria-hidden="true" tabindex="-1"></a>      <span class="at">x =</span> <span class="fu">mean</span>(xlims[[mid_gene]]),</span>
<span id="cb41-330"><a href="#cb41-330" aria-hidden="true" tabindex="-1"></a>      <span class="at">y =</span> <span class="fl">2.5</span>,</span>
<span id="cb41-331"><a href="#cb41-331" aria-hidden="true" tabindex="-1"></a>      <span class="at">sector.index =</span> mid_gene,</span>
<span id="cb41-332"><a href="#cb41-332" aria-hidden="true" tabindex="-1"></a>      <span class="at">track.index =</span> <span class="dv">1</span>,</span>
<span id="cb41-333"><a href="#cb41-333" aria-hidden="true" tabindex="-1"></a>      <span class="at">labels =</span> <span class="st">"Gene Targets"</span>,</span>
<span id="cb41-334"><a href="#cb41-334" aria-hidden="true" tabindex="-1"></a>      <span class="at">facing =</span> <span class="st">"bending.outside"</span>,</span>
<span id="cb41-335"><a href="#cb41-335" aria-hidden="true" tabindex="-1"></a>      <span class="at">niceFacing =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-336"><a href="#cb41-336" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex =</span> <span class="fl">1.0</span>,</span>
<span id="cb41-337"><a href="#cb41-337" aria-hidden="true" tabindex="-1"></a>      <span class="at">font =</span> <span class="dv">2</span>,</span>
<span id="cb41-338"><a href="#cb41-338" aria-hidden="true" tabindex="-1"></a>      <span class="at">col =</span> <span class="st">"darkred"</span></span>
<span id="cb41-339"><a href="#cb41-339" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb41-340"><a href="#cb41-340" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb41-341"><a href="#cb41-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-342"><a href="#cb41-342" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add title and species labels in the center (outer to inner: Peve, Apul, Ptuh)</span></span>
<span id="cb41-343"><a href="#cb41-343" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="dv">0</span>, <span class="fl">0.04</span>, <span class="st">"miR-100 Targets"</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">1.1</span>, <span class="at">col =</span> <span class="st">"black"</span>)</span>
<span id="cb41-344"><a href="#cb41-344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="dv">0</span>, <span class="sc">-</span><span class="fl">0.04</span>, <span class="st">"by Conservation"</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">1.1</span>, <span class="at">col =</span> <span class="st">"black"</span>)</span>
<span id="cb41-345"><a href="#cb41-345" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="sc">-</span><span class="fl">0.09</span>, <span class="fl">0.9</span>, <span class="st">"Peve"</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">0.9</span>, <span class="at">col =</span> <span class="st">"grey30"</span>)</span>
<span id="cb41-346"><a href="#cb41-346" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="sc">-</span><span class="fl">0.09</span>, <span class="fl">0.71</span>, <span class="st">"Apul"</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">0.9</span>, <span class="at">col =</span> <span class="st">"grey30"</span>)</span>
<span id="cb41-347"><a href="#cb41-347" aria-hidden="true" tabindex="-1"></a>  <span class="fu">text</span>(<span class="sc">-</span><span class="fl">0.09</span>, <span class="fl">0.53</span>, <span class="st">"Ptuh"</span>, <span class="at">font =</span> <span class="dv">2</span>, <span class="at">cex =</span> <span class="fl">0.9</span>, <span class="at">col =</span> <span class="st">"grey30"</span>)</span>
<span id="cb41-348"><a href="#cb41-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-349"><a href="#cb41-349" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add legend</span></span>
<span id="cb41-350"><a href="#cb41-350" aria-hidden="true" tabindex="-1"></a>  <span class="fu">legend</span>(</span>
<span id="cb41-351"><a href="#cb41-351" aria-hidden="true" tabindex="-1"></a>    <span class="st">"bottomright"</span>,</span>
<span id="cb41-352"><a href="#cb41-352" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">"Positive PCC"</span>, <span class="st">"Negative PCC"</span>, <span class="st">"Conserved region"</span>),</span>
<span id="cb41-353"><a href="#cb41-353" aria-hidden="true" tabindex="-1"></a>    <span class="at">pch =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">21</span>, <span class="dv">22</span>),</span>
<span id="cb41-354"><a href="#cb41-354" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.bg =</span> <span class="fu">c</span>(pos_col, neg_col, highlight_col),</span>
<span id="cb41-355"><a href="#cb41-355" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> <span class="fu">c</span>(pos_col, neg_col, border_col),</span>
<span id="cb41-356"><a href="#cb41-356" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.cex =</span> <span class="fu">c</span>(<span class="fl">1.5</span>, <span class="fl">1.5</span>, <span class="fl">2.2</span>),</span>
<span id="cb41-357"><a href="#cb41-357" aria-hidden="true" tabindex="-1"></a>    <span class="at">bty =</span> <span class="st">"n"</span>,</span>
<span id="cb41-358"><a href="#cb41-358" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.85</span></span>
<span id="cb41-359"><a href="#cb41-359" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-360"><a href="#cb41-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-361"><a href="#cb41-361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">circos.clear</span>()</span>
<span id="cb41-362"><a href="#cb41-362" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-363"><a href="#cb41-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-364"><a href="#cb41-364" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the compact plot</span></span>
<span id="cb41-365"><a href="#cb41-365" aria-hidden="true" tabindex="-1"></a><span class="fu">create_circos_plot_v3</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_unconserved',
track '1'.
Note: 1 point is out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 11 points are out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_all_3', track
'1'.
Note: 1 point is out of plotting region in sector 'gene_all_3', track
'1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 5 points are out of plotting region in sector 'gene_all_3', track
'1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_peve',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_peve',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_apul_peve',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_peve_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_peve_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_peve_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_unconserved',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 11 points are out of plotting region in sector
'gene_unconserved', track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_unconserved',
track '1'.
Note: 1 point is out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 14 points are out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 12 points are out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2025_12_08_dde_plot_edits_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save to png</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">png</span>(<span class="st">"output/2025_12_08_dde_plot_edits/circos_plot.png"</span>, <span class="at">width =</span> <span class="dv">2000</span>, <span class="at">height =</span> <span class="dv">2000</span>, <span class="at">res =</span> <span class="dv">300</span>)</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="fu">create_circos_plot_v3</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 11 points are out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_all_3', track
'1'.
Note: 1 point is out of plotting region in sector 'gene_all_3', track
'1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 5 points are out of plotting region in sector 'gene_all_3', track
'1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_peve',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_peve',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_apul_peve',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_peve_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_peve_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 9 points are out of plotting region in sector 'gene_peve_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_unconserved',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 11 points are out of plotting region in sector
'gene_unconserved', track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'lnc_unconserved',
track '1'.
Note: 1 point is out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 14 points are out of plotting region in sector 'lnc_unconserved',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.
Note: 1 point is out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Note: 12 points are out of plotting region in sector 'gene_apul_ptuh',
track '1'.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>png 
  2 </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">This notebook is built using <a href="https://quarto.org/">Quarto</a></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>