<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kathleen Durkin">
<meta name="dcterms.date" content="2025-04-18">

<title>Kathleen’s Lab Notebook - Timeseries molecular: A.pul, epigenetic features to predict expression. Machine learning – Part 1</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kathleen’s Lab Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../daily_log.html" rel="" target="">
 <span class="menu-text">Daily Notebook Posts</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../projects.html" rel="" target="">
 <span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/shedurkin" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:kdurkin1@uw.edu" rel="" target=""><i class="bi bi-envelope" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <h1 class="title">Timeseries molecular: A.pul, epigenetic features to predict expression. Machine learning – Part 1</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">E5-coral</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Kathleen Durkin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 18, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#set-up" id="toc-set-up" class="nav-link active" data-scroll-target="#set-up">1 Set up</a></li>
  <li><a href="#load-libraries" id="toc-load-libraries" class="nav-link" data-scroll-target="#load-libraries">2 Load libraries</a></li>
  <li><a href="#load-and-format-data" id="toc-load-and-format-data" class="nav-link" data-scroll-target="#load-and-format-data">3 Load and format data</a>
  <ul class="collapse">
  <li><a href="#rna-seq-data-mrna-mirna-lncrna" id="toc-rna-seq-data-mrna-mirna-lncrna" class="nav-link" data-scroll-target="#rna-seq-data-mrna-mirna-lncrna">3.1 RNA-seq data (mRNA, miRNA, lncRNA)</a></li>
  <li><a href="#wgbs-data" id="toc-wgbs-data" class="nav-link" data-scroll-target="#wgbs-data">3.2 WGBS data</a></li>
  <li><a href="#filter-data-sets" id="toc-filter-data-sets" class="nav-link" data-scroll-target="#filter-data-sets">3.3 Filter data sets</a></li>
  <li><a href="#transform-data" id="toc-transform-data" class="nav-link" data-scroll-target="#transform-data">3.4 Transform data</a></li>
  <li><a href="#islolate-gene-sets" id="toc-islolate-gene-sets" class="nav-link" data-scroll-target="#islolate-gene-sets">3.5 Islolate gene sets</a></li>
  </ul></li>
  <li><a href="#feature-selection" id="toc-feature-selection" class="nav-link" data-scroll-target="#feature-selection">4 Feature selection</a>
  <ul class="collapse">
  <li><a href="#all-genes" id="toc-all-genes" class="nav-link" data-scroll-target="#all-genes">4.1 All genes</a></li>
  <li><a href="#lncrna" id="toc-lncrna" class="nav-link" data-scroll-target="#lncrna">4.2 lncRNA</a></li>
  <li><a href="#mirna" id="toc-mirna" class="nav-link" data-scroll-target="#mirna">4.3 miRNA</a></li>
  <li><a href="#wgbs" id="toc-wgbs" class="nav-link" data-scroll-target="#wgbs">4.4 WGBS</a></li>
  <li><a href="#host_afdw" id="toc-host_afdw" class="nav-link" data-scroll-target="#host_afdw">4.5 Host_AFDW</a></li>
  <li><a href="#am" id="toc-am" class="nav-link" data-scroll-target="#am">4.6 Am</a></li>
  <li><a href="#atp-production-go-terms" id="toc-atp-production-go-terms" class="nav-link" data-scroll-target="#atp-production-go-terms">4.7 ATP production GO terms</a></li>
  <li><a href="#energy-usagestorage-go-terms" id="toc-energy-usagestorage-go-terms" class="nav-link" data-scroll-target="#energy-usagestorage-go-terms">4.8 Energy usage/storage GO terms</a></li>
  </ul></li>
  <li><a href="#merge-predictor-features" id="toc-merge-predictor-features" class="nav-link" data-scroll-target="#merge-predictor-features">5 Merge predictor features</a>
  <ul class="collapse">
  <li><a href="#merge-reduced-datasets" id="toc-merge-reduced-datasets" class="nav-link" data-scroll-target="#merge-reduced-datasets">5.1 Merge reduced datasets</a></li>
  <li><a href="#merge-raw-datsets" id="toc-merge-raw-datsets" class="nav-link" data-scroll-target="#merge-raw-datsets">5.2 Merge raw datsets</a></li>
  </ul></li>
  <li><a href="#all-genes-1" id="toc-all-genes-1" class="nav-link" data-scroll-target="#all-genes-1">7 All genes</a>
  <ul class="collapse">
  <li><a href="#the-model" id="toc-the-model" class="nav-link" data-scroll-target="#the-model">7.1 The model</a></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">7.2 Results</a></li>
  </ul></li>
  <li><a href="#host-biomass-host_afdw" id="toc-host-biomass-host_afdw" class="nav-link" data-scroll-target="#host-biomass-host_afdw">8 Host biomass (Host_AFDW)</a>
  <ul class="collapse">
  <li><a href="#the-model-1" id="toc-the-model-1" class="nav-link" data-scroll-target="#the-model-1">8.1 The model</a></li>
  <li><a href="#results-1" id="toc-results-1" class="nav-link" data-scroll-target="#results-1">8.2 Results</a></li>
  </ul></li>
  <li><a href="#symbiont-photsynthesis-am" id="toc-symbiont-photsynthesis-am" class="nav-link" data-scroll-target="#symbiont-photsynthesis-am">9 Symbiont photsynthesis (Am)</a>
  <ul class="collapse">
  <li><a href="#the-model-2" id="toc-the-model-2" class="nav-link" data-scroll-target="#the-model-2">9.1 The model</a></li>
  <li><a href="#results-2" id="toc-results-2" class="nav-link" data-scroll-target="#results-2">9.2 Results</a></li>
  </ul></li>
  <li><a href="#atp-production-go-terms-1" id="toc-atp-production-go-terms-1" class="nav-link" data-scroll-target="#atp-production-go-terms-1">10 ATP production (GO terms)</a>
  <ul class="collapse">
  <li><a href="#the-model-3" id="toc-the-model-3" class="nav-link" data-scroll-target="#the-model-3">10.1 The model</a></li>
  <li><a href="#results-3" id="toc-results-3" class="nav-link" data-scroll-target="#results-3">10.2 Results</a></li>
  </ul></li>
  <li><a href="#energy-usagestorage-go-terms-1" id="toc-energy-usagestorage-go-terms-1" class="nav-link" data-scroll-target="#energy-usagestorage-go-terms-1">11 Energy Usage/Storage (GO terms)</a>
  <ul class="collapse">
  <li><a href="#the-model-4" id="toc-the-model-4" class="nav-link" data-scroll-target="#the-model-4">11.1 The model</a></li>
  <li><a href="#results-4" id="toc-results-4" class="nav-link" data-scroll-target="#results-4">11.2 Results</a></li>
  </ul></li>
  <li><a href="#compare" id="toc-compare" class="nav-link" data-scroll-target="#compare">12 Compare</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">




<p><a href="https://github.com/urol-e5/timeseries_molecular/blob/main/D-Apul/code/22.2-Apul-multiomic-machine-learning.Rmd">Code</a></p>
<p><a href="https://github.com/urol-e5/timeseries_molecular/blob/main/D-Apul/code/22.2-Apul-multiomic-machine-learning.md">Rendered code</a></p>
<p>Applying ML model using multiomic predictors – miRNA + lncRNA + methylation as predictors of gene expression</p>
<p>Inputs:</p>
<ul>
<li><p>RNA counts matrix (raw): <code>../output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-gene_count_matrix.csv</code></p></li>
<li><p>Gene sets of interest: <code>../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/</code></p></li>
<li><p>sRNA/miRNA counts matrix (raw): <code>../output/03.10-D-Apul-sRNAseq-expression-DESeq2/Apul_miRNA_ShortStack_counts_formatted.txt</code></p></li>
<li><p>lncRNA counts matrix (raw): <code>../output/08-Apul-lncNRA/counts.txt</code></p></li>
<li><p>WGBS data (processed): Performed in <code>/timeseries_molecular/D-Apul/output/15.5-Apul-bismark/</code>, data in <a href="https://gannet.fish.washington.edu/seashell/bu-github/timeseries_molecular/D-Apul/output/15.5-Apul-bismark/">large-file storage</a>.</p></li>
<li><p>sample metadata: <code>../../M-multi-species/data/rna_metadata.csv</code></p></li>
</ul>
<section id="set-up" class="level1">
<h1>1 Set up</h1>
</section>
<section id="load-libraries" class="level1">
<h1>2 Load libraries</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(psych)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidygraph)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rvest)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genefilter)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The model includes random processes, so set a seed for reproducability. Note: for final results, should we consider bootstrapping to ensure reported results are representative?</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">703</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="load-and-format-data" class="level1">
<h1>3 Load and format data</h1>
<section id="rna-seq-data-mrna-mirna-lncrna" class="level2">
<h2 class="anchored" data-anchor-id="rna-seq-data-mrna-mirna-lncrna">3.1 RNA-seq data (mRNA, miRNA, lncRNA)</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">### mRNA </span><span class="al">###</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># raw gene counts data (will filter and variance stabilize)</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>Apul_genes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-gene_count_matrix.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Rows: 44371 Columns: 41
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: ","
## chr  (1): gene_id
## dbl (40): 1A1, 1A10, 1A12, 1A2, 1A8, 1A9, 1B1, 1B10, 1B2, 1B5, 1B9, 1C10, 1C...
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>Apul_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_genes)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># format gene IDs as rownames (instead of a column)</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Apul_genes) <span class="ot">&lt;-</span> Apul_genes<span class="sc">$</span>gene_id</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>Apul_genes <span class="ot">&lt;-</span> Apul_genes<span class="sc">%&gt;%</span><span class="fu">select</span>(<span class="sc">!</span>gene_id)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="do">### miRNA </span><span class="al">###</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># raw miRNA counts (will filter and variance stabilize)</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>Apul_miRNA <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="at">file =</span> <span class="st">"../output/03.10-D-Apul-sRNAseq-expression-DESeq2/Apul_miRNA_ShortStack_counts_formatted.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="do">### lncRNA </span><span class="al">###</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co"># raw lncRNA counts (will filter and variance stabilize)</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA_full <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/08-Apul-lncRNA/counts.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove info on genomic location, set lncRNA IDs as rownames</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Apul_lncRNA_full) <span class="ot">&lt;-</span> Apul_lncRNA_full<span class="sc">$</span>Geneid</span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA <span class="ot">&lt;-</span> Apul_lncRNA_full <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>Geneid, <span class="sc">-</span>Chr, <span class="sc">-</span>Start, <span class="sc">-</span>End, <span class="sc">-</span>Strand, <span class="sc">-</span>Length)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="do">### load and format metadata </span><span class="al">###</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../../M-multi-species/data/rna_metadata.csv"</span>)<span class="sc">%&gt;%</span><span class="fu">select</span>(AzentaSampleName, ColonyID, Timepoint) <span class="sc">%&gt;%</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"ACR"</span>, ColonyID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## New names:
## Rows: 117 Columns: 19
## ── Column specification
## ──────────────────────────────────────────────────────── Delimiter: "," chr
## (13): SampleName, WellNumber, AzentaSampleName, ColonyID, Timepoint, Sam... dbl
## (5): SampleNumber, Plate, TotalAmount-ng, Volume-uL, Conc-ng.uL lgl (1):
## MethodUsedForSpectrophotometry
## ℹ Use `spec()` to retrieve the full column specification for this data. ℹ
## Specify the column types or set `show_col_types = FALSE` to quiet this message.
## • `` -&gt; `...19`</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>metadata<span class="sc">$</span>Sample <span class="ot">&lt;-</span> <span class="fu">paste0</span>(metadata<span class="sc">$</span>ColonyID, <span class="st">"-"</span>, metadata<span class="sc">$</span>Timepoint)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Sample</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Setting row names on a tibble is deprecated.</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>colonies <span class="ot">&lt;-</span> <span class="fu">unique</span>(metadata<span class="sc">$</span>ColonyID)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename gene column names to include full sample info</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_genes) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Sample[<span class="fu">match</span>(<span class="fu">colnames</span>(Apul_genes), metadata<span class="sc">$</span>AzentaSampleName)]</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename miRNA column names to match formatting</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_miRNA) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_.*"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(Apul_miRNA))</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_miRNA) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Sample[<span class="fu">match</span>(<span class="fu">colnames</span>(Apul_miRNA), metadata<span class="sc">$</span>AzentaSampleName)]</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co"># rename lncRNA colin names to include full sample info</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_lncRNA) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"...data."</span>, <span class="st">""</span>, <span class="fu">colnames</span>(Apul_lncRNA))</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_lncRNA) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">".sorted.bam"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(Apul_lncRNA))</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Apul_lncRNA) <span class="ot">&lt;-</span> metadata<span class="sc">$</span>Sample[<span class="fu">match</span>(<span class="fu">colnames</span>(Apul_lncRNA), metadata<span class="sc">$</span>AzentaSampleName)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="wgbs-data" class="level2">
<h2 class="anchored" data-anchor-id="wgbs-data">3.2 WGBS data</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#pull processed files from Gannet </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: Unfortunately we can't use the `cache` feature to make this process more time efficient, as it doesn't support long vectors</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the base URL</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>base_url <span class="ot">&lt;-</span> <span class="st">"https://gannet.fish.washington.edu/seashell/bu-github/timeseries_molecular/D-Apul/output/15.5-Apul-bismark/"</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the HTML page</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>page <span class="ot">&lt;-</span> <span class="fu">read_html</span>(base_url)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract links to files</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>file_links <span class="ot">&lt;-</span> page <span class="sc">%&gt;%</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">html_nodes</span>(<span class="st">"a"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">html_attr</span>(<span class="st">"href"</span>)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter for files ending in "processed.txt"</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>processed_files <span class="ot">&lt;-</span> file_links[<span class="fu">grepl</span>(<span class="st">"processed</span><span class="sc">\\</span><span class="st">.txt$"</span>, file_links)]</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Create full URLs</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>file_urls <span class="ot">&lt;-</span> <span class="fu">paste0</span>(base_url, processed_files)</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to read a file from URL</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a>read_processed_file <span class="ot">&lt;-</span> <span class="cf">function</span>(url) {</span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_table</span>(url, <span class="at">col_types =</span> <span class="fu">cols</span>(<span class="at">.default =</span> <span class="st">"c"</span>))  <span class="co"># Read as character to avoid parsing issues</span></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Import all processed files into a list</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>processed_data <span class="ot">&lt;-</span> <span class="fu">lapply</span>(file_urls, read_processed_file)</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Name the list elements by file name</span></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_data) <span class="ot">&lt;-</span> processed_files</span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Print structure of imported data</span></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(processed_data)</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a><span class="co"># add a header row that has "CpG" for the first column and "sample" for the second column, which will be populated by the file name </span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a>processed_data <span class="ot">&lt;-</span> <span class="fu">Map</span>(<span class="cf">function</span>(df, filename) {</span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CpG"</span>, filename)  <span class="co"># Rename columns</span></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>}, processed_data, <span class="fu">names</span>(processed_data))  <span class="co"># Use stored file names</span></span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a><span class="co">#merge files together by "CpG"</span></span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">reduce</span>(processed_data, full_join, <span class="at">by =</span> <span class="st">"CpG"</span>)</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Print structure of final merged data</span></span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(merged_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Replace any NA with 0.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert all columns (except "CpG") to numeric and replace NAs with 0</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> merged_data <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="sc">-</span>CpG, as.numeric)) <span class="sc">%&gt;%</span>  <span class="co"># Convert all except CpG to numeric</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="sc">-</span>CpG, <span class="sc">~</span> <span class="fu">replace_na</span>(.x, <span class="dv">0</span>)))  <span class="co"># Replace NA with 0 in numeric columns</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="filter-data-sets" class="level2">
<h2 class="anchored" data-anchor-id="filter-data-sets">3.3 Filter data sets</h2>
<p>Only keep CpGs that have a non-zero value in all samples.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> merged_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">if_all</span>(<span class="sc">-</span>CpG, <span class="sc">~</span> .x <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure it's formatted as a data frame</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(filtered_wgbs)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Only keep the sample information in the column name. </span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(filtered_wgbs) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"^(.*?)_.*$"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, <span class="fu">colnames</span>(filtered_wgbs))</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Set CpG IDs to rownames</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(filtered_wgbs) <span class="ot">&lt;-</span> filtered_wgbs<span class="sc">$</span>CpG</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> filtered_wgbs <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>CpG)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(merged_data)</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(filtered_wgbs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>We had 12,093,025 CpGs before filtering and have only 507 after filtering. This makes sense because most CpGs were not methylated in all samples.</p>
<p>Save filtered set to make code reruns/knitting quicker</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(filtered_wgbs, <span class="st">"../output/22.2-Apul-multiomic-machine-learning/filtered-WGBS-CpG-counts.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>If knitting/rerunning code, we can load in this filtered data here, instead of loading raw counts and reprocessing.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/22.2-Apul-multiomic-machine-learning/filtered-WGBS-CpG-counts.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only keep genes, miRNA, and lncRNA that are present in at least one sample</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># genes</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>Apul_genes_red <span class="ot">&lt;-</span> Apul_genes[<span class="fu">rowSums</span>(Apul_genes) <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co"># miRNA</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>Apul_miRNA_red <span class="ot">&lt;-</span> Apul_miRNA[<span class="fu">rowSums</span>(Apul_miRNA) <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># lncRNA</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA_red <span class="ot">&lt;-</span> Apul_lncRNA[<span class="fu">rowSums</span>(Apul_lncRNA) <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Retained "</span>, <span class="fu">nrow</span>(Apul_genes_red), <span class="st">" of "</span>, <span class="fu">nrow</span>(Apul_genes), <span class="st">"genes; "</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>       <span class="fu">nrow</span>(Apul_miRNA_red), <span class="st">" of "</span>, <span class="fu">nrow</span>(Apul_miRNA), <span class="st">" miRNA; and "</span>, </span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>       <span class="fu">nrow</span>(Apul_lncRNA_red), <span class="st">" of "</span>, <span class="fu">nrow</span>(Apul_lncRNA), <span class="st">" lncRNA"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Retained  35869  of  44371 genes;  51  of  51  miRNA; and  20559  of  24181  lncRNA</code></pre>
<p><em>pOverA</em>: Specifying the minimum count for a proportion of samples for each gene. Setting 3/40 = 0.08. This would retain genes that are only expressed in a single season in a couple of the colonies. Additionally, setting the minimum count so that the minimum number of samples must have a gene count above a certain threshold.</p>
<p>genes:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>filt <span class="ot">&lt;-</span> <span class="fu">filterfun</span>(<span class="fu">pOverA</span>(<span class="fl">0.08</span>, <span class="dv">5</span>))</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#create filter for the counts data</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>gfilt <span class="ot">&lt;-</span> <span class="fu">genefilter</span>(Apul_genes_red, filt)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">#identify genes to keep by count filter</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>gkeep <span class="ot">&lt;-</span> Apul_genes_red[gfilt,]</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="co">#identify gene lists</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>gn.keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(gkeep)</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co">#gene count data filtered in PoverA, P percent of the samples have counts over A</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>Apul_genes_filt <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_genes_red[<span class="fu">which</span>(<span class="fu">rownames</span>(Apul_genes_red) <span class="sc">%in%</span> gn.keep),])</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a><span class="co">#How many rows do we have before and after filtering?</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Pre-filtering:"</span>, <span class="fu">nrow</span>(Apul_genes_red), <span class="st">"; Post-filtering:"</span>, <span class="fu">nrow</span>(Apul_genes_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Pre-filtering: 35869 ; Post-filtering: 25730</code></pre>
<p>miRNA:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>mifilt <span class="ot">&lt;-</span> <span class="fu">filterfun</span>(<span class="fu">pOverA</span>(<span class="fl">0.08</span>, <span class="dv">5</span>))</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="co">#create filter for the counts data</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>mifilt <span class="ot">&lt;-</span> <span class="fu">genefilter</span>(Apul_miRNA_red, mifilt)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co">#identify miRNA to keep by count filter</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>mikeep <span class="ot">&lt;-</span> Apul_miRNA_red[mifilt,]</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a><span class="co">#identify miRNA to keep by count filter</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>mikeep <span class="ot">&lt;-</span> Apul_miRNA_red[mifilt,]</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co">#identify miRNA lists</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>mi.keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mikeep)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="co">#miRNA count data filtered in PoverA, P percent of the samples have counts over A</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>Apul_miRNA_filt <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_miRNA_red[<span class="fu">which</span>(<span class="fu">rownames</span>(Apul_miRNA_red) <span class="sc">%in%</span> mi.keep),])</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co">#How many rows do we have before and after filtering?</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Pre-filtering:"</span>, <span class="fu">nrow</span>(Apul_miRNA_red), <span class="st">"; Post-filtering:"</span>, <span class="fu">nrow</span>(Apul_miRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Pre-filtering: 51 ; Post-filtering: 47</code></pre>
<p>Of the 51 miRNA, 47 were retained. Which were removed?</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(<span class="fu">rownames</span>(Apul_miRNA_red), <span class="fu">rownames</span>(Apul_miRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] "Cluster_5685"  "Cluster_11565" "Cluster_13647" "Cluster_14633"</code></pre>
<p>lncRNA:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>lncfilt <span class="ot">&lt;-</span> <span class="fu">filterfun</span>(<span class="fu">pOverA</span>(<span class="fl">0.08</span>, <span class="dv">5</span>))</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#create filter for the counts data</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>lncfilt <span class="ot">&lt;-</span> <span class="fu">genefilter</span>(Apul_lncRNA_red, lncfilt)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="co">#identify lncRNA to keep by count filter</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>lnckeep <span class="ot">&lt;-</span> Apul_lncRNA_red[lncfilt,]</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="co">#identify lncRNA to keep by count filter</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>lnckeep <span class="ot">&lt;-</span> Apul_lncRNA_red[lncfilt,]</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a><span class="co">#identify lncRNA lists</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>lnc.keep <span class="ot">&lt;-</span> <span class="fu">rownames</span>(lnckeep)</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a><span class="co">#lncRNA count data filtered in PoverA, P percent of the samples have counts over A</span></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA_filt <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_lncRNA_red[<span class="fu">which</span>(<span class="fu">rownames</span>(Apul_lncRNA_red) <span class="sc">%in%</span> lnc.keep),])</span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a><span class="co">#How many rows do we have before and after filtering?</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Pre-filtering:"</span>, <span class="fu">nrow</span>(Apul_lncRNA_red), <span class="st">"; Post-filtering:"</span>, <span class="fu">nrow</span>(Apul_lncRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Pre-filtering: 20559 ; Post-filtering: 16006</code></pre>
</section>
<section id="transform-data" class="level2">
<h2 class="anchored" data-anchor-id="transform-data">3.4 Transform data</h2>
<p>Set the order of genes, miRNA, lncRNA, wgbs, and metadata to all be the same.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure rownames of metadata are used as the desired column order</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>desired_order <span class="ot">&lt;-</span> <span class="fu">rownames</span>(metadata)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder data frame columns</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>Apul_genes_filt <span class="ot">&lt;-</span> Apul_genes_filt[, desired_order]</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>Apul_miRNA_filt <span class="ot">&lt;-</span> Apul_miRNA_filt[, desired_order]</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>Apul_lncRNA_filt <span class="ot">&lt;-</span> Apul_lncRNA_filt[, desired_order]</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>filtered_wgbs <span class="ot">&lt;-</span> filtered_wgbs[, desired_order]</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Check they all match</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(Apul_genes_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(Apul_miRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(Apul_lncRNA_filt))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(filtered_wgbs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>Use a variance stabilized transformation for all four data sets. Variance stabilization essentially tries to make variance independent of the mean</p>
<p>(Is this the most appropriate design to use?)</p>
<p>genes:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>dds_genes <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> Apul_genes_filt, </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span>Timepoint<span class="sc">+</span>ColonyID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## converting counts to integer mode

## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors

##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance Stabilizing Transformation</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>vsd_genes <span class="ot">&lt;-</span> <span class="fu">assay</span>(<span class="fu">vst</span>(dds_genes, <span class="at">blind =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>miRNA:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>dds_miRNA <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> Apul_miRNA_filt, </span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span>Timepoint<span class="sc">+</span>ColonyID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors

##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance Stabilizing Transformation</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>vsd_miRNA <span class="ot">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span>(dds_miRNA, <span class="at">blind=</span><span class="cn">TRUE</span>) <span class="co"># Must use varianceStabilizingTransformation() instead of vst() due to few input genes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>vsd_miRNA <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd_miRNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>lncRNA:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>dds_lncRNA <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> Apul_lncRNA_filt, </span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span>Timepoint<span class="sc">+</span>ColonyID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors

##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance Stabilizing Transformation</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>vsd_lncRNA <span class="ot">&lt;-</span> <span class="fu">assay</span>(<span class="fu">vst</span>(dds_lncRNA, <span class="at">blind =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Must round wgbs data to whole integers for normalization - need to return to this to decide if this is appropriate.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co">#round to integers </span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>filtered_wgbs<span class="ot">&lt;-</span>filtered_wgbs <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), round))</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>dds_wgbs <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> filtered_wgbs, </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> Timepoint<span class="sc">+</span>ColonyID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## converting counts to integer mode

## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors

##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance Stabilizing Transformation</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>vsd_wgbs <span class="ot">&lt;-</span> <span class="fu">assay</span>(<span class="fu">varianceStabilizingTransformation</span>(dds_wgbs, <span class="at">blind =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##   Note: levels of factors in the design contain characters other than
##   letters, numbers, '_' and '.'. It is recommended (but not required) to use
##   only letters, numbers, and delimiters '_' or '.', as these are safe characters
##   for column names in R. [This is a message, not a warning or an error]

## -- note: fitType='parametric', but the dispersion trend was not well captured by the
##    function: y = a/x + b, and a local regression fit was automatically substituted.
##    specify fitType='local' or 'mean' to avoid this message next time.</code></pre>
</section>
<section id="islolate-gene-sets" class="level2">
<h2 class="anchored" data-anchor-id="islolate-gene-sets">3.5 Islolate gene sets</h2>
<p>Read in gene set tables</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># genes from WGCNA modules significantly correlated with host biomass</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>Host_AFDW <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/Host_AFDW.mg.cm2_gene_counts.tab"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co"># genes from WGCNA modules significantly correlated with symbiont photosynthesis</span></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>Am <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/Am_gene_counts.tab"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co"># GO terms related to energy production/usage</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a>ATP_production_GO <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/ATP_production_GO_terms_gene_counts.tab"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>energy_GO <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/energy_usage_storage_GO_terms_gene_counts.tab"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Isolate filtered counts by gene set</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>vsd_Host_AFDW <span class="ot">&lt;-</span> vsd_genes[<span class="fu">rownames</span>(vsd_genes) <span class="sc">%in%</span> Host_AFDW<span class="sc">$</span>gene_id,]</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>vsd_Am <span class="ot">&lt;-</span> vsd_genes[<span class="fu">rownames</span>(vsd_genes) <span class="sc">%in%</span> Am<span class="sc">$</span>gene_id,]</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>vsd_ATP_production_GO <span class="ot">&lt;-</span> vsd_genes[<span class="fu">rownames</span>(vsd_genes) <span class="sc">%in%</span> ATP_production_GO<span class="sc">$</span>gene_id,]</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>vsd_energy_GO <span class="ot">&lt;-</span> vsd_genes[<span class="fu">rownames</span>(vsd_genes) <span class="sc">%in%</span> energy_GO<span class="sc">$</span>gene_id,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="feature-selection" class="level1">
<h1>4 Feature selection</h1>
<p>Reduce dimensionality through feature selection.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(vsd_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 25730</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(vsd_miRNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 47</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(vsd_lncRNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 16006</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(vsd_wgbs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 507</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(vsd_Host_AFDW)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 793</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(vsd_Am)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 5746</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(vsd_ATP_production_GO)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 22</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(vsd_energy_GO)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 284</code></pre>
<p>The genes, lncRNA, and WGBS sets are very large, so we need to reduce using dimensionality reduction. We’ll need to do the same for the gene sets, which are also fairly large.</p>
<section id="all-genes" class="level2">
<h2 class="anchored" data-anchor-id="all-genes">4.1 All genes</h2>
<div class="sourceCode" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on gene expression matrix</span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>pca_genes <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(vsd_genes), <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top PCs that explain most variance (e.g., top 50 PCs)</span></span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>explained_var_genes <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_genes)<span class="sc">$</span>importance[<span class="dv">2</span>, ]  <span class="co"># Cumulative variance explained</span></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>num_pcs_genes <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(<span class="fu">cumsum</span>(explained_var_genes) <span class="sc">&gt;</span> <span class="fl">0.95</span>))  <span class="co"># Keep PCs that explain 95% variance</span></span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a>genes_pcs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_genes<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span>num_pcs_genes])  <span class="co"># Extract selected PCs</span></span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(genes_pcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 40 30</code></pre>
<p>30 PCs summarize 95% of the explained variance in all gene expression</p>
</section>
<section id="lncrna" class="level2">
<h2 class="anchored" data-anchor-id="lncrna">4.2 lncRNA</h2>
<div class="sourceCode" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on lncRNA expression matrix</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>pca_lncRNA <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(vsd_lncRNA), <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top PCs that explain most variance (e.g., top 50 PCs)</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>explained_var_lncRNA <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_lncRNA)<span class="sc">$</span>importance[<span class="dv">2</span>, ]  <span class="co"># Cumulative variance explained</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>num_pcs_lncRNA <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(<span class="fu">cumsum</span>(explained_var_lncRNA) <span class="sc">&gt;</span> <span class="fl">0.95</span>))  <span class="co"># Keep PCs that explain 95% variance</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>lncRNA_pcs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_lncRNA<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span>num_pcs_lncRNA])  <span class="co"># Extract selected PCs</span></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(lncRNA_pcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 40 31</code></pre>
<p>31 PCs summarize 95% of the explained variance in lncRNA expression</p>
</section>
<section id="mirna" class="level2">
<h2 class="anchored" data-anchor-id="mirna">4.3 miRNA</h2>
<div class="sourceCode" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on miRNA expression matrix</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>pca_miRNA <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(vsd_miRNA), <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top PCs that explain most variance (e.g., top 50 PCs)</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>explained_var_miRNA <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_miRNA)<span class="sc">$</span>importance[<span class="dv">2</span>, ]  <span class="co"># Cumulative variance explained</span></span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>num_pcs_miRNA <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(<span class="fu">cumsum</span>(explained_var_miRNA) <span class="sc">&gt;</span> <span class="fl">0.95</span>))  <span class="co"># Keep PCs that explain 95% variance</span></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>miRNA_pcs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_miRNA<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span>num_pcs_miRNA])  <span class="co"># Extract selected PCs</span></span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(miRNA_pcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 40 20</code></pre>
<p>20 PCs summarize 95% of the explained variance in miRNA expression</p>
</section>
<section id="wgbs" class="level2">
<h2 class="anchored" data-anchor-id="wgbs">4.4 WGBS</h2>
<div class="sourceCode" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on WGBS CpG matrix</span></span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>pca_wgbs <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(vsd_wgbs), <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top PCs that explain most variance (e.g., top 50 PCs)</span></span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a>explained_var_wgbs <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_wgbs)<span class="sc">$</span>importance[<span class="dv">2</span>, ]  <span class="co"># Cumulative variance explained</span></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>num_pcs_wgbs <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(<span class="fu">cumsum</span>(explained_var_wgbs) <span class="sc">&gt;</span> <span class="fl">0.95</span>))  <span class="co"># Keep PCs that explain 95% variance</span></span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-8"><a href="#cb73-8" aria-hidden="true" tabindex="-1"></a>wgbs_pcs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_wgbs<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span>num_pcs_wgbs])  <span class="co"># Extract selected PCs</span></span>
<span id="cb73-9"><a href="#cb73-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(wgbs_pcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 40 30</code></pre>
<p>30 PCs summarize 95% of the explained variance in methylation</p>
</section>
<section id="host_afdw" class="level2">
<h2 class="anchored" data-anchor-id="host_afdw">4.5 Host_AFDW</h2>
<div class="sourceCode" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on gene set matrix</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>pca_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(vsd_Host_AFDW), <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top PCs that explain most variance (e.g., top 50 PCs)</span></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>explained_var_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_Host_AFDW)<span class="sc">$</span>importance[<span class="dv">2</span>, ]  <span class="co"># Cumulative variance explained</span></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>num_pcs_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(<span class="fu">cumsum</span>(explained_var_Host_AFDW) <span class="sc">&gt;</span> <span class="fl">0.95</span>))  <span class="co"># Keep PCs that explain 95% cumulative variance</span></span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>Host_AFDW_pcs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_Host_AFDW<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span>num_pcs_Host_AFDW])  <span class="co"># Extract selected PCs</span></span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Host_AFDW_pcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 40 29</code></pre>
<p>29 PCs summarize 95% of the explained variance in genes associated with host biomass (Host AFDW)</p>
</section>
<section id="am" class="level2">
<h2 class="anchored" data-anchor-id="am">4.6 Am</h2>
<p>Reduce dimensionality</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on gene set expression matrix</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>pca_Am <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(vsd_Am), <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top PCs that explain most variance (e.g., top 50 PCs)</span></span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>explained_var_Am <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_Am)<span class="sc">$</span>importance[<span class="dv">2</span>, ]  <span class="co"># Cumulative variance explained</span></span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>num_pcs_Am <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(<span class="fu">cumsum</span>(explained_var_Am) <span class="sc">&gt;</span> <span class="fl">0.95</span>))  <span class="co"># Keep PCs that explain 95% cumulative variance</span></span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>Am_pcs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_Am<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span>num_pcs_Am])  <span class="co"># Extract selected PCs</span></span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Am_pcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 40 30</code></pre>
<p>30 PCs summarize 95% of the explained variance in genes associated with symbiont photosynthesis (Am)</p>
</section>
<section id="atp-production-go-terms" class="level2">
<h2 class="anchored" data-anchor-id="atp-production-go-terms">4.7 ATP production GO terms</h2>
<p>Reduce dimensionality</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on gene set expression matrix</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>pca_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(vsd_ATP_production_GO), <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top PCs that explain most variance (e.g., top 50 PCs)</span></span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>explained_var_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_ATP_prod_GO)<span class="sc">$</span>importance[<span class="dv">2</span>, ]  <span class="co"># Cumulative variance explained</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>num_pcs_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(<span class="fu">cumsum</span>(explained_var_ATP_prod_GO) <span class="sc">&gt;</span> <span class="fl">0.95</span>))  <span class="co"># Keep PCs that explain 95% cumulative variance</span></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>ATP_prod_GO_pcs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_ATP_prod_GO<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span>num_pcs_ATP_prod_GO])  <span class="co"># Extract selected PCs</span></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ATP_prod_GO_pcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 40 11</code></pre>
<p>11 PCs summarize 95% of the explained variance in genes annotated with ATP production GO terms</p>
</section>
<section id="energy-usagestorage-go-terms" class="level2">
<h2 class="anchored" data-anchor-id="energy-usagestorage-go-terms">4.8 Energy usage/storage GO terms</h2>
<p>Reduce dimensionality</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform PCA on gene set expression matrix</span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>pca_energy_GO <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(vsd_energy_GO), <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top PCs that explain most variance (e.g., top 50 PCs)</span></span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>explained_var_energy_GO <span class="ot">&lt;-</span> <span class="fu">summary</span>(pca_energy_GO)<span class="sc">$</span>importance[<span class="dv">2</span>, ]  <span class="co"># Cumulative variance explained</span></span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>num_pcs_energy_GO <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(<span class="fu">cumsum</span>(explained_var_energy_GO) <span class="sc">&gt;</span> <span class="fl">0.95</span>))  <span class="co"># Keep PCs that explain 95% cumulative variance</span></span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>energy_GO_pcs <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca_energy_GO<span class="sc">$</span>x[, <span class="dv">1</span><span class="sc">:</span>num_pcs_energy_GO])  <span class="co"># Extract selected PCs</span></span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-10"><a href="#cb81-10" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(energy_GO_pcs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 40 27</code></pre>
<p>27 PCs summarize 95% of the explained variance in genes annotated with ATP production GO terms</p>
</section>
</section>
<section id="merge-predictor-features" class="level1">
<h1>5 Merge predictor features</h1>
<section id="merge-reduced-datasets" class="level2">
<h2 class="anchored" data-anchor-id="merge-reduced-datasets">5.1 Merge reduced datasets</h2>
<p>I want to use miRNA, lncRNA, and methylation (WGBS) as predictors for gene expression, so I need to merge the features of these three data sets.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Note which dataset each lncRNA and WGBS PC is associated with</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(lncRNA_pcs) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">colnames</span>(lncRNA_pcs), <span class="st">"_lncRNA"</span>)</span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(miRNA_pcs) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">colnames</span>(miRNA_pcs), <span class="st">"_miRNA"</span>)</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(wgbs_pcs) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">colnames</span>(wgbs_pcs), <span class="st">"_WGBS"</span>)</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Triple check that all three data frames have sample names in the same order</span></span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(lncRNA_pcs), <span class="fu">rownames</span>(wgbs_pcs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(lncRNA_pcs), <span class="fu">rownames</span>(miRNA_pcs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Bind (stack dataframes horizontally)</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>full_pred <span class="ot">&lt;-</span> <span class="fu">cbind</span>(lncRNA_pcs, miRNA_pcs, wgbs_pcs)</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(full_pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##             PC1_lncRNA PC2_lncRNA PC3_lncRNA PC4_lncRNA  PC5_lncRNA PC6_lncRNA
## ACR-225-TP1 -16.652946  73.328741  -4.900662   23.63265 -15.5434633 18.6993987
## ACR-225-TP2 -64.415273 -39.989770  25.849109   31.36694  -2.7418550 16.8817935
## ACR-225-TP3   7.426632   3.836323   2.613068  -18.79839  -2.1512861  8.4294017
## ACR-225-TP4 -44.525340 -22.405730 -26.727351  -12.11801   9.4110284  5.1441162
## ACR-229-TP1 -56.226873   1.285908  -4.624646  -17.85891   0.4800111  0.9084579
## ACR-229-TP2 -23.046364  25.786839  21.772068  -18.73687  -7.1031312 -0.7275392
##             PC7_lncRNA PC8_lncRNA PC9_lncRNA PC10_lncRNA PC11_lncRNA
## ACR-225-TP1  -6.606146   3.907170 -24.668541    2.700613  -19.570934
## ACR-225-TP2  -6.297825  10.732377  42.589155   24.811017  -26.751535
## ACR-225-TP3  -1.467871 -17.039753  27.992788  -71.853887  -42.845320
## ACR-225-TP4  18.329047  -4.863639   6.499017  -61.577003  -25.414126
## ACR-229-TP1  25.162031 -37.787739  -4.995677   10.091432    5.796247
## ACR-229-TP2   9.430113 -63.615657  14.032926   20.235658   -1.191159
##             PC12_lncRNA PC13_lncRNA PC14_lncRNA PC15_lncRNA PC16_lncRNA
## ACR-225-TP1   14.521305   -2.262914  -41.957968    5.820290   8.3988888
## ACR-225-TP2   23.931986    6.942925    1.163011   -9.992572   8.1105349
## ACR-225-TP3  -13.762339    1.103723   -5.710279   -5.183437  -3.1364670
## ACR-225-TP4  -11.896056   22.104165    2.714092    2.601415   0.1158628
## ACR-229-TP1   -9.434515    6.262263   -1.934178    7.166601   7.3463125
## ACR-229-TP2  -12.910408   -6.829857    2.837974   18.804798  10.0055494
##             PC17_lncRNA PC18_lncRNA PC19_lncRNA PC20_lncRNA PC21_lncRNA
## ACR-225-TP1  -11.414071   -8.144753   -7.129114   -5.766714   0.3039167
## ACR-225-TP2    8.257330  -12.779865    2.255193   14.444535  -3.5473589
## ACR-225-TP3    5.373520    4.439530    4.525741   -6.271834   2.2093061
## ACR-225-TP4   -7.026531   -6.954813   -7.076966   10.860342  -3.9469359
## ACR-229-TP1    3.321744    4.570894    3.865441    6.583016  -1.3653023
## ACR-229-TP2   -7.968622   12.486084    7.597807    5.465330  -3.8226527
##             PC22_lncRNA PC23_lncRNA  PC24_lncRNA PC25_lncRNA PC26_lncRNA
## ACR-225-TP1  3.33621967   6.2624463   7.12577766  -1.1624725  -2.4803030
## ACR-225-TP2 10.36360595  -2.2710427   0.09666598   0.8515688  -0.3120913
## ACR-225-TP3  1.10178240  -3.6705225 -10.74023891  14.7092812 -31.1962830
## ACR-225-TP4 -0.01032393   9.5566511   8.46526275 -15.0129144  38.2439731
## ACR-229-TP1  9.56966429   0.5830298   4.66792421  -5.1182417   0.3595885
## ACR-229-TP2  0.79390943  15.1861516  15.35476257  -4.6161473   8.6759452
##             PC27_lncRNA PC28_lncRNA PC29_lncRNA PC30_lncRNA PC31_lncRNA
## ACR-225-TP1   -3.354632   -1.019635  -3.4401091    1.043338    1.360794
## ACR-225-TP2    1.365050   -1.387810   0.7383812   -1.806439   -1.272689
## ACR-225-TP3    7.629537   25.437233   5.4302405   13.490468   -3.264919
## ACR-225-TP4   -7.281209  -28.557006  -3.6872263  -16.900518    5.777254
## ACR-229-TP1   -3.147646   -6.305934  11.2241848    3.291629   -5.355760
## ACR-229-TP2    6.612615    5.732498  32.6675771   -2.783756    1.239237
##             PC1_miRNA   PC2_miRNA  PC3_miRNA   PC4_miRNA   PC5_miRNA  PC6_miRNA
## ACR-225-TP1  2.798740 -1.60103790  0.3902826 -2.78416979  2.89850841 -1.9126175
## ACR-225-TP2 -2.006672  2.35384472  3.3094108  0.68396712  1.00642619 -2.7188868
## ACR-225-TP3  3.402678  0.03989579  2.7920820 -0.98471667 -0.04969853  0.5269023
## ACR-225-TP4  1.743162 -0.74508972  1.7601194 -1.62529700  2.43570856 -1.1716437
## ACR-229-TP1 -1.684434 -3.73740125 -0.7669801 -2.64001620  2.58499410 -1.9776859
## ACR-229-TP2 -3.660540 -1.78878170  1.2071028 -0.01069494  0.49089503 -0.8033196
##              PC7_miRNA  PC8_miRNA   PC9_miRNA PC10_miRNA  PC11_miRNA PC12_miRNA
## ACR-225-TP1  0.4024505 -1.7868230  1.17633110 -0.8139795  0.84528773 -0.5757136
## ACR-225-TP2 -0.2493649 -3.5015635 -0.01719534 -0.3481687 -1.19223681 -0.5321468
## ACR-225-TP3  3.1542881 -3.1946856  0.81511665 -2.1479615  0.02257447 -0.3650203
## ACR-225-TP4 -0.6827463 -1.1612550  1.71987368 -1.2097857 -0.69841499 -0.4210997
## ACR-229-TP1  1.3091331  0.6514011 -1.91405432  0.4606718  1.11677702 -1.7419935
## ACR-229-TP2  1.2042832  1.1837645 -2.59581592  0.8979452 -0.30985309  1.1755465
##             PC13_miRNA PC14_miRNA PC15_miRNA  PC16_miRNA  PC17_miRNA PC18_miRNA
## ACR-225-TP1  0.2554555 -0.1297357 -0.4883509  1.38110641  0.16749476 -0.5965978
## ACR-225-TP2 -0.1981750 -0.9609219 -0.1396524 -1.67384952  1.62178054 -0.4466742
## ACR-225-TP3 -1.5374296  0.6401065 -0.5767027 -0.68668664 -1.05804473 -0.3272770
## ACR-225-TP4  1.8524932  0.6855648 -0.5112841  0.07760664 -0.18793380  0.9265410
## ACR-229-TP1  0.8215141  1.3028887  0.4432213 -0.30498022  0.09989403  0.2487335
## ACR-229-TP2 -0.6203428  0.6040737 -0.4276845 -1.61714671  0.63347585 -0.5473325
##              PC19_miRNA  PC20_miRNA   PC1_WGBS   PC2_WGBS   PC3_WGBS   PC4_WGBS
## ACR-225-TP1 -0.17349545  0.03586059 -47.371386 -7.4281666 -15.911669 -1.9801404
## ACR-225-TP2 -0.89143615  0.46667292   3.763531 -8.1177745   5.056844 -0.5106781
## ACR-225-TP3  0.34320992 -0.41047389   3.784780 -7.2719899   6.449627  2.5534316
## ACR-225-TP4  0.37380836 -0.74593072   3.149372 -6.2894599   3.109440  3.4838126
## ACR-229-TP1 -0.39021763 -0.16141186   4.620443  0.7863283  -1.764767 -5.1060003
## ACR-229-TP2  0.08006026  0.19192730   4.614269 -0.2214307  -3.036966 -4.6587985
##             PC5_WGBS    PC6_WGBS  PC7_WGBS    PC8_WGBS   PC9_WGBS  PC10_WGBS
## ACR-225-TP1 4.817743  -1.2987551  2.268259  -0.5752089  2.1437705 -0.1523134
## ACR-225-TP2 1.632951 -10.3927327  4.563777  -4.8349365  1.9806328 -6.6701787
## ACR-225-TP3 6.117338  -5.9982021  4.682503 -13.7502728  7.4823613  1.5927329
## ACR-225-TP4 2.596055  -7.8141716  4.917917  -9.3518752  2.0215227  0.8340144
## ACR-229-TP1 3.500710   0.7757339 12.436617   5.2231857 -2.6032216  4.1532102
## ACR-229-TP2 3.681378  -0.4584829 10.458070   5.2523072 -0.6308566 -0.2433186
##              PC11_WGBS  PC12_WGBS  PC13_WGBS  PC14_WGBS  PC15_WGBS  PC16_WGBS
## ACR-225-TP1  1.1999889 -0.2133351 -0.2022223  1.1702454 -0.3240623 -0.4613926
## ACR-225-TP2 -2.0992700  3.7117384 -2.7824001  1.8365265 -3.3164335 -1.6319654
## ACR-225-TP3 -2.0633762 -5.4812552  3.0632119 -2.9768045 -2.4804053 -1.7654961
## ACR-225-TP4 -0.2796565  5.7409197 -0.9415802  1.2707304  7.0430354  2.5553601
## ACR-229-TP1 -0.2271934 -0.7732254  0.2543079  2.5133914  0.4451549  4.8256273
## ACR-229-TP2  1.6555274 -0.6643908  1.0330861 -0.2804019  1.0525279  0.4234749
##              PC17_WGBS   PC18_WGBS  PC19_WGBS PC20_WGBS  PC21_WGBS  PC22_WGBS
## ACR-225-TP1  0.2556796  0.04848013  0.1379598  0.163844  0.4163461 -0.2478828
## ACR-225-TP2  4.0691392  6.21452411 -5.3580039 -2.696985  1.7539712  1.0939080
## ACR-225-TP3  2.3378453 -6.08085670  2.2829177  3.437195 -0.2178858  3.0976112
## ACR-225-TP4 -2.2242573  2.59441480  3.1965907 -1.718827 -1.3134491 -6.1534216
## ACR-229-TP1  1.9901032  1.68799711 -2.4572699  4.012089 -1.8905085  1.9925271
## ACR-229-TP2  1.3819023  0.49627020  0.2958164 -1.487058  2.5702578  1.6684459
##               PC23_WGBS  PC24_WGBS  PC25_WGBS   PC26_WGBS   PC27_WGBS
## ACR-225-TP1 -0.01477822 -0.4024602  0.2823349  0.09620127  0.19132873
## ACR-225-TP2 -2.83840621  0.3542538  2.0758895  1.60663820  0.05388139
## ACR-225-TP3 -0.23924425 -0.4659744 -0.3007130  0.26198619 -0.23169049
## ACR-225-TP4  2.94984959  0.9442884 -2.2553056 -0.51859589 -0.02708711
## ACR-229-TP1  2.35415772 -2.9270652  0.4592204  2.00858242 -0.73890630
## ACR-229-TP2  0.25382606  1.4273242 -0.6150447 -2.84497287  0.11903672
##               PC28_WGBS   PC29_WGBS  PC30_WGBS
## ACR-225-TP1  0.02296127  0.31068945 -0.3600702
## ACR-225-TP2  2.45735398 -0.01454505  0.6784057
## ACR-225-TP3 -1.85458312 -0.53013267 -0.3520069
## ACR-225-TP4  0.20395801 -0.80341258 -1.4896686
## ACR-229-TP1  0.24333281 -2.07909037 -2.5953092
## ACR-229-TP2 -3.70723373 -2.71780970  3.1118242</code></pre>
</section>
<section id="merge-raw-datsets" class="level2">
<h2 class="anchored" data-anchor-id="merge-raw-datsets">5.2 Merge raw datsets</h2>
<p>I also want to try using individiual miRNA/lncRNA/CpGs as predictors, instead of the reduced PCs. Let’s also create a merged dataset that contains the variance stabilized counts for all miRNA, lncRNA, and CpGs.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Triple check that all three data frames have sample names in the same order</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">colnames</span>(vsd_lncRNA), <span class="fu">colnames</span>(vsd_miRNA))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">colnames</span>(vsd_lncRNA), <span class="fu">colnames</span>(vsd_wgbs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Bind (stack dataframes vertically, so that they match by column/sample)</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>full_pred_counts <span class="ot">&lt;-</span> <span class="fu">rbind</span>(vsd_lncRNA, vsd_miRNA, vsd_wgbs)</span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform so that samples are on rows and features are in columns</span></span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a>full_pred_counts <span class="ot">&lt;-</span> <span class="fu">t</span>(full_pred_counts)</span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-7"><a href="#cb93-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(full_pred_counts)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1]    40 16560</code></pre>
<p>Ok, now we can run the model!</p>
<p>The below code uses both predictors (miRNA+lncRNA+CpGs) and responses (genes) that have been reduced using PCs.</p>
</section>
</section>
<section id="all-genes-1" class="level1">
<h1>7 All genes</h1>
<p>Let’s start by just throwing all our genes in.</p>
<p>miRNA + lncRNA + methylation as predictors of expression for all genes</p>
<section id="the-model" class="level2">
<h2 class="anchored" data-anchor-id="the-model">7.1 The model</h2>
<div class="sourceCode" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure sample matching between gene and miRNA+lncRNA+methylation PCs</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(genes_pcs), <span class="fu">rownames</span>(full_pred))</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>genes_pcs <span class="ot">&lt;-</span> genes_pcs[common_samples, ]</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>full_pred <span class="ot">&lt;-</span> full_pred[common_samples, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Train elastic models to predict gene expression PCs from miRNA+lncRNA+methylation PCs.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>train_models <span class="ot">&lt;-</span> <span class="cf">function</span>(response_pcs, predictor_pcs) {</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>  models <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (pc <span class="cf">in</span> <span class="fu">colnames</span>(response_pcs)) {</span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> response_pcs[[pc]]  <span class="co"># Gene expression PC</span></span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>    X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(predictor_pcs)  <span class="co"># miRNA+lncRNA+methylation PCs as predictors</span></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Train elastic net model (alpha = 0.5 for mix of LASSO &amp; Ridge)</span></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>    model <span class="ot">&lt;-</span> <span class="fu">cv.glmnet</span>(X, y, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a>    models[[pc]] <span class="ot">&lt;-</span> model</span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(models)</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression PCs from miRNA+lncRNA+methylation PCs</span></span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a>models_all <span class="ot">&lt;-</span> <span class="fu">train_models</span>(genes_pcs, full_pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance <span class="ot">&lt;-</span> <span class="cf">function</span>(models) {</span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>  importance_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(models, <span class="cf">function</span>(model) {</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>    coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Convert to regular matrix &amp; remove intercept</span></span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Convert to data frame</span></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>    coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">as.numeric</span>(coefs))</span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(coefs_df)</span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine feature importance across all predicted gene PCs</span></span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>  importance_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(importance_list) <span class="sc">%&gt;%</span></span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span></span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">MeanImportance =</span> <span class="fu">mean</span>(<span class="fu">abs</span>(Importance)), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(MeanImportance))</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(importance_df)</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>feature_importance_all <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_all)</span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_all, <span class="dv">20</span>)  <span class="co"># Top 20 predictive epigenetic features</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 2
##    Feature    MeanImportance
##    &lt;chr&gt;               &lt;dbl&gt;
##  1 PC16_miRNA          1.03 
##  2 PC18_miRNA          0.895
##  3 PC9_miRNA           0.687
##  4 PC17_miRNA          0.626
##  5 PC14_miRNA          0.618
##  6 PC11_miRNA          0.614
##  7 PC15_miRNA          0.600
##  8 PC7_miRNA           0.595
##  9 PC12_miRNA          0.586
## 10 PC10_miRNA          0.577
## 11 PC19_miRNA          0.576
## 12 PC20_miRNA          0.562
## 13 PC5_miRNA           0.448
## 14 PC6_miRNA           0.442
## 15 PC13_miRNA          0.438
## 16 PC4_miRNA           0.418
## 17 PC2_miRNA           0.410
## 18 PC8_miRNA           0.397
## 19 PC3_miRNA           0.365
## 20 PC1_miRNA           0.327</code></pre>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>evaluate_model_performance <span class="ot">&lt;-</span> <span class="cf">function</span>(models, response_pcs, predictor_pcs) {</span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC =</span> <span class="fu">colnames</span>(response_pcs), <span class="at">R2 =</span> <span class="cn">NA</span>)</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (pc <span class="cf">in</span> <span class="fu">colnames</span>(response_pcs)) {</span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> response_pcs[[pc]]</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>    X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(predictor_pcs)</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>    model <span class="ot">&lt;-</span> models[[pc]]</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>    preds <span class="ot">&lt;-</span> <span class="fu">predict</span>(model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>    R2 <span class="ot">&lt;-</span> <span class="fu">cor</span>(y, preds)<span class="sc">^</span><span class="dv">2</span>  <span class="co"># R-squared metric</span></span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a>    results[results<span class="sc">$</span>PC <span class="sc">==</span> pc, <span class="st">"R2"</span>] <span class="ot">&lt;-</span> R2</span>
<span id="cb99-13"><a href="#cb99-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb99-14"><a href="#cb99-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-15"><a href="#cb99-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(results)</span>
<span id="cb99-16"><a href="#cb99-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb99-17"><a href="#cb99-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-18"><a href="#cb99-18" aria-hidden="true" tabindex="-1"></a>performance_results_all <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_all, genes_pcs, full_pred)</span>
<span id="cb99-19"><a href="#cb99-19" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_all<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
##  0.1811  0.8549  0.9812  0.8504  0.9984  0.9992       4</code></pre>
</section>
<section id="results" class="level2">
<h2 class="anchored" data-anchor-id="results">7.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top 20 predictive features</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a>top_features_all <span class="ot">&lt;-</span> feature_importance_all <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">20</span>, MeanImportance)</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-4"><a href="#cb101-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb101-5"><a href="#cb101-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_all, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb101-6"><a href="#cb101-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb101-7"><a href="#cb101-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb101-8"><a href="#cb101-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb101-9"><a href="#cb101-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Top 20 Predictive Epigenetic Features"</span>,</span>
<span id="cb101-10"><a href="#cb101-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb101-11"><a href="#cb101-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-35-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_all, <span class="fu">aes</span>(<span class="at">x =</span> PC, <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_all<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Gene Expression PCs"</span>,</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene Expression PC"</span>,</span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Removed 4 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-36-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean explained variance</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(<span class="sc">!</span><span class="fu">is.na</span>(performance_results_all<span class="sc">$</span>R2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 0.8666667</code></pre>
<p>View components associated with PCs (need to fix this code to work with the hybrid predictive input)</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # Get the PCA rotation (loadings) matrix from each PCA</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="co"># lncRNA_loadings &lt;- pca_lncRNA$rotation  # Each column corresponds to a PC</span></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="co"># miRNA_loadings &lt;- pca_miRNA$rotation  # Each column corresponds to a PC</span></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="co"># wgbs_loadings &lt;- pca_wgbs$rotation  # Each column corresponds to a PC</span></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # Identify the top predictive PCs (from feature importance)</span></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a><span class="co"># all_top_predictive_pcs &lt;- feature_importance_all$Feature[1:5]  # Select top 5 most predictive PCs</span></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="co"># # Extract the loadings for those PCs</span></span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a><span class="co"># top_loadings &lt;- wgbs_loadings[, top_predictive_pcs, drop = FALSE]</span></span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a><span class="co"># # Convert to data frame and reshape for plotting</span></span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a><span class="co"># top_loadings_df &lt;- as.data.frame(top_loadings) %&gt;%</span></span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   rownames_to_column(var = "CpG") %&gt;%</span></span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   pivot_longer(-CpG, names_to = "Methylation_PC", values_to = "Loading")</span></span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a><span class="co"># # View top CpGs contributing most to each PC</span></span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a><span class="co"># top_cpgs &lt;- top_loadings_df %&gt;%</span></span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(Methylation_PC) %&gt;%</span></span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   arrange(desc(abs(Loading))) %&gt;%</span></span>
<span id="cb106-21"><a href="#cb106-21" aria-hidden="true" tabindex="-1"></a><span class="co">#   slice_head(n = 20)  # Select top 10 CpGs per PC</span></span>
<span id="cb106-22"><a href="#cb106-22" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb106-23"><a href="#cb106-23" aria-hidden="true" tabindex="-1"></a><span class="co"># print(top_cpgs)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>View top 20 CpGs associated with PC9 (the most important PC)</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># print(top_cpgs%&gt;%filter(Methylation_PC=="PC9"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(top_cpgs, aes(x = reorder(CpG, abs(Loading)), y = Loading, fill = Methylation_PC)) +</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_bar(stat = "identity") +</span></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   coord_flip() +  </span></span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme_minimal() +</span></span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   labs(title = "Top CpGs Contributing to Most Predictive Methylation PCs",</span></span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a><span class="co">#        x = "CpG Site",</span></span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a><span class="co">#        y = "Loading Strength") +</span></span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   facet_grid(~Methylation_PC, scales = "free_y")  # Separate plots for each PC</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>View predicted vs actual gene expression values to evaluate model.</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a gene expression PC to visualize (e.g., the most predictable one)</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>best_pc <span class="ot">&lt;-</span> performance_results_all<span class="sc">$</span>PC[<span class="fu">which.max</span>(performance_results_all<span class="sc">$</span>R2)]</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>actual_values <span class="ot">&lt;-</span> genes_pcs[[best_pc]]</span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a>predicted_values <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_all[[best_pc]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame</span></span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a>prediction_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Actual =</span> actual_values,</span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Predicted =</span> predicted_values</span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb109-13"><a href="#cb109-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-14"><a href="#cb109-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot with regression line</span></span>
<span id="cb109-15"><a href="#cb109-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(prediction_df, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb109-16"><a href="#cb109-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb109-17"><a href="#cb109-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb109-18"><a href="#cb109-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb109-19"><a href="#cb109-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, best_pc),</span>
<span id="cb109-20"><a href="#cb109-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb109-21"><a href="#cb109-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb109-22"><a href="#cb109-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values), <span class="at">y =</span> <span class="fu">max</span>(predicted_values), </span>
<span id="cb109-23"><a href="#cb109-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_all<span class="sc">$</span>R2, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb109-24"><a href="#cb109-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-41-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_all, <span class="fu">aes</span>(<span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">fill =</span> <span class="st">"lightblue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of Predictive Performance (R²) Across PCs"</span>,</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Removed 4 rows containing non-finite outside the scale range
## (`stat_boxplot()`).</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-42-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute correlation between actual and predicted gene expression PCs</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> <span class="fu">sapply</span>(models_all, <span class="cf">function</span>(m) <span class="fu">predict</span>(m, <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>))</span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure matrices are the same size</span></span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> predicted_matrix[, <span class="fu">colnames</span>(genes_pcs), drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Align columns</span></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a><span class="co"># remove 0 variance columns</span></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>predicted_matrix <span class="ot">&lt;-</span> predicted_matrix[, <span class="fu">apply</span>(predicted_matrix, <span class="dv">2</span>, <span class="cf">function</span>(col) <span class="fu">sd</span>(col, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>)]</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute correlation matrix, handling missing values</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>cor_matrix <span class="ot">&lt;-</span> <span class="fu">cor</span>(predicted_matrix, <span class="fu">as.matrix</span>(genes_pcs), <span class="at">use =</span> <span class="st">"complete.obs"</span>)</span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace NA or Inf values with zero</span></span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>cor_matrix[<span class="fu">is.na</span>(cor_matrix) <span class="sc">|</span> <span class="fu">is.infinite</span>(cor_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span>  </span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(cor_matrix, <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">100</span>),</span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"Correlation Between Actual and Predicted Gene Expression PCs"</span>,</span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-43-1.png" class="img-fluid"><!-- --></p>
</section>
</section>
<section id="host-biomass-host_afdw" class="level1">
<h1>8 Host biomass (Host_AFDW)</h1>
<section id="the-model-1" class="level2">
<h2 class="anchored" data-anchor-id="the-model-1">8.1 The model</h2>
<p>Train elastic models to predict gene expression PCs from miRNA expression.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>train_models <span class="ot">&lt;-</span> <span class="cf">function</span>(response_pcs, predictor_pcs) {</span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>  models <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (pc <span class="cf">in</span> <span class="fu">colnames</span>(response_pcs)) {</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> response_pcs[[pc]]  <span class="co"># Gene expression PC</span></span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>    X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(predictor_pcs)  <span class="co"># miRNA+lncRNA+methylation PCs as predictors</span></span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Train elastic net model (alpha = 0.5 for mix of LASSO &amp; Ridge)</span></span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>    model <span class="ot">&lt;-</span> <span class="fu">cv.glmnet</span>(X, y, <span class="at">alpha =</span> <span class="fl">0.5</span>)</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb114-11"><a href="#cb114-11" aria-hidden="true" tabindex="-1"></a>    models[[pc]] <span class="ot">&lt;-</span> model</span>
<span id="cb114-12"><a href="#cb114-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb114-13"><a href="#cb114-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-14"><a href="#cb114-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(models)</span>
<span id="cb114-15"><a href="#cb114-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb114-16"><a href="#cb114-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-17"><a href="#cb114-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression PCs from miRNA+lncRNA+methylation PCs</span></span>
<span id="cb114-18"><a href="#cb114-18" aria-hidden="true" tabindex="-1"></a>models_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">train_models</span>(Host_AFDW_pcs, full_pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance <span class="ot">&lt;-</span> <span class="cf">function</span>(models) {</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>  importance_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(models, <span class="cf">function</span>(model) {</span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>    coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Convert to regular matrix &amp; remove intercept</span></span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Convert to data frame</span></span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>    coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">as.numeric</span>(coefs))</span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(coefs_df)</span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine feature importance across all predicted gene PCs</span></span>
<span id="cb115-12"><a href="#cb115-12" aria-hidden="true" tabindex="-1"></a>  importance_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(importance_list) <span class="sc">%&gt;%</span></span>
<span id="cb115-13"><a href="#cb115-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(Feature) <span class="sc">%&gt;%</span></span>
<span id="cb115-14"><a href="#cb115-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarize</span>(<span class="at">MeanImportance =</span> <span class="fu">mean</span>(<span class="fu">abs</span>(Importance)), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb115-15"><a href="#cb115-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(MeanImportance))</span>
<span id="cb115-16"><a href="#cb115-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-17"><a href="#cb115-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(importance_df)</span>
<span id="cb115-18"><a href="#cb115-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb115-19"><a href="#cb115-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-20"><a href="#cb115-20" aria-hidden="true" tabindex="-1"></a>feature_importance_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_Host_AFDW)</span>
<span id="cb115-21"><a href="#cb115-21" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_Host_AFDW, <span class="dv">20</span>)  <span class="co"># Top predictive features</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 2
##    Feature    MeanImportance
##    &lt;chr&gt;               &lt;dbl&gt;
##  1 PC20_miRNA         0.0906
##  2 PC18_miRNA         0.0771
##  3 PC17_miRNA         0.0762
##  4 PC19_miRNA         0.0760
##  5 PC16_miRNA         0.0753
##  6 PC8_miRNA          0.0689
##  7 PC3_miRNA          0.0674
##  8 PC13_miRNA         0.0672
##  9 PC11_miRNA         0.0669
## 10 PC7_miRNA          0.0663
## 11 PC9_miRNA          0.0581
## 12 PC27_WGBS          0.0527
## 13 PC26_WGBS          0.0516
## 14 PC2_miRNA          0.0505
## 15 PC14_miRNA         0.0471
## 16 PC5_miRNA          0.0463
## 17 PC12_miRNA         0.0440
## 18 PC21_WGBS          0.0430
## 19 PC13_WGBS          0.0430
## 20 PC11_WGBS          0.0354</code></pre>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>evaluate_model_performance <span class="ot">&lt;-</span> <span class="cf">function</span>(models, response_pcs, predictor_pcs) {</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>  results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC =</span> <span class="fu">colnames</span>(response_pcs), <span class="at">R2 =</span> <span class="cn">NA</span>)</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (pc <span class="cf">in</span> <span class="fu">colnames</span>(response_pcs)) {</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> response_pcs[[pc]]</span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>    X <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(predictor_pcs)</span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>    model <span class="ot">&lt;-</span> models[[pc]]</span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>    preds <span class="ot">&lt;-</span> <span class="fu">predict</span>(model, X, <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a>    R2 <span class="ot">&lt;-</span> <span class="fu">cor</span>(y, preds)<span class="sc">^</span><span class="dv">2</span>  <span class="co"># R-squared metric</span></span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>    results[results<span class="sc">$</span>PC <span class="sc">==</span> pc, <span class="st">"R2"</span>] <span class="ot">&lt;-</span> R2</span>
<span id="cb117-13"><a href="#cb117-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb117-14"><a href="#cb117-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-15"><a href="#cb117-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(results)</span>
<span id="cb117-16"><a href="#cb117-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb117-17"><a href="#cb117-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-18"><a href="#cb117-18" aria-hidden="true" tabindex="-1"></a>performance_results_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_Host_AFDW, Host_AFDW_pcs, full_pred)</span>
<span id="cb117-19"><a href="#cb117-19" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_Host_AFDW<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
##  0.2360  0.8370  0.9701  0.8500  0.9960  0.9997       7</code></pre>
</section>
<section id="results-1" class="level2">
<h2 class="anchored" data-anchor-id="results-1">8.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top predictive features</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="co"># few enough miRNA that we can show all</span></span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>top_features_Host_AFDW <span class="ot">&lt;-</span> feature_importance_Host_AFDW <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">50</span>, MeanImportance)</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_Host_AFDW, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb119-7"><a href="#cb119-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb119-8"><a href="#cb119-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb119-9"><a href="#cb119-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb119-10"><a href="#cb119-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"lncRNA, miRNA, and methylation as predictive features"</span>,</span>
<span id="cb119-11"><a href="#cb119-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Features"</span>,</span>
<span id="cb119-12"><a href="#cb119-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-47-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_Host_AFDW, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(PC), <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_Host_AFDW<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Gene Expression PCs"</span>,</span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene Expression PC"</span>,</span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb120-8"><a href="#cb120-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Removed 7 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-48-1.png" class="img-fluid"><!-- --></p>
<p>View components associated with gene PCs</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the PCA rotation (loadings) matrix from the original gene PCA</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>loadings_Host_AFDW <span class="ot">&lt;-</span> pca_Host_AFDW<span class="sc">$</span>rotation  <span class="co"># Each column corresponds to a PC</span></span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and reshape for plotting</span></span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>loadings_Host_AFDW_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(loadings_Host_AFDW) <span class="sc">%&gt;%</span></span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>gene, <span class="at">names_to =</span> <span class="st">"Host_AFDW_PC"</span>, <span class="at">values_to =</span> <span class="st">"Loading"</span>)</span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a><span class="co"># View top genes contributing most to each PC</span></span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a>top_genes_Host_AFDW <span class="ot">&lt;-</span> loadings_Host_AFDW_df <span class="sc">%&gt;%</span></span>
<span id="cb122-11"><a href="#cb122-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Host_AFDW_PC) <span class="sc">%&gt;%</span></span>
<span id="cb122-12"><a href="#cb122-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(Loading))) <span class="sc">%&gt;%</span></span>
<span id="cb122-13"><a href="#cb122-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)  <span class="co"># Select top 20 genes per PC</span></span>
<span id="cb122-14"><a href="#cb122-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-15"><a href="#cb122-15" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_Host_AFDW)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 800 × 3
## # Groups:   Host_AFDW_PC [40]
##    gene       Host_AFDW_PC Loading
##    &lt;chr&gt;      &lt;chr&gt;          &lt;dbl&gt;
##  1 FUN_010504 PC1           0.0593
##  2 FUN_033160 PC1           0.0587
##  3 FUN_042402 PC1           0.0580
##  4 FUN_026248 PC1           0.0579
##  5 FUN_010505 PC1           0.0578
##  6 FUN_030089 PC1           0.0576
##  7 FUN_027385 PC1           0.0575
##  8 FUN_013949 PC1           0.0573
##  9 FUN_008069 PC1           0.0572
## 10 FUN_036450 PC1           0.0570
## # ℹ 790 more rows</code></pre>
<p>View predicted vs actual gene expression values to evaluate model.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a gene expression PC to visualize (e.g., the most predictable one)</span></span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>best_pc_Host_AFDW <span class="ot">&lt;-</span> performance_results_Host_AFDW<span class="sc">$</span>PC[<span class="fu">which.max</span>(performance_results_Host_AFDW<span class="sc">$</span>R2)]</span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>actual_values_Host_AFDW <span class="ot">&lt;-</span> Host_AFDW_pcs[[best_pc_Host_AFDW]]</span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>predicted_values_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_Host_AFDW[[best_pc_Host_AFDW]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame</span></span>
<span id="cb124-9"><a href="#cb124-9" aria-hidden="true" tabindex="-1"></a>prediction_df_Host_AFDW <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb124-10"><a href="#cb124-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Actual =</span> actual_values_Host_AFDW,</span>
<span id="cb124-11"><a href="#cb124-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Predicted =</span> predicted_values_Host_AFDW</span>
<span id="cb124-12"><a href="#cb124-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb124-13"><a href="#cb124-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-14"><a href="#cb124-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot with regression line</span></span>
<span id="cb124-15"><a href="#cb124-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(prediction_df_Host_AFDW, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb124-16"><a href="#cb124-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb124-17"><a href="#cb124-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb124-18"><a href="#cb124-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb124-19"><a href="#cb124-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, best_pc_Host_AFDW),</span>
<span id="cb124-20"><a href="#cb124-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb124-21"><a href="#cb124-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb124-22"><a href="#cb124-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values_Host_AFDW), <span class="at">y =</span> <span class="fu">max</span>(predicted_values_Host_AFDW), </span>
<span id="cb124-23"><a href="#cb124-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_Host_AFDW<span class="sc">$</span>R2, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb124-24"><a href="#cb124-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-50-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="do">## `geom_smooth()` using formula = 'y ~ x'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>View top 20 genes associated with the PC with the highest R^2</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_Host_AFDW<span class="sc">%&gt;%</span><span class="fu">filter</span>(Host_AFDW_PC<span class="sc">==</span>best_pc_Host_AFDW))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 3
## # Groups:   Host_AFDW_PC [1]
##    gene       Host_AFDW_PC Loading
##    &lt;chr&gt;      &lt;chr&gt;          &lt;dbl&gt;
##  1 FUN_032266 PC11         -0.121 
##  2 FUN_001078 PC11         -0.120 
##  3 FUN_002256 PC11         -0.110 
##  4 FUN_036908 PC11         -0.105 
##  5 FUN_038819 PC11         -0.102 
##  6 FUN_041309 PC11          0.0960
##  7 FUN_011431 PC11          0.0955
##  8 FUN_026266 PC11          0.0935
##  9 FUN_035522 PC11          0.0903
## 10 FUN_004295 PC11         -0.0870
## 11 FUN_001845 PC11          0.0857
## 12 FUN_035669 PC11         -0.0854
## 13 FUN_008411 PC11          0.0853
## 14 FUN_035580 PC11          0.0849
## 15 FUN_037480 PC11         -0.0848
## 16 FUN_021024 PC11          0.0840
## 17 FUN_012009 PC11          0.0831
## 18 FUN_029326 PC11          0.0812
## 19 FUN_029077 PC11          0.0791
## 20 FUN_027127 PC11          0.0789</code></pre>
<p>Plot performance for all PCs</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all PCs with R^2 values above 0.75</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>all_pcs_Host_AFDW <span class="ot">&lt;-</span> performance_results_Host_AFDW <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;</span> <span class="fl">0.75</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(PC)</span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pc <span class="cf">in</span> all_pcs_Host_AFDW) {</span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a>  actual_values <span class="ot">&lt;-</span> Host_AFDW_pcs[[pc]]</span>
<span id="cb129-8"><a href="#cb129-8" aria-hidden="true" tabindex="-1"></a>  predicted_values <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_Host_AFDW[[pc]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb129-9"><a href="#cb129-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-10"><a href="#cb129-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create data frame</span></span>
<span id="cb129-11"><a href="#cb129-11" aria-hidden="true" tabindex="-1"></a>  prediction_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb129-12"><a href="#cb129-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">Actual =</span> actual_values,</span>
<span id="cb129-13"><a href="#cb129-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">Predicted =</span> predicted_values</span>
<span id="cb129-14"><a href="#cb129-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb129-15"><a href="#cb129-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-16"><a href="#cb129-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scatter plot with regression line</span></span>
<span id="cb129-17"><a href="#cb129-17" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(prediction_df, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb129-18"><a href="#cb129-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb129-19"><a href="#cb129-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb129-20"><a href="#cb129-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb129-21"><a href="#cb129-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, pc),</span>
<span id="cb129-22"><a href="#cb129-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb129-23"><a href="#cb129-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb129-24"><a href="#cb129-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values), <span class="at">y =</span> <span class="fu">max</span>(predicted_values), </span>
<span id="cb129-25"><a href="#cb129-25" aria-hidden="true" tabindex="-1"></a>             <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_Host_AFDW[performance_results_Host_AFDW<span class="sc">$</span>PC<span class="sc">==</span>pc,<span class="dv">2</span>], <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb129-26"><a href="#cb129-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb129-27"><a href="#cb129-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb129-28"><a href="#cb129-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb129-29"><a href="#cb129-29" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-1.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-2.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-3.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-4.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-5.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-6.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-7.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-8.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-9.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-10.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-11.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-12.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-13.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-14.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-15.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-16.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-52-17.png" class="img-fluid"><!-- --></p>
<p>We can also look at which feature(s) contributed most to predicting gene PCs of interest</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance_for_pc <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb147-2"><a href="#cb147-2" aria-hidden="true" tabindex="-1"></a>  coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Remove intercept</span></span>
<span id="cb147-3"><a href="#cb147-3" aria-hidden="true" tabindex="-1"></a>  coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">abs</span>(<span class="fu">as.numeric</span>(coefs)))</span>
<span id="cb147-4"><a href="#cb147-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-5"><a href="#cb147-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(coefs_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Importance)))  <span class="co"># Sort by importance</span></span>
<span id="cb147-6"><a href="#cb147-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb147-7"><a href="#cb147-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-8"><a href="#cb147-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pc <span class="cf">in</span> all_pcs_Host_AFDW) {</span>
<span id="cb147-9"><a href="#cb147-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract feature importance for the most predictable PC</span></span>
<span id="cb147-10"><a href="#cb147-10" aria-hidden="true" tabindex="-1"></a>  best_pc_model <span class="ot">&lt;-</span> models_Host_AFDW[[pc]]</span>
<span id="cb147-11"><a href="#cb147-11" aria-hidden="true" tabindex="-1"></a>  best_pc_importance <span class="ot">&lt;-</span> <span class="fu">get_feature_importance_for_pc</span>(best_pc_model)</span>
<span id="cb147-12"><a href="#cb147-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-13"><a href="#cb147-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot top most important feature for predicting this PC</span></span>
<span id="cb147-14"><a href="#cb147-14" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(best_pc_importance <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, Importance), <span class="at">y =</span> Importance)) <span class="sc">+</span></span>
<span id="cb147-15"><a href="#cb147-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb147-16"><a href="#cb147-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb147-17"><a href="#cb147-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb147-18"><a href="#cb147-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top miRNA/lncRNA/methylation Predictors for"</span>, pc),</span>
<span id="cb147-19"><a href="#cb147-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb147-20"><a href="#cb147-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Importance Score"</span>)</span>
<span id="cb147-21"><a href="#cb147-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb147-22"><a href="#cb147-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb147-23"><a href="#cb147-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-1.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-2.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-3.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-4.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-5.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-6.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-7.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-8.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-9.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-10.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-11.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-12.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-13.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-14.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-15.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-16.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-53-17.png" class="img-fluid"><!-- --></p>
</section>
</section>
<section id="symbiont-photsynthesis-am" class="level1">
<h1>9 Symbiont photsynthesis (Am)</h1>
<section id="the-model-2" class="level2">
<h2 class="anchored" data-anchor-id="the-model-2">9.1 The model</h2>
<p>Train elastic models to predict gene expression PCs from miRNA expression</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression PCs from miRNA+lncRNA+methylation</span></span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>models_Am <span class="ot">&lt;-</span> <span class="fu">train_models</span>(Am_pcs, full_pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a>feature_importance_Am <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_Am)</span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_Am, <span class="dv">20</span>)  <span class="co"># Top predictive feature</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 2
##    Feature    MeanImportance
##    &lt;chr&gt;               &lt;dbl&gt;
##  1 PC20_miRNA         0.327 
##  2 PC10_miRNA         0.210 
##  3 PC18_miRNA         0.201 
##  4 PC16_miRNA         0.197 
##  5 PC12_miRNA         0.172 
##  6 PC9_miRNA          0.172 
##  7 PC19_miRNA         0.162 
##  8 PC14_miRNA         0.155 
##  9 PC5_miRNA          0.150 
## 10 PC7_miRNA          0.146 
## 11 PC11_miRNA         0.138 
## 12 PC23_WGBS          0.127 
## 13 PC13_miRNA         0.125 
## 14 PC17_miRNA         0.124 
## 15 PC30_WGBS          0.121 
## 16 PC19_WGBS          0.116 
## 17 PC1_miRNA          0.104 
## 18 PC15_miRNA         0.103 
## 19 PC8_miRNA          0.0926
## 20 PC6_miRNA          0.0910</code></pre>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>performance_results_Am <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_Am, Am_pcs, full_pred)</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_Am<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
##  0.2020  0.8016  0.9304  0.8191  0.9905  0.9995       5</code></pre>
</section>
<section id="results-2" class="level2">
<h2 class="anchored" data-anchor-id="results-2">9.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top predictive features</span></span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>top_features_Am <span class="ot">&lt;-</span> feature_importance_Am <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">50</span>, MeanImportance)</span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_Am, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb153-6"><a href="#cb153-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb153-7"><a href="#cb153-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb153-8"><a href="#cb153-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb153-9"><a href="#cb153-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"miRNA/lncRNA/methylation as Predictive Features"</span>,</span>
<span id="cb153-10"><a href="#cb153-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb153-11"><a href="#cb153-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-57-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_Am, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(PC), <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_Am<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Gene Expression PCs"</span>,</span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene Expression PC"</span>,</span>
<span id="cb154-7"><a href="#cb154-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb154-8"><a href="#cb154-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Removed 5 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-58-1.png" class="img-fluid"><!-- --></p>
<p>View components associated with gene PCs</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the PCA rotation (loadings) matrix from the original gene PCA</span></span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a>loadings_Am <span class="ot">&lt;-</span> pca_Am<span class="sc">$</span>rotation  <span class="co"># Each column corresponds to a PC</span></span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-4"><a href="#cb156-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and reshape for plotting</span></span>
<span id="cb156-5"><a href="#cb156-5" aria-hidden="true" tabindex="-1"></a>loadings_Am_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(loadings_Am) <span class="sc">%&gt;%</span></span>
<span id="cb156-6"><a href="#cb156-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb156-7"><a href="#cb156-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>gene, <span class="at">names_to =</span> <span class="st">"Am_PC"</span>, <span class="at">values_to =</span> <span class="st">"Loading"</span>)</span>
<span id="cb156-8"><a href="#cb156-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-9"><a href="#cb156-9" aria-hidden="true" tabindex="-1"></a><span class="co"># View top genes contributing most to each PC</span></span>
<span id="cb156-10"><a href="#cb156-10" aria-hidden="true" tabindex="-1"></a>top_genes_Am <span class="ot">&lt;-</span> loadings_Am_df <span class="sc">%&gt;%</span></span>
<span id="cb156-11"><a href="#cb156-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Am_PC) <span class="sc">%&gt;%</span></span>
<span id="cb156-12"><a href="#cb156-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(Loading))) <span class="sc">%&gt;%</span></span>
<span id="cb156-13"><a href="#cb156-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)  <span class="co"># Select top 20 genes per PC</span></span>
<span id="cb156-14"><a href="#cb156-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-15"><a href="#cb156-15" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_Am)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 800 × 3
## # Groups:   Am_PC [40]
##    gene       Am_PC Loading
##    &lt;chr&gt;      &lt;chr&gt;   &lt;dbl&gt;
##  1 FUN_011681 PC1   -0.0250
##  2 FUN_040949 PC1    0.0244
##  3 FUN_027962 PC1    0.0244
##  4 FUN_023373 PC1    0.0244
##  5 FUN_000239 PC1    0.0244
##  6 FUN_014926 PC1    0.0241
##  7 FUN_001784 PC1   -0.0240
##  8 FUN_016798 PC1   -0.0239
##  9 FUN_023033 PC1   -0.0239
## 10 FUN_016083 PC1   -0.0238
## # ℹ 790 more rows</code></pre>
<p>View predicted vs actual gene expression values to evaluate model.</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a gene expression PC to visualize (e.g., the most predictable one)</span></span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>best_pc_Am <span class="ot">&lt;-</span> performance_results_Am<span class="sc">$</span>PC[<span class="fu">which.max</span>(performance_results_Am<span class="sc">$</span>R2)]</span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a>actual_values_Am <span class="ot">&lt;-</span> Am_pcs[[best_pc_Am]]</span>
<span id="cb158-6"><a href="#cb158-6" aria-hidden="true" tabindex="-1"></a>predicted_values_Am <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_Am[[best_pc_Am]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb158-7"><a href="#cb158-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-8"><a href="#cb158-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame</span></span>
<span id="cb158-9"><a href="#cb158-9" aria-hidden="true" tabindex="-1"></a>prediction_df_Am <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb158-10"><a href="#cb158-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Actual =</span> actual_values_Am,</span>
<span id="cb158-11"><a href="#cb158-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Predicted =</span> predicted_values_Am</span>
<span id="cb158-12"><a href="#cb158-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb158-13"><a href="#cb158-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-14"><a href="#cb158-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot with regression line</span></span>
<span id="cb158-15"><a href="#cb158-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(prediction_df_Am, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb158-16"><a href="#cb158-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb158-17"><a href="#cb158-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb158-18"><a href="#cb158-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb158-19"><a href="#cb158-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, best_pc_Am),</span>
<span id="cb158-20"><a href="#cb158-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb158-21"><a href="#cb158-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb158-22"><a href="#cb158-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values_Am), <span class="at">y =</span> <span class="fu">max</span>(predicted_values_Am), </span>
<span id="cb158-23"><a href="#cb158-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_Am<span class="sc">$</span>R2, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb158-24"><a href="#cb158-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-60-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="do">## `geom_smooth()` using formula = 'y ~ x'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>View top 20 genes associated with the PC with the highest R^2</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_Am<span class="sc">%&gt;%</span><span class="fu">filter</span>(Am_PC<span class="sc">==</span>best_pc_Am))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 3
## # Groups:   Am_PC [1]
##    gene       Am_PC Loading
##    &lt;chr&gt;      &lt;chr&gt;   &lt;dbl&gt;
##  1 FUN_005990 PC18   0.0578
##  2 FUN_005864 PC18  -0.0489
##  3 FUN_025766 PC18  -0.0441
##  4 FUN_040041 PC18   0.0440
##  5 FUN_017855 PC18  -0.0439
##  6 FUN_043319 PC18   0.0431
##  7 FUN_001332 PC18   0.0428
##  8 FUN_008277 PC18  -0.0424
##  9 FUN_015347 PC18  -0.0424
## 10 FUN_001477 PC18   0.0422
## 11 FUN_026571 PC18   0.0421
## 12 FUN_027848 PC18  -0.0417
## 13 FUN_005860 PC18  -0.0414
## 14 FUN_024487 PC18  -0.0405
## 15 FUN_033692 PC18  -0.0403
## 16 FUN_041284 PC18  -0.0403
## 17 FUN_000995 PC18  -0.0402
## 18 FUN_001762 PC18  -0.0402
## 19 FUN_014703 PC18   0.0395
## 20 FUN_032441 PC18   0.0395</code></pre>
<p>Plot performance for all PCs</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all PCs with R^2 values above 0.75</span></span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>all_pcs_Am <span class="ot">&lt;-</span> performance_results_Am <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;</span> <span class="fl">0.75</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(PC)</span>
<span id="cb163-3"><a href="#cb163-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb163-4"><a href="#cb163-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pc <span class="cf">in</span> all_pcs_Am) {</span>
<span id="cb163-5"><a href="#cb163-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb163-6"><a href="#cb163-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb163-7"><a href="#cb163-7" aria-hidden="true" tabindex="-1"></a>  actual_values <span class="ot">&lt;-</span> Am_pcs[[pc]]</span>
<span id="cb163-8"><a href="#cb163-8" aria-hidden="true" tabindex="-1"></a>  predicted_values <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_Am[[pc]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb163-9"><a href="#cb163-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb163-10"><a href="#cb163-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create data frame</span></span>
<span id="cb163-11"><a href="#cb163-11" aria-hidden="true" tabindex="-1"></a>  prediction_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb163-12"><a href="#cb163-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">Actual =</span> actual_values,</span>
<span id="cb163-13"><a href="#cb163-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">Predicted =</span> predicted_values</span>
<span id="cb163-14"><a href="#cb163-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb163-15"><a href="#cb163-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb163-16"><a href="#cb163-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scatter plot with regression line</span></span>
<span id="cb163-17"><a href="#cb163-17" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(prediction_df, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb163-18"><a href="#cb163-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb163-19"><a href="#cb163-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb163-20"><a href="#cb163-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb163-21"><a href="#cb163-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, pc),</span>
<span id="cb163-22"><a href="#cb163-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb163-23"><a href="#cb163-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb163-24"><a href="#cb163-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values), <span class="at">y =</span> <span class="fu">max</span>(predicted_values), </span>
<span id="cb163-25"><a href="#cb163-25" aria-hidden="true" tabindex="-1"></a>             <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_Am[performance_results_Am<span class="sc">$</span>PC<span class="sc">==</span>pc,<span class="dv">2</span>], <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb163-26"><a href="#cb163-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb163-27"><a href="#cb163-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb163-28"><a href="#cb163-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb163-29"><a href="#cb163-29" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-1.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-2.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-3.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-4.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-5.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-6.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-7.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-8.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-9.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-10.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-11.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-12.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-13.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-14.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-15.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-16.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-17.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-18.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-62-19.png" class="img-fluid"><!-- --></p>
<p>We can also look at which miRNA(s) contributed most to predicting gene PCs of interest</p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance_for_pc <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb183-2"><a href="#cb183-2" aria-hidden="true" tabindex="-1"></a>  coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Remove intercept</span></span>
<span id="cb183-3"><a href="#cb183-3" aria-hidden="true" tabindex="-1"></a>  coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">abs</span>(<span class="fu">as.numeric</span>(coefs)))</span>
<span id="cb183-4"><a href="#cb183-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb183-5"><a href="#cb183-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(coefs_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Importance)))  <span class="co"># Sort by importance</span></span>
<span id="cb183-6"><a href="#cb183-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb183-7"><a href="#cb183-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-8"><a href="#cb183-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pc <span class="cf">in</span> all_pcs_Am) {</span>
<span id="cb183-9"><a href="#cb183-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract feature importance for the most predictable PC</span></span>
<span id="cb183-10"><a href="#cb183-10" aria-hidden="true" tabindex="-1"></a>  best_pc_model <span class="ot">&lt;-</span> models_Am[[pc]]</span>
<span id="cb183-11"><a href="#cb183-11" aria-hidden="true" tabindex="-1"></a>  best_pc_importance <span class="ot">&lt;-</span> <span class="fu">get_feature_importance_for_pc</span>(best_pc_model)</span>
<span id="cb183-12"><a href="#cb183-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb183-13"><a href="#cb183-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot top most important miRNA/lncRNA/methylation for predicting this PC</span></span>
<span id="cb183-14"><a href="#cb183-14" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(best_pc_importance <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, Importance), <span class="at">y =</span> Importance)) <span class="sc">+</span></span>
<span id="cb183-15"><a href="#cb183-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb183-16"><a href="#cb183-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb183-17"><a href="#cb183-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb183-18"><a href="#cb183-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top miRNA/lncRNA/methylation Predictors for"</span>, pc),</span>
<span id="cb183-19"><a href="#cb183-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Features"</span>,</span>
<span id="cb183-20"><a href="#cb183-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Importance Score"</span>)</span>
<span id="cb183-21"><a href="#cb183-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb183-22"><a href="#cb183-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb183-23"><a href="#cb183-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-1.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-2.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-3.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-4.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-5.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-6.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-7.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-8.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-9.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-10.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-11.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-12.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-13.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-14.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-15.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-16.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-17.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-18.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-63-19.png" class="img-fluid"><!-- --></p>
</section>
</section>
<section id="atp-production-go-terms-1" class="level1">
<h1>10 ATP production (GO terms)</h1>
<section id="the-model-3" class="level2">
<h2 class="anchored" data-anchor-id="the-model-3">10.1 The model</h2>
<p>Train elastic models to predict gene expression PCs from miRNA expression</p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression PCs from miRNA/lncRNA/methylation PCs</span></span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a>models_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">train_models</span>(ATP_prod_GO_pcs, full_pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a>feature_importance_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_ATP_prod_GO)</span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_ATP_prod_GO, <span class="dv">20</span>)  <span class="co"># Top predictive feature PCs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 2
##    Feature    MeanImportance
##    &lt;chr&gt;               &lt;dbl&gt;
##  1 PC19_miRNA        0.0437 
##  2 PC12_miRNA        0.0380 
##  3 PC5_miRNA         0.0298 
##  4 PC18_miRNA        0.0251 
##  5 PC20_miRNA        0.0236 
##  6 PC30_WGBS         0.0224 
##  7 PC16_miRNA        0.0223 
##  8 PC1_miRNA         0.0209 
##  9 PC2_miRNA         0.0169 
## 10 PC24_WGBS         0.0154 
## 11 PC12_WGBS         0.0145 
## 12 PC23_WGBS         0.0143 
## 13 PC7_miRNA         0.0139 
## 14 PC3_miRNA         0.0137 
## 15 PC4_miRNA         0.00962
## 16 PC11_WGBS         0.00828
## 17 PC15_miRNA        0.00820
## 18 PC21_WGBS         0.00786
## 19 PC9_WGBS          0.00761
## 20 PC14_miRNA        0.00712</code></pre>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a>performance_results_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_ATP_prod_GO, ATP_prod_GO_pcs, full_pred)</span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_ATP_prod_GO<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
##  0.1980  0.7518  0.9177  0.7903  0.9953  0.9994       3</code></pre>
</section>
<section id="results-3" class="level2">
<h2 class="anchored" data-anchor-id="results-3">10.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top predictive features</span></span>
<span id="cb189-2"><a href="#cb189-2" aria-hidden="true" tabindex="-1"></a>top_features_ATP_prod_GO <span class="ot">&lt;-</span> feature_importance_ATP_prod_GO <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">50</span>, MeanImportance)</span>
<span id="cb189-3"><a href="#cb189-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb189-4"><a href="#cb189-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb189-5"><a href="#cb189-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_ATP_prod_GO, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb189-6"><a href="#cb189-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb189-7"><a href="#cb189-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb189-8"><a href="#cb189-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb189-9"><a href="#cb189-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"miRNA/lncRNA/methylation as Predictive Features"</span>,</span>
<span id="cb189-10"><a href="#cb189-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb189-11"><a href="#cb189-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-67-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb190-1"><a href="#cb190-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_ATP_prod_GO, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(PC), <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb190-2"><a href="#cb190-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb190-3"><a href="#cb190-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_ATP_prod_GO<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb190-4"><a href="#cb190-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb190-5"><a href="#cb190-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Gene Expression PCs"</span>,</span>
<span id="cb190-6"><a href="#cb190-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene Expression PC"</span>,</span>
<span id="cb190-7"><a href="#cb190-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb190-8"><a href="#cb190-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Removed 3 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-68-1.png" class="img-fluid"><!-- --></p>
<p>View components associated with gene PCs</p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the PCA rotation (loadings) matrix from the original gene PCA</span></span>
<span id="cb192-2"><a href="#cb192-2" aria-hidden="true" tabindex="-1"></a>loadings_ATP_prod_GO <span class="ot">&lt;-</span> pca_ATP_prod_GO<span class="sc">$</span>rotation  <span class="co"># Each column corresponds to a PC</span></span>
<span id="cb192-3"><a href="#cb192-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-4"><a href="#cb192-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and reshape for plotting</span></span>
<span id="cb192-5"><a href="#cb192-5" aria-hidden="true" tabindex="-1"></a>loadings_ATP_prod_GO_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(loadings_ATP_prod_GO) <span class="sc">%&gt;%</span></span>
<span id="cb192-6"><a href="#cb192-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb192-7"><a href="#cb192-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>gene, <span class="at">names_to =</span> <span class="st">"ATP_prod_GO_PC"</span>, <span class="at">values_to =</span> <span class="st">"Loading"</span>)</span>
<span id="cb192-8"><a href="#cb192-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-9"><a href="#cb192-9" aria-hidden="true" tabindex="-1"></a><span class="co"># View top genes contributing most to each PC</span></span>
<span id="cb192-10"><a href="#cb192-10" aria-hidden="true" tabindex="-1"></a>top_genes_ATP_prod_GO <span class="ot">&lt;-</span> loadings_ATP_prod_GO_df <span class="sc">%&gt;%</span></span>
<span id="cb192-11"><a href="#cb192-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(ATP_prod_GO_PC) <span class="sc">%&gt;%</span></span>
<span id="cb192-12"><a href="#cb192-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(Loading))) <span class="sc">%&gt;%</span></span>
<span id="cb192-13"><a href="#cb192-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)  <span class="co"># Select top 20 genes per PC</span></span>
<span id="cb192-14"><a href="#cb192-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-15"><a href="#cb192-15" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_ATP_prod_GO)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 440 × 3
## # Groups:   ATP_prod_GO_PC [22]
##    gene       ATP_prod_GO_PC Loading
##    &lt;chr&gt;      &lt;chr&gt;            &lt;dbl&gt;
##  1 FUN_025802 PC1              0.328
##  2 FUN_025367 PC1              0.316
##  3 FUN_000960 PC1              0.313
##  4 FUN_031975 PC1              0.311
##  5 FUN_031686 PC1              0.290
##  6 FUN_014565 PC1              0.277
##  7 FUN_007016 PC1              0.246
##  8 FUN_039808 PC1              0.243
##  9 FUN_038166 PC1              0.242
## 10 FUN_025823 PC1              0.208
## # ℹ 430 more rows</code></pre>
<p>View predicted vs actual gene expression values to evaluate model.</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a gene expression PC to visualize (e.g., the most predictable one)</span></span>
<span id="cb194-2"><a href="#cb194-2" aria-hidden="true" tabindex="-1"></a>best_pc_ATP_prod_GO <span class="ot">&lt;-</span> performance_results_ATP_prod_GO<span class="sc">$</span>PC[<span class="fu">which.max</span>(performance_results_ATP_prod_GO<span class="sc">$</span>R2)]</span>
<span id="cb194-3"><a href="#cb194-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-4"><a href="#cb194-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb194-5"><a href="#cb194-5" aria-hidden="true" tabindex="-1"></a>actual_values_ATP_prod_GO <span class="ot">&lt;-</span> ATP_prod_GO_pcs[[best_pc_ATP_prod_GO]]</span>
<span id="cb194-6"><a href="#cb194-6" aria-hidden="true" tabindex="-1"></a>predicted_values_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_ATP_prod_GO[[best_pc_ATP_prod_GO]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb194-7"><a href="#cb194-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-8"><a href="#cb194-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame</span></span>
<span id="cb194-9"><a href="#cb194-9" aria-hidden="true" tabindex="-1"></a>prediction_df_ATP_prod_GO <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb194-10"><a href="#cb194-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Actual =</span> actual_values_ATP_prod_GO,</span>
<span id="cb194-11"><a href="#cb194-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Predicted =</span> predicted_values_ATP_prod_GO</span>
<span id="cb194-12"><a href="#cb194-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb194-13"><a href="#cb194-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb194-14"><a href="#cb194-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot with regression line</span></span>
<span id="cb194-15"><a href="#cb194-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(prediction_df_ATP_prod_GO, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb194-16"><a href="#cb194-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb194-17"><a href="#cb194-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb194-18"><a href="#cb194-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb194-19"><a href="#cb194-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, best_pc_ATP_prod_GO),</span>
<span id="cb194-20"><a href="#cb194-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb194-21"><a href="#cb194-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb194-22"><a href="#cb194-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values_ATP_prod_GO), <span class="at">y =</span> <span class="fu">max</span>(predicted_values_ATP_prod_GO), </span>
<span id="cb194-23"><a href="#cb194-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_ATP_prod_GO<span class="sc">$</span>R2, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb194-24"><a href="#cb194-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-70-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb196-1"><a href="#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="do">## `geom_smooth()` using formula = 'y ~ x'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>View top 20 genes associated with the PC with the highest R^2</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb197-1"><a href="#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_ATP_prod_GO<span class="sc">%&gt;%</span><span class="fu">filter</span>(ATP_prod_GO_PC<span class="sc">==</span>best_pc_ATP_prod_GO))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 3
## # Groups:   ATP_prod_GO_PC [1]
##    gene       ATP_prod_GO_PC Loading
##    &lt;chr&gt;      &lt;chr&gt;            &lt;dbl&gt;
##  1 FUN_014563 PC4             0.371 
##  2 FUN_040783 PC4            -0.361 
##  3 FUN_015065 PC4             0.328 
##  4 FUN_014565 PC4             0.319 
##  5 FUN_038688 PC4             0.309 
##  6 FUN_007016 PC4             0.258 
##  7 FUN_038727 PC4            -0.249 
##  8 FUN_032701 PC4            -0.236 
##  9 FUN_014564 PC4             0.236 
## 10 FUN_033885 PC4             0.209 
## 11 FUN_028263 PC4            -0.197 
## 12 FUN_038166 PC4            -0.183 
## 13 FUN_025802 PC4            -0.137 
## 14 FUN_038738 PC4             0.122 
## 15 FUN_009532 PC4            -0.109 
## 16 FUN_025367 PC4            -0.0883
## 17 FUN_031686 PC4             0.0695
## 18 FUN_039808 PC4            -0.0625
## 19 FUN_036898 PC4            -0.0576
## 20 FUN_000960 PC4            -0.0524</code></pre>
<p>Plot performance for all PCs</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb199-1"><a href="#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all PCs with R^2 values above line in plot</span></span>
<span id="cb199-2"><a href="#cb199-2" aria-hidden="true" tabindex="-1"></a>all_pcs_ATP_prod_GO <span class="ot">&lt;-</span> performance_results_ATP_prod_GO <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;</span> <span class="fl">0.75</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(PC)</span>
<span id="cb199-3"><a href="#cb199-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-4"><a href="#cb199-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pc <span class="cf">in</span> all_pcs_ATP_prod_GO) {</span>
<span id="cb199-5"><a href="#cb199-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb199-6"><a href="#cb199-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb199-7"><a href="#cb199-7" aria-hidden="true" tabindex="-1"></a>  actual_values <span class="ot">&lt;-</span> ATP_prod_GO_pcs[[pc]]</span>
<span id="cb199-8"><a href="#cb199-8" aria-hidden="true" tabindex="-1"></a>  predicted_values <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_ATP_prod_GO[[pc]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb199-9"><a href="#cb199-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb199-10"><a href="#cb199-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create data frame</span></span>
<span id="cb199-11"><a href="#cb199-11" aria-hidden="true" tabindex="-1"></a>  prediction_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb199-12"><a href="#cb199-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">Actual =</span> actual_values,</span>
<span id="cb199-13"><a href="#cb199-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">Predicted =</span> predicted_values</span>
<span id="cb199-14"><a href="#cb199-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb199-15"><a href="#cb199-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb199-16"><a href="#cb199-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scatter plot with regression line</span></span>
<span id="cb199-17"><a href="#cb199-17" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(prediction_df, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb199-18"><a href="#cb199-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb199-19"><a href="#cb199-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb199-20"><a href="#cb199-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb199-21"><a href="#cb199-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, pc),</span>
<span id="cb199-22"><a href="#cb199-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb199-23"><a href="#cb199-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb199-24"><a href="#cb199-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values), <span class="at">y =</span> <span class="fu">max</span>(predicted_values), </span>
<span id="cb199-25"><a href="#cb199-25" aria-hidden="true" tabindex="-1"></a>             <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_ATP_prod_GO[performance_results_ATP_prod_GO<span class="sc">$</span>PC<span class="sc">==</span>pc,<span class="dv">2</span>], <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb199-26"><a href="#cb199-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb199-27"><a href="#cb199-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb199-28"><a href="#cb199-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb199-29"><a href="#cb199-29" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-72-1.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-72-2.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-72-3.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-72-4.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-72-5.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-72-6.png" class="img-fluid"><!-- --></p>
<p>We can also look at which miRNA(s) contributed most to predicting gene PCs of interest</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance_for_pc <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a>  coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Remove intercept</span></span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a>  coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">abs</span>(<span class="fu">as.numeric</span>(coefs)))</span>
<span id="cb206-4"><a href="#cb206-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb206-5"><a href="#cb206-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(coefs_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Importance)))  <span class="co"># Sort by importance</span></span>
<span id="cb206-6"><a href="#cb206-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb206-7"><a href="#cb206-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb206-8"><a href="#cb206-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pc <span class="cf">in</span> all_pcs_ATP_prod_GO) {</span>
<span id="cb206-9"><a href="#cb206-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract feature importance for the most predictable PC</span></span>
<span id="cb206-10"><a href="#cb206-10" aria-hidden="true" tabindex="-1"></a>  best_pc_model <span class="ot">&lt;-</span> models_ATP_prod_GO[[pc]]</span>
<span id="cb206-11"><a href="#cb206-11" aria-hidden="true" tabindex="-1"></a>  best_pc_importance <span class="ot">&lt;-</span> <span class="fu">get_feature_importance_for_pc</span>(best_pc_model)</span>
<span id="cb206-12"><a href="#cb206-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb206-13"><a href="#cb206-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot top most important miRNA/lncRNA/methylation for predicting this PC</span></span>
<span id="cb206-14"><a href="#cb206-14" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(best_pc_importance <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, Importance), <span class="at">y =</span> Importance)) <span class="sc">+</span></span>
<span id="cb206-15"><a href="#cb206-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb206-16"><a href="#cb206-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb206-17"><a href="#cb206-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb206-18"><a href="#cb206-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top miRNA/lncRNA/methylation Predictors for"</span>, pc),</span>
<span id="cb206-19"><a href="#cb206-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb206-20"><a href="#cb206-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Importance Score"</span>)</span>
<span id="cb206-21"><a href="#cb206-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb206-22"><a href="#cb206-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb206-23"><a href="#cb206-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-73-1.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-73-2.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-73-3.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-73-4.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-73-5.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-73-6.png" class="img-fluid"><!-- --></p>
</section>
</section>
<section id="energy-usagestorage-go-terms-1" class="level1">
<h1>11 Energy Usage/Storage (GO terms)</h1>
<section id="the-model-4" class="level2">
<h2 class="anchored" data-anchor-id="the-model-4">11.1 The model</h2>
<p>Train elastic models to predict gene expression PCs from miRNA expression</p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Train models predicting gene expression PCs from miRNA/lncRNA/methylation PCs</span></span>
<span id="cb207-2"><a href="#cb207-2" aria-hidden="true" tabindex="-1"></a>models_energy_GO <span class="ot">&lt;-</span> <span class="fu">train_models</span>(energy_GO_pcs, full_pred)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Extract feature importance.</p>
<div class="sourceCode" id="cb208"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb208-1"><a href="#cb208-1" aria-hidden="true" tabindex="-1"></a>feature_importance_energy_GO <span class="ot">&lt;-</span> <span class="fu">get_feature_importance</span>(models_energy_GO)</span>
<span id="cb208-2"><a href="#cb208-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(feature_importance_energy_GO, <span class="dv">20</span>)  <span class="co"># Top predictive miRNA/lncRNA/methylation PCs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 2
##    Feature    MeanImportance
##    &lt;chr&gt;               &lt;dbl&gt;
##  1 PC10_miRNA         0.0556
##  2 PC16_miRNA         0.0544
##  3 PC20_miRNA         0.0542
##  4 PC11_miRNA         0.0516
##  5 PC12_miRNA         0.0449
##  6 PC18_miRNA         0.0423
##  7 PC9_miRNA          0.0395
##  8 PC17_miRNA         0.0378
##  9 PC23_WGBS          0.0330
## 10 PC6_miRNA          0.0327
## 11 PC14_miRNA         0.0317
## 12 PC5_miRNA          0.0307
## 13 PC7_miRNA          0.0262
## 14 PC20_WGBS          0.0255
## 15 PC27_WGBS          0.0245
## 16 PC19_miRNA         0.0240
## 17 PC18_WGBS          0.0235
## 18 PC29_WGBS          0.0215
## 19 PC4_miRNA          0.0215
## 20 PC15_miRNA         0.0208</code></pre>
<p>Evaluate performance.</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a>performance_results_energy_GO <span class="ot">&lt;-</span> <span class="fu">evaluate_model_performance</span>(models_energy_GO, energy_GO_pcs, full_pred)</span>
<span id="cb210-2"><a href="#cb210-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(performance_results_energy_GO<span class="sc">$</span>R2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max.    NA's 
##  0.1689  0.6786  0.9219  0.7595  0.9822  0.9996       4</code></pre>
</section>
<section id="results-4" class="level2">
<h2 class="anchored" data-anchor-id="results-4">11.2 Results</h2>
<p>Plot results.</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top predictive features</span></span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a><span class="co"># few enough miRNA that we can show all</span></span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a>top_features_energy_GO <span class="ot">&lt;-</span> feature_importance_energy_GO <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="dv">50</span>, MeanImportance)</span>
<span id="cb212-4"><a href="#cb212-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-5"><a href="#cb212-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb212-6"><a href="#cb212-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(top_features_energy_GO, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, MeanImportance), <span class="at">y =</span> MeanImportance)) <span class="sc">+</span></span>
<span id="cb212-7"><a href="#cb212-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb212-8"><a href="#cb212-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span>  <span class="co"># Flip for readability</span></span>
<span id="cb212-9"><a href="#cb212-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb212-10"><a href="#cb212-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"miRNA/lncRNA/methylation as Predictive Features"</span>,</span>
<span id="cb212-11"><a href="#cb212-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb212-12"><a href="#cb212-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Importance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-77-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(performance_results_energy_GO, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(PC), <span class="at">y =</span> R2)) <span class="sc">+</span></span>
<span id="cb213-2"><a href="#cb213-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb213-3"><a href="#cb213-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">mean</span>(performance_results_energy_GO<span class="sc">$</span>R2, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb213-4"><a href="#cb213-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb213-5"><a href="#cb213-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Model Performance Across Gene Expression PCs"</span>,</span>
<span id="cb213-6"><a href="#cb213-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Gene Expression PC"</span>,</span>
<span id="cb213-7"><a href="#cb213-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"R² (Variance Explained)"</span>) <span class="sc">+</span></span>
<span id="cb213-8"><a href="#cb213-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))  <span class="co"># Rotate labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## Warning: Removed 4 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-78-1.png" class="img-fluid"><!-- --></p>
<p>View components associated with gene PCs</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the PCA rotation (loadings) matrix from the original gene PCA</span></span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a>loadings_energy_GO <span class="ot">&lt;-</span> pca_energy_GO<span class="sc">$</span>rotation  <span class="co"># Each column corresponds to a PC</span></span>
<span id="cb215-3"><a href="#cb215-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-4"><a href="#cb215-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame and reshape for plotting</span></span>
<span id="cb215-5"><a href="#cb215-5" aria-hidden="true" tabindex="-1"></a>loadings_energy_GO_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(loadings_energy_GO) <span class="sc">%&gt;%</span></span>
<span id="cb215-6"><a href="#cb215-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb215-7"><a href="#cb215-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>gene, <span class="at">names_to =</span> <span class="st">"energy_GO_PC"</span>, <span class="at">values_to =</span> <span class="st">"Loading"</span>)</span>
<span id="cb215-8"><a href="#cb215-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-9"><a href="#cb215-9" aria-hidden="true" tabindex="-1"></a><span class="co"># View top genes contributing most to each PC</span></span>
<span id="cb215-10"><a href="#cb215-10" aria-hidden="true" tabindex="-1"></a>top_genes_energy_GO <span class="ot">&lt;-</span> loadings_energy_GO_df <span class="sc">%&gt;%</span></span>
<span id="cb215-11"><a href="#cb215-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(energy_GO_PC) <span class="sc">%&gt;%</span></span>
<span id="cb215-12"><a href="#cb215-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(Loading))) <span class="sc">%&gt;%</span></span>
<span id="cb215-13"><a href="#cb215-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>)  <span class="co"># Select top 20 genes per PC</span></span>
<span id="cb215-14"><a href="#cb215-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb215-15"><a href="#cb215-15" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_energy_GO)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 800 × 3
## # Groups:   energy_GO_PC [40]
##    gene       energy_GO_PC Loading
##    &lt;chr&gt;      &lt;chr&gt;          &lt;dbl&gt;
##  1 FUN_028200 PC1            0.115
##  2 FUN_040444 PC1            0.113
##  3 FUN_017915 PC1            0.112
##  4 FUN_001396 PC1           -0.109
##  5 FUN_026618 PC1           -0.108
##  6 FUN_029673 PC1           -0.107
##  7 FUN_023596 PC1            0.107
##  8 FUN_000370 PC1           -0.106
##  9 FUN_039293 PC1            0.106
## 10 FUN_001160 PC1           -0.105
## # ℹ 790 more rows</code></pre>
<p>View predicted vs actual gene expression values to evaluate model.</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb217-1"><a href="#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose a gene expression PC to visualize (e.g., the most predictable one)</span></span>
<span id="cb217-2"><a href="#cb217-2" aria-hidden="true" tabindex="-1"></a>best_pc_energy_GO <span class="ot">&lt;-</span> performance_results_energy_GO<span class="sc">$</span>PC[<span class="fu">which.max</span>(performance_results_energy_GO<span class="sc">$</span>R2)]</span>
<span id="cb217-3"><a href="#cb217-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-4"><a href="#cb217-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb217-5"><a href="#cb217-5" aria-hidden="true" tabindex="-1"></a>actual_values_energy_GO <span class="ot">&lt;-</span> energy_GO_pcs[[best_pc_energy_GO]]</span>
<span id="cb217-6"><a href="#cb217-6" aria-hidden="true" tabindex="-1"></a>predicted_values_energy_GO <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_energy_GO[[best_pc_energy_GO]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb217-7"><a href="#cb217-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-8"><a href="#cb217-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create data frame</span></span>
<span id="cb217-9"><a href="#cb217-9" aria-hidden="true" tabindex="-1"></a>prediction_df_energy_GO <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb217-10"><a href="#cb217-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">Actual =</span> actual_values_energy_GO,</span>
<span id="cb217-11"><a href="#cb217-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">Predicted =</span> predicted_values_energy_GO</span>
<span id="cb217-12"><a href="#cb217-12" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb217-13"><a href="#cb217-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-14"><a href="#cb217-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Scatter plot with regression line</span></span>
<span id="cb217-15"><a href="#cb217-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(prediction_df_energy_GO, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb217-16"><a href="#cb217-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb217-17"><a href="#cb217-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb217-18"><a href="#cb217-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb217-19"><a href="#cb217-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, best_pc_energy_GO),</span>
<span id="cb217-20"><a href="#cb217-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb217-21"><a href="#cb217-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb217-22"><a href="#cb217-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values_energy_GO), <span class="at">y =</span> <span class="fu">max</span>(predicted_values_energy_GO), </span>
<span id="cb217-23"><a href="#cb217-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_energy_GO<span class="sc">$</span>R2, <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb217-24"><a href="#cb217-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-80-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb219-1"><a href="#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="do">## `geom_smooth()` using formula = 'y ~ x'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>View top 20 genes associated with the PC with the highest R^2</p>
<div class="sourceCode" id="cb220"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb220-1"><a href="#cb220-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(top_genes_energy_GO<span class="sc">%&gt;%</span><span class="fu">filter</span>(energy_GO_PC<span class="sc">==</span>best_pc_energy_GO))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## # A tibble: 20 × 3
## # Groups:   energy_GO_PC [1]
##    gene       energy_GO_PC Loading
##    &lt;chr&gt;      &lt;chr&gt;          &lt;dbl&gt;
##  1 FUN_022041 PC24           0.205
##  2 FUN_037231 PC24          -0.189
##  3 FUN_033720 PC24           0.187
##  4 FUN_041903 PC24          -0.172
##  5 FUN_009346 PC24          -0.163
##  6 FUN_042618 PC24          -0.151
##  7 FUN_012471 PC24          -0.150
##  8 FUN_017849 PC24           0.150
##  9 FUN_041699 PC24          -0.137
## 10 FUN_011931 PC24           0.137
## 11 FUN_011927 PC24           0.132
## 12 FUN_036984 PC24           0.132
## 13 FUN_023089 PC24           0.128
## 14 FUN_012329 PC24           0.126
## 15 FUN_014897 PC24           0.124
## 16 FUN_001205 PC24           0.122
## 17 FUN_004635 PC24           0.121
## 18 FUN_006975 PC24          -0.120
## 19 FUN_025406 PC24          -0.118
## 20 FUN_043629 PC24           0.113</code></pre>
<p>Plot performance for all PCs</p>
<div class="sourceCode" id="cb222"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb222-1"><a href="#cb222-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all PCs with R^2 values above line in plot</span></span>
<span id="cb222-2"><a href="#cb222-2" aria-hidden="true" tabindex="-1"></a>all_pcs_energy_GO <span class="ot">&lt;-</span> performance_results_energy_GO <span class="sc">%&gt;%</span> <span class="fu">filter</span>(R2 <span class="sc">&gt;</span> <span class="fl">0.75</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(PC)</span>
<span id="cb222-3"><a href="#cb222-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb222-4"><a href="#cb222-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pc <span class="cf">in</span> all_pcs_energy_GO) {</span>
<span id="cb222-5"><a href="#cb222-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb222-6"><a href="#cb222-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract actual and predicted values for that PC</span></span>
<span id="cb222-7"><a href="#cb222-7" aria-hidden="true" tabindex="-1"></a>  actual_values <span class="ot">&lt;-</span> energy_GO_pcs[[pc]]</span>
<span id="cb222-8"><a href="#cb222-8" aria-hidden="true" tabindex="-1"></a>  predicted_values <span class="ot">&lt;-</span> <span class="fu">predict</span>(models_energy_GO[[pc]], <span class="fu">as.matrix</span>(full_pred), <span class="at">s =</span> <span class="st">"lambda.min"</span>)</span>
<span id="cb222-9"><a href="#cb222-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb222-10"><a href="#cb222-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create data frame</span></span>
<span id="cb222-11"><a href="#cb222-11" aria-hidden="true" tabindex="-1"></a>  prediction_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb222-12"><a href="#cb222-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">Actual =</span> actual_values,</span>
<span id="cb222-13"><a href="#cb222-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">Predicted =</span> predicted_values</span>
<span id="cb222-14"><a href="#cb222-14" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb222-15"><a href="#cb222-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb222-16"><a href="#cb222-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Scatter plot with regression line</span></span>
<span id="cb222-17"><a href="#cb222-17" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(prediction_df, <span class="fu">aes</span>(<span class="at">x =</span> Actual, <span class="at">y =</span> lambda.min)) <span class="sc">+</span></span>
<span id="cb222-18"><a href="#cb222-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb222-19"><a href="#cb222-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb222-20"><a href="#cb222-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb222-21"><a href="#cb222-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Predicted vs. Actual for"</span>, pc),</span>
<span id="cb222-22"><a href="#cb222-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Actual Gene Expression PC"</span>,</span>
<span id="cb222-23"><a href="#cb222-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Predicted Gene Expression PC"</span>) <span class="sc">+</span></span>
<span id="cb222-24"><a href="#cb222-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="fu">min</span>(actual_values), <span class="at">y =</span> <span class="fu">max</span>(predicted_values), </span>
<span id="cb222-25"><a href="#cb222-25" aria-hidden="true" tabindex="-1"></a>             <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">"R² ="</span>, <span class="fu">round</span>(<span class="fu">max</span>(performance_results_energy_GO[performance_results_energy_GO<span class="sc">$</span>PC<span class="sc">==</span>pc,<span class="dv">2</span>], <span class="at">na.rm=</span><span class="cn">TRUE</span>), <span class="dv">3</span>)), </span>
<span id="cb222-26"><a href="#cb222-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb222-27"><a href="#cb222-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb222-28"><a href="#cb222-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb222-29"><a href="#cb222-29" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-1.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-2.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-3.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-4.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-5.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-6.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-7.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-8.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-9.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-10.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-11.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-12.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-13.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-14.png" class="img-fluid"><!-- --></p>
<pre><code>## `geom_smooth()` using formula = 'y ~ x'</code></pre>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-82-15.png" class="img-fluid"><!-- --></p>
<p>We can also look at which miRNA(s) contributed most to predicting gene PCs of interest</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a>get_feature_importance_for_pc <span class="ot">&lt;-</span> <span class="cf">function</span>(model) {</span>
<span id="cb238-2"><a href="#cb238-2" aria-hidden="true" tabindex="-1"></a>  coefs <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">coef</span>(model, <span class="at">s =</span> <span class="st">"lambda.min"</span>))[<span class="sc">-</span><span class="dv">1</span>, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]  <span class="co"># Remove intercept</span></span>
<span id="cb238-3"><a href="#cb238-3" aria-hidden="true" tabindex="-1"></a>  coefs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Feature =</span> <span class="fu">rownames</span>(coefs), <span class="at">Importance =</span> <span class="fu">abs</span>(<span class="fu">as.numeric</span>(coefs)))</span>
<span id="cb238-4"><a href="#cb238-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-5"><a href="#cb238-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(coefs_df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Importance)))  <span class="co"># Sort by importance</span></span>
<span id="cb238-6"><a href="#cb238-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb238-7"><a href="#cb238-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb238-8"><a href="#cb238-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (pc <span class="cf">in</span> all_pcs_energy_GO) {</span>
<span id="cb238-9"><a href="#cb238-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract feature importance for the most predictable PC</span></span>
<span id="cb238-10"><a href="#cb238-10" aria-hidden="true" tabindex="-1"></a>  best_pc_model <span class="ot">&lt;-</span> models_energy_GO[[pc]]</span>
<span id="cb238-11"><a href="#cb238-11" aria-hidden="true" tabindex="-1"></a>  best_pc_importance <span class="ot">&lt;-</span> <span class="fu">get_feature_importance_for_pc</span>(best_pc_model)</span>
<span id="cb238-12"><a href="#cb238-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-13"><a href="#cb238-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot top most important miRNA/lncRNA/methylation PCs for predicting this PC</span></span>
<span id="cb238-14"><a href="#cb238-14" aria-hidden="true" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(best_pc_importance <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Feature, Importance), <span class="at">y =</span> Importance)) <span class="sc">+</span></span>
<span id="cb238-15"><a href="#cb238-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb238-16"><a href="#cb238-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb238-17"><a href="#cb238-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb238-18"><a href="#cb238-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"Top miRNA/lncRNA/methylation PC Predictors for"</span>, pc),</span>
<span id="cb238-19"><a href="#cb238-19" aria-hidden="true" tabindex="-1"></a>         <span class="at">x =</span> <span class="st">"Feature"</span>,</span>
<span id="cb238-20"><a href="#cb238-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Importance Score"</span>)</span>
<span id="cb238-21"><a href="#cb238-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb238-22"><a href="#cb238-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(plot)</span>
<span id="cb238-23"><a href="#cb238-23" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-1.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-2.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-3.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-4.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-5.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-6.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-7.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-8.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-9.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-10.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-11.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-12.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-13.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-14.png" class="img-fluid"><!-- --><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-83-15.png" class="img-fluid"><!-- --></p>
</section>
</section>
<section id="compare" class="level1">
<h1>12 Compare</h1>
<p>Visualize the relative importance of miRNA in predicting expression for these different gene sets:</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb239-1"><a href="#cb239-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform min-max normalization on the mean importance of miRNA/lncRNA/methylation PC for each group</span></span>
<span id="cb239-2"><a href="#cb239-2" aria-hidden="true" tabindex="-1"></a><span class="co"># This will place all along a 0-1 range for comparison purposes</span></span>
<span id="cb239-3"><a href="#cb239-3" aria-hidden="true" tabindex="-1"></a>normalize <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb239-4"><a href="#cb239-4" aria-hidden="true" tabindex="-1"></a>  (x <span class="sc">-</span> <span class="fu">min</span>(x)) <span class="sc">/</span> (<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x))</span>
<span id="cb239-5"><a href="#cb239-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb239-6"><a href="#cb239-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-7"><a href="#cb239-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize</span></span>
<span id="cb239-8"><a href="#cb239-8" aria-hidden="true" tabindex="-1"></a>top_features_Host_AFDW<span class="sc">$</span>MeanImportance_norm <span class="ot">&lt;-</span> <span class="fu">normalize</span>(top_features_Host_AFDW<span class="sc">$</span>MeanImportance)</span>
<span id="cb239-9"><a href="#cb239-9" aria-hidden="true" tabindex="-1"></a>top_features_Am<span class="sc">$</span>MeanImportance_norm <span class="ot">&lt;-</span> <span class="fu">normalize</span>(top_features_Am<span class="sc">$</span>MeanImportance)</span>
<span id="cb239-10"><a href="#cb239-10" aria-hidden="true" tabindex="-1"></a>top_features_ATP_prod_GO<span class="sc">$</span>MeanImportance_norm <span class="ot">&lt;-</span> <span class="fu">normalize</span>(top_features_ATP_prod_GO<span class="sc">$</span>MeanImportance)</span>
<span id="cb239-11"><a href="#cb239-11" aria-hidden="true" tabindex="-1"></a>top_features_energy_GO<span class="sc">$</span>MeanImportance_norm <span class="ot">&lt;-</span> <span class="fu">normalize</span>(top_features_energy_GO<span class="sc">$</span>MeanImportance)</span>
<span id="cb239-12"><a href="#cb239-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-13"><a href="#cb239-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Add group labels</span></span>
<span id="cb239-14"><a href="#cb239-14" aria-hidden="true" tabindex="-1"></a>top_features_Host_AFDW <span class="ot">&lt;-</span> top_features_Host_AFDW <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"Host_AFDW"</span>)</span>
<span id="cb239-15"><a href="#cb239-15" aria-hidden="true" tabindex="-1"></a>top_features_Am <span class="ot">&lt;-</span> top_features_Am <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"Am"</span>)</span>
<span id="cb239-16"><a href="#cb239-16" aria-hidden="true" tabindex="-1"></a>top_features_ATP_prod_GO <span class="ot">&lt;-</span> top_features_ATP_prod_GO <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"ATP_prod_GO"</span>)</span>
<span id="cb239-17"><a href="#cb239-17" aria-hidden="true" tabindex="-1"></a>top_features_energy_GO <span class="ot">&lt;-</span> top_features_energy_GO <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">"energy_GO"</span>)</span>
<span id="cb239-18"><a href="#cb239-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-19"><a href="#cb239-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Set rows in same order</span></span>
<span id="cb239-20"><a href="#cb239-20" aria-hidden="true" tabindex="-1"></a>top_features_Am <span class="ot">&lt;-</span> top_features_Am[<span class="fu">rownames</span>(top_features_Host_AFDW),]</span>
<span id="cb239-21"><a href="#cb239-21" aria-hidden="true" tabindex="-1"></a>top_features_ATP_prod_GO <span class="ot">&lt;-</span> top_features_ATP_prod_GO[<span class="fu">rownames</span>(top_features_Host_AFDW),]</span>
<span id="cb239-22"><a href="#cb239-22" aria-hidden="true" tabindex="-1"></a>top_features_energy_GO <span class="ot">&lt;-</span> top_features_energy_GO[<span class="fu">rownames</span>(top_features_Host_AFDW),]</span>
<span id="cb239-23"><a href="#cb239-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-24"><a href="#cb239-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine</span></span>
<span id="cb239-25"><a href="#cb239-25" aria-hidden="true" tabindex="-1"></a>all_gene_sets <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(top_features_Host_AFDW, top_features_Am, top_features_ATP_prod_GO, top_features_energy_GO)</span>
<span id="cb239-26"><a href="#cb239-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove raw mean importance</span></span>
<span id="cb239-27"><a href="#cb239-27" aria-hidden="true" tabindex="-1"></a>all_gene_sets <span class="ot">&lt;-</span> all_gene_sets <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">!</span>MeanImportance)</span>
<span id="cb239-28"><a href="#cb239-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-29"><a href="#cb239-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Wide format: rows = miRNAs, columns = groups</span></span>
<span id="cb239-30"><a href="#cb239-30" aria-hidden="true" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> all_gene_sets <span class="sc">%&gt;%</span></span>
<span id="cb239-31"><a href="#cb239-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> group, <span class="at">values_from =</span> MeanImportance_norm)</span>
<span id="cb239-32"><a href="#cb239-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-33"><a href="#cb239-33" aria-hidden="true" tabindex="-1"></a>heatmap_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(heatmap_df)</span>
<span id="cb239-34"><a href="#cb239-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-35"><a href="#cb239-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Melt into long format for ggplot</span></span>
<span id="cb239-36"><a href="#cb239-36" aria-hidden="true" tabindex="-1"></a>heatmap_long <span class="ot">&lt;-</span> <span class="fu">melt</span>(heatmap_df, <span class="at">id.vars =</span> <span class="st">"Feature"</span>)</span>
<span id="cb239-37"><a href="#cb239-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb239-38"><a href="#cb239-38" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(heatmap_long, <span class="fu">aes</span>(<span class="at">x =</span> variable, <span class="at">y =</span> Feature, <span class="at">fill =</span> value)) <span class="sc">+</span></span>
<span id="cb239-39"><a href="#cb239-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb239-40"><a href="#cb239-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb239-41"><a href="#cb239-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb239-42"><a href="#cb239-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Group"</span>, <span class="at">y =</span> <span class="st">"Feature"</span>, <span class="at">fill =</span> <span class="st">"Importance"</span>) <span class="sc">+</span></span>
<span id="cb239-43"><a href="#cb239-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-84-1.png" class="img-fluid"><!-- --></p>
<p>Cluster by miRNA importance</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make Feature column the rownames and convert to matrix</span></span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatmap_df) <span class="ot">&lt;-</span> heatmap_df<span class="sc">$</span>Feature</span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a>heatmap_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(heatmap_df[, <span class="sc">-</span><span class="dv">1</span>])  <span class="co"># Removes the 'Feature' column</span></span>
<span id="cb240-4"><a href="#cb240-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb240-5"><a href="#cb240-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace NAs with 0</span></span>
<span id="cb240-6"><a href="#cb240-6" aria-hidden="true" tabindex="-1"></a>heatmap_matrix[<span class="fu">is.na</span>(heatmap_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb240-7"><a href="#cb240-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb240-8"><a href="#cb240-8" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb240-9"><a href="#cb240-9" aria-hidden="true" tabindex="-1"></a>  heatmap_matrix, </span>
<span id="cb240-10"><a href="#cb240-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>,  <span class="co"># Clustering miRNAs (rows) by similarity in importance</span></span>
<span id="cb240-11"><a href="#cb240-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,  <span class="co"># Clustering groups (columns)</span></span>
<span id="cb240-12"><a href="#cb240-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">scale =</span> <span class="st">"none"</span>,  <span class="co"># No scaling (since data is already normalized)</span></span>
<span id="cb240-13"><a href="#cb240-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">TRUE</span>,  <span class="co"># Show miRNA names</span></span>
<span id="cb240-14"><a href="#cb240-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_colnames =</span> <span class="cn">TRUE</span>,  <span class="co"># Show group names</span></span>
<span id="cb240-15"><a href="#cb240-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"red"</span>))(<span class="dv">100</span>),  <span class="co"># Red gradient for importance</span></span>
<span id="cb240-16"><a href="#cb240-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"miRNA/lncRNA/methylation PCs' Importance Across Groups"</span>  <span class="co"># Title of the heatmap</span></span>
<span id="cb240-17"><a href="#cb240-17" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="./images/22.2-Apul-multiomic-machine-learning_files/figure-gfm/unnamed-chunk-85-1.png" class="img-fluid"><!-- --></p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">This notebook is built using <a href="https://quarto.org/">Quarto</a></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>