<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.433">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kathleen Durkin">
<meta name="dcterms.date" content="2025-04-01">

<title>Kathleen’s Lab Notebook - Timeseries molecular: A.pul, Functional Annotation and Enrichment Analyses</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kathleen’s Lab Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../daily_log.html" rel="" target="">
 <span class="menu-text">Daily Notebook Posts</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../projects.html" rel="" target="">
 <span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/shedurkin" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:kdurkin1@uw.edu" rel="" target=""><i class="bi bi-envelope" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <h1 class="title">Timeseries molecular: A.pul, Functional Annotation and Enrichment Analyses</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">E5-coral</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Kathleen Durkin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 1, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-packages" id="toc-load-packages" class="nav-link active" data-scroll-target="#load-packages">Load packages</a></li>
  <li><a href="#load-and-format-annotation-files" id="toc-load-and-format-annotation-files" class="nav-link" data-scroll-target="#load-and-format-annotation-files">Load and format annotation files</a></li>
  <li><a href="#annotate-wgcna-modules" id="toc-annotate-wgcna-modules" class="nav-link" data-scroll-target="#annotate-wgcna-modules">Annotate WGCNA modules</a></li>
  <li><a href="#functional-enrichment-of-modules-topgo" id="toc-functional-enrichment-of-modules-topgo" class="nav-link" data-scroll-target="#functional-enrichment-of-modules-topgo">Functional Enrichment of modules (topGO)</a>
  <ul class="collapse">
  <li><a href="#format-gene-ontology-go-annotations" id="toc-format-gene-ontology-go-annotations" class="nav-link" data-scroll-target="#format-gene-ontology-go-annotations">Format Gene Ontology (GO) annotations</a></li>
  <li><a href="#extract-list-of-significant-wgcna-modules" id="toc-extract-list-of-significant-wgcna-modules" class="nav-link" data-scroll-target="#extract-list-of-significant-wgcna-modules">Extract list of significant WGCNA modules</a></li>
  <li><a href="#topgo-function" id="toc-topgo-function" class="nav-link" data-scroll-target="#topgo-function">topGO function</a></li>
  <li><a href="#heatmap-of-significant-go-terms-by-trait" id="toc-heatmap-of-significant-go-terms-by-trait" class="nav-link" data-scroll-target="#heatmap-of-significant-go-terms-by-trait">Heatmap of significant GO terms by trait</a></li>
  <li><a href="#isolate-enriched-terms-by-associated-traits" id="toc-isolate-enriched-terms-by-associated-traits" class="nav-link" data-scroll-target="#isolate-enriched-terms-by-associated-traits">Isolate enriched terms by associated trait(s)</a></li>
  </ul></li>
  <li><a href="#annotate-putative-binding" id="toc-annotate-putative-binding" class="nav-link" data-scroll-target="#annotate-putative-binding">Annotate putative binding</a>
  <ul class="collapse">
  <li><a href="#load-module-module-correlations" id="toc-load-module-module-correlations" class="nav-link" data-scroll-target="#load-module-module-correlations">Load module-module correlations</a></li>
  <li><a href="#annotate-whether-a-given-module-contains-mirna" id="toc-annotate-whether-a-given-module-contains-mirna" class="nav-link" data-scroll-target="#annotate-whether-a-given-module-contains-mirna">Annotate whether a given module contains miRNA</a></li>
  <li><a href="#id-which-mirna-mrna-pairs-putatively-bind" id="toc-id-which-mirna-mrna-pairs-putatively-bind" class="nav-link" data-scroll-target="#id-which-mirna-mrna-pairs-putatively-bind">ID which miRNA-mRNA pairs putatively bind</a></li>
  <li><a href="#join-with-fa-table-to-annotate-each-binding-mrna" id="toc-join-with-fa-table-to-annotate-each-binding-mrna" class="nav-link" data-scroll-target="#join-with-fa-table-to-annotate-each-binding-mrna">Join with FA table to annotate each binding mRNA</a></li>
  </ul></li>
  <li><a href="#functional-enrichment-of-putative-binding" id="toc-functional-enrichment-of-putative-binding" class="nav-link" data-scroll-target="#functional-enrichment-of-putative-binding">Functional Enrichment of putative binding</a></li>
  <li><a href="#summary-figures" id="toc-summary-figures" class="nav-link" data-scroll-target="#summary-figures">Summary figures</a></li>
  <li><a href="#save-gene-sets" id="toc-save-gene-sets" class="nav-link" data-scroll-target="#save-gene-sets">Save gene sets</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">




<p>Note that I began this work before my time off (Thurs 3/20) and finished after returning, so some portions of this post are retroactive</p>
<p><a href="https://github.com/urol-e5/timeseries_molecular/blob/main/D-Apul/code/21-Apul-annotate-miRNA-mRNA-WGCNA.Rmd">Code</a><br>
<a href="https://github.com/urol-e5/timeseries_molecular/blob/main/D-Apul/code/21-Apul-annotate-miRNA-mRNA-WGCNA.md">Rendered code</a><br>
<a href="https://github.com/urol-e5/timeseries_molecular/tree/main/D-Apul/output/21-Apul-annotate-miRNA-mRNA-WGCNA">Output files</a></p>
<p>This work is based on the mRNA-miRNA WGCNA and the correlation of WGCNA modules with traits of interest (e.g.&nbsp;physiological metrics, seasonal/environmental metrics). In it, I:</p>
<ul>
<li><p>Functionally annotated all WGCNA modules</p></li>
<li><p>Performed functional enrichment analysis (topGO) on all WGCNA modules that are significantly correlated with a trait to inform what genes and functions are associated with phenotype and season.</p></li>
<li><p>Identified overlaps in module-module correlation <em>and</em> miRanda-based putative binding to try to identify putative miRNA-mRNA interactions.</p></li>
<li><p>For putative miRNA-mRNA interactions, performed functional annotation</p></li>
<li><p>For putative miRNA-mRNA interactions, performed functional enrichment analysis on the genes a given miRNA putatively interacts with to investigate putative miRNA function</p></li>
<li><p>For genes of interest (e.g.&nbsp;genes contained within modules that are significantly correlated with a trait), saved filtered gene sets, both an FA table and raw counts.</p></li>
</ul>
<p>Frankly, this is sooooo much data and I didn’t figure out a great way to summarize all of the results in a visual or digestible way. Here a few summary figures:</p>
<p><strong>Which traits have the most significantly correlated modules/genes?</strong></p>
<p><img src="https://github.com/urol-e5/timeseries_molecular/blob/main/D-Apul/code/21-Apul-annotate-miRNA-mRNA-WGCNA_files/figure-gfm/unnamed-chunk-28-1.png?raw=true" class="img-fluid"></p>
<p><strong>For traits of interest, what biological processes are overrepresented, and what overlap is there among traits of interest?</strong></p>
<p><img src="https://github.com/urol-e5/timeseries_molecular/blob/main/D-Apul/code/21-Apul-annotate-miRNA-mRNA-WGCNA_files/figure-gfm/unnamed-chunk-14-1.png?raw=true" class="img-fluid"></p>
<p>The true value of these analyses will require identifying a trait/module/miRNA of interest and the looking at what functions are enriched for it. I’ve been playing around with some interaction network visualizations, similarly to what I did with the miRanda/Pcor results, but this includes many more genes and is thus much more computationally intensive and would make the visualization less helpful</p>
<p>Code copied below in case of file path changes:</p>
<p>After running WGCNA to evaluate modules of mRNA and miRNA with correlated expression, we need to functionally annotate genes to evaluate module function(s).</p>
<section id="load-packages" class="level1">
<h1>Load packages</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(topGO)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-and-format-annotation-files" class="level1">
<h1>Load and format annotation files</h1>
<p>Already annotated the A.pulchra genome as part of the <code>deep-dive expression</code> project (see <a href="https://github.com/urol-e5/deep-dive-expression/blob/main/D-Apul/code/02-Apul-reference-annotation.Rmd"><code>deep-dive-expression/D-Apul/code/02-Apul-reference-annotation.Rmd</code></a>), so we’ll pull the annotation file from that</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Can access file stored here if needed</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># https://gannet.fish.washington.edu/kdurkin1/ravenbackups/deep-dive-expression/D-Apul/output/02-Apul-reference-annotation/Apulcra-genome-mRNA-IDmapping-2024_12_12.tab</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> ../../../deep-dive-expression/D-Apul/output/02-Apul-reference-annotation/Apulcra-genome-mRNA-IDmapping-2024_12_12.tab ../output/21-Apul-annotate-miRNA-mRNA-WGCNA</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Read mapping file into R</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>IDmapping_dirty <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/Apulcra-genome-mRNA-IDmapping-2024_12_12.tab"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Need to remove quotations surrounding each entry</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>IDmapping_locations <span class="ot">&lt;-</span> IDmapping_dirty</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>IDmapping_locations[] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(IDmapping_locations, <span class="cf">function</span>(x) <span class="fu">gsub</span>(<span class="st">'^"(.*)"$'</span>, <span class="st">'</span><span class="sc">\\</span><span class="st">1'</span>, x))</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove unneeded columns </span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>IDmapping_locations <span class="ot">&lt;-</span> IDmapping_locations <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>X, <span class="sc">-</span>V13)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure there are no duplicate rows</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>IDmapping_locations <span class="ot">&lt;-</span> IDmapping_locations <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(IDmapping_locations)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Looks good!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This file shows each gene as it’s genomic location. We want to use gene IDs to associate genes, so add gene IDs to this annotation table</p>
<p>Read in file that associates each mRNA genomic location with corresponding gene ID</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>mRNA_FUNids <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/05-Apul-annotate-UTRs/Apul-mRNA-FUNids.txt"</span>, <span class="at">header=</span><span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="fu">c</span>(<span class="st">"location"</span>, <span class="st">"type"</span>, <span class="st">"mRNA_ID"</span>, <span class="st">"gene_ID"</span>, <span class="st">"product"</span>), <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove unwanted text from parent column</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>mRNA_FUNids<span class="sc">$</span>gene_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Parent="</span>, <span class="st">""</span>, mRNA_FUNids<span class="sc">$</span>gene_ID)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Only need to keep mRNA location and gene ID</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>mRNA_FUNids <span class="ot">&lt;-</span> mRNA_FUNids <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(location, gene_ID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>join with annotation file</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>IDmapping <span class="ot">&lt;-</span> <span class="fu">left_join</span>(IDmapping_locations, mRNA_FUNids, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"V1"</span> <span class="ot">=</span> <span class="st">"location"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="annotate-wgcna-modules" class="level1">
<h1>Annotate WGCNA modules</h1>
<p>Load file that shows which module each gene/miRNA is in</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>module_membership <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/12-Apul-miRNA-mRNA-WGCNA/WGCNA-module-membership.tab"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>join with annotation file</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>module_FA <span class="ot">&lt;-</span> <span class="fu">left_join</span>(module_membership, IDmapping, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"gene_ID"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Many of the genes don’t have available annotations, so reduce to only keep those that are annotated (note this will also remove lncRNA and miRNA).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>module_FA_available <span class="ot">&lt;-</span> module_FA[<span class="sc">!</span><span class="fu">is.na</span>(module_FA<span class="sc">$</span>Gene.Ontology.IDs),]</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>module_FA_available <span class="ot">&lt;-</span> module_FA_available[module_FA_available<span class="sc">$</span>Gene.Ontology.IDs <span class="sc">!=</span> <span class="st">""</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="functional-enrichment-of-modules-topgo" class="level1">
<h1>Functional Enrichment of modules (topGO)</h1>
<p>I want to see whether our modules of coexpressed genes/miRNA represent specific, over-represented functionalities (i.e.&nbsp;does any module contain more metabolism-related genes than expected?)</p>
<p>To do this we’ll perform functional enrichment of each module using the R package topGO</p>
<section id="format-gene-ontology-go-annotations" class="level2">
<h2 class="anchored" data-anchor-id="format-gene-ontology-go-annotations">Format Gene Ontology (GO) annotations</h2>
<p>Want to isolate a list of GO terms per gene</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>gene_to_GO <span class="ot">&lt;-</span> IDmapping <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Gene.Ontology.IDs)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_ID, Gene.Ontology.IDs)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co">#head(gene_to_GO)</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Needs to be formatted as a named list for use in topGO</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert data frame to a named list</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co"># gene_to_GO_list &lt;- gene_to_GO %&gt;%</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(GO_terms = strsplit(Gene.Ontology.IDs, ";")) %&gt;%</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   tibble::deframe()</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>gene_to_GO_list <span class="ot">&lt;-</span> <span class="fu">setNames</span>(</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">strsplit</span>(<span class="fu">as.character</span>(gene_to_GO<span class="sc">$</span>Gene.Ontology.IDs), <span class="st">";"</span>), </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  gene_to_GO<span class="sc">$</span>gene_ID</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Note: I think this means genes that had a Uniprot ID but no GO terms are excluded from this analysis</p>
<p>Define reference set. This should be all genes <em>found in our samples</em>, NOT all genes in the A.pulchra genome. Some genes (e.g., reproduction pathways) may not be found/expected in our samples for valid biological reasons.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>reference_genes <span class="ot">&lt;-</span> module_membership<span class="sc">$</span>gene</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(reference_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extract-list-of-significant-wgcna-modules" class="level2">
<h2 class="anchored" data-anchor-id="extract-list-of-significant-wgcna-modules">Extract list of significant WGCNA modules</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>WGCNA_pvals <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/12-Apul-miRNA-mRNA-WGCNA/pval-cor-WGCNA_module-phys_envir.tab"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># filter to only keep rows and columns with at least one significant pval</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>filtered_rows <span class="ot">&lt;-</span> WGCNA_pvals[<span class="fu">rowSums</span>(WGCNA_pvals <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>, ]</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>WGCNA_pvals_significant <span class="ot">&lt;-</span> filtered_rows[, <span class="fu">colSums</span>(filtered_rows <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>]</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create list of modules that are significantly correlated with at least one trait</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>significant_modules <span class="ot">&lt;-</span> <span class="fu">row.names</span>(WGCNA_pvals_significant)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="topgo-function" class="level2">
<h2 class="anchored" data-anchor-id="topgo-function">topGO function</h2>
<p>Create a function to run topGO functional enrichment analysis for an input module (we’ll want to use this for many modules of interest)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>module_topGO_FE <span class="ot">&lt;-</span> <span class="cf">function</span>(module.name) {</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Isolate genes in our input module of interest</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  genes_in_module <span class="ot">&lt;-</span> module_membership <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(module <span class="sc">==</span> module.name) <span class="sc">%&gt;%</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(gene)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create factor for all reference genes, where 1 represents module membership and 0 means the gene is not in module of interest</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  gene_list <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(reference_genes <span class="sc">%in%</span> genes_in_module))</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(gene_list) <span class="ot">&lt;-</span> reference_genes</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create topGO object</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  GO_BP <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">"topGOdata"</span>,</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>              <span class="at">description =</span> <span class="st">"Functional Enrichment Analysis"</span>, </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">ontology =</span> <span class="st">"BP"</span>, <span class="co"># Biological Process</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">allGenes =</span> gene_list,</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">gene2GO =</span> gene_to_GO_list)</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run GO enrichment test</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>  GO_BP_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_BP, <span class="at">algorithm =</span> <span class="st">"weight01"</span>, <span class="at">statistic =</span> <span class="st">"fisher"</span>)</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># View the results</span></span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>  GO_BP_results <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_BP, <span class="at">Fisher =</span> GO_BP_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter by significant results</span></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>  GO_BP_results<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_BP_results<span class="sc">$</span>Fisher)</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>  GO_BP_results_sig<span class="ot">&lt;-</span>GO_BP_results[GO_BP_results<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return</span></span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(GO_BP_results_sig)</span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(module <span class="cf">in</span> significant_modules) {</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run topGo enrichent function</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  module_results <span class="ot">&lt;-</span> <span class="fu">module_topGO_FE</span>(module)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Only keep results if not empty</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(module_results) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>    <span class="co"># note the source column</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>    module_results<span class="sc">$</span>module <span class="ot">&lt;-</span> module</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    <span class="co"># append to results list</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>    results[[module]] <span class="ot">&lt;-</span> module_results</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all the resuts data frames into one</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>combined_GO_BP_results_sig <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, results)</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co"># View</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(combined_GO_BP_results_sig)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="heatmap-of-significant-go-terms-by-trait" class="level2">
<h2 class="anchored" data-anchor-id="heatmap-of-significant-go-terms-by-trait">Heatmap of significant GO terms by trait</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract significant modules for each trait</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>significant_modules_per_trait <span class="ot">&lt;-</span> <span class="fu">apply</span>(WGCNA_pvals_significant, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">rownames</span>(WGCNA_pvals_significant)[x <span class="sc">&lt;</span> <span class="fl">0.05</span>])</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify traits of interest</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co">#traits &lt;- colnames(WGCNA_pvals_significant)</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"timepoint2"</span>, <span class="st">"timepoint3"</span>, <span class="st">"mean_Temp_mean"</span>, <span class="st">"mean_solar_rad_kwpm2_mean"</span>, <span class="st">"cumulative_rainfall_mm_mean"</span>)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create empty matrix to store values for hetmap</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>go_term_counts <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> <span class="fu">length</span>(traits), <span class="at">ncol =</span> <span class="fu">length</span>(<span class="fu">unique</span>(combined_GO_BP_results_sig<span class="sc">$</span>Term)))</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(go_term_counts) <span class="ot">&lt;-</span> traits  <span class="co"># Trait names</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(go_term_counts) <span class="ot">&lt;-</span> <span class="fu">unique</span>(combined_GO_BP_results_sig<span class="sc">$</span>Term)  <span class="co"># Unique GO terms</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each trait, calculate the number of significant modules enriched for each GO term</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (trait <span class="cf">in</span> traits) {</span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get the significant modules for this trait</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  significant_modules <span class="ot">&lt;-</span> significant_modules_per_trait[[trait]]</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Check if a module is enriched for each GO term</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (go_term <span class="cf">in</span> <span class="fu">unique</span>(combined_GO_BP_results_sig<span class="sc">$</span>Term)) {</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if the module is in the list of significant modules for the trait and if it's enriched for this GO term</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>    enriched_modules <span class="ot">&lt;-</span> combined_GO_BP_results_sig <span class="sc">%&gt;%</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Term <span class="sc">==</span> go_term <span class="sc">&amp;</span> module <span class="sc">%in%</span> significant_modules) <span class="sc">%&gt;%</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>      <span class="fu">nrow</span>()  <span class="co"># Count of modules enriched for this GO term</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Count number of modules that contain each GO term</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>    go_term_counts[trait, go_term] <span class="ot">&lt;-</span> enriched_modules</span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert matrix to df for plotting</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>go_term_counts_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(go_term_counts) <span class="sc">%&gt;%</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"Trait"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gather</span>(<span class="at">key =</span> <span class="st">"GO_term"</span>, <span class="at">value =</span> <span class="st">"Module_Count"</span>, <span class="sc">-</span>Trait)</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Truncate GO term names</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a><span class="co">#go_term_counts_df$GO_term &lt;- substr(go_term_counts_df$GO_term, 1, 40)</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(go_term_counts_df, <span class="fu">aes</span>(<span class="at">x =</span> Trait, <span class="at">y =</span> GO_term, <span class="at">fill =</span> Module_Count)) <span class="sc">+</span></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="at">color =</span> <span class="st">"grey"</span>) <span class="sc">+</span>  <span class="co"># Adds grid lines</span></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"blue"</span>) <span class="sc">+</span>  <span class="co"># Color gradient from white (low) to blue (high)</span></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>, <span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust=</span><span class="fl">0.5</span>, <span class="at">vjust=</span><span class="fl">0.5</span>), </span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>        ) <span class="sc">+</span> </span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"# of Enriched Modules"</span>, <span class="at">title =</span> <span class="st">"Significant Modules Enriched for GO Terms"</span>) <span class="sc">+</span></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/go_term_heatmap_envir_traits.png"</span>, <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="isolate-enriched-terms-by-associated-traits" class="level2">
<h2 class="anchored" data-anchor-id="isolate-enriched-terms-by-associated-traits">Isolate enriched terms by associated trait(s)</h2>
<p>Terms enriched in modules correlated with TP2</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>modules_TP2 <span class="ot">&lt;-</span> significant_modules_per_trait[[<span class="st">"timepoint2"</span>]]</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>GO_TP2 <span class="ot">&lt;-</span> combined_GO_BP_results_sig[combined_GO_BP_results_sig<span class="sc">$</span>module <span class="sc">%in%</span> modules_TP2,]</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>GO_TP2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Terms enriched in modules correlated with TP3</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>modules_TP3 <span class="ot">&lt;-</span> significant_modules_per_trait[[<span class="st">"timepoint3"</span>]]</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>GO_TP3 <span class="ot">&lt;-</span> combined_GO_BP_results_sig[combined_GO_BP_results_sig<span class="sc">$</span>module <span class="sc">%in%</span> modules_TP3,]</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>GO_TP3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Terms enriched in modules correlated with temp</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>modules_temp <span class="ot">&lt;-</span> significant_modules_per_trait[[<span class="st">"mean_Temp_mean"</span>]]</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>GO_temp <span class="ot">&lt;-</span> combined_GO_BP_results_sig[combined_GO_BP_results_sig<span class="sc">$</span>module <span class="sc">%in%</span> modules_temp,]</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>GO_temp</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Terms enriched in modules correlated with solar</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>modules_solar <span class="ot">&lt;-</span> significant_modules_per_trait[[<span class="st">"mean_solar_rad_kwpm2_mean"</span>]]</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>GO_solar <span class="ot">&lt;-</span> combined_GO_BP_results_sig[combined_GO_BP_results_sig<span class="sc">$</span>module <span class="sc">%in%</span> modules_solar,]</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>GO_solar</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Terms enriched in modules correlated with rain</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>modules_rain <span class="ot">&lt;-</span> significant_modules_per_trait[[<span class="st">"cumulative_rainfall_mm_mean"</span>]]</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>GO_rain <span class="ot">&lt;-</span> combined_GO_BP_results_sig[combined_GO_BP_results_sig<span class="sc">$</span>module <span class="sc">%in%</span> modules_rain,]</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>GO_rain</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="annotate-putative-binding" class="level1">
<h1>Annotate putative binding</h1>
<p>Outline:</p>
<ol type="1">
<li>Have table module pair and correlation values</li>
</ol>
<section id="load-module-module-correlations" class="level2">
<h2 class="anchored" data-anchor-id="load-module-module-correlations">Load module-module correlations</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>modules_cor <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/12-Apul-miRNA-mRNA-WGCNA/WGCNA-modules-correlations.tab"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove instances where a module is just correlating to itself</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>modules_cor <span class="ot">&lt;-</span> modules_cor[modules_cor<span class="sc">$</span>module_A <span class="sc">!=</span> modules_cor<span class="sc">$</span>module_B,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="annotate-whether-a-given-module-contains-mirna" class="level2">
<h2 class="anchored" data-anchor-id="annotate-whether-a-given-module-contains-mirna">Annotate whether a given module contains miRNA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate module membership into only mRNA and only miRNA</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>module_membership_mRNA <span class="ot">&lt;-</span> module_membership <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"FUN"</span>, gene))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>module_membership_miRNA <span class="ot">&lt;-</span> module_membership <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"Cluster"</span>, gene))</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Expand each miRNA_module to show all contained miRNAa</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>modules_cor_miRNA <span class="ot">&lt;-</span> <span class="fu">left_join</span>(modules_cor, module_membership_miRNA, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"module_A"</span> <span class="ot">=</span> <span class="st">"module"</span>))</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename columns appropriately</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(modules_cor_miRNA) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"miRNA_module"</span>, <span class="st">"mRNA_module"</span>, <span class="st">"correlation"</span>, <span class="st">"miRNA"</span>)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove instances where the module_A (miRNA_module) doesn't contain miRNA</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>modules_cor_miRNA <span class="ot">&lt;-</span> modules_cor_miRNA <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(miRNA))</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Expand each mRNA module to show all contained genes</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>modules_cor_miRNA_mRNA <span class="ot">&lt;-</span> <span class="fu">left_join</span>(modules_cor_miRNA, module_membership_mRNA, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"mRNA_module"</span> <span class="ot">=</span> <span class="st">"module"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="id-which-mirna-mrna-pairs-putatively-bind" class="level2">
<h2 class="anchored" data-anchor-id="id-which-mirna-mrna-pairs-putatively-bind">ID which miRNA-mRNA pairs putatively bind</h2>
<p>Load and format miRanda tables</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load ##</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>miRanda_3UTR <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/07-Apul-miRNA-mRNA-miRanda/Apul-miRanda-3UTR-strict-parsed-geneIDs.txt"</span>, <span class="at">header=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>miRanda_5UTR <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-5UTR_1kb-strict-parsed.txt"</span>, <span class="at">header=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>miRanda_mRNA <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/07.1-Apul-miRNA-mRNA-miRanda-additional_inputs/Apul-miRanda-mRNA_full-strict-parsed.txt"</span>, <span class="at">header=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Format ##</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove superfluous text in miRNA name column</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>miRanda_3UTR <span class="ot">&lt;-</span> miRanda_3UTR <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">V1 =</span> <span class="fu">str_extract</span>(V1, <span class="st">"Cluster[^.]+"</span>))</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>miRanda_5UTR <span class="ot">&lt;-</span> miRanda_5UTR <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">V1 =</span> <span class="fu">str_extract</span>(V1, <span class="st">"Cluster[^.]+"</span>))</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>miRanda_mRNA <span class="ot">&lt;-</span> miRanda_mRNA <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">V1 =</span> <span class="fu">str_extract</span>(V1, <span class="st">"Cluster[^.]+"</span>))</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="co"># For 5UTR and mRNA, need to add gene_IDs associated with each genomic location</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Load and format table of 5UTR geneIDs</span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>FUNids_5UTR <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/05-Apul-annotate-UTRs/Apul-5UTR-FUNids.txt"</span>, <span class="at">header=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>FUNids_5UTR<span class="sc">$</span>V4 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Parent="</span>, <span class="st">""</span>, FUNids_5UTR<span class="sc">$</span>V4)</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a>FUNids_5UTR <span class="ot">&lt;-</span> FUNids_5UTR <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(V1, V4)</span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Already have table of mRNA geneIDs</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene IDs to the miRanda tables</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>miRanda_5UTR <span class="ot">&lt;-</span> <span class="fu">left_join</span>(miRanda_5UTR, FUNids_5UTR, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"V2"</span> <span class="ot">=</span> <span class="st">"V1"</span>))</span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>miRanda_mRNA <span class="ot">&lt;-</span> <span class="fu">left_join</span>(miRanda_mRNA, mRNA_FUNids, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"V2"</span><span class="ot">=</span><span class="st">"location"</span>))</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure no unwanted duplicate rows</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a>miRanda_3UTR <span class="ot">&lt;-</span> miRanda_3UTR <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a>miRanda_5UTR <span class="ot">&lt;-</span> miRanda_5UTR <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a>miRanda_mRNA <span class="ot">&lt;-</span> miRanda_mRNA <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Add column to identify whether putative binding is in 3UTR, 5UTR, or CDS</span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a>miRanda_3UTR<span class="sc">$</span>binding_loc <span class="ot">&lt;-</span> <span class="st">"3UTR"</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a>miRanda_5UTR<span class="sc">$</span>binding_loc <span class="ot">&lt;-</span> <span class="st">"5UTR"</span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>miRanda_mRNA<span class="sc">$</span>binding_loc <span class="ot">&lt;-</span> <span class="st">"mRNA"</span></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine into single data frame</span></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(miRanda_3UTR) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"V1"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>, <span class="st">"V4"</span>, <span class="st">"V5"</span>, <span class="st">"V6"</span>, <span class="st">"V7"</span>, <span class="st">"V8"</span>, <span class="st">"V9"</span>, <span class="st">"V10"</span>, <span class="st">"binding_loc"</span>)</span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(miRanda_5UTR) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"V1"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>, <span class="st">"V4"</span>, <span class="st">"V5"</span>, <span class="st">"V6"</span>, <span class="st">"V7"</span>, <span class="st">"V8"</span>, <span class="st">"V9"</span>, <span class="st">"V10"</span>, <span class="st">"binding_loc"</span>)</span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(miRanda_mRNA) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"V1"</span>, <span class="st">"V2"</span>, <span class="st">"V3"</span>, <span class="st">"V4"</span>, <span class="st">"V5"</span>, <span class="st">"V6"</span>, <span class="st">"V7"</span>, <span class="st">"V8"</span>, <span class="st">"V9"</span>, <span class="st">"V10"</span>, <span class="st">"binding_loc"</span>)</span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a>miRanda_combined <span class="ot">&lt;-</span> <span class="fu">rbind</span>(miRanda_3UTR, miRanda_5UTR, miRanda_mRNA)</span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(miRanda_combined)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Join with table of correlating modules and their members to annotate, for each miRNA, which of the mRNAs in its correlated module it may bind to</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Join </span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>modules_cor_miRNA_mRNA_bind <span class="ot">&lt;-</span> <span class="fu">left_join</span>(modules_cor_miRNA_mRNA, miRanda_combined, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"V10"</span>))</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Only keep rows where the full miRNA-mRNA interaction matches, not just the mRNA</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>modules_cor_miRNA_mRNA_bind <span class="ot">&lt;-</span> modules_cor_miRNA_mRNA_bind[modules_cor_miRNA_mRNA_bind<span class="sc">$</span>miRNA <span class="sc">==</span> modules_cor_miRNA_mRNA_bind<span class="sc">$</span>V1,] <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This is now a table that, for each module that contians miRNA (miRNA_module), shows each module it’s correlated with (mRNA_module), the genes in that module, and whether that gene putatively binds with the miRNA. It still includes, though, instances of multiple putative binding positions for the same miRNA-mRNA pair. Since we just care about the high-level miRNA-mRNA pair, let’s remove this level of detail</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>mod_cor_bind <span class="ot">&lt;-</span> modules_cor_miRNA_mRNA_bind <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(miRNA_module, mRNA_module, correlation, miRNA, gene, binding_loc) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(mod_cor_bind)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(mod_cor_bind[mod_cor_bind<span class="sc">$</span>correlation <span class="sc">&lt;</span> <span class="dv">0</span>,])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>So there are 41,633 instances of an miRNA putatively binding to a gene that is contained in a correlated module. 21,274 of these (roughly half) are putative binding between modules that are <em>negatively</em> correlated with each other, which is the expected behavior for canonical miRNA function.</p>
</section>
<section id="join-with-fa-table-to-annotate-each-binding-mrna" class="level2">
<h2 class="anchored" data-anchor-id="join-with-fa-table-to-annotate-each-binding-mrna">Join with FA table to annotate each binding mRNA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>mod_cor_bind_FA <span class="ot">&lt;-</span> <span class="fu">left_join</span>(mod_cor_bind, IDmapping, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"gene"</span> <span class="ot">=</span> <span class="st">"gene_ID"</span>)) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(mod_cor_bind_FA <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Gene.Ontology.IDs)))</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(mod_cor_bind_FA <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Gene.Ontology.IDs) <span class="sc">&amp;</span> correlation <span class="sc">&lt;</span> <span class="dv">0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Of these 41,633 instances of an miRNA putatively binding to a gene contained in a correlated module, only 12,367 (roughly a quarter) of the putatively bound genes are annotated with GO terms. Roughly half of the annotated genes are associated with negative module correlation.</p>
</section>
</section>
<section id="functional-enrichment-of-putative-binding" class="level1">
<h1>Functional Enrichment of putative binding</h1>
<p>In the above annotation we’ve functionally annotated all miRNA-mRNA pairs that are in correlated modules <em>and</em> putatively bind in the 3UTR, 5UTR and/or coding sequence (based on miRanda). The ultimate goal is to elucidate putative function of our miRNAs by evaluating the functions of the mRNAs they putatively interact with.</p>
<p>To aid this we can run functional enrichment on the mRNAs that each miRNA putatively interacts with.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Modify topGO function for use with miRNA names</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>miRNA_topGO_FE <span class="ot">&lt;-</span> <span class="cf">function</span>(miRNA.name) {</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Isolate genes in our input module of interest</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  interacting_genes <span class="ot">&lt;-</span> mod_cor_bind_FA <span class="sc">%&gt;%</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(miRNA <span class="sc">==</span> miRNA.name) <span class="sc">%&gt;%</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(gene)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(interacting_genes) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create factor for all reference genes, where 1 represents module membership and 0 means the gene is not in module of interest</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>    gene_list <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(reference_genes <span class="sc">%in%</span> interacting_genes))</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(gene_list) <span class="ot">&lt;-</span> reference_genes</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str</span>(gene_list)</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create topGO object</span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>    GO_BP <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">"topGOdata"</span>,</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>                <span class="at">description =</span> <span class="st">"Functional Enrichment Analysis"</span>, </span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>                <span class="at">ontology =</span> <span class="st">"BP"</span>, <span class="co"># Biological Process</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>                <span class="at">allGenes =</span> gene_list,</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>                <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>                <span class="at">gene2GO =</span> gene_to_GO_list)</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Run GO enrichment test</span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>    GO_BP_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_BP, <span class="at">algorithm =</span> <span class="st">"weight01"</span>, <span class="at">statistic =</span> <span class="st">"fisher"</span>)</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># View the results</span></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>    GO_BP_results <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_BP, <span class="at">Fisher =</span> GO_BP_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Filter by significant results</span></span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>    GO_BP_results<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_BP_results<span class="sc">$</span>Fisher)</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>    GO_BP_results_sig<span class="ot">&lt;-</span>GO_BP_results[GO_BP_results<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Return</span></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(GO_BP_results_sig)</span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>interacting_miRNAs <span class="ot">&lt;-</span> <span class="fu">unique</span>(mod_cor_bind_FA<span class="sc">$</span>miRNA) <span class="sc">%&gt;%</span> na.omit</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(miRNA <span class="cf">in</span> interacting_miRNAs) {</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run topGo enrichment function</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  miRNA_results <span class="ot">&lt;-</span> <span class="fu">miRNA_topGO_FE</span>(miRNA)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(miRNA_results)</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Only keep results if not empty</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(miRNA_results) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># note the source column</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>    miRNA_results<span class="sc">$</span>miRNA <span class="ot">&lt;-</span> miRNA</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>    <span class="co"># append to results list</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>    results[[miRNA]] <span class="ot">&lt;-</span> miRNA_results</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all the resuts data frames into one</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>combined_GO_BP_results_miRNA <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, results)</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a><span class="co"># View</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(combined_GO_BP_results_miRNA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="summary-figures" class="level1">
<h1>Summary figures</h1>
<p>Bubbleplot showing number of significantly correlated modules for eacvh trait and the total genes represented by those modules</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Example significance threshold</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>pval_threshold <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter significant modules for each trait</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>significant_modules_long <span class="ot">&lt;-</span> WGCNA_pvals <span class="sc">%&gt;%</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"module"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>module, <span class="at">names_to =</span> <span class="st">"Trait"</span>, <span class="at">values_to =</span> <span class="st">"Pvalue"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of significant modules per trait</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>module_counts <span class="ot">&lt;-</span> significant_modules_long <span class="sc">%&gt;%</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span></span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">NumModules =</span> <span class="fu">n</span>())</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of genes in significant modules</span></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>gene_counts <span class="ot">&lt;-</span> module_membership <span class="sc">%&gt;%</span></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(module <span class="sc">%in%</span> significant_modules_long<span class="sc">$</span>module) <span class="sc">%&gt;%</span></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(module) <span class="sc">%&gt;%</span></span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">NumGenes =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(significant_modules_long, <span class="at">by =</span> <span class="st">"module"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">TotalGenes =</span> <span class="fu">sum</span>(NumGenes))</span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge data</span></span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> module_counts <span class="sc">%&gt;%</span></span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(gene_counts, <span class="at">by =</span> <span class="st">"Trait"</span>)</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to certain traits if desired</span></span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a><span class="co"># traits &lt;- c("timepoint2", "timepoint3", "mean_Temp_mean", "mean_solar_rad_kwpm2_mean", "cumulative_rainfall_mm_mean")</span></span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_data &lt;- plot_data[plot_data$Trait %in% traits,]</span></span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Bubble plot</span></span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> Trait, <span class="at">y =</span> <span class="fl">0.5</span>, <span class="at">size =</span> TotalGenes, <span class="at">color =</span> NumModules)) <span class="sc">+</span></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>, <span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">15</span>)) <span class="sc">+</span> </span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Significant Modules per Trait"</span>,</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Trait"</span>,</span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">""</span>,</span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>       <span class="at">size =</span> <span class="st">"Total Genes"</span>,</span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Num Modules"</span>) <span class="sc">+</span></span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),  <span class="co"># Hide y-axis labels</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.ticks.y =</span> <span class="fu">element_blank</span>())</span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/bubble_signif_modules.png"</span>, <span class="at">width =</span> <span class="dv">12</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Interaction network showing, for a variable of interest, all the correlated modules, their genes, and miRNA interactions</p>
<ol type="1">
<li>Which modules are in variable of interest WGCNA_pvals[WGCNA_pvals$trait &lt; 0.05,] %&gt;% rownames()</li>
<li>Which genes/miRNA are in each module <code>module_membership</code></li>
<li>which genes each miRNA interacts with <code>modules_cor_miRNA_mRNA_bind</code> ^includes Correlation between modules</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>trait <span class="ot">&lt;-</span> <span class="st">"timepoint"</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract correlated modules</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>sig_modules <span class="ot">&lt;-</span> <span class="fu">rownames</span>(WGCNA_pvals[WGCNA_pvals<span class="sc">$</span>timepoint2 <span class="sc">&lt;</span> <span class="fl">0.05</span>, ])</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract genes and miRNA n each module</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>module_genes <span class="ot">&lt;-</span> module_membership[module_membership<span class="sc">$</span>module <span class="sc">%in%</span> sig_modules, ]</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract miRNA-mRNA interactions for these modules</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>miRNA_interactions <span class="ot">&lt;-</span> modules_cor_miRNA_mRNA_bind[</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  modules_cor_miRNA_mRNA_bind<span class="sc">$</span>miRNA_module <span class="sc">%in%</span> sig_modules <span class="sc">|</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  modules_cor_miRNA_mRNA_bind<span class="sc">$</span>mRNA_module <span class="sc">%in%</span> sig_modules, </span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Create edge list for the network</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="co"># edges &lt;- rbind(</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   data.frame(from = module_genes$gene, to = module_genes$module, type = "gene-module"),</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   data.frame(from = miRNA_interactions$miRNA, to = miRNA_interactions$mRNA, type = "miRNA-mRNA"),</span></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   data.frame(from = miRNA_interactions$miRNA_module, to = miRNA_interactions$mRNA_module, type = "module-module")</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a><span class="co"># # Gene-to-Module edges (Genes and miRNAs assigned to modules)</span></span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a><span class="co"># edges_genes &lt;- data.frame(</span></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   from = module_membership$gene, </span></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   to = module_membership$module, </span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   type = "gene-module",</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   correlation = NA  # No correlation needed for membership edges</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="co"># # miRNA-Gene interaction edges (including correlation from module correlation)</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a><span class="co"># edges_miRNA_gene &lt;- data.frame(</span></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a><span class="co">#   from = modules_cor_miRNA_mRNA_bind$miRNA, </span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a><span class="co">#   to = modules_cor_miRNA_mRNA_bind$gene, </span></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a><span class="co">#   type = "miRNA-gene",</span></span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a><span class="co">#   correlation = modules_cor_miRNA_mRNA_bind$correlation  # Assign correlation value</span></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="co"># )</span></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a><span class="co"># # Combine all edges into a single dataframe</span></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a><span class="co"># edges &lt;- rbind(edges_genes, edges_miRNA_gene) %&gt;% distinct()</span></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a>  <span class="at">from =</span> modules_cor_miRNA_mRNA_bind<span class="sc">$</span>miRNA, </span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a>  <span class="at">to =</span> modules_cor_miRNA_mRNA_bind<span class="sc">$</span>gene, </span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"miRNA-gene"</span>,</span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a>  <span class="at">correlation =</span> modules_cor_miRNA_mRNA_bind<span class="sc">$</span>correlation  <span class="co"># Correlation between miRNA and gene parent modules</span></span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Create node data frame with miRNAs and genes</span></span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a>miRNA_nodes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="fu">unique</span>(modules_cor_miRNA_mRNA_bind<span class="sc">$</span>miRNA),  <span class="co"># Unique miRNAs</span></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a>  <span class="at">module =</span> modules_cor_miRNA_mRNA_bind<span class="sc">$</span>miRNA_module[<span class="fu">match</span>(<span class="fu">unique</span>(modules_cor_miRNA_mRNA_bind<span class="sc">$</span>miRNA), modules_cor_miRNA_mRNA_bind<span class="sc">$</span>miRNA)],  <span class="co"># Corresponding miRNA module</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"miRNA"</span></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Create gene-node dataframe with the filtered module_membership</span></span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a>gene_nodes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">name =</span> <span class="fu">unique</span>(modules_cor_miRNA_mRNA_bind<span class="sc">$</span>gene),  <span class="co"># Unique genes</span></span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">module =</span> modules_cor_miRNA_mRNA_bind<span class="sc">$</span>mRNA_module[<span class="fu">match</span>(<span class="fu">unique</span>(modules_cor_miRNA_mRNA_bind<span class="sc">$</span>gene), modules_cor_miRNA_mRNA_bind<span class="sc">$</span>gene)],  <span class="co"># Corresponding gene module</span></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"gene"</span></span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine both miRNA and gene nodes</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(miRNA_nodes, gene_nodes)</span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot</span></span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(edges, <span class="at">directed =</span> <span class="cn">FALSE</span>, <span class="at">vertices =</span> nodes)</span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a><span class="fu">ggraph</span>(g, <span class="at">layout =</span> <span class="st">"fr"</span>) <span class="sc">+</span></span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Nodes colored by their module (categorical)</span></span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_node_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> module), <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-73"><a href="#cb29-73" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Edges colored by correlation (continuous)</span></span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_edge_link</span>(<span class="fu">aes</span>(<span class="at">color =</span> correlation), <span class="at">width =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Discrete color scale for modules</span></span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">rainbow</span>(<span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">V</span>(g)<span class="sc">$</span>module)))) <span class="sc">+</span></span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Continuous gradient color scale for correlation values</span></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_edge_color_gradient2</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"red"</span>, <span class="at">midpoint =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(edges, <span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/edges.csv"</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(nodes, <span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/nodes.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="save-gene-sets" class="level1">
<h1>Save gene sets</h1>
<p>We’ll likely want to use function-specific gene sets in later analyses, so we’ll save those here. Gene sets should be saved as</p>
<ol type="a">
<li>a df of the genes, the WGCNA module they fall in, and the annotated GO term(s), AND</li>
<li>raw count matrices, with samples in the columns and genes in the rows.</li>
</ol>
<p>Sets will be named for the filter applied (e.g.&nbsp;“respiration” for genes annotated with respiration-related GO terms). Note that, since these gene sets will be used with a functional focus, we’ll exclude miRNAs.</p>
<p>Load raw gene counts</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>Apul_genes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../output/02.20-D-Apul-RNAseq-alignment-HiSat2/apul-gene_count_matrix.csv"</span>)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>Apul_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Apul_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># For each trait of interest:</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (trait <span class="cf">in</span> <span class="fu">names</span>(significant_modules_per_trait)) {</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="do">## a) a df of the genes, the WGCNA module they fall in, and the annotated GO term(s)</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select modules that are significantly correlated with this trait</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  trait_sig_modules <span class="ot">&lt;-</span> significant_modules_per_trait[[trait]]</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter the annotated list of genes to only keep those in selected modules, keep only genes (not miRNA)</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  trait_genes_FA <span class="ot">&lt;-</span> module_FA <span class="sc">%&gt;%</span></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(module <span class="sc">%in%</span> trait_sig_modules <span class="sc">&amp;</span> <span class="fu">grepl</span>(<span class="st">"FUN"</span>, gene))</span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="do">## b) raw count matrices, with samples in the columns and genes in the rows. </span></span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter the raw gene counts to keep only genes in selected modules</span></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>  trait_gene_counts <span class="ot">&lt;-</span> Apul_genes <span class="sc">%&gt;%</span></span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(gene_id <span class="sc">%in%</span> trait_genes_FA<span class="sc">$</span>gene)</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set file names</span></span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a>  file_name_genes_FA <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/"</span>, trait, <span class="st">"_genes_FA.tab"</span>)</span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a>  file_name_gene_counts <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/"</span>, trait, <span class="st">"_gene_counts.tab"</span>)</span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save</span></span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(trait_genes_FA, file_name_genes_FA, <span class="at">row.names=</span><span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(trait_gene_counts, file_name_gene_counts, <span class="at">row.names=</span><span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We also want to be able to save genes associated with functions (GO terms) of interest.</p>
<p>For example, Steven identified a several GO terms related to ATP production a <a href="https://sr320.github.io/tumbling-oysters/posts/41-Apul-GO/">notebook post</a>: - Aerobic respiration (GO:0009060) - Oxidative phosphorylation (GO:0006119) - Canonical glycolysis (GO:0061621) - Tricarboxylic Acid Cycle (GO:0006099)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List GO terms of interest (CHANGE as desired)</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>GO_terms_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GO:0009060"</span>, <span class="st">"GO:0006119"</span>, <span class="st">"GO:0061621"</span>, <span class="st">"GO:0006099"</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select genes that have been functionally annotated with at least on of the listed GO terms of interest</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>GO_terms_genes_FA <span class="ot">&lt;-</span> module_FA <span class="sc">%&gt;%</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">sapply</span>(Gene.Ontology.IDs, <span class="cf">function</span>(x) <span class="fu">any</span>(<span class="fu">sapply</span>(GO_terms_interest, grepl, x))))</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="co"># For these selected genes, get raw gene counts</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>GO_terms_gene_counts <span class="ot">&lt;-</span> Apul_genes <span class="sc">%&gt;%</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene_id <span class="sc">%in%</span> GO_terms_genes_FA<span class="sc">$</span>gene)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Set file name (CHANGE as desired)</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>file_name_GO_FA <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/"</span>, <span class="st">"ATP_production"</span>, <span class="st">"_GO_terms_genes_FA.tab"</span>)</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a>file_name_GO_counts <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"../output/21-Apul-annotate-miRNA-mRNA-WGCNA/filtered-gene-sets/"</span>, <span class="st">"ATP_production"</span>, <span class="st">"_GO_terms_gene_counts.tab"</span>)</span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(GO_terms_genes_FA, file_name_GO_FA, <span class="at">row.names=</span><span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(go_term_counts_df, file_name_GO_counts, <span class="at">row.names=</span><span class="cn">FALSE</span>, <span class="at">quote=</span><span class="cn">FALSE</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">This notebook is built using <a href="https://quarto.org/">Quarto</a></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>



</body></html>