<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kathleen Durkin">
<meta name="dcterms.date" content="2025-04-08">

<title>Deep-Dive Expression: Functional enrichment of miRNA-mRNA interactions – Kathleen’s Lab Notebook</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-2fef5ea3f8957b3e4ecc936fc74692ca.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-b515dea5baab39ffcfcd4bae179e3bdb.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark-195886546b2eb1f67ede645fd062ce19.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../../site_libs/bootstrap/bootstrap-b515dea5baab39ffcfcd4bae179e3bdb.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../../styles.css">
</head>

<body class="nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../../index.html">
    <span class="navbar-title">Kathleen’s Lab Notebook</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../daily_log.html"> 
<span class="menu-text">Daily Notebook Posts</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../projects.html"> 
<span class="menu-text">Projects</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/shedurkin"> <i class="bi bi-github" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="mailto:kdurkin1@uw.edu"> <i class="bi bi-envelope" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-page-left">
      <h1 class="title">Deep-Dive Expression: Functional enrichment of miRNA-mRNA interactions</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">E5-coral</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta column-page-left">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Kathleen Durkin </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">April 8, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#results" id="toc-results" class="nav-link active" data-scroll-target="#results">Results</a></li>
  <li><a href="#code" id="toc-code" class="nav-link" data-scroll-target="#code">Code</a></li>
  <li><a href="#mrna-mirna-interactions-functional-enrichment" id="toc-mrna-mirna-interactions-functional-enrichment" class="nav-link" data-scroll-target="#mrna-mirna-interactions-functional-enrichment">mRNA-miRNA interactions functional enrichment</a></li>
  <li><a href="#format-topgo-files" id="toc-format-topgo-files" class="nav-link" data-scroll-target="#format-topgo-files">1 Format topGO files</a>
  <ul class="collapse">
  <li><a href="#read-in-and-format-annotation-files" id="toc-read-in-and-format-annotation-files" class="nav-link" data-scroll-target="#read-in-and-format-annotation-files">1.1 Read in and format annotation files</a></li>
  <li><a href="#set-up-gene2go-object" id="toc-set-up-gene2go-object" class="nav-link" data-scroll-target="#set-up-gene2go-object">1.2 Set up gene2GO object</a></li>
  <li><a href="#define-reference-set" id="toc-define-reference-set" class="nav-link" data-scroll-target="#define-reference-set">1.3 Define reference set</a></li>
  <li><a href="#read-in-pccmiranda-data" id="toc-read-in-pccmiranda-data" class="nav-link" data-scroll-target="#read-in-pccmiranda-data">1.4 Read in PCC/miranda data</a></li>
  </ul></li>
  <li><a href="#fe-of-all-targets-as-predicted-by-miranda" id="toc-fe-of-all-targets-as-predicted-by-miranda" class="nav-link" data-scroll-target="#fe-of-all-targets-as-predicted-by-miranda">2 FE of all targets, as predicted by miranda</a></li>
  <li><a href="#fe-of-targets-negatively-correlated-with-mirnas-regardless-of-correlation-significance" id="toc-fe-of-targets-negatively-correlated-with-mirnas-regardless-of-correlation-significance" class="nav-link" data-scroll-target="#fe-of-targets-negatively-correlated-with-mirnas-regardless-of-correlation-significance">3 FE of targets negatively correlated with miRNAs (regardless of correlation significance)</a>
  <ul class="collapse">
  <li><a href="#biological-processes" id="toc-biological-processes" class="nav-link" data-scroll-target="#biological-processes">3.0.1 Biological Processes</a></li>
  <li><a href="#cellular-components" id="toc-cellular-components" class="nav-link" data-scroll-target="#cellular-components">3.0.2 Cellular Components</a></li>
  <li><a href="#molecular-functions" id="toc-molecular-functions" class="nav-link" data-scroll-target="#molecular-functions">3.0.3 Molecular Functions</a></li>
  <li><a href="#join-ontologies" id="toc-join-ontologies" class="nav-link" data-scroll-target="#join-ontologies">3.0.4 Join ontologies</a></li>
  </ul></li>
  <li><a href="#fe-of-targets-positively-correlated-with-mirnas-regardless-of-correlation-significance" id="toc-fe-of-targets-positively-correlated-with-mirnas-regardless-of-correlation-significance" class="nav-link" data-scroll-target="#fe-of-targets-positively-correlated-with-mirnas-regardless-of-correlation-significance">4 FE of targets positively correlated with miRNAs (regardless of correlation significance)</a>
  <ul class="collapse">
  <li><a href="#biological-processes-1" id="toc-biological-processes-1" class="nav-link" data-scroll-target="#biological-processes-1">4.0.1 Biological Processes</a></li>
  <li><a href="#cellular-components-1" id="toc-cellular-components-1" class="nav-link" data-scroll-target="#cellular-components-1">4.0.2 Cellular Components</a></li>
  <li><a href="#molecular-functions-1" id="toc-molecular-functions-1" class="nav-link" data-scroll-target="#molecular-functions-1">4.0.3 Molecular Functions</a></li>
  <li><a href="#join-ontologies-1" id="toc-join-ontologies-1" class="nav-link" data-scroll-target="#join-ontologies-1">4.0.4 Join ontologies</a></li>
  </ul></li>
  <li><a href="#fe-of-targets-significantly-negatively-correlated-with-mirnas" id="toc-fe-of-targets-significantly-negatively-correlated-with-mirnas" class="nav-link" data-scroll-target="#fe-of-targets-significantly-negatively-correlated-with-mirnas">5 FE of targets significantly negatively correlated with miRNAs</a>
  <ul class="collapse">
  <li><a href="#biological-processes-2" id="toc-biological-processes-2" class="nav-link" data-scroll-target="#biological-processes-2">5.0.1 Biological Processes</a></li>
  <li><a href="#cellular-components-2" id="toc-cellular-components-2" class="nav-link" data-scroll-target="#cellular-components-2">5.0.2 Cellular Components</a></li>
  <li><a href="#molecular-functions-2" id="toc-molecular-functions-2" class="nav-link" data-scroll-target="#molecular-functions-2">5.0.3 Molecular Functions</a></li>
  <li><a href="#join-ontologies-2" id="toc-join-ontologies-2" class="nav-link" data-scroll-target="#join-ontologies-2">5.0.4 Join ontologies</a></li>
  </ul></li>
  <li><a href="#fe-of-targets-significantly-positively-correlated-with-mirnas" id="toc-fe-of-targets-significantly-positively-correlated-with-mirnas" class="nav-link" data-scroll-target="#fe-of-targets-significantly-positively-correlated-with-mirnas">6 FE of targets significantly positively correlated with miRNAs</a>
  <ul class="collapse">
  <li><a href="#biological-processes-3" id="toc-biological-processes-3" class="nav-link" data-scroll-target="#biological-processes-3">6.0.1 Biological Processes</a></li>
  <li><a href="#cellular-components-3" id="toc-cellular-components-3" class="nav-link" data-scroll-target="#cellular-components-3">6.0.2 Cellular Components</a></li>
  <li><a href="#molecular-functions-3" id="toc-molecular-functions-3" class="nav-link" data-scroll-target="#molecular-functions-3">6.0.3 Molecular Functions</a></li>
  <li><a href="#join-ontologies-3" id="toc-join-ontologies-3" class="nav-link" data-scroll-target="#join-ontologies-3">6.0.4 Join ontologies</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block column-page-left" id="quarto-document-content">






<p>We’ve identified putative miRNA-mRNA interactions by using miRanda to predict miRNA-mRNA binding, then using Pearson’s correlation coefficient to evaluate coexpression of putatively binding miRNA-mRNA pairs.</p>
<p>This code performs functional enrichment analysis to investigate the function(s) of these putative miRNA targets</p>
<p><a href="https://github.com/urol-e5/deep-dive-expression/blob/main/D-Apul/code/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment.Rmd">Code</a><br>
<a href="https://github.com/urol-e5/deep-dive-expression/blob/main/D-Apul/code/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment.md">Rendered Code</a><br>
<a href="https://github.com/urol-e5/deep-dive-expression/tree/main/D-Apul/output">Output files</a></p>
<section id="results" class="level1">
<h1>Results</h1>
<p><strong>miRNA targets with significant negative correlation:</strong></p>
<table class="caption-top table">
<colgroup>
<col style="width: 14%">
<col style="width: 45%">
<col style="width: 16%">
<col style="width: 23%">
</colgroup>
<thead>
<tr class="header">
<th>GO ID</th>
<th>Term</th>
<th>Fisher p_val</th>
<th>Type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GO:0016980</td>
<td>creatinase&nbsp;activity</td>
<td>0.0035</td>
<td>Molecular Function</td>
</tr>
<tr class="even">
<td>GO:0004126</td>
<td>cytidine&nbsp;deaminase&nbsp;activity</td>
<td>0.0175</td>
<td>Molecular Function</td>
</tr>
<tr class="odd">
<td>GO:0003924</td>
<td>GTPase&nbsp;activity</td>
<td>0.0218</td>
<td>Molecular Function</td>
</tr>
<tr class="even">
<td>GO:0004966</td>
<td>galanin&nbsp;receptor&nbsp;activity</td>
<td>0.0415</td>
<td>Molecular Function</td>
</tr>
<tr class="odd">
<td>GO:0000272</td>
<td>polysaccharide catabolic process</td>
<td>0.038</td>
<td>Biological Process</td>
</tr>
<tr class="even">
<td>GO:0000082</td>
<td>G1/S transition of mitotic cell cycle</td>
<td>0.046</td>
<td>Biological Process</td>
</tr>
</tbody>
</table>
<p><strong>miRNA targets with significant positive correlation:</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>GO ID</th>
<th>Term</th>
<th>Fisher p_val</th>
<th>Type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GO:0003924</td>
<td>GTPase activity</td>
<td>0.0218</td>
<td>Molecular Function</td>
</tr>
</tbody>
</table>
</section>
<section id="code" class="level1">
<h1>Code</h1>
</section>
<section id="mrna-mirna-interactions-functional-enrichment" class="level1">
<h1>mRNA-miRNA interactions functional enrichment</h1>
<p>Code used below was created by <code>Jill Ashey</code>, modified for use with A.pulchra datasets by <code>Kathleen Durkin</code></p>
</section>
<section id="format-topgo-files" class="level1">
<h1>1 Format topGO files</h1>
<section id="read-in-and-format-annotation-files" class="level2">
<h2 class="anchored" data-anchor-id="read-in-and-format-annotation-files">1.1 Read in and format annotation files</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in Apul annotations</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>annot_locations <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/02-Apul-reference-annotation/Apulcra-genome-mRNA-IDmapping-2024_12_12.tab"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove unneeded columns </span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>annot_locations <span class="ot">&lt;-</span> annot_locations <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>X, <span class="sc">-</span>V13)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure there are no duplicate rows</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>annot_locations <span class="ot">&lt;-</span> annot_locations <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(annot_locations)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##                     V1     V3 Protein.names
## 1   ntLink_4:1155-1537 P35061   Histone H2A
## 2   ntLink_4:2660-3441 P84239    Histone H3
## 3   ntLink_4:4515-6830 P35061   Histone H2A
## 4   ntLink_4:7096-7859 P84239    Histone H3
## 5   ntLink_4:8474-9669 P35061   Histone H2A
## 6 ntLink_4:11162-11925 P84239    Histone H3
##                                     Organism Gene.Ontology..biological.process.
## 1          Acropora formosa (Staghorn coral)                                   
## 2 Urechis caupo (Innkeeper worm) (Spoonworm)                                   
## 3          Acropora formosa (Staghorn coral)                                   
## 4 Urechis caupo (Innkeeper worm) (Spoonworm)                                   
## 5          Acropora formosa (Staghorn coral)                                   
## 6 Urechis caupo (Innkeeper worm) (Spoonworm)                                   
##                                            Gene.Ontology.IDs
## 1 GO:0000786; GO:0003677; GO:0005634; GO:0030527; GO:0046982
## 2 GO:0000786; GO:0003677; GO:0005634; GO:0030527; GO:0046982
## 3 GO:0000786; GO:0003677; GO:0005634; GO:0030527; GO:0046982
## 4 GO:0000786; GO:0003677; GO:0005634; GO:0030527; GO:0046982
## 5 GO:0000786; GO:0003677; GO:0005634; GO:0030527; GO:0046982
## 6 GO:0000786; GO:0003677; GO:0005634; GO:0030527; GO:0046982</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Looks good!</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This file shows each gene as it’s genomic location. We want to use gene IDs to associate genes, so add gene IDs to this annotation table</p>
<p>Read in file that associates each mRNA genomic location with corresponding gene ID</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>mRNA_FUNids <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"../output/15-Apul-annotate-UTRs/Apul-mRNA-FUNids.txt"</span>, <span class="at">header=</span><span class="cn">FALSE</span>, <span class="at">col.names=</span><span class="fu">c</span>(<span class="st">"location"</span>, <span class="st">"type"</span>, <span class="st">"mRNA_ID"</span>, <span class="st">"gene_ID"</span>, <span class="st">"product"</span>), <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove unwanted text from parent column</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>mRNA_FUNids<span class="sc">$</span>gene_ID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Parent="</span>, <span class="st">""</span>, mRNA_FUNids<span class="sc">$</span>gene_ID)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Only need to keep mRNA location and gene ID</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>mRNA_FUNids <span class="ot">&lt;-</span> mRNA_FUNids <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(location, gene_ID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>join with annotation file</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># join</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> <span class="fu">left_join</span>(annot_locations, mRNA_FUNids, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"V1"</span> <span class="ot">=</span> <span class="st">"location"</span>))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ensure there are no duplicate rows</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> annot <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="set-up-gene2go-object" class="level2">
<h2 class="anchored" data-anchor-id="set-up-gene2go-object">1.2 Set up gene2GO object</h2>
<p>Want to isolate a list of GO terms per gene</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gene2go <span class="ot">&lt;-</span> annot <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Gene.Ontology.IDs)) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(gene_ID, Gene.Ontology.IDs)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">GO.ID =</span> Gene.Ontology.IDs)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>gene2go_list <span class="ot">&lt;-</span> <span class="fu">setNames</span>(</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">strsplit</span>(<span class="fu">as.character</span>(gene2go<span class="sc">$</span>GO.ID), <span class="st">";"</span>), </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  gene2go<span class="sc">$</span>gene_ID</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Note: I think this means genes that had a Uniprot ID but no GO terms are excluded from this analysis</p>
</section>
<section id="define-reference-set" class="level2">
<h2 class="anchored" data-anchor-id="define-reference-set">1.3 Define reference set</h2>
<p>Define reference set of genes. This should be all genes <em>found in our samples</em>, NOT all genes in the A.pulchra genome. Some genes (e.g., reproduction pathways) may not be found/expected in our samples for valid biological reasons.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in counts matrix</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>Apul_counts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/07-Apul-Hisat/Apul-gene_count_matrix.csv"</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Exclude genes with all 0 counts</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>Apul_counts <span class="ot">&lt;-</span> Apul_counts[<span class="fu">rowSums</span>(Apul_counts[, <span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]) <span class="sc">!=</span> <span class="dv">0</span>, ]</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Select gene IDs of the genes present in our samples</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> Apul_counts<span class="sc">$</span>gene_id</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(all_genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] 33624</code></pre>
<p>So we have a reference set of 33624 genes present in our samples.</p>
</section>
<section id="read-in-pccmiranda-data" class="level2">
<h2 class="anchored" data-anchor-id="read-in-pccmiranda-data">1.4 Read in PCC/miranda data</h2>
<p>This is a table of all putative miRNA-mRNA binding predicted by miRanda, plus Pearsons correlation coefficients for coexpression of each putative binding pair.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/09-Apul-mRNA-miRNA-interactions/miranda_PCC_miRNA_mRNA.csv"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>##   X         miRNA       mRNA    PCC.cor   p_value adjusted_p_value score energy
## 1 1  Cluster_5981 FUN_028147  0.6825537 0.2041707        0.9986496   146 -22.19
## 2 2 Cluster_15340 FUN_013332  0.6371070 0.2476393        0.9986496   158 -23.15
## 3 3  Cluster_5981 FUN_041253 -0.2250869 0.7158492        0.9986496   153 -20.50
## 4 4  Cluster_3366 FUN_010827  0.3671005 0.5433145        0.9986496   163 -22.14
## 5 5  Cluster_3367 FUN_010827  0.5369304 0.3507987        0.9986496   163 -22.14
## 6 6 Cluster_15340 FUN_003342  0.1096213 0.8607058        0.9986496   154 -20.65
##   query_start_end subject_start_end total_bp_shared query_similar
## 1            2 21           185 209              21        66.67%
## 2            2 20           198 220              19        68.42%
## 3            2 21           699 719              19        73.68%
## 4            2 18           346 368              16        81.25%
## 5            2 18           346 368              16        81.25%
## 6            2 20           562 585              20        65.00%
##   subject_similar
## 1          71.43%
## 2          84.21%
## 3          73.68%
## 4          93.75%
## 5          93.75%
## 6          80.00%</code></pre>
<p>Set function to select genes of interest (ie those that have pvalue &lt; 0.05)</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>topDiffGenes <span class="ot">&lt;-</span> <span class="cf">function</span>(allScore) {</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">return</span>(allScore <span class="sc">&lt;</span> <span class="fl">0.05</span>)}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="fe-of-all-targets-as-predicted-by-miranda" class="level1">
<h1>2 FE of all targets, as predicted by miranda</h1>
<p>TO DO</p>
</section>
<section id="fe-of-targets-negatively-correlated-with-mirnas-regardless-of-correlation-significance" class="level1">
<h1>3 FE of targets negatively correlated with miRNAs (regardless of correlation significance)</h1>
<p>Filter PCC miranda data</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter so that only negative correlations remain </span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>neg_corr_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PCC.cor <span class="sc">&lt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Make list of target genes for input to topGO</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes of interest - ie those targeted by miRNAs</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>target_genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(neg_corr_data<span class="sc">$</span>mRNA))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply 1 or 0 if gene is gene of interest </span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>GeneList <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(all_genes <span class="sc">%in%</span> target_genes))</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(GeneList) <span class="ot">&lt;-</span> all_genes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The following code will perform GO enrichment using the weighted Fisher’s exact test to assess whether specific GO terms are overrepresented in the genes targeted by miRNAs, regardless of correlation significance.</p>
<section id="biological-processes" class="level3">
<h3 class="anchored" data-anchor-id="biological-processes">3.0.1 Biological Processes</h3>
<p>Create <code>topGOdata</code> object, which is required for topGO analysis</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>GO_BP <span class="ot">&lt;-</span><span class="fu">new</span>(<span class="st">"topGOdata"</span>, </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">ontology=</span><span class="st">"BP"</span>, </span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene2GO=</span>gene2go_list, </span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">allGenes=</span>GeneList, </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">geneSel=</span>topDiffGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## Building most specific GOs .....

##  ( 273 GO terms found. )

## 
## Build GO DAG topology ..........

##  ( 1231 GO terms and 2377 relations. )

## 
## Annotating nodes ...............

##  ( 1427 genes annotated to the GO terms. )</code></pre>
<p>Run GO enrichment test</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>GO_BP_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_BP, <span class="at">algorithm=</span><span class="st">"weight01"</span>, <span class="at">statistic=</span><span class="st">"fisher"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 373 nontrivial nodes
##       parameters: 
##           test statistic: fisher

## 
##   Level 14:  1 nodes to be scored    (0 eliminated genes)

## 
##   Level 13:  5 nodes to be scored    (0 eliminated genes)

## 
##   Level 12:  5 nodes to be scored    (11 eliminated genes)

## 
##   Level 11:  7 nodes to be scored    (157 eliminated genes)

## 
##   Level 10:  16 nodes to be scored   (211 eliminated genes)

## 
##   Level 9:   27 nodes to be scored   (226 eliminated genes)

## 
##   Level 8:   34 nodes to be scored   (337 eliminated genes)

## 
##   Level 7:   47 nodes to be scored   (451 eliminated genes)

## 
##   Level 6:   65 nodes to be scored   (627 eliminated genes)

## 
##   Level 5:   72 nodes to be scored   (908 eliminated genes)

## 
##   Level 4:   50 nodes to be scored   (1025 eliminated genes)

## 
##   Level 3:   32 nodes to be scored   (1219 eliminated genes)

## 
##   Level 2:   11 nodes to be scored   (1343 eliminated genes)

## 
##   Level 1:   1 nodes to be scored    (1417 eliminated genes)</code></pre>
<p>Generate results table</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>GO_BP_En <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_BP, <span class="at">Fisher =</span> GO_BP_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only taking the top 100 GO terms</p>
<p>Filter by significant results</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>GO_BP_En<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_BP_En<span class="sc">$</span>Fisher)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig<span class="ot">&lt;-</span>GO_BP_En[GO_BP_En<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge <code>GO_BP_En_sig</code> with GO and gene info.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate GO terms </span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>neg_cor_gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(GO.ID, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure GO terms in both datasets are formatted similarly (trim whitespaces)</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>neg_cor_gene2go<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(neg_cor_gene2go<span class="sc">$</span>GO.ID)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(GO_BP_En_sig<span class="sc">$</span>GO.ID)</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Join the datasets based on GO term</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene <span class="ot">&lt;-</span> neg_cor_gene2go <span class="sc">%&gt;%</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(GO_BP_En_sig, <span class="at">by =</span> <span class="st">"GO.ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ontology column </span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene<span class="sc">$</span>ontology <span class="ot">&lt;-</span> <span class="st">"Biological Processes"</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only unique rows </span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene <span class="ot">&lt;-</span> GO_BP_En_sig_gene <span class="sc">%&gt;%</span> <span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="cellular-components" class="level3">
<h3 class="anchored" data-anchor-id="cellular-components">3.0.2 Cellular Components</h3>
</section>
<section id="molecular-functions" class="level3">
<h3 class="anchored" data-anchor-id="molecular-functions">3.0.3 Molecular Functions</h3>
<p>Create <code>topGOdata</code> object, which is required for topGO analysis</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>GO_MF <span class="ot">&lt;-</span><span class="fu">new</span>(<span class="st">"topGOdata"</span>, </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">ontology=</span><span class="st">"MF"</span>, </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene2GO=</span>gene2go_list, </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">allGenes=</span>GeneList, </span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">geneSel=</span>topDiffGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## Building most specific GOs .....

##  ( 461 GO terms found. )

## 
## Build GO DAG topology ..........

##  ( 910 GO terms and 1191 relations. )

## 
## Annotating nodes ...............

##  ( 2843 genes annotated to the GO terms. )</code></pre>
<p>Run GO enrichment test</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>GO_MF_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_MF, <span class="at">algorithm=</span><span class="st">"weight01"</span>, <span class="at">statistic=</span><span class="st">"fisher"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 243 nontrivial nodes
##       parameters: 
##           test statistic: fisher

## 
##   Level 10:  4 nodes to be scored    (0 eliminated genes)

## 
##   Level 9:   12 nodes to be scored   (0 eliminated genes)

## 
##   Level 8:   19 nodes to be scored   (146 eliminated genes)

## 
##   Level 7:   30 nodes to be scored   (481 eliminated genes)

## 
##   Level 6:   47 nodes to be scored   (621 eliminated genes)

## 
##   Level 5:   47 nodes to be scored   (930 eliminated genes)

## 
##   Level 4:   50 nodes to be scored   (1352 eliminated genes)

## 
##   Level 3:   24 nodes to be scored   (2004 eliminated genes)

## 
##   Level 2:   9 nodes to be scored    (2371 eliminated genes)

## 
##   Level 1:   1 nodes to be scored    (2681 eliminated genes)</code></pre>
<p>Generate results table</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>GO_MF_En <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_MF, <span class="at">Fisher =</span> GO_MF_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only taking the top 100 GO terms</p>
<p>Filter by significant results</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>GO_MF_En<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_MF_En<span class="sc">$</span>Fisher)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig<span class="ot">&lt;-</span>GO_MF_En[GO_MF_En<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge <code>GO_MF_En_sig</code> with GO and gene info.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate GO terms </span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>neg_cor_gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(GO.ID, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure GO terms in both datasets are formatted similarly (trim whitespaces)</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>neg_cor_gene2go<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(neg_cor_gene2go<span class="sc">$</span>GO.ID)</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(GO_MF_En_sig<span class="sc">$</span>GO.ID)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Join the datasets based on GO term</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene <span class="ot">&lt;-</span> neg_cor_gene2go <span class="sc">%&gt;%</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(GO_MF_En_sig, <span class="at">by =</span> <span class="st">"GO.ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ontology column </span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene<span class="sc">$</span>ontology <span class="ot">&lt;-</span> <span class="st">"Molecular Functions"</span></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only unique rows </span></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(GO_MF_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="join-ontologies" class="level3">
<h3 class="anchored" data-anchor-id="join-ontologies">3.0.4 Join ontologies</h3>
<p>Bind so there is a df that has significantly enriched GO terms for all ontologies</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>GO_neg_corr_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(GO_BP_En_sig_gene, GO_MF_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge with <code>GO_neg_corr_df</code></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> GO_neg_corr_df <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(neg_corr_data, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_ID"</span> <span class="ot">=</span> <span class="st">"mRNA"</span>)) <span class="co">#%&gt;%</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(direction = ifelse(PCC.cor &gt; 0, "Positive", "Negative")) #%&gt;%</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(ontology != "Cellular Components") #%&gt;%</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(p_value &lt; 0.1)</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Save as csv </span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(test, <span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_neg_corr_target_enrichment.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Plot!</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test, <span class="fu">aes</span>(<span class="at">x =</span> Term, <span class="at">y =</span> Fisher, <span class="at">fill =</span> p_value)) <span class="sc">+</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0, 1) +</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 </span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> </span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_size(range = c(2, 20)) + </span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fisher p-value"</span>) <span class="sc">+</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">24</span>) <span class="sc">+</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(ontology), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(); plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="09.1-Apul-mRNA-miRNA-interactions-functional-enrichment_files/figure-gfm/unnamed-chunk-23-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot </span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_neg_corr_target_enrichment.pdf"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">35</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_neg_corr_target_enrichment.png"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">35</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Examine the top 10 most significant GO terms for BP and MF</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to get top 5 unique terms</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>get_top_10_unique <span class="ot">&lt;-</span> <span class="cf">function</span>(data, ontology_type) {</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  data <span class="sc">%&gt;%</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ontology <span class="sc">==</span> ontology_type) <span class="sc">%&gt;%</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Fisher) <span class="sc">%&gt;%</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Term, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 5 unique Biological Processes</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>top_10_BP <span class="ot">&lt;-</span> <span class="fu">get_top_10_unique</span>(test, <span class="st">"Biological Processes"</span>)</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 5 unique Molecular Functions</span></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>top_10_MF <span class="ot">&lt;-</span> <span class="fu">get_top_10_unique</span>(test, <span class="st">"Molecular Functions"</span>)</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>top_10_combined <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(top_10_BP, top_10_MF)</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(top_10_combined<span class="sc">$</span>Term)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] "auxin-activated signaling pathway"         
## [2] "protein ubiquitination"                    
## [3] "MAPK cascade"                              
## [4] "alternative mRNA splicing, via spliceosome"
## [5] "chromatin binding"</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot </span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>top_10_combined <span class="ot">&lt;-</span> top_10_combined <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Fisher)) <span class="sc">%&gt;%</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Term =</span> <span class="fu">factor</span>(Term, <span class="at">levels =</span> <span class="fu">unique</span>(Term)))</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(top_10_combined, <span class="fu">aes</span>(<span class="at">x =</span> Term, <span class="at">y =</span> Fisher, <span class="at">fill =</span> p_value)) <span class="sc">+</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0, 1) +</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 </span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span> </span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">20</span>)) <span class="sc">+</span> </span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fisher p-value"</span>) <span class="sc">+</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">35</span>) <span class="sc">+</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(ontology), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(); plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="09.1-Apul-mRNA-miRNA-interactions-functional-enrichment_files/figure-gfm/unnamed-chunk-24-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot </span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/top10GO_neg_corr_target_enrichment.pdf"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/top10GO_neg_corr_target_enrichment.png"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="fe-of-targets-positively-correlated-with-mirnas-regardless-of-correlation-significance" class="level1">
<h1>4 FE of targets positively correlated with miRNAs (regardless of correlation significance)</h1>
<p>Filter PCC miranda data</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter so that only positive correlations remain </span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>pos_corr_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PCC.cor <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Make list of target genes for input to topGO</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes of interest - ie those targeted by miRNAs</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>target_genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(pos_corr_data<span class="sc">$</span>mRNA))</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply 1 or 0 if gene is gene of interest </span></span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>GeneList <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(all_genes <span class="sc">%in%</span> target_genes))</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(GeneList) <span class="ot">&lt;-</span> all_genes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The following code will perform GO enrichment using the weighted Fisher’s exact test to assess whether specific GO terms are overrepresented in the genes targeted by miRNAs with positively correlated expression, regardless of correlation significance.</p>
<section id="biological-processes-1" class="level3">
<h3 class="anchored" data-anchor-id="biological-processes-1">4.0.1 Biological Processes</h3>
<p>Create <code>topGOdata</code> object, which is required for topGO analysis</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>GO_BP <span class="ot">&lt;-</span><span class="fu">new</span>(<span class="st">"topGOdata"</span>, </span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">ontology=</span><span class="st">"BP"</span>, </span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene2GO=</span>gene2go_list, </span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">allGenes=</span>GeneList, </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">geneSel=</span>topDiffGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## Building most specific GOs .....

##  ( 273 GO terms found. )

## 
## Build GO DAG topology ..........

##  ( 1231 GO terms and 2377 relations. )

## 
## Annotating nodes ...............

##  ( 1427 genes annotated to the GO terms. )</code></pre>
<p>Run GO enrichment test</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>GO_BP_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_BP, <span class="at">algorithm=</span><span class="st">"weight01"</span>, <span class="at">statistic=</span><span class="st">"fisher"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 456 nontrivial nodes
##       parameters: 
##           test statistic: fisher

## 
##   Level 13:  2 nodes to be scored    (0 eliminated genes)

## 
##   Level 12:  9 nodes to be scored    (0 eliminated genes)

## 
##   Level 11:  13 nodes to be scored   (141 eliminated genes)

## 
##   Level 10:  21 nodes to be scored   (219 eliminated genes)

## 
##   Level 9:   38 nodes to be scored   (235 eliminated genes)

## 
##   Level 8:   50 nodes to be scored   (331 eliminated genes)

## 
##   Level 7:   58 nodes to be scored   (466 eliminated genes)

## 
##   Level 6:   78 nodes to be scored   (664 eliminated genes)

## 
##   Level 5:   85 nodes to be scored   (939 eliminated genes)

## 
##   Level 4:   53 nodes to be scored   (1067 eliminated genes)

## 
##   Level 3:   37 nodes to be scored   (1276 eliminated genes)

## 
##   Level 2:   11 nodes to be scored   (1342 eliminated genes)

## 
##   Level 1:   1 nodes to be scored    (1402 eliminated genes)</code></pre>
<p>Generate results table</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>GO_BP_En <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_BP, <span class="at">Fisher =</span> GO_BP_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only taking the top 100 GO terms</p>
<p>Filter by significant results</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>GO_BP_En<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_BP_En<span class="sc">$</span>Fisher)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig<span class="ot">&lt;-</span>GO_BP_En[GO_BP_En<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge <code>GO_BP_En_sig</code> with GO and gene info.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate GO terms </span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>pos_cor_gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(GO.ID, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure GO terms in both datasets are formatted similarly (trim whitespaces)</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>pos_cor_gene2go<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(pos_cor_gene2go<span class="sc">$</span>GO.ID)</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(GO_BP_En_sig<span class="sc">$</span>GO.ID)</span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Join the datasets based on GO term</span></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene <span class="ot">&lt;-</span> pos_cor_gene2go <span class="sc">%&gt;%</span></span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(GO_BP_En_sig, <span class="at">by =</span> <span class="st">"GO.ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ontology column </span></span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene<span class="sc">$</span>ontology <span class="ot">&lt;-</span> <span class="st">"Biological Processes"</span></span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only unique rows </span></span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(GO_BP_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="cellular-components-1" class="level3">
<h3 class="anchored" data-anchor-id="cellular-components-1">4.0.2 Cellular Components</h3>
</section>
<section id="molecular-functions-1" class="level3">
<h3 class="anchored" data-anchor-id="molecular-functions-1">4.0.3 Molecular Functions</h3>
<p>Create <code>topGOdata</code> object, which is required for topGO analysis</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>GO_MF <span class="ot">&lt;-</span><span class="fu">new</span>(<span class="st">"topGOdata"</span>, </span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">ontology=</span><span class="st">"MF"</span>, </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene2GO=</span>gene2go_list, </span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">allGenes=</span>GeneList, </span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">geneSel=</span>topDiffGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## Building most specific GOs .....

##  ( 461 GO terms found. )

## 
## Build GO DAG topology ..........

##  ( 910 GO terms and 1191 relations. )

## 
## Annotating nodes ...............

##  ( 2843 genes annotated to the GO terms. )</code></pre>
<p>Run GO enrichment test</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>GO_MF_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_MF, <span class="at">algorithm=</span><span class="st">"weight01"</span>, <span class="at">statistic=</span><span class="st">"fisher"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 320 nontrivial nodes
##       parameters: 
##           test statistic: fisher

## 
##   Level 13:  1 nodes to be scored    (0 eliminated genes)

## 
##   Level 12:  1 nodes to be scored    (0 eliminated genes)

## 
##   Level 11:  1 nodes to be scored    (2 eliminated genes)

## 
##   Level 10:  5 nodes to be scored    (2 eliminated genes)

## 
##   Level 9:   12 nodes to be scored   (4 eliminated genes)

## 
##   Level 8:   20 nodes to be scored   (155 eliminated genes)

## 
##   Level 7:   38 nodes to be scored   (499 eliminated genes)

## 
##   Level 6:   71 nodes to be scored   (635 eliminated genes)

## 
##   Level 5:   69 nodes to be scored   (1005 eliminated genes)

## 
##   Level 4:   63 nodes to be scored   (1501 eliminated genes)

## 
##   Level 3:   27 nodes to be scored   (2192 eliminated genes)

## 
##   Level 2:   11 nodes to be scored   (2507 eliminated genes)

## 
##   Level 1:   1 nodes to be scored    (2692 eliminated genes)</code></pre>
<p>Generate results table</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>GO_MF_En <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_MF, <span class="at">Fisher =</span> GO_MF_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only taking the top 100 GO terms</p>
<p>Filter by significant results</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>GO_MF_En<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_MF_En<span class="sc">$</span>Fisher)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig<span class="ot">&lt;-</span>GO_MF_En[GO_MF_En<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge <code>GO_MF_En_sig</code> with GO and gene info.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate GO terms </span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>pos_cor_gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(GO.ID, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure GO terms in both datasets are formatted similarly (trim whitespaces)</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>pos_cor_gene2go<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(pos_cor_gene2go<span class="sc">$</span>GO.ID)</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(GO_MF_En_sig<span class="sc">$</span>GO.ID)</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Join the datasets based on GO term</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene <span class="ot">&lt;-</span> pos_cor_gene2go <span class="sc">%&gt;%</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(GO_MF_En_sig, <span class="at">by =</span> <span class="st">"GO.ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ontology column </span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene<span class="sc">$</span>ontology <span class="ot">&lt;-</span> <span class="st">"Molecular Functions"</span></span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only unique rows </span></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(GO_MF_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="join-ontologies-1" class="level3">
<h3 class="anchored" data-anchor-id="join-ontologies-1">4.0.4 Join ontologies</h3>
<p>Bind so there is a df that has significantly enriched GO terms for all ontologies</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>GO_pos_corr_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(GO_BP_En_sig_gene, GO_MF_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge with <code>GO_pos_corr_df</code></p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> GO_pos_corr_df <span class="sc">%&gt;%</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(pos_corr_data, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_ID"</span> <span class="ot">=</span> <span class="st">"mRNA"</span>)) <span class="co">#%&gt;%</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(direction = ifelse(PCC.cor &gt; 0, "Positive", "Negative")) #%&gt;%</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(ontology != "Cellular Components") #%&gt;%</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(p_value &lt; 0.1)</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Save as csv </span></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(test, <span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_pos_corr_target_enrichment.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Plot!</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test, <span class="fu">aes</span>(<span class="at">x =</span> Term, <span class="at">y =</span> Fisher, <span class="at">fill =</span> p_value)) <span class="sc">+</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0, 1) +</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 </span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> </span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_size(range = c(2, 20)) + </span></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fisher p-value"</span>) <span class="sc">+</span></span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">24</span>) <span class="sc">+</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(ontology), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(); plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="09.1-Apul-mRNA-miRNA-interactions-functional-enrichment_files/figure-gfm/unnamed-chunk-38-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot </span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_pos_corr_target_enrichment.pdf"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">35</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_pos_corr_target_enrichment.png"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">35</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Examine the top 10 most significant GO terms for BP and MF</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to get top 5 unique terms</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>get_top_10_unique <span class="ot">&lt;-</span> <span class="cf">function</span>(data, ontology_type) {</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  data <span class="sc">%&gt;%</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ontology <span class="sc">==</span> ontology_type) <span class="sc">%&gt;%</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Fisher) <span class="sc">%&gt;%</span></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Term, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>)</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 5 unique Biological Processes</span></span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>top_10_BP <span class="ot">&lt;-</span> <span class="fu">get_top_10_unique</span>(test, <span class="st">"Biological Processes"</span>)</span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 5 unique Molecular Functions</span></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>top_10_MF <span class="ot">&lt;-</span> <span class="fu">get_top_10_unique</span>(test, <span class="st">"Molecular Functions"</span>)</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results</span></span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>top_10_combined <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(top_10_BP, top_10_MF)</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(top_10_combined<span class="sc">$</span>Term)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] "polyamine catabolic process"                           
## [2] "negative regulation of protein phosphorylation"        
## [3] "RNA polymerase II cis-regulatory region sequence-sp..."
## [4] "cytidine deaminase activity"                           
## [5] "glycosaminoglycan binding"</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot </span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>top_10_combined <span class="ot">&lt;-</span> top_10_combined <span class="sc">%&gt;%</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Fisher)) <span class="sc">%&gt;%</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Term =</span> <span class="fu">factor</span>(Term, <span class="at">levels =</span> <span class="fu">unique</span>(Term)))</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(top_10_combined, <span class="fu">aes</span>(<span class="at">x =</span> Term, <span class="at">y =</span> Fisher, <span class="at">fill =</span> p_value)) <span class="sc">+</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0, 1) +</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 </span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span> </span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">20</span>)) <span class="sc">+</span> </span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fisher p-value"</span>) <span class="sc">+</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">35</span>) <span class="sc">+</span></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(ontology), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(); plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="09.1-Apul-mRNA-miRNA-interactions-functional-enrichment_files/figure-gfm/unnamed-chunk-39-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot </span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/top10GO_pos_corr_target_enrichment.pdf"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/top10GO_pos_corr_target_enrichment.png"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="fe-of-targets-significantly-negatively-correlated-with-mirnas" class="level1">
<h1>5 FE of targets significantly negatively correlated with miRNAs</h1>
<p>Filter PCC miranda data</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter so that only significant negative correlations remain </span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>sig_neg_corr_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PCC.cor <span class="sc">&lt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_value <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Make list of target genes for input to topGO</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes of interest - ie those targeted by miRNAs</span></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>target_genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(sig_neg_corr_data<span class="sc">$</span>mRNA))</span>
<span id="cb60-9"><a href="#cb60-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-10"><a href="#cb60-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply 1 or 0 if gene is gene of interest </span></span>
<span id="cb60-11"><a href="#cb60-11" aria-hidden="true" tabindex="-1"></a>GeneList <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(all_genes <span class="sc">%in%</span> target_genes))</span>
<span id="cb60-12"><a href="#cb60-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(GeneList) <span class="ot">&lt;-</span> all_genes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The following code will perform GO enrichment using the weighted Fisher’s exact test to assess whether specific GO terms are overrepresented in the genes targeted by miRNAs with significantly negatively correlated expression.</p>
<section id="biological-processes-2" class="level3">
<h3 class="anchored" data-anchor-id="biological-processes-2">5.0.1 Biological Processes</h3>
<p>Create <code>topGOdata</code> object, which is required for topGO analysis</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>GO_BP <span class="ot">&lt;-</span><span class="fu">new</span>(<span class="st">"topGOdata"</span>, </span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">ontology=</span><span class="st">"BP"</span>, </span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene2GO=</span>gene2go_list, </span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">allGenes=</span>GeneList, </span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">geneSel=</span>topDiffGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## Building most specific GOs .....

##  ( 273 GO terms found. )

## 
## Build GO DAG topology ..........

##  ( 1231 GO terms and 2377 relations. )

## 
## Annotating nodes ...............

##  ( 1427 genes annotated to the GO terms. )</code></pre>
<p>Run GO enrichment test</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>GO_BP_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_BP, <span class="at">algorithm=</span><span class="st">"weight01"</span>, <span class="at">statistic=</span><span class="st">"fisher"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 40 nontrivial nodes
##       parameters: 
##           test statistic: fisher

## 
##   Level 10:  1 nodes to be scored    (0 eliminated genes)

## 
##   Level 9:   1 nodes to be scored    (0 eliminated genes)

## 
##   Level 8:   1 nodes to be scored    (30 eliminated genes)

## 
##   Level 7:   3 nodes to be scored    (37 eliminated genes)

## 
##   Level 6:   5 nodes to be scored    (37 eliminated genes)

## 
##   Level 5:   8 nodes to be scored    (181 eliminated genes)

## 
##   Level 4:   7 nodes to be scored    (253 eliminated genes)

## 
##   Level 3:   8 nodes to be scored    (369 eliminated genes)

## 
##   Level 2:   5 nodes to be scored    (708 eliminated genes)

## 
##   Level 1:   1 nodes to be scored    (806 eliminated genes)</code></pre>
<p>Generate results table</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>GO_BP_En <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_BP, <span class="at">Fisher =</span> GO_BP_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only taking the top 100 GO terms</p>
<p>Filter by significant results</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>GO_BP_En<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_BP_En<span class="sc">$</span>Fisher)</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig<span class="ot">&lt;-</span>GO_BP_En[GO_BP_En<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge <code>GO_BP_En_sig</code> with GO and gene info.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate GO terms </span></span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>sig_neg_cor_gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(GO.ID, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure GO terms in both datasets are formatted similarly (trim whitespaces)</span></span>
<span id="cb67-6"><a href="#cb67-6" aria-hidden="true" tabindex="-1"></a>sig_neg_cor_gene2go<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(sig_neg_cor_gene2go<span class="sc">$</span>GO.ID)</span>
<span id="cb67-7"><a href="#cb67-7" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(GO_BP_En_sig<span class="sc">$</span>GO.ID)</span>
<span id="cb67-8"><a href="#cb67-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-9"><a href="#cb67-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Join the datasets based on GO term</span></span>
<span id="cb67-10"><a href="#cb67-10" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene <span class="ot">&lt;-</span> sig_neg_cor_gene2go <span class="sc">%&gt;%</span></span>
<span id="cb67-11"><a href="#cb67-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(GO_BP_En_sig, <span class="at">by =</span> <span class="st">"GO.ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb67-12"><a href="#cb67-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb67-13"><a href="#cb67-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-14"><a href="#cb67-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ontology column </span></span>
<span id="cb67-15"><a href="#cb67-15" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene<span class="sc">$</span>ontology <span class="ot">&lt;-</span> <span class="st">"Biological Processes"</span></span>
<span id="cb67-16"><a href="#cb67-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-17"><a href="#cb67-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only unique rows </span></span>
<span id="cb67-18"><a href="#cb67-18" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(GO_BP_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="cellular-components-2" class="level3">
<h3 class="anchored" data-anchor-id="cellular-components-2">5.0.2 Cellular Components</h3>
</section>
<section id="molecular-functions-2" class="level3">
<h3 class="anchored" data-anchor-id="molecular-functions-2">5.0.3 Molecular Functions</h3>
<p>Create <code>topGOdata</code> object, which is required for topGO analysis</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>GO_MF <span class="ot">&lt;-</span><span class="fu">new</span>(<span class="st">"topGOdata"</span>, </span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">ontology=</span><span class="st">"MF"</span>, </span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene2GO=</span>gene2go_list, </span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">allGenes=</span>GeneList, </span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">geneSel=</span>topDiffGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## Building most specific GOs .....

##  ( 461 GO terms found. )

## 
## Build GO DAG topology ..........

##  ( 910 GO terms and 1191 relations. )

## 
## Annotating nodes ...............

##  ( 2843 genes annotated to the GO terms. )</code></pre>
<p>Run GO enrichment test</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>GO_MF_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_MF, <span class="at">algorithm=</span><span class="st">"weight01"</span>, <span class="at">statistic=</span><span class="st">"fisher"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 60 nontrivial nodes
##       parameters: 
##           test statistic: fisher

## 
##   Level 10:  1 nodes to be scored    (0 eliminated genes)

## 
##   Level 9:   3 nodes to be scored    (0 eliminated genes)

## 
##   Level 8:   4 nodes to be scored    (101 eliminated genes)

## 
##   Level 7:   6 nodes to be scored    (145 eliminated genes)

## 
##   Level 6:   10 nodes to be scored   (238 eliminated genes)

## 
##   Level 5:   10 nodes to be scored   (469 eliminated genes)

## 
##   Level 4:   13 nodes to be scored   (518 eliminated genes)

## 
##   Level 3:   8 nodes to be scored    (862 eliminated genes)

## 
##   Level 2:   4 nodes to be scored    (1296 eliminated genes)

## 
##   Level 1:   1 nodes to be scored    (2156 eliminated genes)</code></pre>
<p>Generate results table</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>GO_MF_En <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_MF, <span class="at">Fisher =</span> GO_MF_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only taking the top 100 GO terms</p>
<p>Filter by significant results</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>GO_MF_En<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_MF_En<span class="sc">$</span>Fisher)</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig<span class="ot">&lt;-</span>GO_MF_En[GO_MF_En<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge <code>GO_MF_En_sig</code> with GO and gene info.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate GO terms </span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>sig_neg_cor_gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(GO.ID, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure GO terms in both datasets are formatted similarly (trim whitespaces)</span></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>sig_neg_cor_gene2go<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(sig_neg_cor_gene2go<span class="sc">$</span>GO.ID)</span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(GO_MF_En_sig<span class="sc">$</span>GO.ID)</span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Join the datasets based on GO term</span></span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene <span class="ot">&lt;-</span> sig_neg_cor_gene2go <span class="sc">%&gt;%</span></span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(GO_MF_En_sig, <span class="at">by =</span> <span class="st">"GO.ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ontology column </span></span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene<span class="sc">$</span>ontology <span class="ot">&lt;-</span> <span class="st">"Molecular Functions"</span></span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only unique rows </span></span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(GO_MF_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="join-ontologies-2" class="level3">
<h3 class="anchored" data-anchor-id="join-ontologies-2">5.0.4 Join ontologies</h3>
<p>Bind so there is a df that has significantly enriched GO terms for all ontologies</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>GO_sig_neg_corr_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(GO_BP_En_sig_gene, GO_MF_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge with <code>GO_sig_neg_corr_df</code></p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> GO_sig_neg_corr_df <span class="sc">%&gt;%</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(sig_neg_corr_data, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_ID"</span> <span class="ot">=</span> <span class="st">"mRNA"</span>)) <span class="co">#%&gt;%</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(direction = ifelse(PCC.cor &gt; 0, "Positive", "Negative")) #%&gt;%</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(ontology != "Cellular Components") #%&gt;%</span></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(p_value &lt; 0.1)</span></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Save as csv </span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(test, <span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_sig_neg_corr_target_enrichment.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Plot!</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test, <span class="fu">aes</span>(<span class="at">x =</span> Term, <span class="at">y =</span> Fisher, <span class="at">fill =</span> p_value)) <span class="sc">+</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0, 1) +</span></span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+</span></span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 </span></span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> </span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_size(range = c(2, 20)) + </span></span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fisher p-value"</span>) <span class="sc">+</span></span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">24</span>) <span class="sc">+</span></span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(ontology), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb77-11"><a href="#cb77-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(); plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="09.1-Apul-mRNA-miRNA-interactions-functional-enrichment_files/figure-gfm/unnamed-chunk-53-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot </span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_sig_neg_corr_target_enrichment.pdf"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">35</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_sig_neg_corr_target_enrichment.png"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">35</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Examine the top 10 most significant GO terms for BP and MF</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to get top 5 unique terms</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>get_top_10_unique <span class="ot">&lt;-</span> <span class="cf">function</span>(data, ontology_type) {</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  data <span class="sc">%&gt;%</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ontology <span class="sc">==</span> ontology_type) <span class="sc">%&gt;%</span></span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Fisher) <span class="sc">%&gt;%</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Term, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>)</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 5 unique Biological Processes</span></span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>top_10_BP <span class="ot">&lt;-</span> <span class="fu">get_top_10_unique</span>(test, <span class="st">"Biological Processes"</span>)</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 5 unique Molecular Functions</span></span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a>top_10_MF <span class="ot">&lt;-</span> <span class="fu">get_top_10_unique</span>(test, <span class="st">"Molecular Functions"</span>)</span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results</span></span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a>top_10_combined <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(top_10_BP, top_10_MF)</span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(top_10_combined<span class="sc">$</span>Term)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] "polysaccharide catabolic process"     
## [2] "G1/S transition of mitotic cell cycle"
## [3] "creatinase activity"                  
## [4] "cytidine deaminase activity"          
## [5] "GTPase activity"                      
## [6] "galanin receptor activity"</code></pre>
<div class="sourceCode" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot </span></span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>top_10_combined <span class="ot">&lt;-</span> top_10_combined <span class="sc">%&gt;%</span></span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Fisher)) <span class="sc">%&gt;%</span></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Term =</span> <span class="fu">factor</span>(Term, <span class="at">levels =</span> <span class="fu">unique</span>(Term)))</span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(top_10_combined, <span class="fu">aes</span>(<span class="at">x =</span> Term, <span class="at">y =</span> Fisher, <span class="at">fill =</span> p_value)) <span class="sc">+</span></span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0, 1) +</span></span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+</span></span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 </span></span>
<span id="cb81-10"><a href="#cb81-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span> </span>
<span id="cb81-11"><a href="#cb81-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">20</span>)) <span class="sc">+</span> </span>
<span id="cb81-12"><a href="#cb81-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb81-13"><a href="#cb81-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fisher p-value"</span>) <span class="sc">+</span></span>
<span id="cb81-14"><a href="#cb81-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">35</span>) <span class="sc">+</span></span>
<span id="cb81-15"><a href="#cb81-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(ontology), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb81-16"><a href="#cb81-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(); plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="09.1-Apul-mRNA-miRNA-interactions-functional-enrichment_files/figure-gfm/unnamed-chunk-54-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot </span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/top10GO_sig_neg_corr_target_enrichment.pdf"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/top10GO_sig_neg_corr_target_enrichment.png"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="fe-of-targets-significantly-positively-correlated-with-mirnas" class="level1">
<h1>6 FE of targets significantly positively correlated with miRNAs</h1>
<p>Filter PCC miranda data</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter so that only significant negative correlations remain </span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>sig_pos_corr_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PCC.cor <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(p_value <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Make list of target genes for input to topGO</span></span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Genes of interest - ie those targeted by miRNAs</span></span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>target_genes <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(sig_neg_corr_data<span class="sc">$</span>mRNA))</span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply 1 or 0 if gene is gene of interest </span></span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>GeneList <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.integer</span>(all_genes <span class="sc">%in%</span> target_genes))</span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(GeneList) <span class="ot">&lt;-</span> all_genes</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>The following code will perform GO enrichment using the weighted Fisher’s exact test to assess whether specific GO terms are overrepresented in the genes targeted by miRNAs with significantly positively correlated expression.</p>
<section id="biological-processes-3" class="level3">
<h3 class="anchored" data-anchor-id="biological-processes-3">6.0.1 Biological Processes</h3>
<p>Create <code>topGOdata</code> object, which is required for topGO analysis</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>GO_BP <span class="ot">&lt;-</span><span class="fu">new</span>(<span class="st">"topGOdata"</span>, </span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">ontology=</span><span class="st">"BP"</span>, </span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene2GO=</span>gene2go_list, </span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">allGenes=</span>GeneList, </span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">geneSel=</span>topDiffGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## Building most specific GOs .....

##  ( 273 GO terms found. )

## 
## Build GO DAG topology ..........

##  ( 1231 GO terms and 2377 relations. )

## 
## Annotating nodes ...............

##  ( 1427 genes annotated to the GO terms. )</code></pre>
<p>Run GO enrichment test</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>GO_BP_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_BP, <span class="at">algorithm=</span><span class="st">"weight01"</span>, <span class="at">statistic=</span><span class="st">"fisher"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 40 nontrivial nodes
##       parameters: 
##           test statistic: fisher

## 
##   Level 10:  1 nodes to be scored    (0 eliminated genes)

## 
##   Level 9:   1 nodes to be scored    (0 eliminated genes)

## 
##   Level 8:   1 nodes to be scored    (30 eliminated genes)

## 
##   Level 7:   3 nodes to be scored    (37 eliminated genes)

## 
##   Level 6:   5 nodes to be scored    (37 eliminated genes)

## 
##   Level 5:   8 nodes to be scored    (181 eliminated genes)

## 
##   Level 4:   7 nodes to be scored    (253 eliminated genes)

## 
##   Level 3:   8 nodes to be scored    (369 eliminated genes)

## 
##   Level 2:   5 nodes to be scored    (708 eliminated genes)

## 
##   Level 1:   1 nodes to be scored    (806 eliminated genes)</code></pre>
<p>Generate results table</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>GO_BP_En <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_BP, <span class="at">Fisher =</span> GO_BP_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only taking the top 100 GO terms</p>
<p>Filter by significant results</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>GO_BP_En<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_BP_En<span class="sc">$</span>Fisher)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig<span class="ot">&lt;-</span>GO_BP_En[GO_BP_En<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge <code>GO_BP_En_sig</code> with GO and gene info.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate GO terms </span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>sig_pos_cor_gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(GO.ID, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure GO terms in both datasets are formatted similarly (trim whitespaces)</span></span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>sig_pos_cor_gene2go<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(sig_pos_cor_gene2go<span class="sc">$</span>GO.ID)</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(GO_BP_En_sig<span class="sc">$</span>GO.ID)</span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Join the datasets based on GO term</span></span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene <span class="ot">&lt;-</span> sig_pos_cor_gene2go <span class="sc">%&gt;%</span></span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(GO_BP_En_sig, <span class="at">by =</span> <span class="st">"GO.ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ontology column </span></span>
<span id="cb90-15"><a href="#cb90-15" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene<span class="sc">$</span>ontology <span class="ot">&lt;-</span> <span class="st">"Biological Processes"</span></span>
<span id="cb90-16"><a href="#cb90-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-17"><a href="#cb90-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only unique rows </span></span>
<span id="cb90-18"><a href="#cb90-18" aria-hidden="true" tabindex="-1"></a>GO_BP_En_sig_gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(GO_BP_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="cellular-components-3" class="level3">
<h3 class="anchored" data-anchor-id="cellular-components-3">6.0.2 Cellular Components</h3>
</section>
<section id="molecular-functions-3" class="level3">
<h3 class="anchored" data-anchor-id="molecular-functions-3">6.0.3 Molecular Functions</h3>
<p>Create <code>topGOdata</code> object, which is required for topGO analysis</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>GO_MF <span class="ot">&lt;-</span><span class="fu">new</span>(<span class="st">"topGOdata"</span>, </span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>            <span class="at">ontology=</span><span class="st">"MF"</span>, </span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">gene2GO=</span>gene2go_list, </span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">allGenes=</span>GeneList, </span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">annot =</span> annFUN.gene2GO, </span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">geneSel=</span>topDiffGenes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
## Building most specific GOs .....

##  ( 461 GO terms found. )

## 
## Build GO DAG topology ..........

##  ( 910 GO terms and 1191 relations. )

## 
## Annotating nodes ...............

##  ( 2843 genes annotated to the GO terms. )</code></pre>
<p>Run GO enrichment test</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a>GO_MF_FE <span class="ot">&lt;-</span> <span class="fu">runTest</span>(GO_MF, <span class="at">algorithm=</span><span class="st">"weight01"</span>, <span class="at">statistic=</span><span class="st">"fisher"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## 
##           -- Weight01 Algorithm -- 
## 
##       the algorithm is scoring 60 nontrivial nodes
##       parameters: 
##           test statistic: fisher

## 
##   Level 10:  1 nodes to be scored    (0 eliminated genes)

## 
##   Level 9:   3 nodes to be scored    (0 eliminated genes)

## 
##   Level 8:   4 nodes to be scored    (101 eliminated genes)

## 
##   Level 7:   6 nodes to be scored    (145 eliminated genes)

## 
##   Level 6:   10 nodes to be scored   (238 eliminated genes)

## 
##   Level 5:   10 nodes to be scored   (469 eliminated genes)

## 
##   Level 4:   13 nodes to be scored   (518 eliminated genes)

## 
##   Level 3:   8 nodes to be scored    (862 eliminated genes)

## 
##   Level 2:   4 nodes to be scored    (1296 eliminated genes)

## 
##   Level 1:   1 nodes to be scored    (2156 eliminated genes)</code></pre>
<p>Generate results table</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>GO_MF_En <span class="ot">&lt;-</span> <span class="fu">GenTable</span>(GO_MF, <span class="at">Fisher =</span> GO_MF_FE, <span class="at">orderBy =</span> <span class="st">"Fisher"</span>,  <span class="at">topNodes =</span> <span class="dv">100</span>, <span class="at">numChar =</span> <span class="dv">51</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Only taking the top 100 GO terms</p>
<p>Filter by significant results</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a>GO_MF_En<span class="sc">$</span>Fisher<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(GO_MF_En<span class="sc">$</span>Fisher)</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig<span class="ot">&lt;-</span>GO_MF_En[GO_MF_En<span class="sc">$</span>Fisher<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge <code>GO_MF_En_sig</code> with GO and gene info.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separate GO terms </span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>sig_pos_cor_gene2go <span class="ot">&lt;-</span> gene2go <span class="sc">%&gt;%</span></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(GO.ID, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure GO terms in both datasets are formatted similarly (trim whitespaces)</span></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>sig_pos_cor_gene2go<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(sig_pos_cor_gene2go<span class="sc">$</span>GO.ID)</span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig<span class="sc">$</span>GO.ID <span class="ot">&lt;-</span> <span class="fu">trimws</span>(GO_MF_En_sig<span class="sc">$</span>GO.ID)</span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Join the datasets based on GO term</span></span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene <span class="ot">&lt;-</span> sig_pos_cor_gene2go <span class="sc">%&gt;%</span></span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(GO_MF_En_sig, <span class="at">by =</span> <span class="st">"GO.ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Add ontology column </span></span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene<span class="sc">$</span>ontology <span class="ot">&lt;-</span> <span class="st">"Molecular Functions"</span></span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only unique rows </span></span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>GO_MF_En_sig_gene <span class="ot">&lt;-</span> <span class="fu">unique</span>(GO_MF_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="join-ontologies-3" class="level3">
<h3 class="anchored" data-anchor-id="join-ontologies-3">6.0.4 Join ontologies</h3>
<p>Bind so there is a df that has significantly enriched GO terms for all ontologies</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>GO_sig_pos_corr_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(GO_BP_En_sig_gene, GO_MF_En_sig_gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Merge with <code>GO_pos_corr_df</code></p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> GO_sig_pos_corr_df <span class="sc">%&gt;%</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(sig_pos_corr_data, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_ID"</span> <span class="ot">=</span> <span class="st">"mRNA"</span>)) <span class="co">#%&gt;%</span></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#mutate(direction = ifelse(PCC.cor &gt; 0, "Positive", "Negative")) #%&gt;%</span></span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(ontology != "Cellular Components") #%&gt;%</span></span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>  <span class="co">#filter(p_value &lt; 0.1)</span></span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Save as csv </span></span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(test, <span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_sig_pos_corr_target_enrichment.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Plot!</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(test, <span class="fu">aes</span>(<span class="at">x =</span> Term, <span class="at">y =</span> Fisher, <span class="at">fill =</span> p_value)) <span class="sc">+</span></span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0, 1) +</span></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+</span></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 </span></span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">5</span>) <span class="sc">+</span> </span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_size(range = c(2, 20)) + </span></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fisher p-value"</span>) <span class="sc">+</span></span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">24</span>) <span class="sc">+</span></span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(ontology), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(); plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="09.1-Apul-mRNA-miRNA-interactions-functional-enrichment_files/figure-gfm/unnamed-chunk-68-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot </span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_sig_pos_corr_target_enrichment.pdf"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">35</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/topGO_sig_pos_corr_target_enrichment.png"</span>, plot, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">35</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Examine the top 10 most significant GO terms for BP and MF</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to get top 5 unique terms</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>get_top_10_unique <span class="ot">&lt;-</span> <span class="cf">function</span>(data, ontology_type) {</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  data <span class="sc">%&gt;%</span></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ontology <span class="sc">==</span> ontology_type) <span class="sc">%&gt;%</span></span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(Fisher) <span class="sc">%&gt;%</span></span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(Term, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>)</span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 5 unique Biological Processes</span></span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a>top_10_BP <span class="ot">&lt;-</span> <span class="fu">get_top_10_unique</span>(test, <span class="st">"Biological Processes"</span>)</span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 5 unique Molecular Functions</span></span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a>top_10_MF <span class="ot">&lt;-</span> <span class="fu">get_top_10_unique</span>(test, <span class="st">"Molecular Functions"</span>)</span>
<span id="cb102-15"><a href="#cb102-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-16"><a href="#cb102-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine results</span></span>
<span id="cb102-17"><a href="#cb102-17" aria-hidden="true" tabindex="-1"></a>top_10_combined <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(top_10_BP, top_10_MF)</span>
<span id="cb102-18"><a href="#cb102-18" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(top_10_combined<span class="sc">$</span>Term)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<pre><code>## [1] "GTPase activity"</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot </span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>top_10_combined <span class="ot">&lt;-</span> top_10_combined <span class="sc">%&gt;%</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Fisher)) <span class="sc">%&gt;%</span></span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Term =</span> <span class="fu">factor</span>(Term, <span class="at">levels =</span> <span class="fu">unique</span>(Term)))</span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>plot<span class="ot">&lt;-</span><span class="fu">ggplot</span>(top_10_combined, <span class="fu">aes</span>(<span class="at">x =</span> Term, <span class="at">y =</span> Fisher, <span class="at">fill =</span> p_value)) <span class="sc">+</span></span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#ylim(0, 1) +</span></span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, linetype = "solid", color = "black", linewidth = 1)+</span></span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_hline(yintercept = 0.05, color = "black", linetype = "solid", linewidth = 0.5) +  # Add line at 0.05 </span></span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">10</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span> </span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size</span>(<span class="at">range =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">20</span>)) <span class="sc">+</span> </span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">''</span>) <span class="sc">+</span> </span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Fisher p-value"</span>) <span class="sc">+</span></span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>(<span class="at">base_size =</span> <span class="dv">35</span>) <span class="sc">+</span></span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="fu">vars</span>(ontology), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>(); plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><img src="09.1-Apul-mRNA-miRNA-interactions-functional-enrichment_files/figure-gfm/unnamed-chunk-69-1.png" class="img-fluid"><!-- --></p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save plot </span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/top10GO_sig_pos_corr_target_enrichment.pdf"</span>, <span class="fu">last_plot</span>(), <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"../output/09.1-Apul-mRNA-miRNA-interactions-functional-enrichment/top10GO_sig_pos_corr_target_enrichment.png"</span>, <span class="fu">last_plot</span>(), <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">25</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/shedurkin\.github\.io\/Roberts-LabNotebook");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>This notebook is built using <a href="https://quarto.org/">Quarto</a></p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>