---
title: "Timeseries: Narrowing down methylation features"
author: "Kathleen Durkin"
date: "2025-10-23"
categories: ["E5-coral"]
format:
  html:
    toc: true
execute: 
  eval: TRUE
engine: knitr
bibliography: ../../../references.bib
---

Want to consider the following options:

> ### **Definition of "methylation density" and "methylation level"**
>
> **Methylation density refers to the number of CpG positions that were deemed as methylated within a predefined stretch of DNA sequence.** Qualitatively, we used "sparse" and "dense" to describe methylation density. Quantitative definitions differ depending on context: In Fig. 1, for instance, we were interested in looking at whether there were more methylated positions in a particular genomic feature (exons, introns, or intergenic regions). In this case, we displayed, and discussed, the abundance of methylation positions per standardized length of the genomic feature with arbitrary units.
>
> However, when we compared methylation densities across genes, we had had to consider the effect of gene length on the number of methylated positions (the longer a gene, the higher the number of methylated positions), and CpG biases present in genes (comparing genes of equal length, the one with more CpGs in it will have more methylated positions). We standardized it by defining methylation density as **the fraction of CpG positions in the gene that were deemed methylated**, that is, if a gene has five CpG dinucleotides, there are 10 positions that could be methylated. If 8 positions are methylated, methylation density = 8/10 = 0.8. The range of methylation densities, for this definition, is \[0, 1\].
>
> **Methylation level refers to the abundance of methylation at a particular methylated position.** Qualitatively, we used "weak" and "strong" to describe methylation level. Quantitatively, if a particular cytosine has a coverage of 10×, with eight Cs and two Ts mapping to it, its methylation level is 8/10 = 0.8. Thus, methylation level also has a range of \[0, 1\].
>
> @liew_epigenome-associated_2018

So I should consider:

-   Methylation density. For the feature (e.g. gene), \# of methylated CpGs/ \# unmethylated CpGs, where "methylated CpG" is a binary (yes for any methylation level, "no" if the CpG is completely unmethylated).\
    Note that this will ignore *how* methylated sites are, so a gene where half of the CpGs had 10% methylation would be considered equivalent to a gene where half of the CpGs had 90% methylation

-   coverage-weighted mean methylation across features (genes/intron/exons/etc.)

    Summarizes total methylated reads over total reads across the region of interest (proportional to coverage, so low-coverage sites won't have undue impact)

    Something like this, where M is methylated and U is unmethylated"

    ```         
    weighted_beta <- sum(M_vector) / sum(M_vector + U_vector)
    ```

​​
